// @react-spectrum/dnd/dist/main.js
var $9OUiQ$reactstatelydnd = require("@react-stately/dnd");
var $9OUiQ$reactariadnd = require("@react-aria/dnd");
var $9OUiQ$react = require("react");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "useDragHooks", () => $bb595e9145c20bae$export$ca50582736fec395);
$parcel$export(module.exports, "useDropHooks", () => $eb31ce34e8964bed$export$c9cc1cdafc7aa18a);



function $bb595e9145c20bae$export$ca50582736fec395(options) {
    return $9OUiQ$react.useMemo(()=>({
            useDraggableCollectionState (props) {
                return $9OUiQ$reactstatelydnd.useDraggableCollectionState({
                    ...props,
                    ...options
                });
            },
            useDraggableItem: $9OUiQ$reactariadnd.useDraggableItem,
            DragPreview: $9OUiQ$reactariadnd.DragPreview
        })
    , [
        options
    ]);
}





function $eb31ce34e8964bed$export$c9cc1cdafc7aa18a(options) {
    return $9OUiQ$react.useMemo(()=>({
            useDroppableCollectionState (props) {
                return $9OUiQ$reactstatelydnd.useDroppableCollectionState({
                    ...props,
                    ...options
                });
            },
            useDroppableItem: $9OUiQ$reactariadnd.useDroppableItem,
            useDroppableCollection (props, state, ref) {
                return $9OUiQ$reactariadnd.useDroppableCollection({
                    ...props,
                    ...options
                }, state, ref);
            },
            useDropIndicator: $9OUiQ$reactariadnd.useDropIndicator
        })
    , [
        options
    ]);
}
