// @react-spectrum/dnd/dist/module.js
import {useDraggableCollectionState as $6PX7k$useDraggableCollectionState, useDroppableCollectionState as $6PX7k$useDroppableCollectionState} from "@react-stately/dnd";
import {useDraggableItem as $6PX7k$useDraggableItem, DragPreview as $6PX7k$DragPreview, useDroppableItem as $6PX7k$useDroppableItem, useDroppableCollection as $6PX7k$useDroppableCollection, useDropIndicator as $6PX7k$useDropIndicator} from "@react-aria/dnd";
import {useMemo as $6PX7k$useMemo} from "react";




function $1d9ad6c92ef06635$export$ca50582736fec395(options) {
    return $6PX7k$useMemo(()=>({
            useDraggableCollectionState (props) {
                return $6PX7k$useDraggableCollectionState({
                    ...props,
                    ...options
                });
            },
            useDraggableItem: $6PX7k$useDraggableItem,
            DragPreview: $6PX7k$DragPreview
        })
    , [
        options
    ]);
}





function $d9620cb8af727810$export$c9cc1cdafc7aa18a(options) {
    return $6PX7k$useMemo(()=>({
            useDroppableCollectionState (props) {
                return $6PX7k$useDroppableCollectionState({
                    ...props,
                    ...options
                });
            },
            useDroppableItem: $6PX7k$useDroppableItem,
            useDroppableCollection (props, state, ref) {
                return $6PX7k$useDroppableCollection({
                    ...props,
                    ...options
                }, state, ref);
            },
            useDropIndicator: $6PX7k$useDropIndicator
        })
    , [
        options
    ]);
}




export {$1d9ad6c92ef06635$export$ca50582736fec395 as useDragHooks, $d9620cb8af727810$export$c9cc1cdafc7aa18a as useDropHooks};
