// @react-aria/meter/dist/module.js
import {useProgressBar as $fmhsW$useProgressBar} from "@react-aria/progress";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $0002d7e1d16de204$exports = {};

$parcel$export($0002d7e1d16de204$exports, "useMeter", () => $0002d7e1d16de204$export$e969dbfa146870ff);

function $0002d7e1d16de204$export$e969dbfa146870ff(props) {
    let { progressBarProps: progressBarProps , labelProps: labelProps  } = $fmhsW$useProgressBar(props);
    return {
        meterProps: {
            ...progressBarProps,
            // Use the meter role if available, but fall back to progressbar if not
            // Chrome currently falls back from meter automatically, and Firefox
            // does not support meter at all. Safari 13+ seems to support meter properly.
            // https://bugs.chromium.org/p/chromium/issues/detail?id=944542
            // https://bugzilla.mozilla.org/show_bug.cgi?id=1460378
            role: 'meter progressbar'
        },
        labelProps: labelProps
    };
}




export {$0002d7e1d16de204$export$e969dbfa146870ff as useMeter};
