// @react-spectrum/slider/dist/main.js
require("./main.css");
var $km6zh$reactariautils = require("@react-aria/utils");
var $km6zh$reactspectrumutils = require("@react-spectrum/utils");
var $km6zh$react = require("react");
var $km6zh$reactariai18n = require("@react-aria/i18n");
var $km6zh$reactstatelyslider = require("@react-stately/slider");
var $km6zh$reactspectrumprovider = require("@react-spectrum/provider");
var $km6zh$reactariaslider = require("@react-aria/slider");
var $km6zh$reactariafocus = require("@react-aria/focus");
var $km6zh$reactariainteractions = require("@react-aria/interactions");
var $km6zh$reactariavisuallyhidden = require("@react-aria/visually-hidden");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $3a761bdd22d32681$exports = {};

$parcel$export($3a761bdd22d32681$exports, "Slider", () => $3a761bdd22d32681$export$472062a354075cee);






var $57893a49ef34c169$exports = {};

$parcel$export($57893a49ef34c169$exports, "spectrum-Slider", () => $57893a49ef34c169$export$fc844a18ff837543, (v) => $57893a49ef34c169$export$fc844a18ff837543 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Dial", () => $57893a49ef34c169$export$db619d28010f1dd7, (v) => $57893a49ef34c169$export$db619d28010f1dd7 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider--positionTop", () => $57893a49ef34c169$export$3d0239f4af6225c0, (v) => $57893a49ef34c169$export$3d0239f4af6225c0 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-controls", () => $57893a49ef34c169$export$eed070f7d409a4eb, (v) => $57893a49ef34c169$export$eed070f7d409a4eb = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Dial-controls", () => $57893a49ef34c169$export$9a4ea6c1b342a483, (v) => $57893a49ef34c169$export$9a4ea6c1b342a483 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider--positionSide", () => $57893a49ef34c169$export$7e6c269c6bbb2ffa, (v) => $57893a49ef34c169$export$7e6c269c6bbb2ffa = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-labelContainer", () => $57893a49ef34c169$export$6a95ef18f88e804b, (v) => $57893a49ef34c169$export$6a95ef18f88e804b = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Dial-labelContainer", () => $57893a49ef34c169$export$f64ec32f85626909, (v) => $57893a49ef34c169$export$f64ec32f85626909 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-valueLabelContainer", () => $57893a49ef34c169$export$2183ef1701a88013, (v) => $57893a49ef34c169$export$2183ef1701a88013 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-value", () => $57893a49ef34c169$export$d1c0699c6d985532, (v) => $57893a49ef34c169$export$d1c0699c6d985532 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-track", () => $57893a49ef34c169$export$16e664b3091a0b47, (v) => $57893a49ef34c169$export$16e664b3091a0b47 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-buffer", () => $57893a49ef34c169$export$8ac5a980ecf86d70, (v) => $57893a49ef34c169$export$8ac5a980ecf86d70 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-ramp", () => $57893a49ef34c169$export$74af90f7b59a4a0b, (v) => $57893a49ef34c169$export$74af90f7b59a4a0b = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-fill", () => $57893a49ef34c169$export$79e384500d3b9c9d, (v) => $57893a49ef34c169$export$79e384500d3b9c9d = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-fill--right", () => $57893a49ef34c169$export$4ad5f741edcaf768, (v) => $57893a49ef34c169$export$4ad5f741edcaf768 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider--range", () => $57893a49ef34c169$export$91bbf9ca430efa38, (v) => $57893a49ef34c169$export$91bbf9ca430efa38 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-handle", () => $57893a49ef34c169$export$4e56e324ec89fe53, (v) => $57893a49ef34c169$export$4e56e324ec89fe53 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Dial-handle", () => $57893a49ef34c169$export$41c59246a9248fa9, (v) => $57893a49ef34c169$export$41c59246a9248fa9 = v);
$parcel$export($57893a49ef34c169$exports, "is-dragged", () => $57893a49ef34c169$export$8778c911bed6c759, (v) => $57893a49ef34c169$export$8778c911bed6c759 = v);
$parcel$export($57893a49ef34c169$exports, "is-focused", () => $57893a49ef34c169$export$e7dc768d35940237, (v) => $57893a49ef34c169$export$e7dc768d35940237 = v);
$parcel$export($57893a49ef34c169$exports, "is-tophandle", () => $57893a49ef34c169$export$e7c8b64dd65a8bd6, (v) => $57893a49ef34c169$export$e7c8b64dd65a8bd6 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-input", () => $57893a49ef34c169$export$fdecbf00e674772b, (v) => $57893a49ef34c169$export$fdecbf00e674772b = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Dial-input", () => $57893a49ef34c169$export$39043e8612543a9e, (v) => $57893a49ef34c169$export$39043e8612543a9e = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-label", () => $57893a49ef34c169$export$87d97f82d3be9540, (v) => $57893a49ef34c169$export$87d97f82d3be9540 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Dial-label", () => $57893a49ef34c169$export$d93200a6e46e9ce7, (v) => $57893a49ef34c169$export$d93200a6e46e9ce7 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Dial-value", () => $57893a49ef34c169$export$be548803c10dddf5, (v) => $57893a49ef34c169$export$be548803c10dddf5 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-ticks", () => $57893a49ef34c169$export$8015a78273599416, (v) => $57893a49ef34c169$export$8015a78273599416 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-tick", () => $57893a49ef34c169$export$daf7ddf4a129876d, (v) => $57893a49ef34c169$export$daf7ddf4a129876d = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider-tickLabel", () => $57893a49ef34c169$export$e045256579534563, (v) => $57893a49ef34c169$export$e045256579534563 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider--color", () => $57893a49ef34c169$export$c0271c7b757e2c6d, (v) => $57893a49ef34c169$export$c0271c7b757e2c6d = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Dial--small", () => $57893a49ef34c169$export$40db493959f853b2, (v) => $57893a49ef34c169$export$40db493959f853b2 = v);
$parcel$export($57893a49ef34c169$exports, "is-disabled", () => $57893a49ef34c169$export$d35bc1e505d1ebbf, (v) => $57893a49ef34c169$export$d35bc1e505d1ebbf = v);
$parcel$export($57893a49ef34c169$exports, "is-hovered", () => $57893a49ef34c169$export$b8813cd5d7824ce7, (v) => $57893a49ef34c169$export$b8813cd5d7824ce7 = v);
$parcel$export($57893a49ef34c169$exports, "u-isGrabbing", () => $57893a49ef34c169$export$af9e6e484daea602, (v) => $57893a49ef34c169$export$af9e6e484daea602 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider--filled", () => $57893a49ef34c169$export$743466ce5e272176, (v) => $57893a49ef34c169$export$743466ce5e272176 = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider--ramp", () => $57893a49ef34c169$export$87c47a8a21ea9aa, (v) => $57893a49ef34c169$export$87c47a8a21ea9aa = v);
$parcel$export($57893a49ef34c169$exports, "spectrum-Slider--label-side", () => $57893a49ef34c169$export$fc5a803c4caf36a, (v) => $57893a49ef34c169$export$fc5a803c4caf36a = v);
var $57893a49ef34c169$export$fc844a18ff837543;
var $57893a49ef34c169$export$db619d28010f1dd7;
var $57893a49ef34c169$export$3d0239f4af6225c0;
var $57893a49ef34c169$export$eed070f7d409a4eb;
var $57893a49ef34c169$export$9a4ea6c1b342a483;
var $57893a49ef34c169$export$7e6c269c6bbb2ffa;
var $57893a49ef34c169$export$6a95ef18f88e804b;
var $57893a49ef34c169$export$f64ec32f85626909;
var $57893a49ef34c169$export$2183ef1701a88013;
var $57893a49ef34c169$export$d1c0699c6d985532;
var $57893a49ef34c169$export$16e664b3091a0b47;
var $57893a49ef34c169$export$8ac5a980ecf86d70;
var $57893a49ef34c169$export$74af90f7b59a4a0b;
var $57893a49ef34c169$export$79e384500d3b9c9d;
var $57893a49ef34c169$export$4ad5f741edcaf768;
var $57893a49ef34c169$export$91bbf9ca430efa38;
var $57893a49ef34c169$export$4e56e324ec89fe53;
var $57893a49ef34c169$export$41c59246a9248fa9;
var $57893a49ef34c169$export$8778c911bed6c759;
var $57893a49ef34c169$export$e7dc768d35940237;
var $57893a49ef34c169$export$e7c8b64dd65a8bd6;
var $57893a49ef34c169$export$fdecbf00e674772b;
var $57893a49ef34c169$export$39043e8612543a9e;
var $57893a49ef34c169$export$87d97f82d3be9540;
var $57893a49ef34c169$export$d93200a6e46e9ce7;
var $57893a49ef34c169$export$be548803c10dddf5;
var $57893a49ef34c169$export$8015a78273599416;
var $57893a49ef34c169$export$daf7ddf4a129876d;
var $57893a49ef34c169$export$e045256579534563;
var $57893a49ef34c169$export$c0271c7b757e2c6d;
var $57893a49ef34c169$export$40db493959f853b2;
var $57893a49ef34c169$export$d35bc1e505d1ebbf;
var $57893a49ef34c169$export$b8813cd5d7824ce7;
var $57893a49ef34c169$export$af9e6e484daea602;
var $57893a49ef34c169$export$743466ce5e272176;
var $57893a49ef34c169$export$87c47a8a21ea9aa;
var $57893a49ef34c169$export$fc5a803c4caf36a;
$57893a49ef34c169$export$fc844a18ff837543 = "spectrum-Slider_e4b6ba";
$57893a49ef34c169$export$db619d28010f1dd7 = "spectrum-Dial_e4b6ba";
$57893a49ef34c169$export$3d0239f4af6225c0 = "spectrum-Slider--positionTop_e4b6ba";
$57893a49ef34c169$export$eed070f7d409a4eb = "spectrum-Slider-controls_e4b6ba";
$57893a49ef34c169$export$9a4ea6c1b342a483 = "spectrum-Dial-controls_e4b6ba";
$57893a49ef34c169$export$7e6c269c6bbb2ffa = "spectrum-Slider--positionSide_e4b6ba";
$57893a49ef34c169$export$6a95ef18f88e804b = "spectrum-Slider-labelContainer_e4b6ba";
$57893a49ef34c169$export$f64ec32f85626909 = "spectrum-Dial-labelContainer_e4b6ba";
$57893a49ef34c169$export$2183ef1701a88013 = "spectrum-Slider-valueLabelContainer_e4b6ba";
$57893a49ef34c169$export$d1c0699c6d985532 = "spectrum-Slider-value_e4b6ba";
$57893a49ef34c169$export$16e664b3091a0b47 = "spectrum-Slider-track_e4b6ba";
$57893a49ef34c169$export$8ac5a980ecf86d70 = "spectrum-Slider-buffer_e4b6ba";
$57893a49ef34c169$export$74af90f7b59a4a0b = "spectrum-Slider-ramp_e4b6ba";
$57893a49ef34c169$export$79e384500d3b9c9d = "spectrum-Slider-fill_e4b6ba";
$57893a49ef34c169$export$4ad5f741edcaf768 = "spectrum-Slider-fill--right_e4b6ba";
$57893a49ef34c169$export$91bbf9ca430efa38 = "spectrum-Slider--range_e4b6ba";
$57893a49ef34c169$export$4e56e324ec89fe53 = "spectrum-Slider-handle_e4b6ba";
$57893a49ef34c169$export$41c59246a9248fa9 = "spectrum-Dial-handle_e4b6ba";
$57893a49ef34c169$export$8778c911bed6c759 = "is-dragged_e4b6ba";
$57893a49ef34c169$export$e7dc768d35940237 = "is-focused_e4b6ba";
$57893a49ef34c169$export$e7c8b64dd65a8bd6 = "is-tophandle_e4b6ba";
$57893a49ef34c169$export$fdecbf00e674772b = "spectrum-Slider-input_e4b6ba";
$57893a49ef34c169$export$39043e8612543a9e = "spectrum-Dial-input_e4b6ba";
$57893a49ef34c169$export$87d97f82d3be9540 = "spectrum-Slider-label_e4b6ba";
$57893a49ef34c169$export$d93200a6e46e9ce7 = "spectrum-Dial-label_e4b6ba";
$57893a49ef34c169$export$be548803c10dddf5 = "spectrum-Dial-value_e4b6ba";
$57893a49ef34c169$export$8015a78273599416 = "spectrum-Slider-ticks_e4b6ba";
$57893a49ef34c169$export$daf7ddf4a129876d = "spectrum-Slider-tick_e4b6ba";
$57893a49ef34c169$export$e045256579534563 = "spectrum-Slider-tickLabel_e4b6ba";
$57893a49ef34c169$export$c0271c7b757e2c6d = "spectrum-Slider--color_e4b6ba";
$57893a49ef34c169$export$40db493959f853b2 = "spectrum-Dial--small_e4b6ba";
$57893a49ef34c169$export$d35bc1e505d1ebbf = "is-disabled_e4b6ba";
$57893a49ef34c169$export$b8813cd5d7824ce7 = "is-hovered_e4b6ba";
$57893a49ef34c169$export$af9e6e484daea602 = "u-isGrabbing_e4b6ba";
$57893a49ef34c169$export$743466ce5e272176 = "spectrum-Slider--filled_e4b6ba";
$57893a49ef34c169$export$87c47a8a21ea9aa = "spectrum-Slider--ramp_e4b6ba";
$57893a49ef34c169$export$fc5a803c4caf36a = "spectrum-Slider--label-side_e4b6ba";





function $ec6ca573d82152ff$var$SliderBase(props, ref) {
    props = $km6zh$reactspectrumprovider.useProviderProps(props);
    let { isDisabled: isDisabled , children: children , classes: classes , style: style , labelPosition: labelPosition = 'top' , getValueLabel: getValueLabel , showValueLabel: showValueLabel = !!props.label , formatOptions: formatOptions , minValue: minValue = 0 , maxValue: maxValue = 100 , ...otherProps } = props;
    let { styleProps: styleProps  } = $km6zh$reactspectrumutils.useStyleProps(otherProps);
    // `Math.abs(Math.sign(a) - Math.sign(b)) === 2` is true if the values have a different sign.
    let alwaysDisplaySign = Math.abs(Math.sign(minValue) - Math.sign(maxValue)) === 2;
    if (alwaysDisplaySign) {
        if (formatOptions != null) {
            if (!('signDisplay' in formatOptions)) formatOptions = {
                ...formatOptions,
                // @ts-ignore
                signDisplay: 'exceptZero'
            };
        } else // @ts-ignore
        formatOptions = {
            signDisplay: 'exceptZero'
        };
    }
    const formatter = $km6zh$reactariai18n.useNumberFormatter(formatOptions);
    const state = $km6zh$reactstatelyslider.useSliderState({
        ...props,
        numberFormatter: formatter,
        minValue: minValue,
        maxValue: maxValue
    });
    let trackRef = $km6zh$react.useRef();
    let { groupProps: groupProps , trackProps: trackProps , labelProps: labelProps , outputProps: outputProps  } = $km6zh$reactariaslider.useSlider(props, state, trackRef);
    let inputRef = $km6zh$react.useRef();
    let domRef = $km6zh$reactspectrumutils.useFocusableRef(ref, inputRef);
    let displayValue = '';
    let maxLabelLength = undefined;
    if (typeof getValueLabel === 'function') {
        displayValue = getValueLabel(state.values);
        switch(state.values.length){
            case 1:
                maxLabelLength = Math.max(getValueLabel([
                    minValue
                ]).length, getValueLabel([
                    maxValue
                ]).length);
                break;
            case 2:
                // Try all possible combinations of min and max values.
                maxLabelLength = Math.max(getValueLabel([
                    minValue,
                    minValue
                ]).length, getValueLabel([
                    minValue,
                    maxValue
                ]).length, getValueLabel([
                    maxValue,
                    minValue
                ]).length, getValueLabel([
                    maxValue,
                    maxValue
                ]).length);
                break;
            default:
                throw new Error('Only sliders with 1 or 2 handles are supported!');
        }
    } else {
        maxLabelLength = Math.max([
            ...formatter.format(minValue)
        ].length, [
            ...formatter.format(maxValue)
        ].length);
        switch(state.values.length){
            case 1:
                displayValue = state.getThumbValueLabel(0);
                break;
            case 2:
                // This should really use the NumberFormat#formatRange proposal...
                // https://github.com/tc39/ecma402/issues/393
                // https://github.com/tc39/proposal-intl-numberformat-v3#formatrange-ecma-402-393
                displayValue = `${state.getThumbValueLabel(0)} – ${state.getThumbValueLabel(1)}`;
                maxLabelLength = 3 + 2 * Math.max(maxLabelLength, [
                    ...formatter.format(minValue)
                ].length, [
                    ...formatter.format(maxValue)
                ].length);
                break;
            default:
                throw new Error('Only sliders with 1 or 2 handles are supported!');
        }
    }
    let labelNode = /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("label", {
        className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-label'),
        ...labelProps
    }, props.label);
    let valueNode = /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("output", {
        ...outputProps,
        className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-value'),
        style: maxLabelLength && {
            width: `${maxLabelLength}ch`,
            minWidth: `${maxLabelLength}ch`
        }
    }, displayValue);
    return(/*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
        ref: domRef,
        className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider', {
            'spectrum-Slider--positionTop': labelPosition === 'top',
            'spectrum-Slider--positionSide': labelPosition === 'side',
            'is-disabled': isDisabled
        }, classes, styleProps.className),
        style: {
            ...style,
            ...styleProps.style
        },
        ...groupProps
    }, props.label && /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
        className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-labelContainer'),
        role: "presentation"
    }, props.label && labelNode, labelPosition === 'top' && showValueLabel && valueNode), /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
        className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-controls'),
        ref: trackRef,
        ...trackProps,
        role: "presentation"
    }, children({
        trackRef: trackRef,
        inputRef: inputRef,
        state: state
    })), labelPosition === 'side' && /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
        className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-valueLabelContainer'),
        role: "presentation"
    }, showValueLabel && valueNode)));
}
const $ec6ca573d82152ff$export$9418495bb635ebde = /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).forwardRef($ec6ca573d82152ff$var$SliderBase);











function $77acc7fb8bfd13d6$export$2c1b491743890dec(props) {
    let { index: index , inputRef: inputRef , state: state  } = props;
    let backupRef = $km6zh$react.useRef();
    inputRef = inputRef || backupRef;
    let { thumbProps: thumbProps , inputProps: inputProps  } = $km6zh$reactariaslider.useSliderThumb({
        ...props,
        inputRef: inputRef
    }, state);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $km6zh$reactariainteractions.useHover({
    });
    let { direction: direction  } = $km6zh$reactariai18n.useLocale();
    let cssDirection = direction === 'rtl' ? 'right' : 'left';
    return(/*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement($km6zh$reactariafocus.FocusRing, {
        within: true,
        focusRingClass: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'is-focused')
    }, /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
        className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-handle', {
            'is-hovered': isHovered,
            'is-dragged': state.isThumbDragging(index),
            'is-tophandle': state.focusedThumb === index
        }),
        style: {
            [cssDirection]: `${state.getThumbPercent(index) * 100}%`
        },
        ...$km6zh$reactariautils.mergeProps(thumbProps, hoverProps),
        role: "presentation"
    }, /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement($km6zh$reactariavisuallyhidden.VisuallyHidden, null, /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("input", {
        className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-input'),
        ref: inputRef,
        ...inputProps
    })))));
}




function $3a761bdd22d32681$var$Slider(props, ref) {
    let { onChange: onChange , onChangeEnd: onChangeEnd , value: value , defaultValue: defaultValue , isFilled: isFilled , fillOffset: fillOffset , trackGradient: trackGradient , getValueLabel: getValueLabel , ...otherProps } = props;
    let baseProps = {
        ...otherProps,
        // Normalize `value: number[]` to `value: number`
        value: value != null ? [
            value
        ] : undefined,
        defaultValue: defaultValue != null ? [
            defaultValue
        ] : undefined,
        onChange: (v)=>{
            onChange === null || onChange === void 0 ? void 0 : onChange(v[0]);
        },
        onChangeEnd: (v)=>{
            onChangeEnd === null || onChangeEnd === void 0 ? void 0 : onChangeEnd(v[0]);
        },
        getValueLabel: getValueLabel ? ([v])=>getValueLabel(v)
         : undefined
    };
    let { direction: direction  } = $km6zh$reactariai18n.useLocale();
    return(/*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement($ec6ca573d82152ff$export$9418495bb635ebde, {
        ...baseProps,
        ref: ref,
        classes: {
            'spectrum-Slider--filled': isFilled && fillOffset == null
        },
        style: // @ts-ignore
        {
            '--spectrum-slider-track-gradient': trackGradient && `linear-gradient(to ${direction === 'ltr' ? 'right' : 'left'}, ${trackGradient.join(', ')})`
        }
    }, ({ trackRef: trackRef , inputRef: inputRef , state: state  })=>{
        fillOffset = fillOffset != null ? $km6zh$reactariautils.clamp(fillOffset, state.getThumbMinValue(0), state.getThumbMaxValue(0)) : fillOffset;
        let cssDirection = direction === 'rtl' ? 'right' : 'left';
        let lowerTrack = /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
            className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${state.getThumbPercent(0) * 100}%`,
                // TODO not sure if it has advantages, but this could also be implemented as CSS calc():
                // .track::before {
                //    background-size: calc((1/ (var(--width)/100)) * 100%);
                //    width: calc(var(--width) * 1%)M
                // }
                // @ts-ignore
                '--spectrum-track-background-size': `${1 / state.getThumbPercent(0) * 100}%`,
                '--spectrum-track-background-position': direction === 'ltr' ? '0' : '100%'
            }
        });
        let upperTrack = /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
            className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${(1 - state.getThumbPercent(0)) * 100}%`,
                // @ts-ignore
                '--spectrum-track-background-size': `${1 / (1 - state.getThumbPercent(0)) * 100}%`,
                '--spectrum-track-background-position': direction === 'ltr' ? '100%' : '0'
            }
        });
        let filledTrack = null;
        if (isFilled && fillOffset != null) {
            let width = state.getThumbPercent(0) - state.getValuePercent(fillOffset);
            let isRightOfOffset = width > 0;
            let offset = isRightOfOffset ? state.getValuePercent(fillOffset) : state.getThumbPercent(0);
            filledTrack = /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
                className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-fill', {
                    'spectrum-Slider-fill--right': isRightOfOffset
                }),
                style: {
                    [cssDirection]: `${offset * 100}%`,
                    width: `${Math.abs(width) * 100}%`
                }
            });
        }
        return(/*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement(($parcel$interopDefault($km6zh$react)).Fragment, null, lowerTrack, /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement($77acc7fb8bfd13d6$export$2c1b491743890dec, {
            index: 0,
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            inputRef: inputRef,
            state: state
        }), upperTrack, filledTrack));
    }));
}
/**
 * Sliders allow users to quickly select a value within a range. They should be used when the upper and lower bounds to the range are invariable.
 */ const $3a761bdd22d32681$export$472062a354075cee = /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).forwardRef($3a761bdd22d32681$var$Slider);


var $f166b52c7f82645c$exports = {};

$parcel$export($f166b52c7f82645c$exports, "RangeSlider", () => $f166b52c7f82645c$export$826424dabc3dd705);

var $afaab5654d50c4e1$exports = {};
var $a2445e74ce1bf08e$exports = {};
$a2445e74ce1bf08e$exports = JSON.parse("{\"maximum\":\"أقصى\",\"minimum\":\"أدنى\"}");


var $7529d1c4650d61a0$exports = {};
$7529d1c4650d61a0$exports = JSON.parse("{\"maximum\":\"Максимум\",\"minimum\":\"Минимум\"}");


var $08ef0c921c77a706$exports = {};
$08ef0c921c77a706$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $f1f4cb7353b3e1a5$exports = {};
$f1f4cb7353b3e1a5$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $dda90fce5a19ad19$exports = {};
$dda90fce5a19ad19$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $ebefaefb60993a30$exports = {};
$ebefaefb60993a30$exports = JSON.parse("{\"maximum\":\"Μέγιστο\",\"minimum\":\"Ελάχιστο\"}");


var $b325b3ab75a907a3$exports = {};
$b325b3ab75a907a3$exports = JSON.parse("{\"minimum\":\"Minimum\",\"maximum\":\"Maximum\"}");


var $574d919a54e97d8f$exports = {};
$574d919a54e97d8f$exports = JSON.parse("{\"maximum\":\"Máximo\",\"minimum\":\"Mínimo\"}");


var $0e4c525eebbd0e98$exports = {};
$0e4c525eebbd0e98$exports = JSON.parse("{\"maximum\":\"Maksimaalne\",\"minimum\":\"Minimaalne\"}");


var $bfde7f11aa8e8020$exports = {};
$bfde7f11aa8e8020$exports = JSON.parse("{\"maximum\":\"Maksimi\",\"minimum\":\"Minimi\"}");


var $36bd43961b8b921c$exports = {};
$36bd43961b8b921c$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $e084b86a8ffaff00$exports = {};
$e084b86a8ffaff00$exports = JSON.parse("{\"maximum\":\"מקסימום\",\"minimum\":\"מינימום\"}");


var $0bd0209c7b29fe5e$exports = {};
$0bd0209c7b29fe5e$exports = JSON.parse("{\"maximum\":\"Najviše\",\"minimum\":\"Najmanje\"}");


var $8adcea5187336731$exports = {};
$8adcea5187336731$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $dcf2e3956abc2a1c$exports = {};
$dcf2e3956abc2a1c$exports = JSON.parse("{\"maximum\":\"Massimo\",\"minimum\":\"Minimo\"}");


var $c18b3d59e84bb56c$exports = {};
$c18b3d59e84bb56c$exports = JSON.parse("{\"maximum\":\"最大\",\"minimum\":\"最小\"}");


var $3e8bc4aa20d59ba3$exports = {};
$3e8bc4aa20d59ba3$exports = JSON.parse("{\"maximum\":\"최대\",\"minimum\":\"최소\"}");


var $42ba332c0a3a3511$exports = {};
$42ba332c0a3a3511$exports = JSON.parse("{\"maximum\":\"Daugiausia\",\"minimum\":\"Mažiausia\"}");


var $9b999d057f45d745$exports = {};
$9b999d057f45d745$exports = JSON.parse("{\"maximum\":\"Maksimālā vērtība\",\"minimum\":\"Minimālā vērtība\"}");


var $dc6590540fd7af94$exports = {};
$dc6590540fd7af94$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $7442f76ae633df1a$exports = {};
$7442f76ae633df1a$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $ff01e4173212df57$exports = {};
$ff01e4173212df57$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $834f55baddab886e$exports = {};
$834f55baddab886e$exports = JSON.parse("{\"maximum\":\"Máximo\",\"minimum\":\"Mínimo\"}");


var $632ef0739d349135$exports = {};
$632ef0739d349135$exports = JSON.parse("{\"maximum\":\"Máximo\",\"minimum\":\"Mínimo\"}");


var $978ce1fa765e0e5d$exports = {};
$978ce1fa765e0e5d$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $eb4754473bf590c4$exports = {};
$eb4754473bf590c4$exports = JSON.parse("{\"maximum\":\"Максимум\",\"minimum\":\"Минимум\"}");


var $1513d8732734033f$exports = {};
$1513d8732734033f$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $9d66b76468ad0360$exports = {};
$9d66b76468ad0360$exports = JSON.parse("{\"maximum\":\"Največji\",\"minimum\":\"Najmanj\"}");


var $ed57e62e39c366c3$exports = {};
$ed57e62e39c366c3$exports = JSON.parse("{\"maximum\":\"Najviše\",\"minimum\":\"Najmanje\"}");


var $d2c79b9c3f18b1b0$exports = {};
$d2c79b9c3f18b1b0$exports = JSON.parse("{\"maximum\":\"Maximum\",\"minimum\":\"Minimum\"}");


var $59dd8e51fc51c2ca$exports = {};
$59dd8e51fc51c2ca$exports = JSON.parse("{\"maximum\":\"Maksimum\",\"minimum\":\"Minimum\"}");


var $6fff324f013d6ab3$exports = {};
$6fff324f013d6ab3$exports = JSON.parse("{\"maximum\":\"Максимум\",\"minimum\":\"Мінімум\"}");


var $f796a700d95b5fc7$exports = {};
$f796a700d95b5fc7$exports = JSON.parse("{\"maximum\":\"最大\",\"minimum\":\"最小\"}");


var $04f898e4cac34434$exports = {};
$04f898e4cac34434$exports = JSON.parse("{\"maximum\":\"最大值\",\"minimum\":\"最小值\"}");


$afaab5654d50c4e1$exports = {
    "ar-AE": $a2445e74ce1bf08e$exports,
    "bg-BG": $7529d1c4650d61a0$exports,
    "cs-CZ": $08ef0c921c77a706$exports,
    "da-DK": $f1f4cb7353b3e1a5$exports,
    "de-DE": $dda90fce5a19ad19$exports,
    "el-GR": $ebefaefb60993a30$exports,
    "en-US": $b325b3ab75a907a3$exports,
    "es-ES": $574d919a54e97d8f$exports,
    "et-EE": $0e4c525eebbd0e98$exports,
    "fi-FI": $bfde7f11aa8e8020$exports,
    "fr-FR": $36bd43961b8b921c$exports,
    "he-IL": $e084b86a8ffaff00$exports,
    "hr-HR": $0bd0209c7b29fe5e$exports,
    "hu-HU": $8adcea5187336731$exports,
    "it-IT": $dcf2e3956abc2a1c$exports,
    "ja-JP": $c18b3d59e84bb56c$exports,
    "ko-KR": $3e8bc4aa20d59ba3$exports,
    "lt-LT": $42ba332c0a3a3511$exports,
    "lv-LV": $9b999d057f45d745$exports,
    "nb-NO": $dc6590540fd7af94$exports,
    "nl-NL": $7442f76ae633df1a$exports,
    "pl-PL": $ff01e4173212df57$exports,
    "pt-BR": $834f55baddab886e$exports,
    "pt-PT": $632ef0739d349135$exports,
    "ro-RO": $978ce1fa765e0e5d$exports,
    "ru-RU": $eb4754473bf590c4$exports,
    "sk-SK": $1513d8732734033f$exports,
    "sl-SI": $9d66b76468ad0360$exports,
    "sr-SP": $ed57e62e39c366c3$exports,
    "sv-SE": $d2c79b9c3f18b1b0$exports,
    "tr-TR": $59dd8e51fc51c2ca$exports,
    "uk-UA": $6fff324f013d6ab3$exports,
    "zh-CN": $f796a700d95b5fc7$exports,
    "zh-TW": $04f898e4cac34434$exports
};







function $f166b52c7f82645c$var$RangeSlider(props, ref) {
    let { onChange: onChange , onChangeEnd: onChangeEnd , value: value , defaultValue: defaultValue , getValueLabel: getValueLabel , ...otherProps } = props;
    var _minValue, _maxValue;
    let baseProps = {
        ...otherProps,
        value: value != null ? [
            value.start,
            value.end
        ] : undefined,
        defaultValue: defaultValue != null ? [
            defaultValue.start,
            defaultValue.end
        ] : [
            (_minValue = props.minValue) !== null && _minValue !== void 0 ? _minValue : 0,
            (_maxValue = props.maxValue) !== null && _maxValue !== void 0 ? _maxValue : 100
        ],
        onChange (v) {
            onChange === null || onChange === void 0 ? void 0 : onChange({
                start: v[0],
                end: v[1]
            });
        },
        onChangeEnd (v) {
            onChangeEnd === null || onChangeEnd === void 0 ? void 0 : onChangeEnd({
                start: v[0],
                end: v[1]
            });
        },
        getValueLabel: getValueLabel ? ([start, end])=>getValueLabel({
                start: start,
                end: end
            })
         : undefined
    };
    let formatter = $km6zh$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($afaab5654d50c4e1$exports)));
    let { direction: direction  } = $km6zh$reactariai18n.useLocale();
    return(/*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement($ec6ca573d82152ff$export$9418495bb635ebde, {
        ...baseProps,
        classes: 'spectrum-Slider--range',
        ref: ref
    }, ({ trackRef: trackRef , inputRef: inputRef , state: state  })=>{
        let cssDirection = direction === 'rtl' ? 'right' : 'left';
        return(/*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement(($parcel$interopDefault($km6zh$react)).Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
            className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${state.getThumbPercent(0) * 100}%`
            }
        }), /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement($77acc7fb8bfd13d6$export$2c1b491743890dec, {
            index: 0,
            "aria-label": formatter('minimum'),
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            inputRef: inputRef,
            state: state
        }), /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
            className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-track'),
            style: {
                [cssDirection]: `${state.getThumbPercent(0) * 100}%`,
                width: `${Math.abs(state.getThumbPercent(0) - state.getThumbPercent(1)) * 100}%`
            }
        }), /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement($77acc7fb8bfd13d6$export$2c1b491743890dec, {
            index: 1,
            "aria-label": formatter('maximum'),
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            state: state
        }), /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).createElement("div", {
            className: $km6zh$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($57893a49ef34c169$exports)), 'spectrum-Slider-track'),
            style: {
                width: `${(1 - state.getThumbPercent(1)) * 100}%`
            }
        })));
    }));
}
/**
 * RangeSliders allow users to quickly select a subset range. They should be used when the upper and lower bounds to the range are invariable.
 */ const $f166b52c7f82645c$export$826424dabc3dd705 = /*#__PURE__*/ ($parcel$interopDefault($km6zh$react)).forwardRef($f166b52c7f82645c$var$RangeSlider);


$parcel$exportWildcard(module.exports, $3a761bdd22d32681$exports);
$parcel$exportWildcard(module.exports, $f166b52c7f82645c$exports);
