// @adobe/react-spectrum-workflow/dist/Deduplication.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uDeduplication = A4uDeduplication;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uDeduplication(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("circle", {
    cx: 7.939,
    cy: 6.25,
    r: 3.75
  }), _react["default"].createElement("path", {
    d: "M21.506 10h-8.75l4.375-7.5 4.375 7.5z"
  }), _react["default"].createElement("circle", {
    cx: 11.939,
    cy: 30.25,
    r: 3.75
  }), _react["default"].createElement("path", {
    d: "M27.603 34h-8.75l4.375-7.5 4.375 7.5zM32.121 10h-8.75l4.375-7.5 4.375 7.5zM27.939 12.058h-20v1.222a1.514 1.514 0 0 0 .723 1.3l5.689 4.02a3.056 3.056 0 0 1 1.114 2.377v4.193a.733.733 0 0 0 .714.75H19.7a.733.733 0 0 0 .714-.75v-4.194a3.056 3.056 0 0 1 1.113-2.376l5.689-4.015a1.514 1.514 0 0 0 .723-1.3z"
  }));
}
