// @adobe/react-spectrum-workflow/dist/DeselectCircular.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uDeselectCircular = A4uDeselectCircular;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uDeselectCircular(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("rect", {
    height: 43.854,
    rx: 0.818,
    ry: 0.818,
    transform: "rotate(-45 18 18)",
    width: 2.455,
    x: 16.772,
    y: -3.927
  }), _react["default"].createElement("path", {
    d: "M31.569 21.45a13.9 13.9 0 0 1-1.661 3.895l1.448 1.448a15.884 15.884 0 0 0 2.152-4.852zM29.1 9.463c.132.17.26.345.382.521l1.642-1.143q-.211-.3-.439-.6a15.985 15.985 0 0 0-3.6-3.42l-1.137 1.648A14.009 14.009 0 0 1 29.1 9.463zM32 17.979h2a15.927 15.927 0 0 0-1.018-5.6l-1.872.7a13.944 13.944 0 0 1 .89 4.9zM10.657 6.094a13.866 13.866 0 0 1 3.811-1.646l-.5-1.935A15.875 15.875 0 0 0 9.21 4.647zM22.844 4.862l.69-1.877A16.174 16.174 0 0 0 17.928 2l.007 2a14.166 14.166 0 0 1 4.909.862zM4.43 14.55a13.929 13.929 0 0 1 1.661-3.9L4.643 9.207a15.9 15.9 0 0 0-2.152 4.852zM6.9 26.537a14.79 14.79 0 0 1-.382-.521L4.88 27.159q.212.3.439.6a16.027 16.027 0 0 0 3.6 3.42l1.136-1.647A13.982 13.982 0 0 1 6.9 26.537zM4 18.021H2a15.927 15.927 0 0 0 1.018 5.6l1.873-.7a13.9 13.9 0 0 1-.891-4.9zM25.343 29.906a13.9 13.9 0 0 1-3.812 1.646l.5 1.935a15.875 15.875 0 0 0 4.754-2.134zM13.155 31.137l-.69 1.878a16.174 16.174 0 0 0 5.606.985l-.007-2a14.144 14.144 0 0 1-4.909-.863z"
  }));
}
