// @adobe/react-spectrum-workflow/dist/GraphPathing.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uGraphPathing = A4uGraphPathing;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uGraphPathing(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("rect", {
    height: 12,
    rx: 0.5,
    ry: 0.5,
    width: 6,
    x: 2,
    y: 2
  }), _react["default"].createElement("rect", {
    height: 8,
    rx: 0.5,
    ry: 0.5,
    width: 8,
    x: 26,
    y: 2
  }), _react["default"].createElement("rect", {
    height: 8,
    rx: 0.5,
    ry: 0.5,
    width: 8,
    x: 26,
    y: 14
  }), _react["default"].createElement("rect", {
    height: 8,
    rx: 0.5,
    ry: 0.5,
    width: 8,
    x: 26,
    y: 26
  }), _react["default"].createElement("path", {
    d: "M24 6.479a.508.508 0 0 1-.513.5 28.045 28.045 0 0 1-7.35-1.088 22.668 22.668 0 0 0-5.639-.9.5.5 0 0 1-.5-.5v-1a.51.51 0 0 1 .518-.5 24.63 24.63 0 0 1 6.115.965A26.4 26.4 0 0 0 23.5 4.982a.5.5 0 0 1 .5.5zM24 18.058a.5.5 0 0 1-.525.5c-2.937-.236-4.214-2.459-5.452-4.612-1.532-2.666-2.982-5.189-7.531-5.358A.5.5 0 0 1 10 8.1v-1a.505.505 0 0 1 .517-.5c5.7.194 7.657 3.606 9.241 6.362 1.225 2.132 2.045 3.412 3.769 3.6a.511.511 0 0 1 .473.5z"
  }), _react["default"].createElement("path", {
    d: "M24 30.452a.51.51 0 0 1-.591.5c-3.2-.431-4.6-4.385-6.079-8.557-1.573-4.437-3.2-9.019-6.858-9.381a.505.505 0 0 1-.472-.499v-1.007a.5.5 0 0 1 .525-.5c5.02.357 6.966 5.851 8.69 10.718 1.249 3.522 2.432 6.862 4.417 7.23a.479.479 0 0 1 .368.481z"
  }));
}
