// @adobe/react-spectrum-workflow/dist/ImageAutoMode.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uImageAutoMode = A4uImageAutoMode;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uImageAutoMode(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("circle", {
    cx: 20.757,
    cy: 19.283,
    r: 2.5
  }), _react["default"].createElement("path", {
    d: "M20.865.409l.1 2.842a2.318 2.318 0 0 0 1.186 1.939l2.482 1.39-2.843.1a2.317 2.317 0 0 0-1.938 1.184l-1.39 2.482-.1-2.843a2.317 2.317 0 0 0-1.184-1.939l-2.482-1.39 2.843-.1a2.318 2.318 0 0 0 1.936-1.184zM29.686 5.541l.133 3.659a2.984 2.984 0 0 0 1.524 2.5l3.2 1.79-3.661.133a2.982 2.982 0 0 0-2.5 1.524l-1.791 3.2-.132-3.661a2.986 2.986 0 0 0-1.525-2.5l-3.2-1.791 3.661-.132a2.987 2.987 0 0 0 2.5-1.525z"
  }), _react["default"].createElement("path", {
    d: "M26 22v6.463l-3.687-3.686a2 2 0 0 0-2.828 0l-3.071 3.071-7.556-7.556a2 2 0 0 0-2.829 0L2 24.321V14h21l-3-2H1a1 1 0 0 0-1 1v18a1 1 0 0 0 1 1h26a1 1 0 0 0 1-1V19z"
  }));
}
