// @adobe/react-spectrum-workflow/dist/Relevance.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uRelevance = A4uRelevance;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uRelevance(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("path", {
    d: "M4.225 15.585a13.987 13.987 0 0 1 11.36-11.36A.494.494 0 0 0 16 3.74V2.721a.5.5 0 0 0-.578-.5 15.992 15.992 0 0 0-13.2 13.2.5.5 0 0 0 .5.578H3.74a.494.494 0 0 0 .485-.414zM20.415 4.225a13.987 13.987 0 0 1 11.36 11.36.494.494 0 0 0 .485.415h1.019a.5.5 0 0 0 .5-.578 15.992 15.992 0 0 0-13.2-13.2.5.5 0 0 0-.578.5V3.74a.494.494 0 0 0 .414.485zM15.585 31.775a13.987 13.987 0 0 1-11.36-11.36A.494.494 0 0 0 3.74 20H2.721a.5.5 0 0 0-.5.578 15.992 15.992 0 0 0 13.2 13.2.5.5 0 0 0 .578-.5V32.26a.494.494 0 0 0-.414-.485zM31.775 20.415a13.987 13.987 0 0 1-11.36 11.36.494.494 0 0 0-.415.485v1.019a.5.5 0 0 0 .578.5 15.992 15.992 0 0 0 13.2-13.2.5.5 0 0 0-.5-.578H32.26a.494.494 0 0 0-.485.414z"
  }), _react["default"].createElement("circle", {
    cx: 18,
    cy: 18,
    r: 6
  }));
}
