// @adobe/react-spectrum-workflow/dist/Sandbox.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uSandbox = A4uSandbox;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uSandbox(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("rect", {
    x: 2,
    y: 2,
    width: 14,
    height: 30,
    rx: 1
  }), _react["default"].createElement("path", {
    d: "M24 2h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M24 2h2v2h-2zM28 2h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M28 2h2v2h-2zM34 4V3a1 1 0 0 0-1-1h-1v2z"
  }), _react["default"].createElement("path", {
    d: "M34 4V3a1 1 0 0 0-1-1h-1v2zM22 4V2h-1a1 1 0 0 0-1 1v1zM20 6h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M20 6h2v2h-2zM20 10h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M20 10h2v2h-2zM20 14h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M20 14h2v2h-2zM20 18h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M20 18h2v2h-2zM20 22h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M20 22h2v2h-2zM20 26h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M20 26h2v2h-2zM22 32v-2h-2v1a1 1 0 0 0 1 1z"
  }), _react["default"].createElement("path", {
    d: "M22 32v-2h-2v1a1 1 0 0 0 1 1z"
  }), _react["default"].createElement("g", null, _react["default"].createElement("path", {
    d: "M24 30h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M24 30h2v2h-2z"
  })), _react["default"].createElement("g", null, _react["default"].createElement("path", {
    d: "M28 30h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M28 30h2v2h-2z"
  })), _react["default"].createElement("g", null, _react["default"].createElement("path", {
    d: "M32 6h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M32 6h2v2h-2z"
  })), _react["default"].createElement("g", null, _react["default"].createElement("path", {
    d: "M32 10h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M32 10h2v2h-2z"
  })), _react["default"].createElement("g", null, _react["default"].createElement("path", {
    d: "M32 14h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M32 14h2v2h-2z"
  })), _react["default"].createElement("g", null, _react["default"].createElement("path", {
    d: "M32 18h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M32 18h2v2h-2z"
  })), _react["default"].createElement("g", null, _react["default"].createElement("path", {
    d: "M32 22h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M32 22h2v2h-2z"
  })), _react["default"].createElement("g", null, _react["default"].createElement("path", {
    d: "M32 26h2v2h-2z"
  }), _react["default"].createElement("path", {
    d: "M32 26h2v2h-2z"
  })), _react["default"].createElement("g", null, _react["default"].createElement("path", {
    d: "M34 31v-1h-2v2h1a1 1 0 0 0 1-1z"
  }), _react["default"].createElement("path", {
    d: "M34 31v-1h-2v2h1a1 1 0 0 0 1-1z"
  })));
}
