// @adobe/react-spectrum-workflow/dist/StraightenOutline.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uStraightenOutline = A4uStraightenOutline;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uStraightenOutline(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("path", {
    d: "M33.5 14H.5a.5.5 0 0 0-.5.5v13a.5.5 0 0 0 .5.5h33a.5.5 0 0 0 .5-.5v-13a.5.5 0 0 0-.5-.5zm-11.286 2l.018.968C22.232 19.05 18.9 20.1 17 20.1s-5.25-1.05-5.25-3.107V16zM6 26H2V16h4zm20 0H8V16h2v1c0 3 3.134 5 7 5s7-2 7-5v-1h2zm6 0h-4V16h4z"
  }), _react["default"].createElement("circle", {
    cx: 7,
    cy: 11,
    r: 1.3
  }), _react["default"].createElement("circle", {
    cx: 27,
    cy: 11,
    r: 1.3
  }), _react["default"].createElement("circle", {
    cx: 17,
    cy: 5,
    r: 1.3
  }), _react["default"].createElement("circle", {
    cx: 11,
    cy: 7,
    r: 1.3
  }), _react["default"].createElement("circle", {
    cx: 23,
    cy: 7,
    r: 1.3
  }));
}
