// @adobe/react-spectrum-workflow/dist/Subscribe.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uSubscribe = A4uSubscribe;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uSubscribe(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("path", {
    d: "M24.779 21.963L36 30.367V13.541l-11.221 8.422zM22.866 23.4l-3.576 2.694a2.172 2.172 0 0 1-2.58 0l-3.628-2.719L0 33.068A.981.981 0 0 0 1 34h34a.884.884 0 0 0 1-.756zM11.165 21.938L0 13.511v16.684l11.165-8.257z"
  }), _react["default"].createElement("path", {
    d: "M19.067.672a2 2 0 0 0-2.133 0L0 11.365V14h6V9a1 1 0 0 1 1-1h22a1 1 0 0 1 1 1v5h6v-2.665z"
  }), _react["default"].createElement("rect", {
    height: 2,
    rx: 0.5,
    ry: 0.5,
    width: 16,
    x: 10,
    y: 12
  }), _react["default"].createElement("path", {
    d: "M21.83 20h-7.66a.5.5 0 0 1-.3-.1l-1.882-1.448a.25.25 0 0 1 .147-.452h11.73a.25.25 0 0 1 .152.448L22.135 19.9a.5.5 0 0 1-.305.1z"
  }));
}
