// @adobe/react-spectrum-workflow/dist/SuccessMetric.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uSuccessMetric = A4uSuccessMetric;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uSuccessMetric(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("rect", {
    height: 8,
    rx: 1,
    ry: 1,
    width: 8,
    x: 4,
    y: 26
  }), _react["default"].createElement("rect", {
    height: 24,
    rx: 1,
    ry: 1,
    width: 8,
    x: 14,
    y: 10
  }), _react["default"].createElement("rect", {
    height: 12,
    rx: 1,
    ry: 1,
    width: 8,
    x: 24,
    y: 22
  }), _react["default"].createElement("path", {
    d: "M12 16H6.5a.5.5 0 0 1-.5-.5v-1a.5.5 0 0 1 .5-.5H12zM7.768 6.27L12 8.979l-1.078 1.684-4.233-2.709a.5.5 0 0 1-.152-.691l.539-.842a.5.5 0 0 1 .692-.151zM16.63 8l-1.9-5.971a.25.25 0 0 0-.314-.163l-1.43.454a.25.25 0 0 0-.163.314L14.532 8zM24 16h5.5a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5H24zM28.232 6.27L24 8.979l1.078 1.684 4.233-2.709a.5.5 0 0 0 .152-.691l-.539-.842a.5.5 0 0 0-.692-.151zM19.37 8l1.9-5.971a.25.25 0 0 1 .314-.163l1.43.454a.25.25 0 0 1 .163.314L21.468 8z"
  }));
}
