// @adobe/react-spectrum-workflow/dist/TextBulletedHierarchyExclude.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uTextBulletedHierarchyExclude = A4uTextBulletedHierarchyExclude;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uTextBulletedHierarchyExclude(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("path", {
    d: "M27 18.1a8.9 8.9 0 1 0 8.9 8.9 8.9 8.9 0 0 0-8.9-8.9zM20 27a6.929 6.929 0 0 1 1.475-4.252l9.777 9.777A6.966 6.966 0 0 1 20 27zm12.525 4.252l-9.777-9.777a6.966 6.966 0 0 1 9.777 9.777z"
  }), _react["default"].createElement("rect", {
    height: 6,
    rx: 2.8,
    ry: 2.8,
    width: 6,
    y: 2
  }), _react["default"].createElement("rect", {
    height: 6,
    rx: 2.8,
    ry: 2.8,
    width: 6,
    x: 6,
    y: 14
  }), _react["default"].createElement("rect", {
    height: 6,
    rx: 2.8,
    ry: 2.8,
    width: 6,
    x: 6,
    y: 26
  }), _react["default"].createElement("rect", {
    height: 4,
    rx: 1,
    ry: 1,
    width: 22,
    x: 8,
    y: 4
  }), _react["default"].createElement("path", {
    d: "M27 16H15a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h3.515A10.975 10.975 0 0 1 27 16zM16.05 28H15a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h2.21a10.942 10.942 0 0 1-1.16-4z"
  }));
}
