// @adobe/react-spectrum-workflow/dist/TextBaselineShift.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uTextBaselineShift = A4uTextBaselineShift;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uTextBaselineShift(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("path", {
    d: "M21.3 23.776L13.061 3c-.037-.129-.071-.16-.212-.16H9.412a.16.16 0 0 0-.176.16 3.073 3.073 0 0 1-.246 1.312L1.345 23.744c-.034.16.036.256.21.256H3.94c.175 0 .247-.064.281-.192L6.488 18h9.428l2.3 5.84a.317.317 0 0 0 .28.16h2.666c.176 0 .21-.1.138-.224zM11.167 5.017h.033c.665 2.176 3.345 8.9 4.117 10.983H7.091c1.333-3.521 3.479-8.935 4.076-10.983z"
  }), _react["default"].createElement("rect", {
    height: 2,
    rx: 0.5,
    ry: 0.5,
    width: 21,
    x: 1,
    y: 26
  }), _react["default"].createElement("rect", {
    height: 2,
    rx: 0.5,
    ry: 0.5,
    width: 12,
    x: 23,
    y: 16
  }), _react["default"].createElement("path", {
    d: "M33.537 11.728a9.194 9.194 0 0 0 .047 1.148c0 .048 0 .071-.047.1A9.872 9.872 0 0 1 29.065 14c-2.536 0-4.449-1.244-4.449-3.755 0-2.535 2.367-3.659 5.334-3.659.883 0 1.386.025 1.65.048V6.06c0-.74-.36-2.391-2.7-2.391a6.414 6.414 0 0 0-3.037.717.117.117 0 0 1-.166-.119V2.808a.21.21 0 0 1 .119-.191 7.9 7.9 0 0 1 3.391-.717c3.061 0 4.33 2.008 4.33 4.5zM31.6 8.212a11.4 11.4 0 0 0-1.58-.1c-2.32 0-3.444.79-3.444 2.129 0 1.076.743 2.129 2.846 2.129a5.614 5.614 0 0 0 2.178-.407zM33.871 25.166l-4-4a.5.5 0 0 0-.743 0l-4 4A.49.49 0 0 0 25 25.5a.5.5 0 0 0 .5.5H28v5.155a.845.845 0 0 0 .845.845h1.31a.845.845 0 0 0 .845-.845V26h2.5a.5.5 0 0 0 .5-.5.49.49 0 0 0-.129-.334z"
  }));
}
