// @adobe/react-spectrum-workflow/dist/ViewAllTags.js
/**
* @adobe/react-spectrum-workflow (c) by Adobe
* 
* @adobe/react-spectrum-workflow is licensed under a
* Creative Commons Attribution-NoDerivatives 4.0 International License.
* 
* You should have received a copy of the license along with this
* work. If not, see <http://creativecommons.org/licenses/by-nd/4.0/>.
**/
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.A4uViewAllTags = A4uViewAllTags;

var _react = _interopRequireDefault(require("react"));

var _util = require("./util");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function A4uViewAllTags(_ref) {
  var _ref$scale = _ref.scale,
      scale = _ref$scale === void 0 ? 'M' : _ref$scale,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? 'LIGHT' : _ref$color,
      props = _objectWithoutProperties(_ref, ["scale", "color"]);

  return _react["default"].createElement("svg", _extends({
    viewBox: _util.viewBoxes[scale]
  }, props, props), _react["default"].createElement("rect", {
    height: 4,
    rx: 1,
    ry: 1,
    width: 4,
    x: 2,
    y: 2
  }), _react["default"].createElement("rect", {
    height: 4,
    rx: 1,
    ry: 1,
    width: 22,
    x: 10,
    y: 2
  }), _react["default"].createElement("rect", {
    height: 4,
    rx: 1,
    ry: 1,
    width: 4,
    x: 2,
    y: 10
  }), _react["default"].createElement("rect", {
    height: 4,
    rx: 1,
    ry: 1,
    width: 22,
    x: 10,
    y: 10
  }), _react["default"].createElement("rect", {
    height: 4,
    rx: 1,
    ry: 1,
    width: 4,
    x: 2,
    y: 18
  }), _react["default"].createElement("rect", {
    height: 4,
    rx: 1,
    ry: 1,
    width: 4,
    x: 2,
    y: 26
  }), _react["default"].createElement("path", {
    d: "M35.668 26.106l-9.88-9.879a.772.772 0 0 0-.546-.227h-8.47a.772.772 0 0 0-.772.772v8.471a.772.772 0 0 0 .226.546l9.879 9.88a.772.772 0 0 0 1.092 0l8.471-8.469a.772.772 0 0 0 0-1.094zM20.4 22.948a2.548 2.548 0 1 1 2.548-2.548 2.548 2.548 0 0 1-2.548 2.548zM14.294 27.2c-.332-.332-.223-.756-.353-1.2H11a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h6.091zM14 18h-3a1 1 0 0 0-1 1v2a1 1 0 0 0 1 1h3z"
  }));
}
