// @react-aria/link/dist/main.js
var $3NNeN$reactariautils = require("@react-aria/utils");
var $3NNeN$reactariafocus = require("@react-aria/focus");
var $3NNeN$reactariainteractions = require("@react-aria/interactions");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $fbe6b19572273b2f$exports = {};

$parcel$export($fbe6b19572273b2f$exports, "useLink", () => $fbe6b19572273b2f$export$dcf14c9974fe2767);



function $fbe6b19572273b2f$export$dcf14c9974fe2767(props, ref) {
    let { elementType: elementType = 'a' , onPress: onPress , onPressStart: onPressStart , onPressEnd: onPressEnd , // @ts-ignore
    onClick: deprecatedOnClick , isDisabled: isDisabled , ...otherProps } = props;
    let linkProps;
    if (elementType !== 'a') linkProps = {
        role: 'link',
        tabIndex: !isDisabled ? 0 : undefined
    };
    let { focusableProps: focusableProps  } = $3NNeN$reactariafocus.useFocusable(props, ref);
    let { pressProps: pressProps , isPressed: isPressed  } = $3NNeN$reactariainteractions.usePress({
        onPress: onPress,
        onPressStart: onPressStart,
        onPressEnd: onPressEnd,
        isDisabled: isDisabled,
        ref: ref
    });
    let domProps = $3NNeN$reactariautils.filterDOMProps(otherProps, {
        labelable: true
    });
    let interactionHandlers = $3NNeN$reactariautils.mergeProps(focusableProps, pressProps);
    return {
        isPressed: isPressed,
        linkProps: $3NNeN$reactariautils.mergeProps(domProps, {
            ...interactionHandlers,
            ...linkProps,
            'aria-disabled': isDisabled || undefined,
            onClick: (e)=>{
                pressProps.onClick(e);
                if (deprecatedOnClick) {
                    deprecatedOnClick(e);
                    console.warn('onClick is deprecated, please use onPress');
                }
            }
        })
    };
}


$parcel$exportWildcard(module.exports, $fbe6b19572273b2f$exports);
