// @react-spectrum/list/dist/main.js
require("./main.css");
var $k9Du2$reactstatelycollections = require("@react-stately/collections");
var $k9Du2$reactspectrumutils = require("@react-spectrum/utils");
var $k9Du2$reactariautils = require("@react-aria/utils");
var $k9Du2$reactariafocus = require("@react-aria/focus");
var $k9Du2$reactstatelylayout = require("@react-stately/layout");
var $k9Du2$reactstatelylist = require("@react-stately/list");
var $k9Du2$reactspectrumprogress = require("@react-spectrum/progress");
var $k9Du2$react = require("react");
var $k9Du2$reactariai18n = require("@react-aria/i18n");
var $k9Du2$reactariagridlist = require("@react-aria/gridlist");
var $k9Du2$reactspectrumprovider = require("@react-spectrum/provider");
var $k9Du2$reactariavirtualizer = require("@react-aria/virtualizer");
var $k9Du2$reactariavisuallyhidden = require("@react-aria/visually-hidden");
var $k9Du2$reactspectrumcheckbox = require("@react-spectrum/checkbox");
var $k9Du2$spectrumiconsuiChevronLeftMedium = require("@spectrum-icons/ui/ChevronLeftMedium");
var $k9Du2$spectrumiconsuiChevronRightMedium = require("@spectrum-icons/ui/ChevronRightMedium");
var $k9Du2$reacttransitiongroup = require("react-transition-group");
var $k9Du2$reactspectrumlayout = require("@react-spectrum/layout");
var $k9Du2$reactariainteractions = require("@react-aria/interactions");
var $k9Du2$spectrumiconsuiListGripper = require("@spectrum-icons/ui/ListGripper");
var $k9Du2$reactspectrumtext = require("@react-spectrum/text");
var $k9Du2$reactariabutton = require("@react-aria/button");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "ListView", () => $60684b441be8e84c$export$84d0dd190d551cd1);
$parcel$export(module.exports, "Item", () => $k9Du2$reactstatelycollections.Item);




var $279507a6f189e509$exports = {};

$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView", () => $279507a6f189e509$export$4ad4709e9aee8e02, (v) => $279507a6f189e509$export$4ad4709e9aee8e02 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView-row", () => $279507a6f189e509$export$f80410436c6352c8, (v) => $279507a6f189e509$export$f80410436c6352c8 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--emphasized", () => $279507a6f189e509$export$eb0d5352b2a55473, (v) => $279507a6f189e509$export$eb0d5352b2a55473 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--dropTarget", () => $279507a6f189e509$export$d9ba3f873d0eaec6, (v) => $279507a6f189e509$export$d9ba3f873d0eaec6 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem", () => $279507a6f189e509$export$ce8c93b50eec2d4f, (v) => $279507a6f189e509$export$ce8c93b50eec2d4f = v);
$parcel$export($279507a6f189e509$exports, "is-selected", () => $279507a6f189e509$export$1e0fb04f31d3c22a, (v) => $279507a6f189e509$export$1e0fb04f31d3c22a = v);
$parcel$export($279507a6f189e509$exports, "is-hovered", () => $279507a6f189e509$export$b8813cd5d7824ce7, (v) => $279507a6f189e509$export$b8813cd5d7824ce7 = v);
$parcel$export($279507a6f189e509$exports, "is-active", () => $279507a6f189e509$export$20fd0f7cd4e6112f, (v) => $279507a6f189e509$export$20fd0f7cd4e6112f = v);
$parcel$export($279507a6f189e509$exports, "focus-ring", () => $279507a6f189e509$export$f39a09f249340e2a, (v) => $279507a6f189e509$export$f39a09f249340e2a = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem--firstRow", () => $279507a6f189e509$export$60162f05499b7560, (v) => $279507a6f189e509$export$60162f05499b7560 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--quiet", () => $279507a6f189e509$export$b01cb9680b0b8d22, (v) => $279507a6f189e509$export$b01cb9680b0b8d22 = v);
$parcel$export($279507a6f189e509$exports, "is-next-selected", () => $279507a6f189e509$export$53aa811f86efbfad, (v) => $279507a6f189e509$export$53aa811f86efbfad = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem--isFlushBottom", () => $279507a6f189e509$export$f9ed481cac4d935a, (v) => $279507a6f189e509$export$f9ed481cac4d935a = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--loadingMore", () => $279507a6f189e509$export$5aa06adeaa92dbca, (v) => $279507a6f189e509$export$5aa06adeaa92dbca = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem--lastRow", () => $279507a6f189e509$export$659c081622dc13b0, (v) => $279507a6f189e509$export$659c081622dc13b0 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--isVerticalScrollbarVisible", () => $279507a6f189e509$export$1bf50375883926, (v) => $279507a6f189e509$export$1bf50375883926 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--isHorizontalScrollbarVisible", () => $279507a6f189e509$export$d9b13621bc5dd4d1, (v) => $279507a6f189e509$export$d9b13621bc5dd4d1 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--wrap", () => $279507a6f189e509$export$68b45d42c77fe1cf, (v) => $279507a6f189e509$export$68b45d42c77fe1cf = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-content", () => $279507a6f189e509$export$dc2b0b08d47555, (v) => $279507a6f189e509$export$dc2b0b08d47555 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-description", () => $279507a6f189e509$export$5d2adfeb248e654a, (v) => $279507a6f189e509$export$5d2adfeb248e654a = v);
$parcel$export($279507a6f189e509$exports, "round-tops", () => $279507a6f189e509$export$3b4577d06eb82ec0, (v) => $279507a6f189e509$export$3b4577d06eb82ec0 = v);
$parcel$export($279507a6f189e509$exports, "round-bottoms", () => $279507a6f189e509$export$8a26470723c97997, (v) => $279507a6f189e509$export$8a26470723c97997 = v);
$parcel$export($279507a6f189e509$exports, "is-focused", () => $279507a6f189e509$export$e7dc768d35940237, (v) => $279507a6f189e509$export$e7dc768d35940237 = v);
$parcel$export($279507a6f189e509$exports, "is-disabled", () => $279507a6f189e509$export$d35bc1e505d1ebbf, (v) => $279507a6f189e509$export$d35bc1e505d1ebbf = v);
$parcel$export($279507a6f189e509$exports, "has-checkbox", () => $279507a6f189e509$export$27f4d14f7fc5b94, (v) => $279507a6f189e509$export$27f4d14f7fc5b94 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-grid", () => $279507a6f189e509$export$aaaf83100348d389, (v) => $279507a6f189e509$export$aaaf83100348d389 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-draghandle-container", () => $279507a6f189e509$export$f385e7456ddfee84, (v) => $279507a6f189e509$export$f385e7456ddfee84 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-draghandle-button", () => $279507a6f189e509$export$4784f0c2ec8bbf7b, (v) => $279507a6f189e509$export$4784f0c2ec8bbf7b = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkboxWrapper", () => $279507a6f189e509$export$41a33ae9818c51be, (v) => $279507a6f189e509$export$41a33ae9818c51be = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkbox", () => $279507a6f189e509$export$f979f76ca45242f2, (v) => $279507a6f189e509$export$f979f76ca45242f2 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-thumbnail", () => $279507a6f189e509$export$38c6ebb7f58dedaa, (v) => $279507a6f189e509$export$38c6ebb7f58dedaa = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--hasDescription", () => $279507a6f189e509$export$7c9efe74c4b77579, (v) => $279507a6f189e509$export$7c9efe74c4b77579 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-actions", () => $279507a6f189e509$export$26480389595d4515, (v) => $279507a6f189e509$export$26480389595d4515 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-actionmenu", () => $279507a6f189e509$export$58dfda3903289699, (v) => $279507a6f189e509$export$58dfda3903289699 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-parentIndicator", () => $279507a6f189e509$export$39b42a8a24290c17, (v) => $279507a6f189e509$export$39b42a8a24290c17 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-dragPreview", () => $279507a6f189e509$export$84286d36e5c00c55, (v) => $279507a6f189e509$export$84286d36e5c00c55 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-badge", () => $279507a6f189e509$export$6498e436231eea33, (v) => $279507a6f189e509$export$6498e436231eea33 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-dragPreview--multiple", () => $279507a6f189e509$export$2fb3c7c2d5c0d720, (v) => $279507a6f189e509$export$2fb3c7c2d5c0d720 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--hasAnyChildren", () => $279507a6f189e509$export$7150e72e440612bf, (v) => $279507a6f189e509$export$7150e72e440612bf = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-parentIndicator--hasChildItems", () => $279507a6f189e509$export$c29dcfa80dab8c88, (v) => $279507a6f189e509$export$c29dcfa80dab8c88 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--compact", () => $279507a6f189e509$export$1f55b27adf96a61e, (v) => $279507a6f189e509$export$1f55b27adf96a61e = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--spacious", () => $279507a6f189e509$export$cbd30c7809acdd81, (v) => $279507a6f189e509$export$cbd30c7809acdd81 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--draggable", () => $279507a6f189e509$export$a2d35348c0cc9bca, (v) => $279507a6f189e509$export$a2d35348c0cc9bca = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem--dropTarget", () => $279507a6f189e509$export$8afa006321ee3a8f, (v) => $279507a6f189e509$export$8afa006321ee3a8f = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView-centeredWrapper", () => $279507a6f189e509$export$1b66ee5cd8394caf, (v) => $279507a6f189e509$export$1b66ee5cd8394caf = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView-centeredWrapper--loadingMore", () => $279507a6f189e509$export$6b3b7792abd9e5f4, (v) => $279507a6f189e509$export$6b3b7792abd9e5f4 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewInsertionIndicator", () => $279507a6f189e509$export$2a6a235e0baff403, (v) => $279507a6f189e509$export$2a6a235e0baff403 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewInsertionIndicator--dropTarget", () => $279507a6f189e509$export$927a243805a411ca, (v) => $279507a6f189e509$export$927a243805a411ca = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkbox--enter", () => $279507a6f189e509$export$9dc11c0e0825a249, (v) => $279507a6f189e509$export$9dc11c0e0825a249 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkbox--enterActive", () => $279507a6f189e509$export$f89865e352230a35, (v) => $279507a6f189e509$export$f89865e352230a35 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkbox--exit", () => $279507a6f189e509$export$867c05a16e5d2a7c, (v) => $279507a6f189e509$export$867c05a16e5d2a7c = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkbox--exitActive", () => $279507a6f189e509$export$61b66671605f1b1a, (v) => $279507a6f189e509$export$61b66671605f1b1a = v);
var $279507a6f189e509$export$4ad4709e9aee8e02;
var $279507a6f189e509$export$f80410436c6352c8;
var $279507a6f189e509$export$eb0d5352b2a55473;
var $279507a6f189e509$export$d9ba3f873d0eaec6;
var $279507a6f189e509$export$ce8c93b50eec2d4f;
var $279507a6f189e509$export$1e0fb04f31d3c22a;
var $279507a6f189e509$export$b8813cd5d7824ce7;
var $279507a6f189e509$export$20fd0f7cd4e6112f;
var $279507a6f189e509$export$f39a09f249340e2a;
var $279507a6f189e509$export$60162f05499b7560;
var $279507a6f189e509$export$b01cb9680b0b8d22;
var $279507a6f189e509$export$53aa811f86efbfad;
var $279507a6f189e509$export$f9ed481cac4d935a;
var $279507a6f189e509$export$5aa06adeaa92dbca;
var $279507a6f189e509$export$659c081622dc13b0;
var $279507a6f189e509$export$1bf50375883926;
var $279507a6f189e509$export$d9b13621bc5dd4d1;
var $279507a6f189e509$export$68b45d42c77fe1cf;
var $279507a6f189e509$export$dc2b0b08d47555;
var $279507a6f189e509$export$5d2adfeb248e654a;
var $279507a6f189e509$export$3b4577d06eb82ec0;
var $279507a6f189e509$export$8a26470723c97997;
var $279507a6f189e509$export$e7dc768d35940237;
var $279507a6f189e509$export$d35bc1e505d1ebbf;
var $279507a6f189e509$export$27f4d14f7fc5b94;
var $279507a6f189e509$export$aaaf83100348d389;
var $279507a6f189e509$export$f385e7456ddfee84;
var $279507a6f189e509$export$4784f0c2ec8bbf7b;
var $279507a6f189e509$export$41a33ae9818c51be;
var $279507a6f189e509$export$f979f76ca45242f2;
var $279507a6f189e509$export$38c6ebb7f58dedaa;
var $279507a6f189e509$export$7c9efe74c4b77579;
var $279507a6f189e509$export$26480389595d4515;
var $279507a6f189e509$export$58dfda3903289699;
var $279507a6f189e509$export$39b42a8a24290c17;
var $279507a6f189e509$export$84286d36e5c00c55;
var $279507a6f189e509$export$6498e436231eea33;
var $279507a6f189e509$export$2fb3c7c2d5c0d720;
var $279507a6f189e509$export$7150e72e440612bf;
var $279507a6f189e509$export$c29dcfa80dab8c88;
var $279507a6f189e509$export$1f55b27adf96a61e;
var $279507a6f189e509$export$cbd30c7809acdd81;
var $279507a6f189e509$export$a2d35348c0cc9bca;
var $279507a6f189e509$export$8afa006321ee3a8f;
var $279507a6f189e509$export$1b66ee5cd8394caf;
var $279507a6f189e509$export$6b3b7792abd9e5f4;
var $279507a6f189e509$export$2a6a235e0baff403;
var $279507a6f189e509$export$927a243805a411ca;
var $279507a6f189e509$export$9dc11c0e0825a249;
var $279507a6f189e509$export$f89865e352230a35;
var $279507a6f189e509$export$867c05a16e5d2a7c;
var $279507a6f189e509$export$61b66671605f1b1a;
$279507a6f189e509$export$4ad4709e9aee8e02 = "react-spectrum-ListView_dd5487";
$279507a6f189e509$export$f80410436c6352c8 = "react-spectrum-ListView-row_dd5487";
$279507a6f189e509$export$eb0d5352b2a55473 = "react-spectrum-ListView--emphasized_dd5487";
$279507a6f189e509$export$d9ba3f873d0eaec6 = "react-spectrum-ListView--dropTarget_dd5487";
$279507a6f189e509$export$ce8c93b50eec2d4f = "react-spectrum-ListViewItem_dd5487";
$279507a6f189e509$export$1e0fb04f31d3c22a = "is-selected_dd5487";
$279507a6f189e509$export$b8813cd5d7824ce7 = "is-hovered_dd5487";
$279507a6f189e509$export$20fd0f7cd4e6112f = "is-active_dd5487";
$279507a6f189e509$export$f39a09f249340e2a = "focus-ring_dd5487";
$279507a6f189e509$export$60162f05499b7560 = "react-spectrum-ListViewItem--firstRow_dd5487";
$279507a6f189e509$export$b01cb9680b0b8d22 = "react-spectrum-ListView--quiet_dd5487";
$279507a6f189e509$export$53aa811f86efbfad = "is-next-selected_dd5487";
$279507a6f189e509$export$f9ed481cac4d935a = "react-spectrum-ListViewItem--isFlushBottom_dd5487";
$279507a6f189e509$export$5aa06adeaa92dbca = "react-spectrum-ListView--loadingMore_dd5487";
$279507a6f189e509$export$659c081622dc13b0 = "react-spectrum-ListViewItem--lastRow_dd5487";
$279507a6f189e509$export$1bf50375883926 = "react-spectrum-ListView--isVerticalScrollbarVisible_dd5487";
$279507a6f189e509$export$d9b13621bc5dd4d1 = "react-spectrum-ListView--isHorizontalScrollbarVisible_dd5487";
$279507a6f189e509$export$68b45d42c77fe1cf = "react-spectrum-ListView--wrap_dd5487";
$279507a6f189e509$export$dc2b0b08d47555 = "react-spectrum-ListViewItem-content_dd5487";
$279507a6f189e509$export$5d2adfeb248e654a = "react-spectrum-ListViewItem-description_dd5487";
$279507a6f189e509$export$3b4577d06eb82ec0 = "round-tops_dd5487";
$279507a6f189e509$export$8a26470723c97997 = "round-bottoms_dd5487";
$279507a6f189e509$export$e7dc768d35940237 = "is-focused_dd5487";
$279507a6f189e509$export$d35bc1e505d1ebbf = "is-disabled_dd5487";
$279507a6f189e509$export$27f4d14f7fc5b94 = "has-checkbox_dd5487";
$279507a6f189e509$export$aaaf83100348d389 = "react-spectrum-ListViewItem-grid_dd5487";
$279507a6f189e509$export$f385e7456ddfee84 = "react-spectrum-ListViewItem-draghandle-container_dd5487";
$279507a6f189e509$export$4784f0c2ec8bbf7b = "react-spectrum-ListViewItem-draghandle-button_dd5487";
$279507a6f189e509$export$41a33ae9818c51be = "react-spectrum-ListViewItem-checkboxWrapper_dd5487";
$279507a6f189e509$export$f979f76ca45242f2 = "react-spectrum-ListViewItem-checkbox_dd5487";
$279507a6f189e509$export$38c6ebb7f58dedaa = "react-spectrum-ListViewItem-thumbnail_dd5487";
$279507a6f189e509$export$7c9efe74c4b77579 = "react-spectrum-ListView--hasDescription_dd5487";
$279507a6f189e509$export$26480389595d4515 = "react-spectrum-ListViewItem-actions_dd5487";
$279507a6f189e509$export$58dfda3903289699 = "react-spectrum-ListViewItem-actionmenu_dd5487";
$279507a6f189e509$export$39b42a8a24290c17 = "react-spectrum-ListViewItem-parentIndicator_dd5487";
$279507a6f189e509$export$84286d36e5c00c55 = "react-spectrum-ListViewItem-dragPreview_dd5487";
$279507a6f189e509$export$6498e436231eea33 = "react-spectrum-ListViewItem-badge_dd5487";
$279507a6f189e509$export$2fb3c7c2d5c0d720 = "react-spectrum-ListViewItem-dragPreview--multiple_dd5487";
$279507a6f189e509$export$7150e72e440612bf = "react-spectrum-ListView--hasAnyChildren_dd5487";
$279507a6f189e509$export$c29dcfa80dab8c88 = "react-spectrum-ListViewItem-parentIndicator--hasChildItems_dd5487";
$279507a6f189e509$export$1f55b27adf96a61e = "react-spectrum-ListView--compact_dd5487";
$279507a6f189e509$export$cbd30c7809acdd81 = "react-spectrum-ListView--spacious_dd5487";
$279507a6f189e509$export$a2d35348c0cc9bca = "react-spectrum-ListView--draggable_dd5487";
$279507a6f189e509$export$8afa006321ee3a8f = "react-spectrum-ListViewItem--dropTarget_dd5487";
$279507a6f189e509$export$1b66ee5cd8394caf = "react-spectrum-ListView-centeredWrapper_dd5487";
$279507a6f189e509$export$6b3b7792abd9e5f4 = "react-spectrum-ListView-centeredWrapper--loadingMore_dd5487";
$279507a6f189e509$export$2a6a235e0baff403 = "react-spectrum-ListViewInsertionIndicator_dd5487";
$279507a6f189e509$export$927a243805a411ca = "react-spectrum-ListViewInsertionIndicator--dropTarget_dd5487";
$279507a6f189e509$export$9dc11c0e0825a249 = "react-spectrum-ListViewItem-checkbox--enter_dd5487";
$279507a6f189e509$export$f89865e352230a35 = "react-spectrum-ListViewItem-checkbox--enterActive_dd5487";
$279507a6f189e509$export$867c05a16e5d2a7c = "react-spectrum-ListViewItem-checkbox--exit_dd5487";
$279507a6f189e509$export$61b66671605f1b1a = "react-spectrum-ListViewItem-checkbox--exitActive_dd5487";





function $fc02fd49ab58c72e$export$2e2bcd8739ae039(props) {
    let { dropState: dropState , dropHooks: dropHooks  } = $k9Du2$react.useContext($60684b441be8e84c$export$870039b0abfe3de0);
    const { target: target , isPresentationOnly: isPresentationOnly  } = props;
    let ref = $k9Du2$react.useRef();
    let { dropIndicatorProps: dropIndicatorProps  } = dropHooks.useDropIndicator(props, dropState, ref);
    let { visuallyHiddenProps: visuallyHiddenProps  } = $k9Du2$reactariavisuallyhidden.useVisuallyHidden();
    let isDropTarget = dropState.isDropTarget(target);
    if (!isDropTarget && dropIndicatorProps['aria-hidden']) return null;
    return(/*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        role: "row",
        "aria-hidden": dropIndicatorProps['aria-hidden']
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        role: "gridcell",
        "aria-selected": "false",
        className: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'react-spectrum-ListViewInsertionIndicator', {
            'react-spectrum-ListViewInsertionIndicator--dropTarget': isDropTarget
        })
    }, !isPresentationOnly && /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        ...visuallyHiddenProps,
        role: "button",
        ...dropIndicatorProps,
        ref: ref
    }))));
}


var $fd0918c0ca138160$exports = {};
var $1beeff0a5a0ccb2a$exports = {};
$1beeff0a5a0ccb2a$exports = {
    "loading": `جارٍ التحميل...`,
    "loadingMore": `جارٍ تحميل المزيد...`
};


var $8d53c5c3a365f51f$exports = {};
$8d53c5c3a365f51f$exports = {
    "loading": `Зареждане...`,
    "loadingMore": `Зареждане на още...`
};


var $99a0008efa578202$exports = {};
$99a0008efa578202$exports = {
    "loading": `Načítání...`,
    "loadingMore": `Načítání dalších...`
};


var $fed6a92f3a9a90c2$exports = {};
$fed6a92f3a9a90c2$exports = {
    "loading": `Indlæser...`,
    "loadingMore": `Indlæser flere...`
};


var $ad6943b7faf38b41$exports = {};
$ad6943b7faf38b41$exports = {
    "loading": `Laden...`,
    "loadingMore": `Mehr laden ...`
};


var $ed6da119df8f733b$exports = {};
$ed6da119df8f733b$exports = {
    "loading": `Φόρτωση...`,
    "loadingMore": `Φόρτωση περισσότερων...`
};


var $a922ddd791b2701d$exports = {};
$a922ddd791b2701d$exports = {
    "loading": `Loading…`,
    "loadingMore": `Loading more…`
};


var $1616ac893d9df40d$exports = {};
$1616ac893d9df40d$exports = {
    "loading": `Cargando…`,
    "loadingMore": `Cargando más…`
};


var $71139980c308673a$exports = {};
$71139980c308673a$exports = {
    "loading": `Laadimine...`,
    "loadingMore": `Laadi rohkem...`
};


var $df400f4d8990deb0$exports = {};
$df400f4d8990deb0$exports = {
    "loading": `Ladataan…`,
    "loadingMore": `Ladataan lisää…`
};


var $4fedac3a4a024339$exports = {};
$4fedac3a4a024339$exports = {
    "loading": `Chargement...`,
    "loadingMore": `Chargement supplémentaire...`
};


var $0c1298be372fa672$exports = {};
$0c1298be372fa672$exports = {
    "loading": `טוען...`,
    "loadingMore": `טוען עוד...`
};


var $aac34baf9a6ccbce$exports = {};
$aac34baf9a6ccbce$exports = {
    "loading": `Učitavam...`,
    "loadingMore": `Učitavam još...`
};


var $6d93c8a72fbf4a90$exports = {};
$6d93c8a72fbf4a90$exports = {
    "loading": `Betöltés folyamatban…`,
    "loadingMore": `Továbbiak betöltése folyamatban…`
};


var $69346d8975e9f344$exports = {};
$69346d8975e9f344$exports = {
    "loading": `Caricamento...`,
    "loadingMore": `Caricamento altri...`
};


var $b263dd2c132d2852$exports = {};
$b263dd2c132d2852$exports = {
    "loading": `読み込み中...`,
    "loadingMore": `さらに読み込み中...`
};


var $b79146096cdb30c7$exports = {};
$b79146096cdb30c7$exports = {
    "loading": `로드 중…`,
    "loadingMore": `추가 로드 중…`
};


var $adc50e375b062a9e$exports = {};
$adc50e375b062a9e$exports = {
    "loading": `Įkeliama...`,
    "loadingMore": `Įkeliama daugiau...`
};


var $20ba6eae003d4c10$exports = {};
$20ba6eae003d4c10$exports = {
    "loading": `Notiek ielāde...`,
    "loadingMore": `Tiek ielādēts vēl...`
};


var $6a03ad2ccd528f82$exports = {};
$6a03ad2ccd528f82$exports = {
    "loading": `Laster inn ...`,
    "loadingMore": `Laster inn flere ...`
};


var $6bdc84276c605774$exports = {};
$6bdc84276c605774$exports = {
    "loading": `Laden...`,
    "loadingMore": `Meer laden...`
};


var $41e9b40a686907b7$exports = {};
$41e9b40a686907b7$exports = {
    "loading": `Ładowanie...`,
    "loadingMore": `Wczytywanie większej liczby...`
};


var $332f285483e392b9$exports = {};
$332f285483e392b9$exports = {
    "loading": `Carregando...`,
    "loadingMore": `Carregando mais...`
};


var $21a7a272ae3b9fd4$exports = {};
$21a7a272ae3b9fd4$exports = {
    "loading": `A carregar...`,
    "loadingMore": `A carregar mais...`
};


var $7f773822d0ecfb7d$exports = {};
$7f773822d0ecfb7d$exports = {
    "loading": `Se încarcă...`,
    "loadingMore": `Se încarcă mai multe...`
};


var $7d8f25b69820e6ce$exports = {};
$7d8f25b69820e6ce$exports = {
    "loading": `Загрузка...`,
    "loadingMore": `Дополнительная загрузка...`
};


var $ef1cbac310feac6e$exports = {};
$ef1cbac310feac6e$exports = {
    "loading": `Načítava sa...`,
    "loadingMore": `Načítava sa viac...`
};


var $a37847ff0890b3f8$exports = {};
$a37847ff0890b3f8$exports = {
    "loading": `Nalaganje ...`,
    "loadingMore": `Nalaganje več vsebine ...`
};


var $f95d6d9be7c5418d$exports = {};
$f95d6d9be7c5418d$exports = {
    "loading": `Učitavam...`,
    "loadingMore": `Učitavam još...`
};


var $eb2bbd94105316d8$exports = {};
$eb2bbd94105316d8$exports = {
    "loading": `Läser in...`,
    "loadingMore": `Läser in mer...`
};


var $afc69bf18d2f3b1a$exports = {};
$afc69bf18d2f3b1a$exports = {
    "loading": `Yükleniyor...`,
    "loadingMore": `Daha fazla yükleniyor...`
};


var $eda3e7326f5e4dcb$exports = {};
$eda3e7326f5e4dcb$exports = {
    "loading": `Завантаження…`,
    "loadingMore": `Завантаження інших об’єктів...`
};


var $6d455ae2db499a05$exports = {};
$6d455ae2db499a05$exports = {
    "loading": `正在加载...`,
    "loadingMore": `正在加载更多...`
};


var $0fa2d17f8a898b48$exports = {};
$0fa2d17f8a898b48$exports = {
    "loading": `載入中…`,
    "loadingMore": `正在載入更多…`
};


$fd0918c0ca138160$exports = {
    "ar-AE": $1beeff0a5a0ccb2a$exports,
    "bg-BG": $8d53c5c3a365f51f$exports,
    "cs-CZ": $99a0008efa578202$exports,
    "da-DK": $fed6a92f3a9a90c2$exports,
    "de-DE": $ad6943b7faf38b41$exports,
    "el-GR": $ed6da119df8f733b$exports,
    "en-US": $a922ddd791b2701d$exports,
    "es-ES": $1616ac893d9df40d$exports,
    "et-EE": $71139980c308673a$exports,
    "fi-FI": $df400f4d8990deb0$exports,
    "fr-FR": $4fedac3a4a024339$exports,
    "he-IL": $0c1298be372fa672$exports,
    "hr-HR": $aac34baf9a6ccbce$exports,
    "hu-HU": $6d93c8a72fbf4a90$exports,
    "it-IT": $69346d8975e9f344$exports,
    "ja-JP": $b263dd2c132d2852$exports,
    "ko-KR": $b79146096cdb30c7$exports,
    "lt-LT": $adc50e375b062a9e$exports,
    "lv-LV": $20ba6eae003d4c10$exports,
    "nb-NO": $6a03ad2ccd528f82$exports,
    "nl-NL": $6bdc84276c605774$exports,
    "pl-PL": $41e9b40a686907b7$exports,
    "pt-BR": $332f285483e392b9$exports,
    "pt-PT": $21a7a272ae3b9fd4$exports,
    "ro-RO": $7f773822d0ecfb7d$exports,
    "ru-RU": $7d8f25b69820e6ce$exports,
    "sk-SK": $ef1cbac310feac6e$exports,
    "sl-SI": $a37847ff0890b3f8$exports,
    "sr-SP": $f95d6d9be7c5418d$exports,
    "sv-SE": $eb2bbd94105316d8$exports,
    "tr-TR": $afc69bf18d2f3b1a$exports,
    "uk-UA": $eda3e7326f5e4dcb$exports,
    "zh-CN": $6d455ae2db499a05$exports,
    "zh-TW": $0fa2d17f8a898b48$exports
};
























function $9eae7a1cb1535a6b$export$c6bde0c04b033c0e(props) {
    var ref, ref1;
    let { item: item , isEmphasized: isEmphasized  } = props;
    let { state: state , dragState: dragState , dropState: dropState , isListDraggable: isListDraggable , isListDroppable: isListDroppable , layout: layout , dragHooks: dragHooks , dropHooks: dropHooks , loadingState: loadingState  } = $k9Du2$react.useContext($60684b441be8e84c$export$870039b0abfe3de0);
    let { direction: direction  } = $k9Du2$reactariai18n.useLocale();
    let rowRef = $k9Du2$react.useRef();
    let { isFocusVisible: isFocusVisibleWithin , focusProps: focusWithinProps  } = $k9Du2$reactariafocus.useFocusRing({
        within: true
    });
    let { isFocusVisible: isFocusVisible , focusProps: focusProps  } = $k9Du2$reactariafocus.useFocusRing();
    let { rowProps: rowProps , gridCellProps: gridCellProps , isPressed: isPressed , descriptionProps: descriptionProps , isSelected: isSelected , isDisabled: isDisabled , allowsSelection: allowsSelection , hasAction: hasAction  } = $k9Du2$reactariagridlist.useGridListItem({
        node: item,
        isVirtualized: true,
        shouldSelectOnPressUp: isListDraggable
    }, state, rowRef);
    let isDroppable = isListDroppable && !isDisabled;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $k9Du2$reactariainteractions.useHover({
        isDisabled: !allowsSelection && !hasAction
    });
    let { checkboxProps: checkboxProps  } = $k9Du2$reactariagridlist.useGridListSelectionCheckbox({
        key: item.key
    }, state);
    let hasDescription = $k9Du2$reactspectrumutils.useHasChild(`.${(/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-description']}`, rowRef);
    let draggableItem;
    if (isListDraggable) {
        // eslint-disable-next-line react-hooks/rules-of-hooks
        draggableItem = dragHooks.useDraggableItem({
            key: item.key
        }, dragState);
        if (isDisabled) draggableItem = null;
    }
    let droppableItem;
    let isDropTarget;
    let dropIndicator;
    let dropIndicatorRef = $k9Du2$react.useRef();
    if (isListDroppable) {
        let target = {
            type: 'item',
            key: item.key,
            dropPosition: 'on'
        };
        isDropTarget = dropState.isDropTarget(target);
        // eslint-disable-next-line react-hooks/rules-of-hooks
        droppableItem = dropHooks.useDroppableItem({
            target: target
        }, dropState, rowRef);
        dropIndicator = dropHooks.useDropIndicator({
            target: target
        }, dropState, dropIndicatorRef);
    }
    let dragButtonRef = ($parcel$interopDefault($k9Du2$react)).useRef();
    let { buttonProps: buttonProps  } = $k9Du2$reactariabutton.useButton({
        ...draggableItem === null || draggableItem === void 0 ? void 0 : draggableItem.dragButtonProps,
        elementType: 'div'
    }, dragButtonRef);
    let chevron = direction === 'ltr' ? /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement(($parcel$interopDefault($k9Du2$spectrumiconsuiChevronRightMedium)), {
        "aria-hidden": "true",
        UNSAFE_className: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'react-spectrum-ListViewItem-parentIndicator', {
            'react-spectrum-ListViewItem-parentIndicator--hasChildItems': item.props.hasChildItems,
            'is-disabled': !hasAction
        })
    }) : /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement(($parcel$interopDefault($k9Du2$spectrumiconsuiChevronLeftMedium)), {
        "aria-hidden": "true",
        UNSAFE_className: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'react-spectrum-ListViewItem-parentIndicator', {
            'react-spectrum-ListViewItem-parentIndicator--hasChildItems': item.props.hasChildItems,
            'is-disabled': !hasAction
        })
    });
    let showCheckbox = state.selectionManager.selectionMode !== 'none' && state.selectionManager.selectionBehavior === 'toggle';
    let { visuallyHiddenProps: visuallyHiddenProps  } = $k9Du2$reactariavisuallyhidden.useVisuallyHidden();
    let dropProps = isDroppable ? droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.dropProps : {
        'aria-hidden': droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.dropProps['aria-hidden']
    };
    const mergedProps = $k9Du2$reactariautils.mergeProps(rowProps, draggableItem === null || draggableItem === void 0 ? void 0 : draggableItem.dragProps, dropProps, hoverProps, focusWithinProps, focusProps);
    let isFirstRow = item.prevKey == null;
    let isLastRow = item.nextKey == null;
    // Figure out if the ListView content is equal or greater in height to the container. If so, we'll need to round the bottom
    // border corners of the last row when selected and we can get rid of the bottom border if it isn't selected to avoid border overlap
    // with bottom border
    let isFlushWithContainerBottom = false;
    if (isLastRow && loadingState !== 'loadingMore') {
        if (((ref = layout.getContentSize()) === null || ref === void 0 ? void 0 : ref.height) >= ((ref1 = layout.virtualizer) === null || ref1 === void 0 ? void 0 : ref1.getVisibleRect().height)) isFlushWithContainerBottom = true;
    }
    // previous item isn't selected
    // and the previous item isn't focused or, if it is focused, then if focus globally isn't visible or just focus isn't in the listview
    let roundTops = !state.selectionManager.isSelected(item.prevKey) && (state.selectionManager.focusedKey !== item.prevKey || !($k9Du2$reactariainteractions.isFocusVisible() && state.selectionManager.isFocused));
    let roundBottoms = !state.selectionManager.isSelected(item.nextKey) && (state.selectionManager.focusedKey !== item.nextKey || !($k9Du2$reactariainteractions.isFocusVisible() && state.selectionManager.isFocused));
    let content = typeof item.rendered === 'string' ? /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactspectrumtext.Text, null, item.rendered) : item.rendered;
    if (isDisabled) content = /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactspectrumprovider.Provider, {
        isDisabled: true
    }, content);
    return(/*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        ...mergedProps,
        className: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'react-spectrum-ListView-row', {
            'focus-ring': isFocusVisible,
            'round-tops': roundTops || isHovered && !isSelected && state.selectionManager.focusedKey !== item.key,
            'round-bottoms': roundBottoms || isHovered && !isSelected && state.selectionManager.focusedKey !== item.key
        }),
        ref: rowRef
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        // TODO: refactor the css here now that we are focusing the row?
        className: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'react-spectrum-ListViewItem', {
            'is-active': isPressed,
            'is-focused': isFocusVisibleWithin,
            'focus-ring': isFocusVisible,
            'is-hovered': isHovered,
            'is-selected': isSelected,
            'is-disabled': isDisabled,
            'is-prev-selected': state.selectionManager.isSelected(item.prevKey),
            'is-next-selected': state.selectionManager.isSelected(item.nextKey),
            'react-spectrum-ListViewItem--highlightSelection': state.selectionManager.selectionBehavior === 'replace' && (isSelected || state.selectionManager.isSelected(item.nextKey)),
            'react-spectrum-ListViewItem--dropTarget': !!isDropTarget,
            'react-spectrum-ListViewItem--firstRow': isFirstRow,
            'react-spectrum-ListViewItem--lastRow': isLastRow,
            'react-spectrum-ListViewItem--isFlushBottom': isFlushWithContainerBottom,
            'react-spectrum-ListViewItem--hasDescription': hasDescription
        }),
        ...gridCellProps
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactspectrumlayout.Grid, {
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-grid']
    }, isListDraggable && /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-draghandle-container']
    }, !isDisabled && /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactariafocus.FocusRing, {
        focusRingClass: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        ...buttonProps,
        className: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'react-spectrum-ListViewItem-draghandle-button'),
        style: !isFocusVisibleWithin ? {
            ...visuallyHiddenProps.style
        } : {
        },
        ref: dragButtonRef,
        draggable: "true"
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement(($parcel$interopDefault($k9Du2$spectrumiconsuiListGripper)), null)))), isDropTarget && !(dropIndicator === null || dropIndicator === void 0 ? void 0 : dropIndicator.dropIndicatorProps['aria-hidden']) && /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        role: "button",
        ...visuallyHiddenProps,
        ...dropIndicator === null || dropIndicator === void 0 ? void 0 : dropIndicator.dropIndicatorProps,
        ref: dropIndicatorRef
    }), /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reacttransitiongroup.CSSTransition, {
        in: showCheckbox,
        unmountOnExit: true,
        classNames: {
            enter: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-checkbox--enter'],
            enterActive: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-checkbox--enterActive'],
            exit: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-checkbox--exit'],
            exitActive: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-checkbox--exitActive']
        },
        timeout: 160
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-checkboxWrapper']
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactspectrumcheckbox.Checkbox, {
        ...checkboxProps,
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-checkbox'],
        isEmphasized: isEmphasized
    }))), /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactspectrumutils.SlotProvider, {
        slots: {
            text: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-content']
            },
            description: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-description'],
                ...descriptionProps
            },
            illustration: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-thumbnail']
            },
            image: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-thumbnail']
            },
            actionButton: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-actions'],
                isQuiet: true
            },
            actionGroup: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-actions'],
                isQuiet: true,
                density: 'compact'
            },
            actionMenu: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-actionmenu'],
                isQuiet: true
            }
        }
    }, content, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactspectrumutils.ClearSlots, null, chevron))))));
}








function $c59d3898f402f50b$export$2e2bcd8739ae039() {
    let { dropState: dropState , dropHooks: dropHooks  } = $k9Du2$react.useContext($60684b441be8e84c$export$870039b0abfe3de0);
    let ref = $k9Du2$react.useRef();
    let { dropIndicatorProps: dropIndicatorProps  } = dropHooks.useDropIndicator({
        target: {
            type: 'root'
        }
    }, dropState, ref);
    let isDropTarget = dropState.isDropTarget({
        type: 'root'
    });
    let { visuallyHiddenProps: visuallyHiddenProps  } = $k9Du2$reactariavisuallyhidden.useVisuallyHidden();
    if (!isDropTarget && dropIndicatorProps['aria-hidden']) return null;
    return(/*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        role: "row",
        "aria-hidden": dropIndicatorProps['aria-hidden']
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        role: "gridcell",
        "aria-selected": "false",
        ...visuallyHiddenProps
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        role: "button",
        ...dropIndicatorProps,
        ref: ref
    }))));
}







function $3d665b19f7865ff9$export$905ab40ac2179daa(props) {
    let { item: item , itemCount: itemCount , itemHeight: itemHeight  } = props;
    let isDraggingMultiple = itemCount > 1;
    return(/*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        style: {
            height: itemHeight
        },
        className: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'react-spectrum-ListViewItem', 'react-spectrum-ListViewItem-dragPreview', {
            'react-spectrum-ListViewItem-dragPreview--multiple': isDraggingMultiple
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactspectrumlayout.Grid, {
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-grid']
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactspectrumutils.SlotProvider, {
        slots: {
            text: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-content']
            },
            description: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-description']
            },
            illustration: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-thumbnail']
            },
            image: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-thumbnail']
            },
            actionButton: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-actions'],
                isQuiet: true
            },
            actionGroup: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-actions'],
                isQuiet: true,
                density: 'compact'
            },
            actionMenu: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))['react-spectrum-ListViewItem-actionmenu'],
                isQuiet: true
            }
        }
    }, typeof item.rendered === 'string' ? /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactspectrumtext.Text, null, item.rendered) : item.rendered, isDraggingMultiple && /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        className: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'react-spectrum-ListViewItem-badge')
    }, itemCount)))));
}






const $60684b441be8e84c$export$870039b0abfe3de0 = /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createContext(null);
const $60684b441be8e84c$var$ROW_HEIGHTS = {
    compact: {
        medium: 32,
        large: 40
    },
    regular: {
        medium: 40,
        large: 50
    },
    spacious: {
        medium: 48,
        large: 60
    }
};
function $60684b441be8e84c$var$useListLayout(state, density, overflowMode) {
    let { scale: scale  } = $k9Du2$reactspectrumprovider.useProvider();
    let collator = $k9Du2$reactariai18n.useCollator({
        usage: 'search',
        sensitivity: 'base'
    });
    let isEmpty = state.collection.size === 0;
    let layout = $k9Du2$react.useMemo(()=>new $k9Du2$reactstatelylayout.ListLayout({
            estimatedRowHeight: $60684b441be8e84c$var$ROW_HEIGHTS[density][scale],
            padding: 0,
            collator: collator,
            loaderHeight: isEmpty ? null : $60684b441be8e84c$var$ROW_HEIGHTS[density][scale]
        })
    , [
        collator,
        scale,
        density,
        isEmpty,
        overflowMode
    ]);
    layout.collection = state.collection;
    layout.disabledKeys = state.disabledKeys;
    return layout;
}
function $60684b441be8e84c$var$ListView(props, ref) {
    var ref1;
    let { density: density = 'regular' , loadingState: loadingState , onLoadMore: onLoadMore , isQuiet: isQuiet , overflowMode: overflowMode = 'truncate' , onAction: onAction , dragHooks: dragHooks , dropHooks: dropHooks , ...otherProps } = props;
    let isListDraggable = !!dragHooks;
    let isListDroppable = !!dropHooks;
    let dragHooksProvided = $k9Du2$react.useRef(isListDraggable);
    let dropHooksProvided = $k9Du2$react.useRef(isListDroppable);
    if (dragHooksProvided.current !== isListDraggable) console.warn('Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
    if (dropHooksProvided.current !== isListDroppable) console.warn('Drop hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.');
    let domRef = $k9Du2$reactspectrumutils.useDOMRef(ref);
    let state = $k9Du2$reactstatelylist.useListState({
        ...props,
        selectionBehavior: props.selectionStyle === 'highlight' ? 'replace' : 'toggle'
    });
    let { collection: collection , selectionManager: selectionManager  } = state;
    let stringFormatter = $k9Du2$reactariai18n.useLocalizedStringFormatter((/*@__PURE__*/$parcel$interopDefault($fd0918c0ca138160$exports)));
    let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
    let { styleProps: styleProps  } = $k9Du2$reactspectrumutils.useStyleProps(props);
    let dragState;
    let preview = $k9Du2$react.useRef(null);
    if (isListDraggable) dragState = dragHooks.useDraggableCollectionState({
        collection: collection,
        selectionManager: selectionManager,
        preview: preview
    });
    let layout = $60684b441be8e84c$var$useListLayout(state, props.density || 'regular', overflowMode);
    // !!0 is false, so we can cast size or undefined and they'll be falsy
    layout.allowDisabledKeyFocus = state.selectionManager.disabledBehavior === 'selection' || !!(dragState === null || dragState === void 0 ? void 0 : dragState.draggingKeys.size);
    let DragPreview = dragHooks === null || dragHooks === void 0 ? void 0 : dragHooks.DragPreview;
    let dropState;
    let droppableCollection;
    let isRootDropTarget;
    if (isListDroppable) {
        dropState = dropHooks.useDroppableCollectionState({
            collection: collection,
            selectionManager: selectionManager
        });
        droppableCollection = dropHooks.useDroppableCollection({
            keyboardDelegate: layout,
            dropTargetDelegate: layout
        }, dropState, domRef);
        isRootDropTarget = dropState.isDropTarget({
            type: 'root'
        });
    }
    let { gridProps: gridProps  } = $k9Du2$reactariagridlist.useGridList({
        ...props,
        isVirtualized: true,
        keyboardDelegate: layout,
        onAction: onAction
    }, state, domRef);
    // Sync loading state into the layout.
    layout.isLoading = isLoading;
    let focusedKey = selectionManager.focusedKey;
    if ((dropState === null || dropState === void 0 ? void 0 : (ref1 = dropState.target) === null || ref1 === void 0 ? void 0 : ref1.type) === 'item') focusedKey = dropState.target.key;
    // wait for layout to get accurate measurements
    let [isVerticalScrollbarVisible, setVerticalScollbarVisible] = $k9Du2$react.useState(false);
    let [isHorizontalScrollbarVisible, setHorizontalScollbarVisible] = $k9Du2$react.useState(false);
    $k9Du2$reactariautils.useLayoutEffect(()=>{
        if (domRef.current) {
            // 2 is the width of the border which is not part of the box size
            setVerticalScollbarVisible(domRef.current.clientWidth + 2 < domRef.current.offsetWidth);
            setHorizontalScollbarVisible(domRef.current.clientHeight + 2 < domRef.current.offsetHeight);
        }
    });
    let hasAnyChildren = $k9Du2$react.useMemo(()=>[
            ...collection
        ].some((item)=>item.hasChildNodes
        )
    , [
        collection
    ]);
    return(/*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($60684b441be8e84c$export$870039b0abfe3de0.Provider, {
        value: {
            state: state,
            dragState: dragState,
            dropState: dropState,
            dragHooks: dragHooks,
            dropHooks: dropHooks,
            onAction: onAction,
            isListDraggable: isListDraggable,
            isListDroppable: isListDroppable,
            layout: layout,
            loadingState: loadingState
        }
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactariafocus.FocusRing, {
        focusRingClass: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactariavirtualizer.Virtualizer, {
        ...$k9Du2$reactariautils.mergeProps(isListDroppable && (droppableCollection === null || droppableCollection === void 0 ? void 0 : droppableCollection.collectionProps), gridProps),
        ...$k9Du2$reactariautils.filterDOMProps(otherProps),
        ...gridProps,
        ...styleProps,
        isLoading: isLoading,
        onLoadMore: onLoadMore,
        ref: domRef,
        focusedKey: focusedKey,
        scrollDirection: "vertical",
        className: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'react-spectrum-ListView', `react-spectrum-ListView--${density}`, 'react-spectrum-ListView--emphasized', {
            'react-spectrum-ListView--quiet': isQuiet,
            'react-spectrum-ListView--loadingMore': loadingState === 'loadingMore',
            'react-spectrum-ListView--draggable': !!isListDraggable,
            'react-spectrum-ListView--dropTarget': !!isRootDropTarget,
            'react-spectrum-ListView--isVerticalScrollbarVisible': isVerticalScrollbarVisible,
            'react-spectrum-ListView--isHorizontalScrollbarVisible': isHorizontalScrollbarVisible,
            'react-spectrum-ListView--hasAnyChildren': hasAnyChildren,
            'react-spectrum-ListView--wrap': overflowMode === 'wrap'
        }, styleProps.className),
        layout: layout,
        collection: collection,
        transitionDuration: isLoading ? 160 : 220
    }, (type, item)=>{
        if (type === 'item') return(/*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement(($parcel$interopDefault($k9Du2$react)).Fragment, null, isListDroppable && collection.getKeyBefore(item.key) == null && /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($c59d3898f402f50b$export$2e2bcd8739ae039, {
            key: "root"
        }), isListDroppable && /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($fc02fd49ab58c72e$export$2e2bcd8739ae039, {
            key: `${item.key}-before`,
            target: {
                key: item.key,
                type: 'item',
                dropPosition: 'before'
            }
        }), /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($9eae7a1cb1535a6b$export$c6bde0c04b033c0e, {
            item: item,
            isEmphasized: true,
            hasActions: !!onAction
        }), isListDroppable && /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($fc02fd49ab58c72e$export$2e2bcd8739ae039, {
            key: `${item.key}-after`,
            target: {
                key: item.key,
                type: 'item',
                dropPosition: 'after'
            },
            isPresentationOnly: collection.getKeyAfter(item.key) != null
        })));
        else if (type === 'loader') return(/*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($60684b441be8e84c$var$CenteredWrapper, null, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($k9Du2$reactspectrumprogress.ProgressCircle, {
            isIndeterminate: true,
            "aria-label": collection.size > 0 ? stringFormatter.format('loadingMore') : stringFormatter.format('loading')
        })));
        else if (type === 'placeholder') {
            let emptyState = props.renderEmptyState ? props.renderEmptyState() : null;
            if (emptyState == null) return null;
            return(/*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($60684b441be8e84c$var$CenteredWrapper, null, emptyState));
        }
    })), DragPreview && isListDraggable && /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement(DragPreview, {
        ref: preview
    }, ()=>{
        let item = state.collection.getItem(dragState.draggedKey);
        let itemCount = dragState.draggingKeys.size;
        let itemHeight = layout.getLayoutInfo(dragState.draggedKey).rect.height;
        return(/*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement($3d665b19f7865ff9$export$905ab40ac2179daa, {
            item: item,
            itemCount: itemCount,
            itemHeight: itemHeight
        }));
    })));
}
function $60684b441be8e84c$var$CenteredWrapper({ children: children  }) {
    let { state: state  } = $k9Du2$react.useContext($60684b441be8e84c$export$870039b0abfe3de0);
    return(/*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        role: "row",
        "aria-rowindex": state.collection.size + 1,
        className: $k9Du2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)), 'react-spectrum-ListView-centeredWrapper', {
            'react-spectrum-ListView-centeredWrapper--loadingMore': state.collection.size > 0
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).createElement("div", {
        role: "gridcell"
    }, children)));
}
/**
 * A ListView displays a list of interactive items, and allows a user to navigate, select, or perform an action.
 */ const $60684b441be8e84c$export$84d0dd190d551cd1 = /*#__PURE__*/ ($parcel$interopDefault($k9Du2$react)).forwardRef($60684b441be8e84c$var$ListView);
