// @react-stately/select/dist/main.js
var $aJE2N$reactstatelymenu = require("@react-stately/menu");
var $aJE2N$reactstatelylist = require("@react-stately/list");
var $aJE2N$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $80ebb60e77198879$exports = {};

$parcel$export($80ebb60e77198879$exports, "useSelectState", () => $80ebb60e77198879$export$5159ec8b34d4ec12);



function $80ebb60e77198879$export$5159ec8b34d4ec12(props) {
    let triggerState = $aJE2N$reactstatelymenu.useMenuTriggerState(props);
    let listState = $aJE2N$reactstatelylist.useSingleSelectListState({
        ...props,
        onSelectionChange: (key)=>{
            if (props.onSelectionChange != null) props.onSelectionChange(key);
            triggerState.close();
        }
    });
    let [isFocused, setFocused] = $aJE2N$react.useState(false);
    return {
        ...listState,
        ...triggerState,
        open () {
            // Don't open if the collection is empty.
            if (listState.collection.size !== 0) triggerState.open();
        },
        toggle (focusStrategy) {
            if (listState.collection.size !== 0) triggerState.toggle(focusStrategy);
        },
        isFocused: isFocused,
        setFocused: setFocused
    };
}


$parcel$exportWildcard(module.exports, $80ebb60e77198879$exports);
