// @react-spectrum/image/dist/main.js
require("./main.css");
var $je8mI$reactspectrumutils = require("@react-spectrum/utils");
var $je8mI$reactariautils = require("@react-aria/utils");
var $je8mI$react = require("react");
var $je8mI$reactspectrumprovider = require("@react-spectrum/provider");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $bda2e7b47de3a920$exports = {};

$parcel$export($bda2e7b47de3a920$exports, "Image", () => $bda2e7b47de3a920$export$3e431a229df88919);



var $ec6587ec6732a958$exports = {};

$parcel$export($ec6587ec6732a958$exports, "spectrum-Image-img", () => $ec6587ec6732a958$export$3238177972e11200, (v) => $ec6587ec6732a958$export$3238177972e11200 = v);
var $ec6587ec6732a958$export$3238177972e11200;
$ec6587ec6732a958$export$3238177972e11200 = "spectrum-Image-img_fdc794";



// incomplete component for show right now
function $bda2e7b47de3a920$var$Image(props, ref) {
    /* Slots should be able to pass an alt for default behavior, but in Images, the child may know better. */ let userProvidedAlt = props.alt;
    props = $je8mI$reactspectrumutils.useSlotProps(props, 'image');
    props = $je8mI$reactspectrumprovider.useProviderProps(props);
    let { objectFit: objectFit , src: src , alt: alt , ...otherProps } = props;
    let { styleProps: styleProps  } = $je8mI$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $je8mI$reactspectrumutils.useDOMRef(ref);
    if (alt == null) console.warn("The `alt` prop was not provided to an image. Add `alt` text for screen readers, or set `alt=\"\"` prop to indicate that the image is decorative or redundant with displayed text and should not be announced by screen readers.");
    return(/*#__PURE__*/ ($parcel$interopDefault($je8mI$react)).createElement("div", {
        ...$je8mI$reactariautils.filterDOMProps(props),
        ...styleProps,
        className: $je8mI$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ec6587ec6732a958$exports)), styleProps.className),
        style: {
            ...styleProps.style,
            overflow: 'hidden'
        },
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($je8mI$react)).createElement("img", {
        src: src,
        alt: userProvidedAlt || alt,
        style: {
            objectFit: objectFit
        },
        className: $je8mI$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($ec6587ec6732a958$exports)), 'spectrum-Image-img')
    })));
}
/**
 * Image is used to insert and display an image within a component.
 */ const $bda2e7b47de3a920$export$3e431a229df88919 = /*#__PURE__*/ ($parcel$interopDefault($je8mI$react)).forwardRef($bda2e7b47de3a920$var$Image);


$parcel$exportWildcard(module.exports, $bda2e7b47de3a920$exports);
