// @react-spectrum/contextualhelp/dist/module.js
import "./main.css";
import {ActionButton as $2FtBM$ActionButton} from "@react-spectrum/button";
import {DialogTrigger as $2FtBM$DialogTrigger, Dialog as $2FtBM$Dialog} from "@react-spectrum/dialog";
import $2FtBM$spectrumiconsworkflowHelpOutline from "@spectrum-icons/workflow/HelpOutline";
import $2FtBM$spectrumiconsworkflowInfoOutline from "@spectrum-icons/workflow/InfoOutline";
import $2FtBM$react from "react";
import {SlotProvider as $2FtBM$SlotProvider} from "@react-spectrum/utils";
import {useMessageFormatter as $2FtBM$useMessageFormatter} from "@react-aria/i18n";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $9d9069cfc9959198$exports = {};

$parcel$export($9d9069cfc9959198$exports, "ContextualHelp", () => $9d9069cfc9959198$export$7d3cdb256c2ba320);



var $d69bde6495703df1$exports = {};

$parcel$export($d69bde6495703df1$exports, "react-spectrum-ContextualHelp-button", () => $d69bde6495703df1$export$20d9583c9f1529c1, (v) => $d69bde6495703df1$export$20d9583c9f1529c1 = v);
$parcel$export($d69bde6495703df1$exports, "react-spectrum-ContextualHelp-dialog", () => $d69bde6495703df1$export$ab2d873674d20d28, (v) => $d69bde6495703df1$export$ab2d873674d20d28 = v);
$parcel$export($d69bde6495703df1$exports, "react-spectrum-ContextualHelp-content", () => $d69bde6495703df1$export$a1f805890cf58c4, (v) => $d69bde6495703df1$export$a1f805890cf58c4 = v);
$parcel$export($d69bde6495703df1$exports, "react-spectrum-ContextualHelp-footer", () => $d69bde6495703df1$export$d901bc3fa82b87a6, (v) => $d69bde6495703df1$export$d901bc3fa82b87a6 = v);
var $d69bde6495703df1$export$20d9583c9f1529c1;
var $d69bde6495703df1$export$ab2d873674d20d28;
var $d69bde6495703df1$export$a1f805890cf58c4;
var $d69bde6495703df1$export$d901bc3fa82b87a6;
$d69bde6495703df1$export$20d9583c9f1529c1 = "react-spectrum-ContextualHelp-button_d88303";
$d69bde6495703df1$export$ab2d873674d20d28 = "react-spectrum-ContextualHelp-dialog_d88303";
$d69bde6495703df1$export$a1f805890cf58c4 = "react-spectrum-ContextualHelp-content_d88303";
$d69bde6495703df1$export$d901bc3fa82b87a6 = "react-spectrum-ContextualHelp-footer_d88303";



var $7c93b3220be85068$exports = {};
var $9cc8b85ee8327c05$exports = {};
$9cc8b85ee8327c05$exports = JSON.parse("{\"help\":\"مساعدة\",\"info\":\"معلومات\"}");


var $56f7d9be41e5dc75$exports = {};
$56f7d9be41e5dc75$exports = JSON.parse("{\"help\":\"Помощ\",\"info\":\"Информация\"}");


var $c427e4a9ab6700b2$exports = {};
$c427e4a9ab6700b2$exports = JSON.parse("{\"help\":\"Nápověda\",\"info\":\"Informace\"}");


var $ee2212e66656667f$exports = {};
$ee2212e66656667f$exports = JSON.parse("{\"help\":\"Hjælp\",\"info\":\"Oplysninger\"}");


var $112e467ad7137275$exports = {};
$112e467ad7137275$exports = JSON.parse("{\"help\":\"Hilfe\",\"info\":\"Informationen\"}");


var $7ad0d6a9c54bdc43$exports = {};
$7ad0d6a9c54bdc43$exports = JSON.parse("{\"help\":\"Βοήθεια\",\"info\":\"Πληροφορίες\"}");


var $704e90e35332a9cb$exports = {};
$704e90e35332a9cb$exports = JSON.parse("{\"info\":\"Information\",\"help\":\"Help\"}");


var $f2e0410dbb606550$exports = {};
$f2e0410dbb606550$exports = JSON.parse("{\"help\":\"Ayuda\",\"info\":\"Información\"}");


var $f4852a97714b8951$exports = {};
$f4852a97714b8951$exports = JSON.parse("{\"help\":\"Spikker\",\"info\":\"Teave\"}");


var $fb4baf837c8a96f6$exports = {};
$fb4baf837c8a96f6$exports = JSON.parse("{\"help\":\"Ohje\",\"info\":\"Tiedot\"}");


var $28b0fbf00803a68f$exports = {};
$28b0fbf00803a68f$exports = JSON.parse("{\"help\":\"Aide\",\"info\":\"Informations\"}");


var $77110a8f3000e559$exports = {};
$77110a8f3000e559$exports = JSON.parse("{\"help\":\"עזרה\",\"info\":\"מידע\"}");


var $dcc3492993669fc3$exports = {};
$dcc3492993669fc3$exports = JSON.parse("{\"help\":\"Pomoć\",\"info\":\"Informacije\"}");


var $d376dd5b16d08137$exports = {};
$d376dd5b16d08137$exports = JSON.parse("{\"help\":\"Súgó\",\"info\":\"Információ\"}");


var $45f88f0321bb5a8f$exports = {};
$45f88f0321bb5a8f$exports = JSON.parse("{\"help\":\"Aiuto\",\"info\":\"Informazioni\"}");


var $8f643aac2e9e9e26$exports = {};
$8f643aac2e9e9e26$exports = JSON.parse("{\"help\":\"ヘルプ\",\"info\":\"情報\"}");


var $784ec4e789b99a71$exports = {};
$784ec4e789b99a71$exports = JSON.parse("{\"help\":\"도움말\",\"info\":\"정보\"}");


var $fcda0e0582088650$exports = {};
$fcda0e0582088650$exports = JSON.parse("{\"help\":\"Žinynas\",\"info\":\"Informacija\"}");


var $9c3b35ecd7c5e348$exports = {};
$9c3b35ecd7c5e348$exports = JSON.parse("{\"help\":\"Palīdzība\",\"info\":\"Informācija\"}");


var $b6518647ee722bc3$exports = {};
$b6518647ee722bc3$exports = JSON.parse("{\"help\":\"Hjelp\",\"info\":\"Informasjon\"}");


var $5cb1939fecdd5f1a$exports = {};
$5cb1939fecdd5f1a$exports = JSON.parse("{\"help\":\"Help\",\"info\":\"Informatie\"}");


var $3ed876b500650109$exports = {};
$3ed876b500650109$exports = JSON.parse("{\"help\":\"Pomoc\",\"info\":\"Informacja\"}");


var $999b0348d4f9e0ae$exports = {};
$999b0348d4f9e0ae$exports = JSON.parse("{\"help\":\"Ajuda\",\"info\":\"Informações\"}");


var $01e38e0d24e067b8$exports = {};
$01e38e0d24e067b8$exports = JSON.parse("{\"help\":\"Ajuda\",\"info\":\"Informação\"}");


var $130ce4093fe216d5$exports = {};
$130ce4093fe216d5$exports = JSON.parse("{\"help\":\"Ajutor\",\"info\":\"Informaţii\"}");


var $3b25696a80a9bfd8$exports = {};
$3b25696a80a9bfd8$exports = JSON.parse("{\"help\":\"Справка\",\"info\":\"Информация\"}");


var $70037a741b152768$exports = {};
$70037a741b152768$exports = JSON.parse("{\"help\":\"Pomoc\",\"info\":\"Informácie\"}");


var $4ce06844b60c40b5$exports = {};
$4ce06844b60c40b5$exports = JSON.parse("{\"help\":\"Pomoč\",\"info\":\"Informacije\"}");


var $78c0c0c4796561ef$exports = {};
$78c0c0c4796561ef$exports = JSON.parse("{\"help\":\"Pomoć\",\"info\":\"Informacije\"}");


var $a701758f24293d53$exports = {};
$a701758f24293d53$exports = JSON.parse("{\"help\":\"Hjälp\",\"info\":\"Information\"}");


var $ac0041dae2f30090$exports = {};
$ac0041dae2f30090$exports = JSON.parse("{\"help\":\"Yardım\",\"info\":\"Bilgiler\"}");


var $e3e68b40fcb16018$exports = {};
$e3e68b40fcb16018$exports = JSON.parse("{\"help\":\"Довідка\",\"info\":\"Інформація\"}");


var $befba97b83d35197$exports = {};
$befba97b83d35197$exports = JSON.parse("{\"help\":\"帮助\",\"info\":\"信息\"}");


var $6051e1035ff73e7b$exports = {};
$6051e1035ff73e7b$exports = JSON.parse("{\"help\":\"說明\",\"info\":\"資訊\"}");


$7c93b3220be85068$exports = {
    "ar-AE": $9cc8b85ee8327c05$exports,
    "bg-BG": $56f7d9be41e5dc75$exports,
    "cs-CZ": $c427e4a9ab6700b2$exports,
    "da-DK": $ee2212e66656667f$exports,
    "de-DE": $112e467ad7137275$exports,
    "el-GR": $7ad0d6a9c54bdc43$exports,
    "en-US": $704e90e35332a9cb$exports,
    "es-ES": $f2e0410dbb606550$exports,
    "et-EE": $f4852a97714b8951$exports,
    "fi-FI": $fb4baf837c8a96f6$exports,
    "fr-FR": $28b0fbf00803a68f$exports,
    "he-IL": $77110a8f3000e559$exports,
    "hr-HR": $dcc3492993669fc3$exports,
    "hu-HU": $d376dd5b16d08137$exports,
    "it-IT": $45f88f0321bb5a8f$exports,
    "ja-JP": $8f643aac2e9e9e26$exports,
    "ko-KR": $784ec4e789b99a71$exports,
    "lt-LT": $fcda0e0582088650$exports,
    "lv-LV": $9c3b35ecd7c5e348$exports,
    "nb-NO": $b6518647ee722bc3$exports,
    "nl-NL": $5cb1939fecdd5f1a$exports,
    "pl-PL": $3ed876b500650109$exports,
    "pt-BR": $999b0348d4f9e0ae$exports,
    "pt-PT": $01e38e0d24e067b8$exports,
    "ro-RO": $130ce4093fe216d5$exports,
    "ru-RU": $3b25696a80a9bfd8$exports,
    "sk-SK": $70037a741b152768$exports,
    "sl-SI": $4ce06844b60c40b5$exports,
    "sr-SP": $78c0c0c4796561ef$exports,
    "sv-SE": $a701758f24293d53$exports,
    "tr-TR": $ac0041dae2f30090$exports,
    "uk-UA": $e3e68b40fcb16018$exports,
    "zh-CN": $befba97b83d35197$exports,
    "zh-TW": $6051e1035ff73e7b$exports
};





function $9d9069cfc9959198$var$ContextualHelp(props, ref) {
    let { variant: variant = 'help' , placement: placement = 'bottom start' , children: children , ...otherProps } = props;
    let formatMessage = $2FtBM$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($7c93b3220be85068$exports)));
    let icon = variant === 'info' ? /*#__PURE__*/ $2FtBM$react.createElement($2FtBM$spectrumiconsworkflowInfoOutline, null) : /*#__PURE__*/ $2FtBM$react.createElement($2FtBM$spectrumiconsworkflowHelpOutline, null);
    let slots = {
        content: {
            UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d69bde6495703df1$exports))['react-spectrum-ContextualHelp-content']
        },
        footer: {
            UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d69bde6495703df1$exports))['react-spectrum-ContextualHelp-footer']
        }
    };
    let ariaLabel = otherProps['aria-label'];
    if (!ariaLabel && !otherProps['aria-labelledby']) ariaLabel = formatMessage(variant);
    return(/*#__PURE__*/ $2FtBM$react.createElement($2FtBM$DialogTrigger, {
        ...otherProps,
        type: "popover",
        placement: placement,
        hideArrow: true
    }, /*#__PURE__*/ $2FtBM$react.createElement($2FtBM$ActionButton, {
        ...otherProps,
        ref: ref,
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d69bde6495703df1$exports))['react-spectrum-ContextualHelp-button'],
        isQuiet: true,
        "aria-label": ariaLabel
    }, icon), /*#__PURE__*/ $2FtBM$react.createElement($2FtBM$SlotProvider, {
        slots: slots
    }, /*#__PURE__*/ $2FtBM$react.createElement($2FtBM$Dialog, {
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($d69bde6495703df1$exports))['react-spectrum-ContextualHelp-dialog']
    }, children))));
}
/**
 * Contextual help shows a user extra information about the state of an adjacent component, or a total view.
 */ let $9d9069cfc9959198$export$7d3cdb256c2ba320 = /*#__PURE__*/ $2FtBM$react.forwardRef($9d9069cfc9959198$var$ContextualHelp);




export {$9d9069cfc9959198$export$7d3cdb256c2ba320 as ContextualHelp};
