// @react-spectrum/combobox/dist/module.js
import "./main.css";
import {Item as $6ea56810956c2335$re_export$Item, Section as $6ea56810956c2335$re_export$Section} from "@react-stately/collections";
import $725Me$spectrumiconsuiChevronDownMedium from "@spectrum-icons/ui/ChevronDownMedium";
import {useIsMobileDevice as $725Me$useIsMobileDevice, useUnwrapDOMRef as $725Me$useUnwrapDOMRef, useFocusableRef as $725Me$useFocusableRef, useResizeObserver as $725Me$useResizeObserver, classNames as $725Me$classNames, unwrapDOMRef as $725Me$unwrapDOMRef} from "@react-spectrum/utils";
import {useOverlayPosition as $725Me$useOverlayPosition, DismissButton as $725Me$DismissButton, useOverlayTrigger as $725Me$useOverlayTrigger} from "@react-aria/overlays";
import {Field as $725Me$Field} from "@react-spectrum/label";
import {FieldButton as $725Me$FieldButton, ClearButton as $725Me$ClearButton} from "@react-spectrum/button";
import {FocusRing as $725Me$FocusRing, focusSafely as $725Me$focusSafely, FocusScope as $725Me$FocusScope} from "@react-aria/focus";
import {useListBoxLayout as $725Me$useListBoxLayout, ListBoxBase as $725Me$ListBoxBase} from "@react-spectrum/listbox";
import {Popover as $725Me$Popover, Tray as $725Me$Tray} from "@react-spectrum/overlays";
import {useHover as $725Me$useHover, PressResponder as $725Me$PressResponder, setInteractionModality as $725Me$setInteractionModality} from "@react-aria/interactions";
import {ProgressCircle as $725Me$ProgressCircle} from "@react-spectrum/progress";
import $725Me$react, {useRef as $725Me$useRef, useState as $725Me$useState, useCallback as $725Me$useCallback, useEffect as $725Me$useEffect} from "react";
import {TextFieldBase as $725Me$TextFieldBase} from "@react-spectrum/textfield";
import {useComboBox as $725Me$useComboBox} from "@react-aria/combobox";
import {useComboBoxState as $725Me$useComboBoxState} from "@react-stately/combobox";
import {useMessageFormatter as $725Me$useMessageFormatter, useFilter as $725Me$useFilter} from "@react-aria/i18n";
import {useLayoutEffect as $725Me$useLayoutEffect, mergeProps as $725Me$mergeProps, useId as $725Me$useId} from "@react-aria/utils";
import {useProviderProps as $725Me$useProviderProps, useProvider as $725Me$useProvider} from "@react-spectrum/provider";
import $725Me$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import $725Me$spectrumiconsuiCheckmarkMedium from "@spectrum-icons/ui/CheckmarkMedium";
import {useButton as $725Me$useButton} from "@react-aria/button";
import {useDialog as $725Me$useDialog} from "@react-aria/dialog";
import {useField as $725Me$useField} from "@react-aria/label";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $2289f375e813f187$exports = {};

$parcel$export($2289f375e813f187$exports, "ComboBox", () => $2289f375e813f187$export$72b9695b8216309a);


var $7a1753adda45f7fb$exports = {};

$parcel$export($7a1753adda45f7fb$exports, "no-results", () => $7a1753adda45f7fb$export$5b1939da91820a47, (v) => $7a1753adda45f7fb$export$5b1939da91820a47 = v);
$parcel$export($7a1753adda45f7fb$exports, "mobile-combobox", () => $7a1753adda45f7fb$export$31f2f19e6c50453d, (v) => $7a1753adda45f7fb$export$31f2f19e6c50453d = v);
$parcel$export($7a1753adda45f7fb$exports, "mobile-input", () => $7a1753adda45f7fb$export$bededa1c0ac414c5, (v) => $7a1753adda45f7fb$export$bededa1c0ac414c5 = v);
$parcel$export($7a1753adda45f7fb$exports, "mobile-value", () => $7a1753adda45f7fb$export$e32ca2bb9f795355, (v) => $7a1753adda45f7fb$export$e32ca2bb9f795355 = v);
$parcel$export($7a1753adda45f7fb$exports, "tray-dialog", () => $7a1753adda45f7fb$export$3d4198eea215577a, (v) => $7a1753adda45f7fb$export$3d4198eea215577a = v);
$parcel$export($7a1753adda45f7fb$exports, "tray-textfield", () => $7a1753adda45f7fb$export$73cc97e04e76a6d0, (v) => $7a1753adda45f7fb$export$73cc97e04e76a6d0 = v);
$parcel$export($7a1753adda45f7fb$exports, "has-label", () => $7a1753adda45f7fb$export$b8e014ff9ad5e5e2, (v) => $7a1753adda45f7fb$export$b8e014ff9ad5e5e2 = v);
$parcel$export($7a1753adda45f7fb$exports, "tray-textfield-input", () => $7a1753adda45f7fb$export$feb70b7bc717e2b8, (v) => $7a1753adda45f7fb$export$feb70b7bc717e2b8 = v);
$parcel$export($7a1753adda45f7fb$exports, "tray-listbox", () => $7a1753adda45f7fb$export$7281dc54d38d4bed, (v) => $7a1753adda45f7fb$export$7281dc54d38d4bed = v);
var $7a1753adda45f7fb$export$5b1939da91820a47;
var $7a1753adda45f7fb$export$31f2f19e6c50453d;
var $7a1753adda45f7fb$export$bededa1c0ac414c5;
var $7a1753adda45f7fb$export$e32ca2bb9f795355;
var $7a1753adda45f7fb$export$3d4198eea215577a;
var $7a1753adda45f7fb$export$73cc97e04e76a6d0;
var $7a1753adda45f7fb$export$b8e014ff9ad5e5e2;
var $7a1753adda45f7fb$export$feb70b7bc717e2b8;
var $7a1753adda45f7fb$export$7281dc54d38d4bed;
$7a1753adda45f7fb$export$5b1939da91820a47 = "no-results_743018";
$7a1753adda45f7fb$export$31f2f19e6c50453d = "mobile-combobox_743018";
$7a1753adda45f7fb$export$bededa1c0ac414c5 = "mobile-input_743018";
$7a1753adda45f7fb$export$e32ca2bb9f795355 = "mobile-value_743018";
$7a1753adda45f7fb$export$3d4198eea215577a = "tray-dialog_743018";
$7a1753adda45f7fb$export$73cc97e04e76a6d0 = "tray-textfield_743018";
$7a1753adda45f7fb$export$b8e014ff9ad5e5e2 = "has-label_743018";
$7a1753adda45f7fb$export$feb70b7bc717e2b8 = "tray-textfield-input_743018";
$7a1753adda45f7fb$export$7281dc54d38d4bed = "tray-listbox_743018";






var $9c69665f9f0b71fd$exports = {};
var $37184f3a1eff2016$exports = {};
$37184f3a1eff2016$exports = JSON.parse("{\"clear\":\"مسح\",\"invalid\":\"(غير صالح)\",\"loading\":\"جارٍ التحميل...\",\"noResults\":\"لا توجد نتائج\"}");


var $292e71851d5fef25$exports = {};
$292e71851d5fef25$exports = JSON.parse("{\"clear\":\"Изчисти\",\"invalid\":\"(невалиден)\",\"loading\":\"Зареждане...\",\"noResults\":\"Няма резултати\"}");


var $e03903e9a2e8a8dc$exports = {};
$e03903e9a2e8a8dc$exports = JSON.parse("{\"clear\":\"Vymazat\",\"invalid\":\"(neplatné)\",\"loading\":\"Načítání...\",\"noResults\":\"Žádné výsledky\"}");


var $e254601d3b112ae8$exports = {};
$e254601d3b112ae8$exports = JSON.parse("{\"clear\":\"Ryd\",\"invalid\":\"(ugyldig)\",\"loading\":\"Indlæser ...\",\"noResults\":\"Ingen resultater\"}");


var $760f46c8bf4d0bd2$exports = {};
$760f46c8bf4d0bd2$exports = JSON.parse("{\"clear\":\"Löschen\",\"invalid\":\"(ungültig)\",\"loading\":\"Wird geladen...\",\"noResults\":\"Keine Ergebnisse\"}");


var $1ea466df78549cf5$exports = {};
$1ea466df78549cf5$exports = JSON.parse("{\"clear\":\"Καθαρισμός\",\"invalid\":\"(δεν ισχύει)\",\"loading\":\"Φόρτωση...\",\"noResults\":\"Χωρίς αποτέλεσμα\"}");


var $2ae24b959273b5c1$exports = {};
$2ae24b959273b5c1$exports = JSON.parse("{\"loading\":\"Loading...\",\"noResults\":\"No results\",\"clear\":\"Clear\",\"invalid\":\"(invalid)\"}");


var $1e422c6caf1d6300$exports = {};
$1e422c6caf1d6300$exports = JSON.parse("{\"clear\":\"Borrar\",\"invalid\":\"(no válido)\",\"loading\":\"Cargando...\",\"noResults\":\"Sin resultados\"}");


var $ec276c10668cb27d$exports = {};
$ec276c10668cb27d$exports = JSON.parse("{\"clear\":\"Puhasta\",\"invalid\":\"(kehtetu)\",\"loading\":\"Laadimine...\",\"noResults\":\"Tulemusi pole\"}");


var $3ee762802ba9edfd$exports = {};
$3ee762802ba9edfd$exports = JSON.parse("{\"clear\":\"Kirkas\",\"invalid\":\"(epäkelpo)\",\"loading\":\"Ladataan...\",\"noResults\":\"Ei tuloksia\"}");


var $4a4c220cbaf33119$exports = {};
$4a4c220cbaf33119$exports = JSON.parse("{\"clear\":\"Effacer\",\"invalid\":\"(non valide)\",\"loading\":\"Chargement en cours...\",\"noResults\":\"Aucun résultat\"}");


var $5768444fcb73d3f4$exports = {};
$5768444fcb73d3f4$exports = JSON.parse("{\"clear\":\"נקי\",\"invalid\":\"(לא חוקי)\",\"loading\":\"טוען...\",\"noResults\":\"אין תוצאות\"}");


var $0c19c7a2667b34f6$exports = {};
$0c19c7a2667b34f6$exports = JSON.parse("{\"clear\":\"Izbriši\",\"invalid\":\"(nevažeće)\",\"loading\":\"Učitavam...\",\"noResults\":\"Nema rezultata\"}");


var $c688fc1a398cdc12$exports = {};
$c688fc1a398cdc12$exports = JSON.parse("{\"clear\":\"Törlés\",\"invalid\":\"(érvénytelen)\",\"loading\":\"Betöltés folyamatban…\",\"noResults\":\"Nincsenek találatok\"}");


var $beee3b800889f449$exports = {};
$beee3b800889f449$exports = JSON.parse("{\"clear\":\"Cancella\",\"invalid\":\"(non valido)\",\"loading\":\"Caricamento in corso...\",\"noResults\":\"Nessun risultato\"}");


var $82c9b25b834471a2$exports = {};
$82c9b25b834471a2$exports = JSON.parse("{\"clear\":\"クリア\",\"invalid\":\"(無効)\",\"loading\":\"読み込み中...\",\"noResults\":\"結果なし\"}");


var $be9cae133bb9c811$exports = {};
$be9cae133bb9c811$exports = JSON.parse("{\"clear\":\"지우기\",\"invalid\":\"(유효하지 않음)\",\"loading\":\"로드 중...\",\"noResults\":\"결과 없음\"}");


var $5ef0bd350b9637e3$exports = {};
$5ef0bd350b9637e3$exports = JSON.parse("{\"clear\":\"Skaidrus\",\"invalid\":\"(netinkama)\",\"loading\":\"Įkeliama...\",\"noResults\":\"Be rezultatų\"}");


var $e897441eed0b833f$exports = {};
$e897441eed0b833f$exports = JSON.parse("{\"clear\":\"Notīrīt\",\"invalid\":\"(nederīgs)\",\"loading\":\"Notiek ielāde...\",\"noResults\":\"Nav rezultātu\"}");


var $a4b49a4502f68578$exports = {};
$a4b49a4502f68578$exports = JSON.parse("{\"clear\":\"Tøm\",\"invalid\":\"(ugyldig)\",\"loading\":\"Laster inn ...\",\"noResults\":\"Ingen resultater\"}");


var $f07280ca05544eb8$exports = {};
$f07280ca05544eb8$exports = JSON.parse("{\"clear\":\"Helder\",\"invalid\":\"(ongeldig)\",\"loading\":\"Laden...\",\"noResults\":\"Geen resultaten\"}");


var $a8ee33871b41ed34$exports = {};
$a8ee33871b41ed34$exports = JSON.parse("{\"clear\":\"Wyczyść\",\"invalid\":\"(nieprawidłowy)\",\"loading\":\"Trwa ładowanie...\",\"noResults\":\"Brak wyników\"}");


var $48778e86f0af7d7f$exports = {};
$48778e86f0af7d7f$exports = JSON.parse("{\"clear\":\"Limpar\",\"invalid\":\"(inválido)\",\"loading\":\"Carregando...\",\"noResults\":\"Nenhum resultado\"}");


var $95c76e6a3ff4645c$exports = {};
$95c76e6a3ff4645c$exports = JSON.parse("{\"clear\":\"Limpar\",\"invalid\":\"(inválido)\",\"loading\":\"A carregar...\",\"noResults\":\"Sem resultados\"}");


var $ea41d6dadb4aad60$exports = {};
$ea41d6dadb4aad60$exports = JSON.parse("{\"clear\":\"Golire\",\"invalid\":\"(nevalid)\",\"loading\":\"Se încarcă...\",\"noResults\":\"Niciun rezultat\"}");


var $ad31a70e68aa4a8e$exports = {};
$ad31a70e68aa4a8e$exports = JSON.parse("{\"clear\":\"Очистить\",\"invalid\":\"(недействительно)\",\"loading\":\"Загрузка...\",\"noResults\":\"Результаты отсутствуют\"}");


var $9700a8010c106d6b$exports = {};
$9700a8010c106d6b$exports = JSON.parse("{\"clear\":\"Vymazať\",\"invalid\":\"(neplatné)\",\"loading\":\"Načítava sa...\",\"noResults\":\"Žiadne výsledky\"}");


var $be219b4f0a2c4a58$exports = {};
$be219b4f0a2c4a58$exports = JSON.parse("{\"clear\":\"Jasen\",\"invalid\":\"(neveljavno)\",\"loading\":\"Nalaganje...\",\"noResults\":\"Ni rezultatov\"}");


var $5d0ff97fe4cea552$exports = {};
$5d0ff97fe4cea552$exports = JSON.parse("{\"clear\":\"Izbriši\",\"invalid\":\"(nevažeće)\",\"loading\":\"Učitavam...\",\"noResults\":\"Nema rezultata\"}");


var $d3b9b386d3cfbc16$exports = {};
$d3b9b386d3cfbc16$exports = JSON.parse("{\"clear\":\"Rensa\",\"invalid\":\"(ogiltigt)\",\"loading\":\"Läser in...\",\"noResults\":\"Inga resultat\"}");


var $50a61df6fac70d29$exports = {};
$50a61df6fac70d29$exports = JSON.parse("{\"clear\":\"Temizle\",\"invalid\":\"(geçersiz)\",\"loading\":\"Yükleniyor...\",\"noResults\":\"Sonuç yok\"}");


var $ef974ff21ee4e13f$exports = {};
$ef974ff21ee4e13f$exports = JSON.parse("{\"clear\":\"Очистити\",\"invalid\":\"(недійсне)\",\"loading\":\"Завантаження...\",\"noResults\":\"Результатів немає\"}");


var $d38897b975d5f445$exports = {};
$d38897b975d5f445$exports = JSON.parse("{\"clear\":\"透明\",\"invalid\":\"（无效）\",\"loading\":\"正在加载...\",\"noResults\":\"无结果\"}");


var $ffc147749052dc62$exports = {};
$ffc147749052dc62$exports = JSON.parse("{\"clear\":\"清除\",\"invalid\":\"(無效)\",\"loading\":\"正在載入...\",\"noResults\":\"無任何結果\"}");


$9c69665f9f0b71fd$exports = {
    "ar-AE": $37184f3a1eff2016$exports,
    "bg-BG": $292e71851d5fef25$exports,
    "cs-CZ": $e03903e9a2e8a8dc$exports,
    "da-DK": $e254601d3b112ae8$exports,
    "de-DE": $760f46c8bf4d0bd2$exports,
    "el-GR": $1ea466df78549cf5$exports,
    "en-US": $2ae24b959273b5c1$exports,
    "es-ES": $1e422c6caf1d6300$exports,
    "et-EE": $ec276c10668cb27d$exports,
    "fi-FI": $3ee762802ba9edfd$exports,
    "fr-FR": $4a4c220cbaf33119$exports,
    "he-IL": $5768444fcb73d3f4$exports,
    "hr-HR": $0c19c7a2667b34f6$exports,
    "hu-HU": $c688fc1a398cdc12$exports,
    "it-IT": $beee3b800889f449$exports,
    "ja-JP": $82c9b25b834471a2$exports,
    "ko-KR": $be9cae133bb9c811$exports,
    "lt-LT": $5ef0bd350b9637e3$exports,
    "lv-LV": $e897441eed0b833f$exports,
    "nb-NO": $a4b49a4502f68578$exports,
    "nl-NL": $f07280ca05544eb8$exports,
    "pl-PL": $a8ee33871b41ed34$exports,
    "pt-BR": $48778e86f0af7d7f$exports,
    "pt-PT": $95c76e6a3ff4645c$exports,
    "ro-RO": $ea41d6dadb4aad60$exports,
    "ru-RU": $ad31a70e68aa4a8e$exports,
    "sk-SK": $9700a8010c106d6b$exports,
    "sl-SI": $be219b4f0a2c4a58$exports,
    "sr-SP": $5d0ff97fe4cea552$exports,
    "sv-SE": $d3b9b386d3cfbc16$exports,
    "tr-TR": $50a61df6fac70d29$exports,
    "uk-UA": $ef974ff21ee4e13f$exports,
    "zh-CN": $d38897b975d5f445$exports,
    "zh-TW": $ffc147749052dc62$exports
};




var $086e93d2352a4536$exports = {};

$parcel$export($086e93d2352a4536$exports, "spectrum-Button", () => $086e93d2352a4536$export$1db4cca5b4ade39a, (v) => $086e93d2352a4536$export$1db4cca5b4ade39a = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ActionButton", () => $086e93d2352a4536$export$53da69f51b770d3, (v) => $086e93d2352a4536$export$53da69f51b770d3 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-LogicButton", () => $086e93d2352a4536$export$36ae67a131730482, (v) => $086e93d2352a4536$export$36ae67a131730482 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-FieldButton", () => $086e93d2352a4536$export$a024c99ecf6b5741, (v) => $086e93d2352a4536$export$a024c99ecf6b5741 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ClearButton", () => $086e93d2352a4536$export$b752ce409e5660c, (v) => $086e93d2352a4536$export$b752ce409e5660c = v);
$parcel$export($086e93d2352a4536$exports, "focus-ring", () => $086e93d2352a4536$export$f39a09f249340e2a, (v) => $086e93d2352a4536$export$f39a09f249340e2a = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-Icon", () => $086e93d2352a4536$export$d374b04f30360026, (v) => $086e93d2352a4536$export$d374b04f30360026 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ActionButton--emphasized", () => $086e93d2352a4536$export$5aabb61a966e266f, (v) => $086e93d2352a4536$export$5aabb61a966e266f = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ActionButton--staticColor", () => $086e93d2352a4536$export$29b7b5d6a5a497c5, (v) => $086e93d2352a4536$export$29b7b5d6a5a497c5 = v);
$parcel$export($086e93d2352a4536$exports, "is-hovered", () => $086e93d2352a4536$export$b8813cd5d7824ce7, (v) => $086e93d2352a4536$export$b8813cd5d7824ce7 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-Button-label", () => $086e93d2352a4536$export$c081ba562e204942, (v) => $086e93d2352a4536$export$c081ba562e204942 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ActionButton-label", () => $086e93d2352a4536$export$ebb4635c66bacdb9, (v) => $086e93d2352a4536$export$ebb4635c66bacdb9 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ActionButton-hold", () => $086e93d2352a4536$export$4a4f01c443eff992, (v) => $086e93d2352a4536$export$4a4f01c443eff992 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ActionButton--quiet", () => $086e93d2352a4536$export$40e1f009544bacf3, (v) => $086e93d2352a4536$export$40e1f009544bacf3 = v);
$parcel$export($086e93d2352a4536$exports, "is-disabled", () => $086e93d2352a4536$export$d35bc1e505d1ebbf, (v) => $086e93d2352a4536$export$d35bc1e505d1ebbf = v);
$parcel$export($086e93d2352a4536$exports, "is-open", () => $086e93d2352a4536$export$a9781837241c946d, (v) => $086e93d2352a4536$export$a9781837241c946d = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-FieldButton--quiet", () => $086e93d2352a4536$export$5ca7b274e4fff9fc, (v) => $086e93d2352a4536$export$5ca7b274e4fff9fc = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ClearButton--overBackground", () => $086e93d2352a4536$export$116f2fccf941d34d, (v) => $086e93d2352a4536$export$116f2fccf941d34d = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ClearButton--small", () => $086e93d2352a4536$export$49280ffa5236ba8b, (v) => $086e93d2352a4536$export$49280ffa5236ba8b = v);
$parcel$export($086e93d2352a4536$exports, "is-focused", () => $086e93d2352a4536$export$e7dc768d35940237, (v) => $086e93d2352a4536$export$e7dc768d35940237 = v);
$parcel$export($086e93d2352a4536$exports, "is-active", () => $086e93d2352a4536$export$20fd0f7cd4e6112f, (v) => $086e93d2352a4536$export$20fd0f7cd4e6112f = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-Button--cta", () => $086e93d2352a4536$export$b818fdf0052a2cab, (v) => $086e93d2352a4536$export$b818fdf0052a2cab = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-Button--primary", () => $086e93d2352a4536$export$c1c3511f5e96e926, (v) => $086e93d2352a4536$export$c1c3511f5e96e926 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-Button--secondary", () => $086e93d2352a4536$export$4f1b75526afd8ff3, (v) => $086e93d2352a4536$export$4f1b75526afd8ff3 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-Button--warning", () => $086e93d2352a4536$export$6f97cef817ff1cac, (v) => $086e93d2352a4536$export$6f97cef817ff1cac = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-Button--overBackground", () => $086e93d2352a4536$export$5cda1128f9112d2e, (v) => $086e93d2352a4536$export$5cda1128f9112d2e = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-Button--quiet", () => $086e93d2352a4536$export$25f049c049371040, (v) => $086e93d2352a4536$export$25f049c049371040 = v);
$parcel$export($086e93d2352a4536$exports, "is-selected", () => $086e93d2352a4536$export$1e0fb04f31d3c22a, (v) => $086e93d2352a4536$export$1e0fb04f31d3c22a = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ActionButton--staticWhite", () => $086e93d2352a4536$export$ccc536e1adbdc059, (v) => $086e93d2352a4536$export$ccc536e1adbdc059 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-ActionButton--staticBlack", () => $086e93d2352a4536$export$8f7c9db66978d20f, (v) => $086e93d2352a4536$export$8f7c9db66978d20f = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-LogicButton--and", () => $086e93d2352a4536$export$e52daab43a62c528, (v) => $086e93d2352a4536$export$e52daab43a62c528 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-LogicButton--or", () => $086e93d2352a4536$export$da76be6126aaa633, (v) => $086e93d2352a4536$export$da76be6126aaa633 = v);
$parcel$export($086e93d2352a4536$exports, "is-placeholder", () => $086e93d2352a4536$export$e5b2f5233e4e5194, (v) => $086e93d2352a4536$export$e5b2f5233e4e5194 = v);
$parcel$export($086e93d2352a4536$exports, "spectrum-FieldButton--invalid", () => $086e93d2352a4536$export$c9f503f672e8a3c1, (v) => $086e93d2352a4536$export$c9f503f672e8a3c1 = v);
var $086e93d2352a4536$export$1db4cca5b4ade39a;
var $086e93d2352a4536$export$53da69f51b770d3;
var $086e93d2352a4536$export$36ae67a131730482;
var $086e93d2352a4536$export$a024c99ecf6b5741;
var $086e93d2352a4536$export$b752ce409e5660c;
var $086e93d2352a4536$export$f39a09f249340e2a;
var $086e93d2352a4536$export$d374b04f30360026;
var $086e93d2352a4536$export$5aabb61a966e266f;
var $086e93d2352a4536$export$29b7b5d6a5a497c5;
var $086e93d2352a4536$export$b8813cd5d7824ce7;
var $086e93d2352a4536$export$c081ba562e204942;
var $086e93d2352a4536$export$ebb4635c66bacdb9;
var $086e93d2352a4536$export$4a4f01c443eff992;
var $086e93d2352a4536$export$40e1f009544bacf3;
var $086e93d2352a4536$export$d35bc1e505d1ebbf;
var $086e93d2352a4536$export$a9781837241c946d;
var $086e93d2352a4536$export$5ca7b274e4fff9fc;
var $086e93d2352a4536$export$116f2fccf941d34d;
var $086e93d2352a4536$export$49280ffa5236ba8b;
var $086e93d2352a4536$export$e7dc768d35940237;
var $086e93d2352a4536$export$20fd0f7cd4e6112f;
var $086e93d2352a4536$export$b818fdf0052a2cab;
var $086e93d2352a4536$export$c1c3511f5e96e926;
var $086e93d2352a4536$export$4f1b75526afd8ff3;
var $086e93d2352a4536$export$6f97cef817ff1cac;
var $086e93d2352a4536$export$5cda1128f9112d2e;
var $086e93d2352a4536$export$25f049c049371040;
var $086e93d2352a4536$export$1e0fb04f31d3c22a;
var $086e93d2352a4536$export$ccc536e1adbdc059;
var $086e93d2352a4536$export$8f7c9db66978d20f;
var $086e93d2352a4536$export$e52daab43a62c528;
var $086e93d2352a4536$export$da76be6126aaa633;
var $086e93d2352a4536$export$e5b2f5233e4e5194;
var $086e93d2352a4536$export$c9f503f672e8a3c1;
$086e93d2352a4536$export$1db4cca5b4ade39a = "spectrum-Button_e2d99e";
$086e93d2352a4536$export$53da69f51b770d3 = "spectrum-ActionButton_e2d99e";
$086e93d2352a4536$export$36ae67a131730482 = "spectrum-LogicButton_e2d99e";
$086e93d2352a4536$export$a024c99ecf6b5741 = "spectrum-FieldButton_e2d99e";
$086e93d2352a4536$export$b752ce409e5660c = "spectrum-ClearButton_e2d99e";
$086e93d2352a4536$export$f39a09f249340e2a = "focus-ring_e2d99e";
$086e93d2352a4536$export$d374b04f30360026 = "spectrum-Icon_e2d99e";
$086e93d2352a4536$export$5aabb61a966e266f = "spectrum-ActionButton--emphasized_e2d99e";
$086e93d2352a4536$export$29b7b5d6a5a497c5 = "spectrum-ActionButton--staticColor_e2d99e";
$086e93d2352a4536$export$b8813cd5d7824ce7 = "is-hovered_e2d99e";
$086e93d2352a4536$export$c081ba562e204942 = "spectrum-Button-label_e2d99e";
$086e93d2352a4536$export$ebb4635c66bacdb9 = "spectrum-ActionButton-label_e2d99e";
$086e93d2352a4536$export$4a4f01c443eff992 = "spectrum-ActionButton-hold_e2d99e";
$086e93d2352a4536$export$40e1f009544bacf3 = "spectrum-ActionButton--quiet_e2d99e";
$086e93d2352a4536$export$d35bc1e505d1ebbf = "is-disabled_e2d99e";
$086e93d2352a4536$export$a9781837241c946d = "is-open_e2d99e";
$086e93d2352a4536$export$5ca7b274e4fff9fc = "spectrum-FieldButton--quiet_e2d99e";
$086e93d2352a4536$export$116f2fccf941d34d = "spectrum-ClearButton--overBackground_e2d99e";
$086e93d2352a4536$export$49280ffa5236ba8b = "spectrum-ClearButton--small_e2d99e";
$086e93d2352a4536$export$e7dc768d35940237 = "is-focused_e2d99e";
$086e93d2352a4536$export$20fd0f7cd4e6112f = "is-active_e2d99e";
$086e93d2352a4536$export$b818fdf0052a2cab = "spectrum-Button--cta_e2d99e";
$086e93d2352a4536$export$c1c3511f5e96e926 = "spectrum-Button--primary_e2d99e";
$086e93d2352a4536$export$4f1b75526afd8ff3 = "spectrum-Button--secondary_e2d99e";
$086e93d2352a4536$export$6f97cef817ff1cac = "spectrum-Button--warning_e2d99e";
$086e93d2352a4536$export$5cda1128f9112d2e = "spectrum-Button--overBackground_e2d99e";
$086e93d2352a4536$export$25f049c049371040 = "spectrum-Button--quiet_e2d99e";
$086e93d2352a4536$export$1e0fb04f31d3c22a = "is-selected_e2d99e";
$086e93d2352a4536$export$ccc536e1adbdc059 = "spectrum-ActionButton--staticWhite_e2d99e";
$086e93d2352a4536$export$8f7c9db66978d20f = "spectrum-ActionButton--staticBlack_e2d99e";
$086e93d2352a4536$export$e52daab43a62c528 = "spectrum-LogicButton--and_e2d99e";
$086e93d2352a4536$export$da76be6126aaa633 = "spectrum-LogicButton--or_e2d99e";
$086e93d2352a4536$export$e5b2f5233e4e5194 = "is-placeholder_e2d99e";
$086e93d2352a4536$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_e2d99e";













var $cd231e464c886c86$exports = {};

$parcel$export($cd231e464c886c86$exports, "spectrum-FieldLabel", () => $cd231e464c886c86$export$493d18e796ae054, (v) => $cd231e464c886c86$export$493d18e796ae054 = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-FieldLabel--positionSide", () => $cd231e464c886c86$export$9e6f19dc21f22f2e, (v) => $cd231e464c886c86$export$9e6f19dc21f22f2e = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-FieldLabel-requiredIcon", () => $cd231e464c886c86$export$7c47642c7d46f3c9, (v) => $cd231e464c886c86$export$7c47642c7d46f3c9 = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-FieldLabel--alignEnd", () => $cd231e464c886c86$export$885efcc08143a987, (v) => $cd231e464c886c86$export$885efcc08143a987 = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-Field", () => $cd231e464c886c86$export$f6d480ae1e56eba0, (v) => $cd231e464c886c86$export$f6d480ae1e56eba0 = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-Field--positionTop", () => $cd231e464c886c86$export$a4ea780a9064d7f9, (v) => $cd231e464c886c86$export$a4ea780a9064d7f9 = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-Field-field", () => $cd231e464c886c86$export$3ab8a3dc7f0563d, (v) => $cd231e464c886c86$export$3ab8a3dc7f0563d = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-Field-field--multiline", () => $cd231e464c886c86$export$f58a1e966a92ba5e, (v) => $cd231e464c886c86$export$f58a1e966a92ba5e = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-Field--positionSide", () => $cd231e464c886c86$export$2963225c91179589, (v) => $cd231e464c886c86$export$2963225c91179589 = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-Field-wrapper", () => $cd231e464c886c86$export$127df0b7290eb8ef, (v) => $cd231e464c886c86$export$127df0b7290eb8ef = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-Form", () => $cd231e464c886c86$export$94d2bcc94afabd89, (v) => $cd231e464c886c86$export$94d2bcc94afabd89 = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-Form--positionSide", () => $cd231e464c886c86$export$7f44db659563c8f4, (v) => $cd231e464c886c86$export$7f44db659563c8f4 = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-Form--positionTop", () => $cd231e464c886c86$export$11b8bc81fe551057, (v) => $cd231e464c886c86$export$11b8bc81fe551057 = v);
$parcel$export($cd231e464c886c86$exports, "spectrum-Form-itemLabel", () => $cd231e464c886c86$export$51ec27881a1e3928, (v) => $cd231e464c886c86$export$51ec27881a1e3928 = v);
$parcel$export($cd231e464c886c86$exports, "is-disabled", () => $cd231e464c886c86$export$d35bc1e505d1ebbf, (v) => $cd231e464c886c86$export$d35bc1e505d1ebbf = v);
var $cd231e464c886c86$export$493d18e796ae054;
var $cd231e464c886c86$export$9e6f19dc21f22f2e;
var $cd231e464c886c86$export$7c47642c7d46f3c9;
var $cd231e464c886c86$export$885efcc08143a987;
var $cd231e464c886c86$export$f6d480ae1e56eba0;
var $cd231e464c886c86$export$a4ea780a9064d7f9;
var $cd231e464c886c86$export$3ab8a3dc7f0563d;
var $cd231e464c886c86$export$f58a1e966a92ba5e;
var $cd231e464c886c86$export$2963225c91179589;
var $cd231e464c886c86$export$127df0b7290eb8ef;
var $cd231e464c886c86$export$94d2bcc94afabd89;
var $cd231e464c886c86$export$7f44db659563c8f4;
var $cd231e464c886c86$export$11b8bc81fe551057;
var $cd231e464c886c86$export$51ec27881a1e3928;
var $cd231e464c886c86$export$d35bc1e505d1ebbf;
$cd231e464c886c86$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$cd231e464c886c86$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$cd231e464c886c86$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$cd231e464c886c86$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$cd231e464c886c86$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$cd231e464c886c86$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$cd231e464c886c86$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$cd231e464c886c86$export$f58a1e966a92ba5e = "spectrum-Field-field--multiline_d2db1f";
$cd231e464c886c86$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$cd231e464c886c86$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$cd231e464c886c86$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$cd231e464c886c86$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$cd231e464c886c86$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$cd231e464c886c86$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$cd231e464c886c86$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";






var $5589e2270a5c2f51$exports = {};

$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search", () => $5589e2270a5c2f51$export$713911d754e1e4ae, (v) => $5589e2270a5c2f51$export$713911d754e1e4ae = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-ClearButton", () => $5589e2270a5c2f51$export$b752ce409e5660c, (v) => $5589e2270a5c2f51$export$b752ce409e5660c = v);
$parcel$export($5589e2270a5c2f51$exports, "is-quiet", () => $5589e2270a5c2f51$export$6d5c5433f0da4241, (v) => $5589e2270a5c2f51$export$6d5c5433f0da4241 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search-input", () => $5589e2270a5c2f51$export$50e02603f2d97ae1, (v) => $5589e2270a5c2f51$export$50e02603f2d97ae1 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search--invalid", () => $5589e2270a5c2f51$export$77a6151a0b46a6a3, (v) => $5589e2270a5c2f51$export$77a6151a0b46a6a3 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search--valid", () => $5589e2270a5c2f51$export$aa81e3ab40f37567, (v) => $5589e2270a5c2f51$export$aa81e3ab40f37567 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search--loadable", () => $5589e2270a5c2f51$export$9c74d64afae3b2d4, (v) => $5589e2270a5c2f51$export$9c74d64afae3b2d4 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search-circleLoader", () => $5589e2270a5c2f51$export$70ae4353602451dc, (v) => $5589e2270a5c2f51$export$70ae4353602451dc = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search-validationIcon", () => $5589e2270a5c2f51$export$938f0981825eae6e, (v) => $5589e2270a5c2f51$export$938f0981825eae6e = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Textfield", () => $5589e2270a5c2f51$export$2c8a10299d8b3418, (v) => $5589e2270a5c2f51$export$2c8a10299d8b3418 = v);
var $5589e2270a5c2f51$export$713911d754e1e4ae;
var $5589e2270a5c2f51$export$b752ce409e5660c;
var $5589e2270a5c2f51$export$6d5c5433f0da4241;
var $5589e2270a5c2f51$export$50e02603f2d97ae1;
var $5589e2270a5c2f51$export$77a6151a0b46a6a3;
var $5589e2270a5c2f51$export$aa81e3ab40f37567;
var $5589e2270a5c2f51$export$9c74d64afae3b2d4;
var $5589e2270a5c2f51$export$70ae4353602451dc;
var $5589e2270a5c2f51$export$938f0981825eae6e;
var $5589e2270a5c2f51$export$2c8a10299d8b3418;
$5589e2270a5c2f51$export$713911d754e1e4ae = "spectrum-Search_d3c0f3";
$5589e2270a5c2f51$export$b752ce409e5660c = "spectrum-ClearButton_d3c0f3";
$5589e2270a5c2f51$export$6d5c5433f0da4241 = "is-quiet_d3c0f3";
$5589e2270a5c2f51$export$50e02603f2d97ae1 = "spectrum-Search-input_d3c0f3";
$5589e2270a5c2f51$export$77a6151a0b46a6a3 = "spectrum-Search--invalid_d3c0f3";
$5589e2270a5c2f51$export$aa81e3ab40f37567 = "spectrum-Search--valid_d3c0f3";
$5589e2270a5c2f51$export$9c74d64afae3b2d4 = "spectrum-Search--loadable_d3c0f3";
$5589e2270a5c2f51$export$70ae4353602451dc = "spectrum-Search-circleLoader_d3c0f3";
$5589e2270a5c2f51$export$938f0981825eae6e = "spectrum-Search-validationIcon_d3c0f3";
$5589e2270a5c2f51$export$2c8a10299d8b3418 = "spectrum-Textfield_d3c0f3";



var $8fb52a80fe52b1c7$exports = {};

$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup", () => $8fb52a80fe52b1c7$export$5be359c1227eedcb, (v) => $8fb52a80fe52b1c7$export$5be359c1227eedcb = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-FieldButton", () => $8fb52a80fe52b1c7$export$a024c99ecf6b5741, (v) => $8fb52a80fe52b1c7$export$a024c99ecf6b5741 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "is-disabled", () => $8fb52a80fe52b1c7$export$d35bc1e505d1ebbf, (v) => $8fb52a80fe52b1c7$export$d35bc1e505d1ebbf = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup--invalid", () => $8fb52a80fe52b1c7$export$3b48509739a9cda0, (v) => $8fb52a80fe52b1c7$export$3b48509739a9cda0 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup--quiet", () => $8fb52a80fe52b1c7$export$ebf265ffc062029b, (v) => $8fb52a80fe52b1c7$export$ebf265ffc062029b = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-field", () => $8fb52a80fe52b1c7$export$ea77459595e26bec, (v) => $8fb52a80fe52b1c7$export$ea77459595e26bec = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-input-validationIcon", () => $8fb52a80fe52b1c7$export$f7ab13ee113f4884, (v) => $8fb52a80fe52b1c7$export$f7ab13ee113f4884 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-input-circleLoader", () => $8fb52a80fe52b1c7$export$dcc9e00cdad967fa, (v) => $8fb52a80fe52b1c7$export$dcc9e00cdad967fa = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-input", () => $8fb52a80fe52b1c7$export$92706f5c6fe4074, (v) => $8fb52a80fe52b1c7$export$92706f5c6fe4074 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-icon", () => $8fb52a80fe52b1c7$export$b91b6af799fd93a8, (v) => $8fb52a80fe52b1c7$export$b91b6af799fd93a8 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-popover--quiet", () => $8fb52a80fe52b1c7$export$5e116178f9b1556a, (v) => $8fb52a80fe52b1c7$export$5e116178f9b1556a = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker--range", () => $8fb52a80fe52b1c7$export$1d8b0b29e11d7cd, (v) => $8fb52a80fe52b1c7$export$1d8b0b29e11d7cd = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker--datetimeRange", () => $8fb52a80fe52b1c7$export$58dd7cd23c92699f, (v) => $8fb52a80fe52b1c7$export$58dd7cd23c92699f = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker-startField", () => $8fb52a80fe52b1c7$export$80f9c838a8406469, (v) => $8fb52a80fe52b1c7$export$80f9c838a8406469 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker-endField", () => $8fb52a80fe52b1c7$export$1c72fb41fa4f21ee, (v) => $8fb52a80fe52b1c7$export$1c72fb41fa4f21ee = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker--rangeDash", () => $8fb52a80fe52b1c7$export$202113e883898dc, (v) => $8fb52a80fe52b1c7$export$202113e883898dc = v);
$parcel$export($8fb52a80fe52b1c7$exports, "is-focused", () => $8fb52a80fe52b1c7$export$e7dc768d35940237, (v) => $8fb52a80fe52b1c7$export$e7dc768d35940237 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker-focusRing", () => $8fb52a80fe52b1c7$export$eb262b15b9261fde, (v) => $8fb52a80fe52b1c7$export$eb262b15b9261fde = v);
$parcel$export($8fb52a80fe52b1c7$exports, "is-hovered", () => $8fb52a80fe52b1c7$export$b8813cd5d7824ce7, (v) => $8fb52a80fe52b1c7$export$b8813cd5d7824ce7 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "focus-ring", () => $8fb52a80fe52b1c7$export$f39a09f249340e2a, (v) => $8fb52a80fe52b1c7$export$f39a09f249340e2a = v);
$parcel$export($8fb52a80fe52b1c7$exports, "is-selected", () => $8fb52a80fe52b1c7$export$1e0fb04f31d3c22a, (v) => $8fb52a80fe52b1c7$export$1e0fb04f31d3c22a = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-FieldButton--invalid", () => $8fb52a80fe52b1c7$export$c9f503f672e8a3c1, (v) => $8fb52a80fe52b1c7$export$c9f503f672e8a3c1 = v);
var $8fb52a80fe52b1c7$export$5be359c1227eedcb;
var $8fb52a80fe52b1c7$export$a024c99ecf6b5741;
var $8fb52a80fe52b1c7$export$d35bc1e505d1ebbf;
var $8fb52a80fe52b1c7$export$3b48509739a9cda0;
var $8fb52a80fe52b1c7$export$ebf265ffc062029b;
var $8fb52a80fe52b1c7$export$ea77459595e26bec;
var $8fb52a80fe52b1c7$export$f7ab13ee113f4884;
var $8fb52a80fe52b1c7$export$dcc9e00cdad967fa;
var $8fb52a80fe52b1c7$export$92706f5c6fe4074;
var $8fb52a80fe52b1c7$export$b91b6af799fd93a8;
var $8fb52a80fe52b1c7$export$5e116178f9b1556a;
var $8fb52a80fe52b1c7$export$1d8b0b29e11d7cd;
var $8fb52a80fe52b1c7$export$58dd7cd23c92699f;
var $8fb52a80fe52b1c7$export$80f9c838a8406469;
var $8fb52a80fe52b1c7$export$1c72fb41fa4f21ee;
var $8fb52a80fe52b1c7$export$202113e883898dc;
var $8fb52a80fe52b1c7$export$e7dc768d35940237;
var $8fb52a80fe52b1c7$export$eb262b15b9261fde;
var $8fb52a80fe52b1c7$export$b8813cd5d7824ce7;
var $8fb52a80fe52b1c7$export$f39a09f249340e2a;
var $8fb52a80fe52b1c7$export$1e0fb04f31d3c22a;
var $8fb52a80fe52b1c7$export$c9f503f672e8a3c1;
$8fb52a80fe52b1c7$export$5be359c1227eedcb = "spectrum-InputGroup_a0942f";
$8fb52a80fe52b1c7$export$a024c99ecf6b5741 = "spectrum-FieldButton_a0942f";
$8fb52a80fe52b1c7$export$d35bc1e505d1ebbf = "is-disabled_a0942f";
$8fb52a80fe52b1c7$export$3b48509739a9cda0 = "spectrum-InputGroup--invalid_a0942f";
$8fb52a80fe52b1c7$export$ebf265ffc062029b = "spectrum-InputGroup--quiet_a0942f";
$8fb52a80fe52b1c7$export$ea77459595e26bec = "spectrum-InputGroup-field_a0942f";
$8fb52a80fe52b1c7$export$f7ab13ee113f4884 = "spectrum-InputGroup-input-validationIcon_a0942f";
$8fb52a80fe52b1c7$export$dcc9e00cdad967fa = "spectrum-InputGroup-input-circleLoader_a0942f";
$8fb52a80fe52b1c7$export$92706f5c6fe4074 = "spectrum-InputGroup-input_a0942f";
$8fb52a80fe52b1c7$export$b91b6af799fd93a8 = "spectrum-InputGroup-icon_a0942f";
$8fb52a80fe52b1c7$export$5e116178f9b1556a = "spectrum-InputGroup-popover--quiet_a0942f";
$8fb52a80fe52b1c7$export$1d8b0b29e11d7cd = "spectrum-Datepicker--range_a0942f";
$8fb52a80fe52b1c7$export$58dd7cd23c92699f = "spectrum-Datepicker--datetimeRange_a0942f";
$8fb52a80fe52b1c7$export$80f9c838a8406469 = "spectrum-Datepicker-startField_a0942f";
$8fb52a80fe52b1c7$export$1c72fb41fa4f21ee = "spectrum-Datepicker-endField_a0942f";
$8fb52a80fe52b1c7$export$202113e883898dc = "spectrum-Datepicker--rangeDash_a0942f";
$8fb52a80fe52b1c7$export$e7dc768d35940237 = "is-focused_a0942f";
$8fb52a80fe52b1c7$export$eb262b15b9261fde = "spectrum-Datepicker-focusRing_a0942f";
$8fb52a80fe52b1c7$export$b8813cd5d7824ce7 = "is-hovered_a0942f";
$8fb52a80fe52b1c7$export$f39a09f249340e2a = "focus-ring_a0942f";
$8fb52a80fe52b1c7$export$1e0fb04f31d3c22a = "is-selected_a0942f";
$8fb52a80fe52b1c7$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_a0942f";



var $c14528ddc146317f$exports = {};

$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-input", () => $c14528ddc146317f$export$1ac009feb2b0bd0c, (v) => $c14528ddc146317f$export$1ac009feb2b0bd0c = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield", () => $c14528ddc146317f$export$2c8a10299d8b3418, (v) => $c14528ddc146317f$export$2c8a10299d8b3418 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield--quiet", () => $c14528ddc146317f$export$bffd5e3d61a81737, (v) => $c14528ddc146317f$export$bffd5e3d61a81737 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield--multiline", () => $c14528ddc146317f$export$9699f1c538a448a0, (v) => $c14528ddc146317f$export$9699f1c538a448a0 = v);
$parcel$export($c14528ddc146317f$exports, "is-placeholder", () => $c14528ddc146317f$export$e5b2f5233e4e5194, (v) => $c14528ddc146317f$export$e5b2f5233e4e5194 = v);
$parcel$export($c14528ddc146317f$exports, "is-hovered", () => $c14528ddc146317f$export$b8813cd5d7824ce7, (v) => $c14528ddc146317f$export$b8813cd5d7824ce7 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield--valid", () => $c14528ddc146317f$export$c2abbc3f970170b7, (v) => $c14528ddc146317f$export$c2abbc3f970170b7 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield--invalid", () => $c14528ddc146317f$export$58e50b10d30123df, (v) => $c14528ddc146317f$export$58e50b10d30123df = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield--loadable", () => $c14528ddc146317f$export$14fd30fea6f1348c, (v) => $c14528ddc146317f$export$14fd30fea6f1348c = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-validationIcon", () => $c14528ddc146317f$export$cf0ead5b44db0da3, (v) => $c14528ddc146317f$export$cf0ead5b44db0da3 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-icon", () => $c14528ddc146317f$export$222d1ccd1870be1d, (v) => $c14528ddc146317f$export$222d1ccd1870be1d = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-inputIcon", () => $c14528ddc146317f$export$c1cdcaa5fe76a871, (v) => $c14528ddc146317f$export$c1cdcaa5fe76a871 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-circleLoader", () => $c14528ddc146317f$export$82a249fb4d6127, (v) => $c14528ddc146317f$export$82a249fb4d6127 = v);
$parcel$export($c14528ddc146317f$exports, "is-focused", () => $c14528ddc146317f$export$e7dc768d35940237, (v) => $c14528ddc146317f$export$e7dc768d35940237 = v);
$parcel$export($c14528ddc146317f$exports, "focus-ring", () => $c14528ddc146317f$export$f39a09f249340e2a, (v) => $c14528ddc146317f$export$f39a09f249340e2a = v);
$parcel$export($c14528ddc146317f$exports, "is-disabled", () => $c14528ddc146317f$export$d35bc1e505d1ebbf, (v) => $c14528ddc146317f$export$d35bc1e505d1ebbf = v);
var $c14528ddc146317f$export$1ac009feb2b0bd0c;
var $c14528ddc146317f$export$2c8a10299d8b3418;
var $c14528ddc146317f$export$bffd5e3d61a81737;
var $c14528ddc146317f$export$9699f1c538a448a0;
var $c14528ddc146317f$export$e5b2f5233e4e5194;
var $c14528ddc146317f$export$b8813cd5d7824ce7;
var $c14528ddc146317f$export$c2abbc3f970170b7;
var $c14528ddc146317f$export$58e50b10d30123df;
var $c14528ddc146317f$export$14fd30fea6f1348c;
var $c14528ddc146317f$export$cf0ead5b44db0da3;
var $c14528ddc146317f$export$222d1ccd1870be1d;
var $c14528ddc146317f$export$c1cdcaa5fe76a871;
var $c14528ddc146317f$export$82a249fb4d6127;
var $c14528ddc146317f$export$e7dc768d35940237;
var $c14528ddc146317f$export$f39a09f249340e2a;
var $c14528ddc146317f$export$d35bc1e505d1ebbf;
$c14528ddc146317f$export$1ac009feb2b0bd0c = "spectrum-Textfield-input_73bc77";
$c14528ddc146317f$export$2c8a10299d8b3418 = "spectrum-Textfield_73bc77";
$c14528ddc146317f$export$bffd5e3d61a81737 = "spectrum-Textfield--quiet_73bc77";
$c14528ddc146317f$export$9699f1c538a448a0 = "spectrum-Textfield--multiline_73bc77";
$c14528ddc146317f$export$e5b2f5233e4e5194 = "is-placeholder_73bc77";
$c14528ddc146317f$export$b8813cd5d7824ce7 = "is-hovered_73bc77";
$c14528ddc146317f$export$c2abbc3f970170b7 = "spectrum-Textfield--valid_73bc77";
$c14528ddc146317f$export$58e50b10d30123df = "spectrum-Textfield--invalid_73bc77";
$c14528ddc146317f$export$14fd30fea6f1348c = "spectrum-Textfield--loadable_73bc77";
$c14528ddc146317f$export$cf0ead5b44db0da3 = "spectrum-Textfield-validationIcon_73bc77";
$c14528ddc146317f$export$222d1ccd1870be1d = "spectrum-Textfield-icon_73bc77";
$c14528ddc146317f$export$c1cdcaa5fe76a871 = "spectrum-Textfield-inputIcon_73bc77";
$c14528ddc146317f$export$82a249fb4d6127 = "spectrum-Textfield-circleLoader_73bc77";
$c14528ddc146317f$export$e7dc768d35940237 = "is-focused_73bc77";
$c14528ddc146317f$export$f39a09f249340e2a = "focus-ring_73bc77";
$c14528ddc146317f$export$d35bc1e505d1ebbf = "is-disabled_73bc77";












const $a1ae4cad4496428f$export$7637df911c069b4d = /*#__PURE__*/ $725Me$react.forwardRef(function $a1ae4cad4496428f$export$7637df911c069b4d(props, ref) {
    props = $725Me$useProviderProps(props);
    let { isQuiet: isQuiet , isDisabled: isDisabled , validationState: validationState , isReadOnly: isReadOnly  } = props;
    let { contains: contains  } = $725Me$useFilter({
        sensitivity: 'base'
    });
    let state = $725Me$useComboBoxState({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: true,
        // Needs to be false here otherwise we double up on commitSelection/commitCustomValue calls when
        // user taps on underlay (i.e. initial tap will call setFocused(false) -> commitSelection/commitCustomValue via onBlur,
        // then the closing of the tray will call setFocused(false) again due to cleanup effect)
        shouldCloseOnBlur: false
    });
    let buttonRef = $725Me$useRef();
    let domRef = $725Me$useFocusableRef(ref, buttonRef);
    let { triggerProps: triggerProps , overlayProps: overlayProps  } = $725Me$useOverlayTrigger({
        type: 'listbox'
    }, state, buttonRef);
    let { labelProps: labelProps , fieldProps: fieldProps  } = $725Me$useField({
        ...props,
        labelElementType: 'span'
    });
    // Focus the button and show focus ring when clicking on the label
    labelProps.onClick = ()=>{
        if (!props.isDisabled) {
            buttonRef.current.focus();
            $725Me$setInteractionModality('keyboard');
        }
    };
    let onClose = ()=>state.commit()
    ;
    return(/*#__PURE__*/ $725Me$react.createElement($725Me$react.Fragment, null, /*#__PURE__*/ $725Me$react.createElement($725Me$Field, {
        ...props,
        labelProps: labelProps,
        elementType: "span",
        ref: domRef,
        includeNecessityIndicatorInAccessibilityName: true
    }, /*#__PURE__*/ $725Me$react.createElement($a1ae4cad4496428f$var$ComboBoxButton, {
        ...$725Me$mergeProps(triggerProps, fieldProps, {
            autoFocus: props.autoFocus
        }),
        ref: buttonRef,
        isQuiet: isQuiet,
        isDisabled: isDisabled,
        isPlaceholder: !state.inputValue,
        validationState: validationState,
        onPress: ()=>!isReadOnly && state.open(null, 'manual')
    }, state.inputValue || props.placeholder || '')), /*#__PURE__*/ $725Me$react.createElement($725Me$Tray, {
        isOpen: state.isOpen,
        onClose: onClose,
        isFixedHeight: true,
        isNonModal: true,
        ...overlayProps
    }, /*#__PURE__*/ $725Me$react.createElement($a1ae4cad4496428f$var$ComboBoxTray, {
        ...props,
        onClose: onClose,
        overlayProps: overlayProps,
        state: state
    }))));
});
const $a1ae4cad4496428f$var$ComboBoxButton = /*#__PURE__*/ $725Me$react.forwardRef(function ComboBoxButton(props, ref) {
    let { isQuiet: isQuiet , isDisabled: isDisabled , isPlaceholder: isPlaceholder , validationState: validationState , children: children , style: style , className: className  } = props;
    let formatMessage = $725Me$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($9c69665f9f0b71fd$exports)));
    let valueId = $725Me$useId();
    let invalidId = $725Me$useId();
    let validationIcon = validationState === 'invalid' ? /*#__PURE__*/ $725Me$react.createElement($725Me$spectrumiconsuiAlertMedium, {
        id: invalidId,
        "aria-label": formatMessage('invalid')
    }) : /*#__PURE__*/ $725Me$react.createElement($725Me$spectrumiconsuiCheckmarkMedium, null);
    let validation = /*#__PURE__*/ $725Me$react.cloneElement(validationIcon, {
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($c14528ddc146317f$exports)), 'spectrum-Textfield-validationIcon', $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-input-validationIcon'))
    });
    let { hoverProps: hoverProps , isHovered: isHovered  } = $725Me$useHover({
    });
    let { buttonProps: buttonProps , isPressed: isPressed  } = $725Me$useButton({
        ...props,
        'aria-labelledby': [
            props['aria-labelledby'],
            props['aria-label'] && !props['aria-labelledby'] ? props.id : null,
            valueId,
            validationState === 'invalid' ? invalidId : null
        ].filter(Boolean).join(' '),
        elementType: 'div'
    }, ref);
    return(/*#__PURE__*/ $725Me$react.createElement($725Me$FocusRing, {
        focusClass: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'is-focused'),
        focusRingClass: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'focus-ring')
    }, /*#__PURE__*/ $725Me$react.createElement("div", {
        ...$725Me$mergeProps(hoverProps, buttonProps),
        "aria-haspopup": "dialog",
        ref: ref,
        style: {
            ...style,
            outline: 'none'
        },
        className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($7a1753adda45f7fb$exports)), 'mobile-combobox'), className)
    }, /*#__PURE__*/ $725Me$react.createElement("div", {
        className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($c14528ddc146317f$exports)), 'spectrum-Textfield', {
            'spectrum-Textfield--invalid': validationState === 'invalid',
            'spectrum-Textfield--valid': validationState === 'valid',
            'spectrum-Textfield--quiet': isQuiet
        }, $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-field'))
    }, /*#__PURE__*/ $725Me$react.createElement("div", {
        className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($c14528ddc146317f$exports)), 'spectrum-Textfield-input', {
            'is-hovered': isHovered,
            'is-placeholder': isPlaceholder,
            'is-disabled': isDisabled
        }, $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-input', $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($cd231e464c886c86$exports)), 'spectrum-Field-field')), $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($7a1753adda45f7fb$exports)), 'mobile-input'))
    }, /*#__PURE__*/ $725Me$react.createElement("span", {
        id: valueId,
        className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($7a1753adda45f7fb$exports)), 'mobile-value')
    }, children)), validationState ? validation : null), /*#__PURE__*/ $725Me$react.createElement("div", {
        className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($086e93d2352a4536$exports)), 'spectrum-FieldButton', {
            'spectrum-FieldButton--quiet': isQuiet,
            'is-active': isPressed,
            'is-disabled': isDisabled,
            'spectrum-FieldButton--invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-FieldButton'))
    }, /*#__PURE__*/ $725Me$react.createElement($725Me$spectrumiconsuiChevronDownMedium, {
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-Dropdown-chevron')
    })))));
});
function $a1ae4cad4496428f$var$ComboBoxTray(props) {
    let { state: // completionMode = 'suggest',
    state , isDisabled: isDisabled , validationState: validationState , label: label , overlayProps: overlayProps , loadingState: loadingState , onLoadMore: onLoadMore , onClose: onClose  } = props;
    let timeout = $725Me$useRef(null);
    let [showLoading, setShowLoading] = $725Me$useState(false);
    let inputRef = $725Me$useRef();
    let buttonRef = $725Me$useRef();
    let popoverRef = $725Me$useRef();
    let listBoxRef = $725Me$useRef();
    let layout = $725Me$useListBoxLayout(state);
    let formatMessage = $725Me$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($9c69665f9f0b71fd$exports)));
    let { inputProps: inputProps , listBoxProps: listBoxProps , labelProps: labelProps  } = $725Me$useComboBox({
        ...props,
        // completionMode,
        keyboardDelegate: layout,
        buttonRef: $725Me$unwrapDOMRef(buttonRef),
        popoverRef: popoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef
    }, state);
    $725Me$react.useEffect(()=>{
        $725Me$focusSafely(inputRef.current);
        // When the tray unmounts, set state.isFocused (i.e. the tray input's focus tracker) to false.
        // This is to prevent state.isFocused from being set to true when the tray closes via tapping on the underlay
        // (FocusScope attempts to restore focus to the tray input when tapping outside the tray due to "contain")
        // Have to do this manually since React doesn't call onBlur when a component is unmounted: https://github.com/facebook/react/issues/12363
        return ()=>{
            state.setFocused(false);
        };
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    let { dialogProps: dialogProps  } = $725Me$useDialog({
        'aria-labelledby': $725Me$useId(labelProps.id)
    }, popoverRef);
    // Override the role of the input to "searchbox" instead of "combobox".
    // Since the listbox is always visible, the combobox role doesn't really give us anything.
    // VoiceOver on iOS reads "double tap to collapse" when focused on the input rather than
    // "double tap to edit text", as with a textbox or searchbox. We'd like double tapping to
    // open the virtual keyboard rather than closing the tray.
    inputProps.role = 'searchbox';
    inputProps['aria-haspopup'] = 'listbox';
    delete inputProps.onTouchEnd;
    let clearButton = /*#__PURE__*/ $725Me$react.createElement($725Me$ClearButton, {
        preventFocus: true,
        "aria-label": formatMessage('clear'),
        excludeFromTabOrder: true,
        onPress: ()=>{
            state.setInputValue('');
            inputRef.current.focus();
        },
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($5589e2270a5c2f51$exports)), 'spectrum-ClearButton'),
        isDisabled: isDisabled
    });
    let loadingCircle = /*#__PURE__*/ $725Me$react.createElement($725Me$ProgressCircle, {
        "aria-label": formatMessage('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($5589e2270a5c2f51$exports)), 'spectrum-Search-circleLoader', $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($c14528ddc146317f$exports)), 'spectrum-Textfield-circleLoader'))
    });
    // Close the software keyboard on scroll to give the user a bigger area to scroll.
    // But only do this if scrolling with touch, otherwise it can cause issues with touch
    // screen readers.
    let isTouchDown = $725Me$useRef(false);
    let onTouchStart = ()=>{
        isTouchDown.current = true;
    };
    let onTouchEnd = ()=>{
        isTouchDown.current = false;
    };
    let onScroll = $725Me$useCallback(()=>{
        if (!inputRef.current || document.activeElement !== inputRef.current || !isTouchDown.current) return;
        popoverRef.current.focus();
    }, [
        inputRef,
        popoverRef,
        isTouchDown
    ]);
    let inputValue = inputProps.value;
    let lastInputValue = $725Me$useRef(inputValue);
    $725Me$useEffect(()=>{
        if (loadingState === 'filtering' && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (loadingState !== 'filtering') {
            // If loading is no longer happening, clear any timers and hide the loading circle
            setShowLoading(false);
            clearTimeout(timeout.current);
            timeout.current = null;
        }
        lastInputValue.current = inputValue;
    }, [
        loadingState,
        inputValue,
        showLoading
    ]);
    let onKeyDown = (e)=>{
        // Close virtual keyboard if user hits Enter w/o any focused options
        if (e.key === 'Enter' && state.selectionManager.focusedKey == null) popoverRef.current.focus();
        else inputProps.onKeyDown(e);
    };
    return(/*#__PURE__*/ $725Me$react.createElement($725Me$FocusScope, {
        restoreFocus: true,
        contain: true
    }, /*#__PURE__*/ $725Me$react.createElement("div", {
        ...$725Me$mergeProps(overlayProps, dialogProps),
        ref: popoverRef,
        className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($7a1753adda45f7fb$exports)), 'tray-dialog')
    }, /*#__PURE__*/ $725Me$react.createElement($725Me$DismissButton, {
        onDismiss: onClose
    }), /*#__PURE__*/ $725Me$react.createElement($725Me$TextFieldBase, {
        label: label,
        labelProps: labelProps,
        inputProps: {
            ...inputProps,
            onKeyDown: onKeyDown
        },
        inputRef: inputRef,
        isDisabled: isDisabled,
        isLoading: showLoading && loadingState === 'filtering',
        loadingIndicator: loadingState != null && loadingCircle,
        validationState: validationState,
        labelAlign: "start",
        labelPosition: "top",
        wrapperChildren: (state.inputValue !== '' || loadingState === 'filtering' || validationState != null) && !props.isReadOnly && clearButton,
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($5589e2270a5c2f51$exports)), 'spectrum-Search', 'spectrum-Textfield', 'spectrum-Search--loadable', {
            'spectrum-Search--invalid': validationState === 'invalid',
            'spectrum-Search--valid': validationState === 'valid'
        }, $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($7a1753adda45f7fb$exports)), 'tray-textfield', {
            'has-label': !!props.label
        })),
        inputClassName: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($7a1753adda45f7fb$exports)), 'tray-textfield-input', $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($5589e2270a5c2f51$exports)), 'spectrum-Search-input')),
        validationIconClassName: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($5589e2270a5c2f51$exports)), 'spectrum-Search-validationIcon')
    }), /*#__PURE__*/ $725Me$react.createElement($725Me$ListBoxBase, {
        ...listBoxProps,
        domProps: {
            onTouchStart: onTouchStart,
            onTouchEnd: onTouchEnd
        },
        disallowEmptySelection: true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        renderEmptyState: ()=>loadingState !== 'loading' && /*#__PURE__*/ $725Me$react.createElement("span", {
                className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($7a1753adda45f7fb$exports)), 'no-results')
            }, formatMessage('noResults'))
        ,
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($7a1753adda45f7fb$exports)), 'tray-listbox'),
        ref: listBoxRef,
        onScroll: onScroll,
        onLoadMore: onLoadMore,
        isLoading: loadingState === 'loading' || loadingState === 'loadingMore'
    }), /*#__PURE__*/ $725Me$react.createElement($725Me$DismissButton, {
        onDismiss: onClose
    }))));
}















function $2289f375e813f187$var$ComboBox(props, ref) {
    props = $725Me$useProviderProps(props);
    let isMobile = $725Me$useIsMobileDevice();
    if (isMobile) // menuTrigger=focus/manual don't apply to mobile combobox
    return(/*#__PURE__*/ $725Me$react.createElement($a1ae4cad4496428f$export$7637df911c069b4d, {
        ...props,
        menuTrigger: "input",
        ref: ref
    }));
    else return(/*#__PURE__*/ $725Me$react.createElement($2289f375e813f187$var$ComboBoxBase, {
        ...props,
        ref: ref
    }));
}
const $2289f375e813f187$var$ComboBoxBase = /*#__PURE__*/ $725Me$react.forwardRef(function ComboBoxBase(props, ref) {
    let { menuTrigger: menuTrigger = 'input' , shouldFlip: shouldFlip = true , direction: direction = 'bottom' , isQuiet: isQuiet , loadingState: loadingState , onLoadMore: onLoadMore  } = props;
    let formatMessage = $725Me$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($9c69665f9f0b71fd$exports)));
    let isAsync = loadingState != null;
    let popoverRef = $725Me$useRef();
    let unwrappedPopoverRef = $725Me$useUnwrapDOMRef(popoverRef);
    let buttonRef = $725Me$useRef();
    let unwrappedButtonRef = $725Me$useUnwrapDOMRef(buttonRef);
    let listBoxRef = $725Me$useRef();
    let inputRef = $725Me$useRef();
    let domRef = $725Me$useFocusableRef(ref, inputRef);
    let { contains: contains  } = $725Me$useFilter({
        sensitivity: 'base'
    });
    let state = $725Me$useComboBoxState({
        ...props,
        defaultFilter: contains,
        allowsEmptyCollection: isAsync
    });
    let layout = $725Me$useListBoxLayout(state);
    let { buttonProps: buttonProps , inputProps: inputProps , listBoxProps: listBoxProps , labelProps: labelProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $725Me$useComboBox({
        ...props,
        keyboardDelegate: layout,
        buttonRef: unwrappedButtonRef,
        popoverRef: unwrappedPopoverRef,
        listBoxRef: listBoxRef,
        inputRef: inputRef,
        menuTrigger: menuTrigger
    }, state);
    let { overlayProps: overlayProps , placement: placement , updatePosition: updatePosition  } = $725Me$useOverlayPosition({
        targetRef: unwrappedButtonRef,
        overlayRef: unwrappedPopoverRef,
        scrollRef: listBoxRef,
        placement: `${direction} end`,
        shouldFlip: shouldFlip,
        isOpen: state.isOpen,
        onClose: state.close
    });
    // Measure the width of the inputfield and the button to inform the width of the menu (below).
    let [menuWidth, setMenuWidth] = $725Me$useState(null);
    let { scale: scale  } = $725Me$useProvider();
    let onResize = $725Me$useCallback(()=>{
        if (unwrappedButtonRef.current && inputRef.current) {
            let buttonWidth = unwrappedButtonRef.current.offsetWidth;
            let inputWidth = inputRef.current.offsetWidth;
            setMenuWidth(buttonWidth + inputWidth);
        }
    }, [
        unwrappedButtonRef,
        inputRef,
        setMenuWidth
    ]);
    $725Me$useResizeObserver({
        ref: domRef,
        onResize: onResize
    });
    $725Me$useLayoutEffect(onResize, [
        scale,
        onResize
    ]);
    // Update position once the ListBox has rendered. This ensures that
    // it flips properly when it doesn't fit in the available space.
    // TODO: add ResizeObserver to useOverlayPosition so we don't need this.
    $725Me$useLayoutEffect(()=>{
        if (state.isOpen) requestAnimationFrame(()=>{
            updatePosition();
        });
    }, [
        state.isOpen,
        updatePosition
    ]);
    let style = {
        ...overlayProps.style,
        width: isQuiet ? null : menuWidth,
        minWidth: isQuiet ? `calc(${menuWidth}px + calc(2 * var(--spectrum-dropdown-quiet-offset)))` : menuWidth
    };
    return(/*#__PURE__*/ $725Me$react.createElement($725Me$react.Fragment, null, /*#__PURE__*/ $725Me$react.createElement($725Me$Field, {
        ...props,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        labelProps: labelProps,
        ref: domRef
    }, /*#__PURE__*/ $725Me$react.createElement($2289f375e813f187$var$ComboBoxInput, {
        ...props,
        isOpen: state.isOpen,
        loadingState: loadingState,
        inputProps: inputProps,
        inputRef: inputRef,
        triggerProps: buttonProps,
        triggerRef: buttonRef
    })), /*#__PURE__*/ $725Me$react.createElement($725Me$Popover, {
        isOpen: state.isOpen,
        UNSAFE_style: style,
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-popover', {
            'spectrum-InputGroup-popover--quiet': isQuiet
        }),
        ref: popoverRef,
        placement: placement,
        hideArrow: true,
        isNonModal: true,
        isDismissable: false
    }, /*#__PURE__*/ $725Me$react.createElement($725Me$ListBoxBase, {
        ...listBoxProps,
        ref: listBoxRef,
        disallowEmptySelection: true,
        autoFocus: state.focusStrategy,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        shouldUseVirtualFocus: true,
        isLoading: loadingState === 'loadingMore',
        onLoadMore: onLoadMore,
        renderEmptyState: ()=>isAsync && /*#__PURE__*/ $725Me$react.createElement("span", {
                className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($7a1753adda45f7fb$exports)), 'no-results')
            }, loadingState === 'loading' ? formatMessage('loading') : formatMessage('noResults'))
    }), /*#__PURE__*/ $725Me$react.createElement($725Me$DismissButton, {
        onDismiss: ()=>state.close()
    }))));
});
const $2289f375e813f187$var$ComboBoxInput = /*#__PURE__*/ $725Me$react.forwardRef(function ComboBoxInput(props, ref) {
    let { isQuiet: isQuiet , isDisabled: isDisabled , validationState: validationState , inputProps: inputProps , inputRef: inputRef , triggerProps: triggerProps , triggerRef: triggerRef , autoFocus: autoFocus , style: style , className: className , loadingState: loadingState , isOpen: isOpen , menuTrigger: menuTrigger  } = props;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $725Me$useHover({
    });
    let formatMessage = $725Me$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($9c69665f9f0b71fd$exports)));
    let timeout = $725Me$useRef(null);
    let [showLoading, setShowLoading] = $725Me$useState(false);
    let loadingCircle = /*#__PURE__*/ $725Me$react.createElement($725Me$ProgressCircle, {
        "aria-label": formatMessage('loading'),
        size: "S",
        isIndeterminate: true,
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($c14528ddc146317f$exports)), 'spectrum-Textfield-circleLoader', $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-input-circleLoader'))
    });
    let isLoading = loadingState === 'loading' || loadingState === 'filtering';
    let inputValue = inputProps.value;
    let lastInputValue = $725Me$useRef(inputValue);
    $725Me$useEffect(()=>{
        if (isLoading && !showLoading) {
            if (timeout.current === null) timeout.current = setTimeout(()=>{
                setShowLoading(true);
            }, 500);
            // If user is typing, clear the timer and restart since it is a new request
            if (inputValue !== lastInputValue.current) {
                clearTimeout(timeout.current);
                timeout.current = setTimeout(()=>{
                    setShowLoading(true);
                }, 500);
            }
        } else if (!isLoading) {
            // If loading is no longer happening, clear any timers and hide the loading circle
            setShowLoading(false);
            clearTimeout(timeout.current);
            timeout.current = null;
        }
        lastInputValue.current = inputValue;
    }, [
        isLoading,
        showLoading,
        inputValue
    ]);
    return(/*#__PURE__*/ $725Me$react.createElement($725Me$FocusRing, {
        within: true,
        isTextInput: true,
        focusClass: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'is-focused'),
        focusRingClass: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'focus-ring'),
        autoFocus: autoFocus
    }, /*#__PURE__*/ $725Me$react.createElement("div", {
        ...hoverProps,
        ref: ref,
        style: style,
        className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup', {
            'spectrum-InputGroup--quiet': isQuiet,
            'is-disabled': isDisabled,
            'spectrum-InputGroup--invalid': validationState === 'invalid',
            'is-hovered': isHovered
        }, className)
    }, /*#__PURE__*/ $725Me$react.createElement($725Me$TextFieldBase, {
        inputProps: inputProps,
        inputRef: inputRef,
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-field'),
        inputClassName: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-input'),
        validationIconClassName: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-input-validationIcon'),
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: validationState,
        // loading circle should only be displayed if menu is open, if menuTrigger is "manual", or first time load (to stop circle from showing up when user selects an option)
        // TODO: add special case for completionMode: complete as well
        isLoading: showLoading && (isOpen || menuTrigger === 'manual' || loadingState === 'loading'),
        loadingIndicator: loadingState != null && loadingCircle
    }), /*#__PURE__*/ $725Me$react.createElement($725Me$PressResponder, {
        preventFocusOnPress: true,
        isPressed: isOpen
    }, /*#__PURE__*/ $725Me$react.createElement($725Me$FieldButton, {
        ...triggerProps,
        ref: triggerRef,
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: validationState
    }, /*#__PURE__*/ $725Me$react.createElement($725Me$spectrumiconsuiChevronDownMedium, {
        UNSAFE_className: $725Me$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-Dropdown-chevron')
    }))))));
});
/**
 * ComboBoxes combine a text entry with a picker menu, allowing users to filter longer lists to only the selections matching a query.
 */ const $2289f375e813f187$export$72b9695b8216309a = /*#__PURE__*/ $725Me$react.forwardRef($2289f375e813f187$var$ComboBox);





export {$6ea56810956c2335$re_export$Item as Item, $6ea56810956c2335$re_export$Section as Section, $2289f375e813f187$export$72b9695b8216309a as ComboBox};
