// @react-spectrum/text/dist/main.js
var $kLXM6$reactariautils = require("@react-aria/utils");
var $kLXM6$react = require("react");
var $kLXM6$reactspectrumutils = require("@react-spectrum/utils");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Text", () => $1efae25dd196d679$export$5f1af8db9871e1d6);
$parcel$export(module.exports, "Heading", () => $c7c750c966e80a46$export$a8a3e93435678ff9);
$parcel$export(module.exports, "Keyboard", () => $961f7ef9e414f5f0$export$16e4d70cc375e707);



function $1efae25dd196d679$var$Text(props, ref) {
    props = $kLXM6$reactspectrumutils.useSlotProps(props, 'text');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $kLXM6$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $kLXM6$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($kLXM6$react)).createElement("span", {
        ...$kLXM6$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, children));
}
/**
 * Text represents text with no specific semantic meaning.
 */ const $1efae25dd196d679$export$5f1af8db9871e1d6 = /*#__PURE__*/ $kLXM6$react.forwardRef($1efae25dd196d679$var$Text);





function $c7c750c966e80a46$var$Heading(props, ref) {
    props = $kLXM6$reactspectrumutils.useSlotProps(props, 'heading');
    let { children: children , level: level = 3 , ...otherProps } = props;
    let { styleProps: styleProps  } = $kLXM6$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $kLXM6$reactspectrumutils.useDOMRef(ref);
    let HeadingTag = `h${level}`;
    return(/*#__PURE__*/ ($parcel$interopDefault($kLXM6$react)).createElement(HeadingTag, {
        ...$kLXM6$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, children));
}
/**
 * Heading is used to create various levels of typographic hierarchies.
 */ const $c7c750c966e80a46$export$a8a3e93435678ff9 = /*#__PURE__*/ $kLXM6$react.forwardRef($c7c750c966e80a46$var$Heading);





function $961f7ef9e414f5f0$var$Keyboard(props, ref) {
    props = $kLXM6$reactspectrumutils.useSlotProps(props, 'keyboard');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $kLXM6$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $kLXM6$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($kLXM6$react)).createElement("kbd", {
        ...$kLXM6$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        dir: "ltr",
        ref: domRef
    }, children));
}
/**
 * Keyboard represents text that specifies a keyboard command.
 */ const $961f7ef9e414f5f0$export$16e4d70cc375e707 = /*#__PURE__*/ $kLXM6$react.forwardRef($961f7ef9e414f5f0$var$Keyboard);
