// @react-spectrum/text/dist/module.js
import {filterDOMProps as $3GX9J$filterDOMProps} from "@react-aria/utils";
import $3GX9J$react, {forwardRef as $3GX9J$forwardRef} from "react";
import {useSlotProps as $3GX9J$useSlotProps, useStyleProps as $3GX9J$useStyleProps, useDOMRef as $3GX9J$useDOMRef} from "@react-spectrum/utils";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $cd449e8defa988f0$exports = {};

$parcel$export($cd449e8defa988f0$exports, "Text", () => $cd449e8defa988f0$export$5f1af8db9871e1d6);



function $cd449e8defa988f0$var$Text(props, ref) {
    props = $3GX9J$useSlotProps(props, 'text');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $3GX9J$useStyleProps(otherProps);
    let domRef = $3GX9J$useDOMRef(ref);
    return(/*#__PURE__*/ $3GX9J$react.createElement("span", {
        ...$3GX9J$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, children));
}
/**
 * Text represents text with no specific semantic meaning.
 */ const $cd449e8defa988f0$export$5f1af8db9871e1d6 = /*#__PURE__*/ $3GX9J$forwardRef($cd449e8defa988f0$var$Text);


var $c382fceb69609a50$exports = {};

$parcel$export($c382fceb69609a50$exports, "Heading", () => $c382fceb69609a50$export$a8a3e93435678ff9);



function $c382fceb69609a50$var$Heading(props, ref) {
    props = $3GX9J$useSlotProps(props, 'heading');
    let { children: children , level: level = 3 , ...otherProps } = props;
    let { styleProps: styleProps  } = $3GX9J$useStyleProps(otherProps);
    let domRef = $3GX9J$useDOMRef(ref);
    let HeadingTag = `h${level}`;
    return(/*#__PURE__*/ $3GX9J$react.createElement(HeadingTag, {
        ...$3GX9J$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, children));
}
/**
 * Heading is used to create various levels of typographic hierarchies.
 */ const $c382fceb69609a50$export$a8a3e93435678ff9 = /*#__PURE__*/ $3GX9J$forwardRef($c382fceb69609a50$var$Heading);


var $a42b6b1607b36926$exports = {};

$parcel$export($a42b6b1607b36926$exports, "Keyboard", () => $a42b6b1607b36926$export$16e4d70cc375e707);



function $a42b6b1607b36926$var$Keyboard(props, ref) {
    props = $3GX9J$useSlotProps(props, 'keyboard');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $3GX9J$useStyleProps(otherProps);
    let domRef = $3GX9J$useDOMRef(ref);
    return(/*#__PURE__*/ $3GX9J$react.createElement("kbd", {
        ...$3GX9J$filterDOMProps(otherProps),
        ...styleProps,
        dir: "ltr",
        ref: domRef
    }, children));
}
/**
 * Keyboard represents text that specifies a keyboard command.
 */ const $a42b6b1607b36926$export$16e4d70cc375e707 = /*#__PURE__*/ $3GX9J$forwardRef($a42b6b1607b36926$var$Keyboard);




export {$cd449e8defa988f0$export$5f1af8db9871e1d6 as Text, $c382fceb69609a50$export$a8a3e93435678ff9 as Heading, $a42b6b1607b36926$export$16e4d70cc375e707 as Keyboard};
