// @react-spectrum/dialog/dist/main.js
require("./main.css");
var $iaoFN$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $iaoFN$reactspectrumbutton = require("@react-spectrum/button");
var $iaoFN$reactspectrumbuttongroup = require("@react-spectrum/buttongroup");
var $iaoFN$reactariautils = require("@react-aria/utils");
var $iaoFN$reactspectrumutils = require("@react-spectrum/utils");
var $iaoFN$reactspectrumview = require("@react-spectrum/view");
var $iaoFN$reactspectrumdivider = require("@react-spectrum/divider");
var $iaoFN$reactspectrumtext = require("@react-spectrum/text");
var $iaoFN$react = require("react");
var $iaoFN$reactariai18n = require("@react-aria/i18n");
var $iaoFN$spectrumiconsuiCrossLarge = require("@spectrum-icons/ui/CrossLarge");
var $iaoFN$reactariaoverlays = require("@react-aria/overlays");
var $iaoFN$reactariafocus = require("@react-aria/focus");
var $iaoFN$reactspectrumlayout = require("@react-spectrum/layout");
var $iaoFN$reactariadialog = require("@react-aria/dialog");
var $iaoFN$reactspectrumoverlays = require("@react-spectrum/overlays");
var $iaoFN$reactstatelyoverlays = require("@react-stately/overlays");
var $iaoFN$reactariainteractions = require("@react-aria/interactions");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $66f8653ff750aa27$exports = {};

$parcel$export($66f8653ff750aa27$exports, "AlertDialog", () => $66f8653ff750aa27$export$de466dd8317b0b75);






var $a57fc2323bf98a33$exports = {};

$parcel$export($a57fc2323bf98a33$exports, "Dialog", () => $a57fc2323bf98a33$export$3ddf2d174ce01153);




const $0eb909ad2388d989$export$8b93a07348a7730c = ($parcel$interopDefault($iaoFN$react)).createContext(null);





var $27a5e9bc18eefb32$exports = {};
var $a55551a02f9dcb5c$exports = {};
$a55551a02f9dcb5c$exports = JSON.parse("{\"alert\":\"تنبيه\",\"dismiss\":\"تجاهل\"}");


var $d9c7f58ad2bbfa4c$exports = {};
$d9c7f58ad2bbfa4c$exports = JSON.parse("{\"alert\":\"Сигнал\",\"dismiss\":\"Отхвърляне\"}");


var $cc8c40dafb8d03b0$exports = {};
$cc8c40dafb8d03b0$exports = JSON.parse("{\"alert\":\"Výstraha\",\"dismiss\":\"Odstranit\"}");


var $d970ec0fa7e711e7$exports = {};
$d970ec0fa7e711e7$exports = JSON.parse("{\"alert\":\"Advarsel\",\"dismiss\":\"Luk\"}");


var $c619cc654d2ca674$exports = {};
$c619cc654d2ca674$exports = JSON.parse("{\"alert\":\"Warnhinweis\",\"dismiss\":\"Schließen\"}");


var $2d258a772c4fa392$exports = {};
$2d258a772c4fa392$exports = JSON.parse("{\"alert\":\"Ειδοποίηση\",\"dismiss\":\"Απόρριψη\"}");


var $88f649eb626e5bda$exports = {};
$88f649eb626e5bda$exports = JSON.parse("{\"dismiss\":\"Dismiss\",\"alert\":\"Alert\"}");


var $4cd7244dbd86e2ba$exports = {};
$4cd7244dbd86e2ba$exports = JSON.parse("{\"alert\":\"Alerta\",\"dismiss\":\"Descartar\"}");


var $86a88b5d274957dd$exports = {};
$86a88b5d274957dd$exports = JSON.parse("{\"alert\":\"Teade\",\"dismiss\":\"Lõpeta\"}");


var $6b6909c1e7df2772$exports = {};
$6b6909c1e7df2772$exports = JSON.parse("{\"alert\":\"Hälytys\",\"dismiss\":\"Hylkää\"}");


var $c05d64ee0d510606$exports = {};
$c05d64ee0d510606$exports = JSON.parse("{\"alert\":\"Alerte\",\"dismiss\":\"Rejeter\"}");


var $b54ffde1ebba1e69$exports = {};
$b54ffde1ebba1e69$exports = JSON.parse("{\"alert\":\"התראה\",\"dismiss\":\"התעלם\"}");


var $6d5732627fd95466$exports = {};
$6d5732627fd95466$exports = JSON.parse("{\"alert\":\"Upozorenje\",\"dismiss\":\"Odbaci\"}");


var $525291eaaaf36ab1$exports = {};
$525291eaaaf36ab1$exports = JSON.parse("{\"alert\":\"Figyelmeztetés\",\"dismiss\":\"Elutasítás\"}");


var $6dc44201330bcb0a$exports = {};
$6dc44201330bcb0a$exports = JSON.parse("{\"alert\":\"Avviso\",\"dismiss\":\"Ignora\"}");


var $38fba7087c48d89b$exports = {};
$38fba7087c48d89b$exports = JSON.parse("{\"alert\":\"アラート\",\"dismiss\":\"閉じる\"}");


var $0d321e557ee0ff2b$exports = {};
$0d321e557ee0ff2b$exports = JSON.parse("{\"alert\":\"경고\",\"dismiss\":\"무시\"}");


var $81ef7c03f6531c38$exports = {};
$81ef7c03f6531c38$exports = JSON.parse("{\"alert\":\"Įspėjimas\",\"dismiss\":\"Atmesti\"}");


var $dab9caf08f2773b2$exports = {};
$dab9caf08f2773b2$exports = JSON.parse("{\"alert\":\"Brīdinājums\",\"dismiss\":\"Nerādīt\"}");


var $8af697b2f49b994d$exports = {};
$8af697b2f49b994d$exports = JSON.parse("{\"alert\":\"Varsel\",\"dismiss\":\"Lukk\"}");


var $94b03262a47d5c5b$exports = {};
$94b03262a47d5c5b$exports = JSON.parse("{\"alert\":\"Melding\",\"dismiss\":\"Negeren\"}");


var $9531f14fa20d8709$exports = {};
$9531f14fa20d8709$exports = JSON.parse("{\"alert\":\"Ostrzeżenie\",\"dismiss\":\"Zignoruj\"}");


var $d9e8b49831ccbf2a$exports = {};
$d9e8b49831ccbf2a$exports = JSON.parse("{\"alert\":\"Alerta\",\"dismiss\":\"Descartar\"}");


var $6a11f56de734f9d5$exports = {};
$6a11f56de734f9d5$exports = JSON.parse("{\"alert\":\"Alerta\",\"dismiss\":\"Dispensar\"}");


var $e6f020c13f71513f$exports = {};
$e6f020c13f71513f$exports = JSON.parse("{\"alert\":\"Alertă\",\"dismiss\":\"Revocare\"}");


var $3268a43e923b3cbf$exports = {};
$3268a43e923b3cbf$exports = JSON.parse("{\"alert\":\"Предупреждение\",\"dismiss\":\"Пропустить\"}");


var $ff45268c9e1a78be$exports = {};
$ff45268c9e1a78be$exports = JSON.parse("{\"alert\":\"Upozornenie\",\"dismiss\":\"Zrušiť\"}");


var $f2cb6f88707095ee$exports = {};
$f2cb6f88707095ee$exports = JSON.parse("{\"alert\":\"Opozorilo\",\"dismiss\":\"Opusti\"}");


var $ece68e97a81612d8$exports = {};
$ece68e97a81612d8$exports = JSON.parse("{\"alert\":\"Upozorenje\",\"dismiss\":\"Odbaci\"}");


var $cce4ae8855cd8246$exports = {};
$cce4ae8855cd8246$exports = JSON.parse("{\"alert\":\"Varning\",\"dismiss\":\"Avvisa\"}");


var $763283ab7e680652$exports = {};
$763283ab7e680652$exports = JSON.parse("{\"alert\":\"Uyarı\",\"dismiss\":\"Kapat\"}");


var $ec9a4b7a09770ea4$exports = {};
$ec9a4b7a09770ea4$exports = JSON.parse("{\"alert\":\"Сигнал тривоги\",\"dismiss\":\"Скасувати\"}");


var $56f81c88b16f2b96$exports = {};
$56f81c88b16f2b96$exports = JSON.parse("{\"alert\":\"警报\",\"dismiss\":\"取消\"}");


var $6fb708b0a8fb135e$exports = {};
$6fb708b0a8fb135e$exports = JSON.parse("{\"alert\":\"警示\",\"dismiss\":\"關閉\"}");


$27a5e9bc18eefb32$exports = {
    "ar-AE": $a55551a02f9dcb5c$exports,
    "bg-BG": $d9c7f58ad2bbfa4c$exports,
    "cs-CZ": $cc8c40dafb8d03b0$exports,
    "da-DK": $d970ec0fa7e711e7$exports,
    "de-DE": $c619cc654d2ca674$exports,
    "el-GR": $2d258a772c4fa392$exports,
    "en-US": $88f649eb626e5bda$exports,
    "es-ES": $4cd7244dbd86e2ba$exports,
    "et-EE": $86a88b5d274957dd$exports,
    "fi-FI": $6b6909c1e7df2772$exports,
    "fr-FR": $c05d64ee0d510606$exports,
    "he-IL": $b54ffde1ebba1e69$exports,
    "hr-HR": $6d5732627fd95466$exports,
    "hu-HU": $525291eaaaf36ab1$exports,
    "it-IT": $6dc44201330bcb0a$exports,
    "ja-JP": $38fba7087c48d89b$exports,
    "ko-KR": $0d321e557ee0ff2b$exports,
    "lt-LT": $81ef7c03f6531c38$exports,
    "lv-LV": $dab9caf08f2773b2$exports,
    "nb-NO": $8af697b2f49b994d$exports,
    "nl-NL": $94b03262a47d5c5b$exports,
    "pl-PL": $9531f14fa20d8709$exports,
    "pt-BR": $d9e8b49831ccbf2a$exports,
    "pt-PT": $6a11f56de734f9d5$exports,
    "ro-RO": $e6f020c13f71513f$exports,
    "ru-RU": $3268a43e923b3cbf$exports,
    "sk-SK": $ff45268c9e1a78be$exports,
    "sl-SI": $f2cb6f88707095ee$exports,
    "sr-SP": $ece68e97a81612d8$exports,
    "sv-SE": $cce4ae8855cd8246$exports,
    "tr-TR": $763283ab7e680652$exports,
    "uk-UA": $ec9a4b7a09770ea4$exports,
    "zh-CN": $56f81c88b16f2b96$exports,
    "zh-TW": $6fb708b0a8fb135e$exports
};




var $a74c0984b1adb651$exports = {};

$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog", () => $a74c0984b1adb651$export$4681f3bcac39b2b1, (v) => $a74c0984b1adb651$export$4681f3bcac39b2b1 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--small", () => $a74c0984b1adb651$export$5f2d9cc47409bd63, (v) => $a74c0984b1adb651$export$5f2d9cc47409bd63 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--medium", () => $a74c0984b1adb651$export$b9b8ef443d0a67ca, (v) => $a74c0984b1adb651$export$b9b8ef443d0a67ca = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--large", () => $a74c0984b1adb651$export$b73c0c2fca84be2b, (v) => $a74c0984b1adb651$export$b73c0c2fca84be2b = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-hero", () => $a74c0984b1adb651$export$d405fb0a0231aa1a, (v) => $a74c0984b1adb651$export$d405fb0a0231aa1a = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-grid", () => $a74c0984b1adb651$export$c03cd5e23af67ff5, (v) => $a74c0984b1adb651$export$c03cd5e23af67ff5 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-heading", () => $a74c0984b1adb651$export$94335d73b7b209ec, (v) => $a74c0984b1adb651$export$94335d73b7b209ec = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-heading--noHeader", () => $a74c0984b1adb651$export$ade8307239736e89, (v) => $a74c0984b1adb651$export$ade8307239736e89 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-header", () => $a74c0984b1adb651$export$1e158eaa6363e131, (v) => $a74c0984b1adb651$export$1e158eaa6363e131 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-typeIcon", () => $a74c0984b1adb651$export$5beee9480fc6ff07, (v) => $a74c0984b1adb651$export$5beee9480fc6ff07 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-divider", () => $a74c0984b1adb651$export$dd7633d642d0bc1c, (v) => $a74c0984b1adb651$export$dd7633d642d0bc1c = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--noDivider", () => $a74c0984b1adb651$export$22425527a8943917, (v) => $a74c0984b1adb651$export$22425527a8943917 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-content", () => $a74c0984b1adb651$export$cb8eccd0f3639238, (v) => $a74c0984b1adb651$export$cb8eccd0f3639238 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-footer", () => $a74c0984b1adb651$export$3ecb5025a06fe08f, (v) => $a74c0984b1adb651$export$3ecb5025a06fe08f = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Button", () => $a74c0984b1adb651$export$1db4cca5b4ade39a, (v) => $a74c0984b1adb651$export$1db4cca5b4ade39a = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-buttonGroup", () => $a74c0984b1adb651$export$570ae852861555b7, (v) => $a74c0984b1adb651$export$570ae852861555b7 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-buttonGroup--noFooter", () => $a74c0984b1adb651$export$39dd75369bf2e830, (v) => $a74c0984b1adb651$export$39dd75369bf2e830 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--dismissable", () => $a74c0984b1adb651$export$9461b285d631d575, (v) => $a74c0984b1adb651$export$9461b285d631d575 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-closeButton", () => $a74c0984b1adb651$export$a8494287d763a75d, (v) => $a74c0984b1adb651$export$a8494287d763a75d = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--error", () => $a74c0984b1adb651$export$6b5beff56972a3b, (v) => $a74c0984b1adb651$export$6b5beff56972a3b = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--fullscreen", () => $a74c0984b1adb651$export$118506da5d3ee7c4, (v) => $a74c0984b1adb651$export$118506da5d3ee7c4 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--fullscreenTakeover", () => $a74c0984b1adb651$export$5c061b2d3664fda9, (v) => $a74c0984b1adb651$export$5c061b2d3664fda9 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--warning", () => $a74c0984b1adb651$export$4e66396adf61534f, (v) => $a74c0984b1adb651$export$4e66396adf61534f = v);
var $a74c0984b1adb651$export$4681f3bcac39b2b1;
var $a74c0984b1adb651$export$5f2d9cc47409bd63;
var $a74c0984b1adb651$export$b9b8ef443d0a67ca;
var $a74c0984b1adb651$export$b73c0c2fca84be2b;
var $a74c0984b1adb651$export$d405fb0a0231aa1a;
var $a74c0984b1adb651$export$c03cd5e23af67ff5;
var $a74c0984b1adb651$export$94335d73b7b209ec;
var $a74c0984b1adb651$export$ade8307239736e89;
var $a74c0984b1adb651$export$1e158eaa6363e131;
var $a74c0984b1adb651$export$5beee9480fc6ff07;
var $a74c0984b1adb651$export$dd7633d642d0bc1c;
var $a74c0984b1adb651$export$22425527a8943917;
var $a74c0984b1adb651$export$cb8eccd0f3639238;
var $a74c0984b1adb651$export$3ecb5025a06fe08f;
var $a74c0984b1adb651$export$1db4cca5b4ade39a;
var $a74c0984b1adb651$export$570ae852861555b7;
var $a74c0984b1adb651$export$39dd75369bf2e830;
var $a74c0984b1adb651$export$9461b285d631d575;
var $a74c0984b1adb651$export$a8494287d763a75d;
var $a74c0984b1adb651$export$6b5beff56972a3b;
var $a74c0984b1adb651$export$118506da5d3ee7c4;
var $a74c0984b1adb651$export$5c061b2d3664fda9;
var $a74c0984b1adb651$export$4e66396adf61534f;
$a74c0984b1adb651$export$4681f3bcac39b2b1 = "spectrum-Dialog_6d8b48";
$a74c0984b1adb651$export$5f2d9cc47409bd63 = "spectrum-Dialog--small_6d8b48";
$a74c0984b1adb651$export$b9b8ef443d0a67ca = "spectrum-Dialog--medium_6d8b48";
$a74c0984b1adb651$export$b73c0c2fca84be2b = "spectrum-Dialog--large_6d8b48";
$a74c0984b1adb651$export$d405fb0a0231aa1a = "spectrum-Dialog-hero_6d8b48";
$a74c0984b1adb651$export$c03cd5e23af67ff5 = "spectrum-Dialog-grid_6d8b48";
$a74c0984b1adb651$export$94335d73b7b209ec = "spectrum-Dialog-heading_6d8b48";
$a74c0984b1adb651$export$ade8307239736e89 = "spectrum-Dialog-heading--noHeader_6d8b48";
$a74c0984b1adb651$export$1e158eaa6363e131 = "spectrum-Dialog-header_6d8b48";
$a74c0984b1adb651$export$5beee9480fc6ff07 = "spectrum-Dialog-typeIcon_6d8b48";
$a74c0984b1adb651$export$dd7633d642d0bc1c = "spectrum-Dialog-divider_6d8b48";
$a74c0984b1adb651$export$22425527a8943917 = "spectrum-Dialog--noDivider_6d8b48";
$a74c0984b1adb651$export$cb8eccd0f3639238 = "spectrum-Dialog-content_6d8b48";
$a74c0984b1adb651$export$3ecb5025a06fe08f = "spectrum-Dialog-footer_6d8b48";
$a74c0984b1adb651$export$1db4cca5b4ade39a = "spectrum-Button_6d8b48";
$a74c0984b1adb651$export$570ae852861555b7 = "spectrum-Dialog-buttonGroup_6d8b48";
$a74c0984b1adb651$export$39dd75369bf2e830 = "spectrum-Dialog-buttonGroup--noFooter_6d8b48";
$a74c0984b1adb651$export$9461b285d631d575 = "spectrum-Dialog--dismissable_6d8b48";
$a74c0984b1adb651$export$a8494287d763a75d = "spectrum-Dialog-closeButton_6d8b48";
$a74c0984b1adb651$export$6b5beff56972a3b = "spectrum-Dialog--error_6d8b48";
$a74c0984b1adb651$export$118506da5d3ee7c4 = "spectrum-Dialog--fullscreen_6d8b48";
$a74c0984b1adb651$export$5c061b2d3664fda9 = "spectrum-Dialog--fullscreenTakeover_6d8b48";
$a74c0984b1adb651$export$4e66396adf61534f = "spectrum-Dialog--warning_6d8b48";




let $a57fc2323bf98a33$var$sizeMap = {
    S: 'small',
    M: 'medium',
    L: 'large',
    fullscreen: 'fullscreen',
    fullscreenTakeover: 'fullscreenTakeover'
};
function $a57fc2323bf98a33$var$Dialog(props, ref) {
    let { type: type = 'modal' , ...contextProps } = $iaoFN$react.useContext($0eb909ad2388d989$export$8b93a07348a7730c) || {
    };
    let { children: children , isDismissable: isDismissable = contextProps.isDismissable , onDismiss: onDismiss = contextProps.onClose , size: size , ...otherProps } = props;
    let formatMessage = $iaoFN$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($27a5e9bc18eefb32$exports)));
    let { styleProps: styleProps  } = $iaoFN$reactspectrumutils.useStyleProps(otherProps);
    size = type === 'popover' ? size || 'S' : size || 'L';
    let domRef = $iaoFN$reactspectrumutils.useDOMRef(ref);
    let gridRef = $iaoFN$react.useRef();
    let sizeVariant = $a57fc2323bf98a33$var$sizeMap[type] || $a57fc2323bf98a33$var$sizeMap[size];
    let { dialogProps: dialogProps , titleProps: titleProps  } = $iaoFN$reactariadialog.useDialog($iaoFN$reactariautils.mergeProps(contextProps, props), domRef);
    let hasHeader = $iaoFN$reactspectrumutils.useHasChild(`.${(/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))['spectrum-Dialog-header']}`, $iaoFN$reactspectrumutils.unwrapDOMRef(gridRef));
    let hasFooter = $iaoFN$reactspectrumutils.useHasChild(`.${(/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))['spectrum-Dialog-footer']}`, $iaoFN$reactspectrumutils.unwrapDOMRef(gridRef));
    let slots = $iaoFN$react.useMemo(()=>({
            hero: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))['spectrum-Dialog-hero']
            },
            header: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))['spectrum-Dialog-header']
            },
            heading: {
                UNSAFE_className: $iaoFN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)), 'spectrum-Dialog-heading', {
                    'spectrum-Dialog-heading--noHeader': !hasHeader
                }),
                level: 2,
                ...titleProps
            },
            typeIcon: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))['spectrum-Dialog-typeIcon']
            },
            divider: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))['spectrum-Dialog-divider'],
                size: 'M'
            },
            content: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))['spectrum-Dialog-content']
            },
            footer: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))['spectrum-Dialog-footer']
            },
            buttonGroup: {
                UNSAFE_className: $iaoFN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)), 'spectrum-Dialog-buttonGroup', {
                    'spectrum-Dialog-buttonGroup--noFooter': !hasFooter
                }),
                align: 'end'
            }
        })
    , [
        (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)),
        hasFooter,
        hasHeader,
        titleProps
    ]);
    // If rendered in a popover or tray there won't be a visible dismiss button,
    // so we render a hidden one for screen readers.
    let dismissButton;
    if (type === 'popover' || type === 'tray') dismissButton = /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactariaoverlays.DismissButton, {
        onDismiss: onDismiss
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactariafocus.FocusScope, {
        contain: true,
        restoreFocus: true
    }, /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement("section", {
        ...styleProps,
        ...dialogProps,
        className: $iaoFN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)), 'spectrum-Dialog', {
            [`spectrum-Dialog--${sizeVariant}`]: sizeVariant,
            'spectrum-Dialog--dismissable': isDismissable
        }, styleProps.className),
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumlayout.Grid, {
        ref: gridRef,
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))['spectrum-Dialog-grid']
    }, /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumutils.SlotProvider, {
        slots: slots
    }, children), isDismissable && /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumbutton.ActionButton, {
        UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))['spectrum-Dialog-closeButton'],
        isQuiet: true,
        "aria-label": formatMessage('dismiss'),
        onPress: onDismiss
    }, /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement(($parcel$interopDefault($iaoFN$spectrumiconsuiCrossLarge)), null))), dismissButton)));
}
/**
 * Dialogs are windows containing contextual information, tasks, or workflows that appear over the user interface.
 * Depending on the kind of Dialog, further interactions may be blocked until the Dialog is acknowledged.
 */ let $a57fc2323bf98a33$export$3ddf2d174ce01153 = /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).forwardRef($a57fc2323bf98a33$var$Dialog);









/**
 * AlertDialogs are a specific type of Dialog. They display important information that users need to acknowledge.
 */ function $66f8653ff750aa27$var$AlertDialog(props, ref) {
    let { onClose: onClose = ()=>{
    }  } = $iaoFN$react.useContext($0eb909ad2388d989$export$8b93a07348a7730c) || {
    };
    let { variant: variant , children: children , primaryActionLabel: primaryActionLabel , secondaryActionLabel: secondaryActionLabel , cancelLabel: cancelLabel , autoFocusButton: autoFocusButton , title: title , isPrimaryActionDisabled: isPrimaryActionDisabled , isSecondaryActionDisabled: isSecondaryActionDisabled , onCancel: onCancel = ()=>{
    } , onPrimaryAction: onPrimaryAction = ()=>{
    } , onSecondaryAction: onSecondaryAction = ()=>{
    } , ...otherProps } = props;
    let { styleProps: styleProps  } = $iaoFN$reactspectrumutils.useStyleProps(otherProps);
    let formatMessage = $iaoFN$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($27a5e9bc18eefb32$exports)));
    let confirmVariant = 'primary';
    if (variant) {
        if (variant === 'confirmation') confirmVariant = 'cta';
        else if (variant === 'destructive') confirmVariant = 'negative';
    }
    return(/*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($a57fc2323bf98a33$export$3ddf2d174ce01153, {
        UNSAFE_style: styleProps.style,
        UNSAFE_className: $iaoFN$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)), {
            [`spectrum-Dialog--${variant}`]: variant
        }, styleProps.className),
        isHidden: styleProps.hidden,
        size: "M",
        role: "alertdialog",
        ref: ref
    }, /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumtext.Heading, null, title), (variant === 'error' || variant === 'warning') && /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement(($parcel$interopDefault($iaoFN$spectrumiconsuiAlertMedium)), {
        slot: "typeIcon",
        "aria-label": formatMessage('alert')
    }), /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumdivider.Divider, null), /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumview.Content, null, children), /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumbuttongroup.ButtonGroup, {
        align: "end"
    }, cancelLabel && /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumbutton.Button, {
        variant: "secondary",
        onPress: ()=>$iaoFN$reactariautils.chain(onClose(), onCancel())
        ,
        autoFocus: autoFocusButton === 'cancel'
    }, cancelLabel), secondaryActionLabel && /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumbutton.Button, {
        variant: "secondary",
        onPress: ()=>$iaoFN$reactariautils.chain(onClose(), onSecondaryAction())
        ,
        isDisabled: isSecondaryActionDisabled,
        autoFocus: autoFocusButton === 'secondary'
    }, secondaryActionLabel), /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumbutton.Button, {
        variant: confirmVariant,
        onPress: ()=>$iaoFN$reactariautils.chain(onClose(), onPrimaryAction())
        ,
        isDisabled: isPrimaryActionDisabled,
        autoFocus: autoFocusButton === 'primary'
    }, primaryActionLabel))));
}
/**
 * AlertDialogs are a specific type of Dialog. They display important information that users need to acknowledge.
 */ let $66f8653ff750aa27$export$de466dd8317b0b75 = /*#__PURE__*/ $iaoFN$react.forwardRef($66f8653ff750aa27$var$AlertDialog);



var $46e54a192dc519c9$exports = {};

$parcel$export($46e54a192dc519c9$exports, "DialogTrigger", () => $46e54a192dc519c9$export$2e1e1122cf0cba88);







function $46e54a192dc519c9$var$DialogTrigger(props) {
    let { children: children , type: type = 'modal' , mobileType: mobileType = type === 'popover' ? 'modal' : type , hideArrow: hideArrow , targetRef: targetRef , isDismissable: isDismissable , isKeyboardDismissDisabled: isKeyboardDismissDisabled , ...positionProps } = props;
    if (!Array.isArray(children) || children.length > 2) throw new Error('DialogTrigger must have exactly 2 children');
    // if a function is passed as the second child, it won't appear in toArray
    let [trigger, content] = children;
    // On small devices, show a modal or tray instead of a popover.
    // TODO: DNA variable?
    let isMobile = $iaoFN$reactspectrumutils.useMediaQuery('(max-width: 700px)');
    if (isMobile) {
        // handle cases where desktop popovers need a close button for the mobile modal view
        if (type !== 'modal' && mobileType === 'modal') isDismissable = true;
        type = mobileType;
    }
    let state = $iaoFN$reactstatelyoverlays.useOverlayTriggerState(props);
    let wasOpen = $iaoFN$react.useRef(false);
    wasOpen.current = state.isOpen;
    let isExiting = $iaoFN$react.useRef(false);
    let onExiting = ()=>isExiting.current = true
    ;
    let onExited = ()=>isExiting.current = false
    ;
    // eslint-disable-next-line arrow-body-style
    $iaoFN$react.useEffect(()=>{
        return ()=>{
            if ((wasOpen.current || isExiting.current) && type !== 'popover' && type !== 'tray') console.warn('A DialogTrigger unmounted while open. This is likely due to being placed within a trigger that unmounts or inside a conditional. Consider using a DialogContainer instead.');
        };
    }, []);
    if (type === 'popover') return(/*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($46e54a192dc519c9$var$PopoverTrigger, {
        ...positionProps,
        state: state,
        targetRef: targetRef,
        trigger: trigger,
        content: content,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled,
        hideArrow: hideArrow
    }));
    let renderOverlay = ()=>{
        switch(type){
            case 'fullscreen':
            case 'fullscreenTakeover':
            case 'modal':
                return(/*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumoverlays.Modal, {
                    isOpen: state.isOpen,
                    isDismissable: type === 'modal' ? isDismissable : false,
                    onClose: state.close,
                    type: type,
                    isKeyboardDismissDisabled: isKeyboardDismissDisabled,
                    onExiting: onExiting,
                    onExited: onExited
                }, typeof content === 'function' ? content(state.close) : content));
            case 'tray':
                return(/*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumoverlays.Tray, {
                    isOpen: state.isOpen,
                    onClose: state.close,
                    isKeyboardDismissDisabled: isKeyboardDismissDisabled
                }, typeof content === 'function' ? content(state.close) : content));
        }
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($46e54a192dc519c9$var$DialogTriggerBase, {
        type: type,
        state: state,
        isDismissable: isDismissable,
        trigger: trigger,
        overlay: renderOverlay()
    }));
}
// Support DialogTrigger inside components using CollectionBuilder.
$46e54a192dc519c9$var$DialogTrigger.getCollectionNode = function*(props) {
    let [trigger] = ($parcel$interopDefault($iaoFN$react)).Children.toArray(props.children);
    let [, content] = props.children;
    yield {
        element: trigger,
        wrapper: (element)=>/*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($46e54a192dc519c9$var$DialogTrigger, {
                key: element.key,
                ...props
            }, element, content)
    };
};
/**
 * DialogTrigger serves as a wrapper around a Dialog and its associated trigger, linking the Dialog's
 * open state with the trigger's press state. Additionally, it allows you to customize the type and
 * positioning of the Dialog.
 */ // We don't want getCollectionNode to show up in the type definition
let $46e54a192dc519c9$export$2e1e1122cf0cba88 = $46e54a192dc519c9$var$DialogTrigger;
function $46e54a192dc519c9$var$PopoverTrigger({ state: state , targetRef: targetRef , trigger: trigger , content: content , hideArrow: hideArrow , isKeyboardDismissDisabled: isKeyboardDismissDisabled , ...props }) {
    let triggerRef = $iaoFN$react.useRef();
    let overlayRef = $iaoFN$react.useRef();
    let { overlayProps: popoverProps , placement: placement , arrowProps: arrowProps  } = $iaoFN$reactariaoverlays.useOverlayPosition({
        targetRef: targetRef || triggerRef,
        overlayRef: $iaoFN$reactspectrumutils.unwrapDOMRef(overlayRef),
        placement: props.placement,
        containerPadding: props.containerPadding,
        offset: props.offset,
        crossOffset: props.crossOffset,
        shouldFlip: props.shouldFlip,
        isOpen: state.isOpen
    });
    let { triggerProps: triggerProps , overlayProps: overlayProps  } = $iaoFN$reactariaoverlays.useOverlayTrigger({
        type: 'dialog'
    }, state, triggerRef);
    let triggerPropsWithRef = {
        ...triggerProps,
        ref: targetRef ? undefined : triggerRef
    };
    let overlay = /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumoverlays.Popover, {
        isOpen: state.isOpen,
        UNSAFE_style: popoverProps.style,
        ref: overlayRef,
        onClose: state.close,
        placement: placement,
        arrowProps: arrowProps,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled,
        hideArrow: hideArrow
    }, typeof content === 'function' ? content(state.close) : content);
    return(/*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($46e54a192dc519c9$var$DialogTriggerBase, {
        type: "popover",
        state: state,
        triggerProps: triggerPropsWithRef,
        dialogProps: overlayProps,
        trigger: trigger,
        overlay: overlay
    }));
}
function $46e54a192dc519c9$var$DialogTriggerBase({ type: type , state: state , isDismissable: isDismissable , dialogProps: dialogProps = {
} , triggerProps: triggerProps = {
} , overlay: overlay , trigger: trigger  }) {
    let context = {
        type: type,
        onClose: state.close,
        isDismissable: isDismissable,
        ...dialogProps
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$react.Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactariainteractions.PressResponder, {
        ...triggerProps,
        onPress: state.toggle,
        isPressed: state.isOpen && type !== 'modal' && type !== 'fullscreen' && type !== 'fullscreenTakeover'
    }, trigger), /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($0eb909ad2388d989$export$8b93a07348a7730c.Provider, {
        value: context
    }, overlay)));
}


var $1b7050825e0eb5b6$exports = {};

$parcel$export($1b7050825e0eb5b6$exports, "DialogContainer", () => $1b7050825e0eb5b6$export$547754aada6e339f);



function $1b7050825e0eb5b6$export$547754aada6e339f(props) {
    let { children: children , type: type = 'modal' , onDismiss: onDismiss , isDismissable: isDismissable , isKeyboardDismissDisabled: isKeyboardDismissDisabled  } = props;
    let childArray = ($parcel$interopDefault($iaoFN$react)).Children.toArray(children);
    if (childArray.length > 1) throw new Error('Only a single child can be passed to DialogContainer.');
    let lastChild = $iaoFN$react.useRef(null);
    let child = /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).isValidElement(childArray[0]) ? childArray[0] : null;
    if (child) lastChild.current = child;
    let context = {
        type: type,
        onClose: onDismiss,
        isDismissable: isDismissable
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($iaoFN$reactspectrumoverlays.Modal, {
        isOpen: !!child,
        onClose: onDismiss,
        type: type,
        isDismissable: isDismissable,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled
    }, /*#__PURE__*/ ($parcel$interopDefault($iaoFN$react)).createElement($0eb909ad2388d989$export$8b93a07348a7730c.Provider, {
        value: context
    }, lastChild.current)));
}


var $1f05ce3abe26c4cc$exports = {};

$parcel$export($1f05ce3abe26c4cc$exports, "useDialogContainer", () => $1f05ce3abe26c4cc$export$a2f2d8fa6720dab1);


function $1f05ce3abe26c4cc$export$a2f2d8fa6720dab1() {
    let context = $iaoFN$react.useContext($0eb909ad2388d989$export$8b93a07348a7730c);
    if (!context) throw new Error('Cannot call useDialogContext outside a <DialogTrigger> or <DialogContainer>.');
    return {
        type: context.type,
        dismiss () {
            context.onClose();
        }
    };
}


$parcel$exportWildcard(module.exports, $66f8653ff750aa27$exports);
$parcel$exportWildcard(module.exports, $a57fc2323bf98a33$exports);
$parcel$exportWildcard(module.exports, $46e54a192dc519c9$exports);
$parcel$exportWildcard(module.exports, $1b7050825e0eb5b6$exports);
$parcel$exportWildcard(module.exports, $1f05ce3abe26c4cc$exports);
