// @react-spectrum/tabs/dist/module.js
import "./main.css";
import {Item as $7e899e8977e9ac07$re_export$Item, useCollection as $cLzzU$useCollection} from "@react-stately/collections";
import {useDOMRef as $cLzzU$useDOMRef, useStyleProps as $cLzzU$useStyleProps, classNames as $cLzzU$classNames, SlotProvider as $cLzzU$SlotProvider, unwrapDOMRef as $cLzzU$unwrapDOMRef} from "@react-spectrum/utils";
import {useValueEffect as $cLzzU$useValueEffect, useResizeObserver as $cLzzU$useResizeObserver, useId as $cLzzU$useId, filterDOMProps as $cLzzU$filterDOMProps, mergeProps as $cLzzU$mergeProps, useLayoutEffect as $cLzzU$useLayoutEffect} from "@react-aria/utils";
import {FocusRing as $cLzzU$FocusRing} from "@react-aria/focus";
import {Picker as $cLzzU$Picker, Item as $cLzzU$Item} from "@react-spectrum/picker";
import {ListCollection as $cLzzU$ListCollection} from "@react-stately/list";
import $cLzzU$react, {useRef as $cLzzU$useRef, useState as $cLzzU$useState, useEffect as $cLzzU$useEffect, useCallback as $cLzzU$useCallback, useContext as $cLzzU$useContext} from "react";
import {useTabListState as $cLzzU$useTabListState} from "@react-stately/tabs";
import {Text as $cLzzU$Text} from "@react-spectrum/text";
import {useHover as $cLzzU$useHover} from "@react-aria/interactions";
import {useLocale as $cLzzU$useLocale} from "@react-aria/i18n";
import {useProviderProps as $cLzzU$useProviderProps, useProvider as $cLzzU$useProvider} from "@react-spectrum/provider";
import {useTab as $cLzzU$useTab, useTabList as $cLzzU$useTabList, useTabPanel as $cLzzU$useTabPanel} from "@react-aria/tabs";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $65ab70ddbed1564b$exports = {};

$parcel$export($65ab70ddbed1564b$exports, "TabList", () => $65ab70ddbed1564b$export$e51a686c67fdaa2d);
$parcel$export($65ab70ddbed1564b$exports, "TabPanels", () => $65ab70ddbed1564b$export$5dae8d435677f210);
$parcel$export($65ab70ddbed1564b$exports, "Tabs", () => $65ab70ddbed1564b$export$b2539bed5023c21c);







var $32d597b82a7aad64$exports = {};

$parcel$export($32d597b82a7aad64$exports, "spectrum-TabsPanel", () => $32d597b82a7aad64$export$34f1a65c57bb7bb, (v) => $32d597b82a7aad64$export$34f1a65c57bb7bb = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-TabsPanel--horizontal", () => $32d597b82a7aad64$export$778fb839eaa0528c, (v) => $32d597b82a7aad64$export$778fb839eaa0528c = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-TabsPanel--vertical", () => $32d597b82a7aad64$export$9a6e2fb72c648072, (v) => $32d597b82a7aad64$export$9a6e2fb72c648072 = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-Tabs", () => $32d597b82a7aad64$export$35c105c0d9c8e56f, (v) => $32d597b82a7aad64$export$35c105c0d9c8e56f = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-Tabs-item", () => $32d597b82a7aad64$export$123e4c01a5440a3e, (v) => $32d597b82a7aad64$export$123e4c01a5440a3e = v);
$parcel$export($32d597b82a7aad64$exports, "is-disabled", () => $32d597b82a7aad64$export$d35bc1e505d1ebbf, (v) => $32d597b82a7aad64$export$d35bc1e505d1ebbf = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-Tabs-itemLabel", () => $32d597b82a7aad64$export$1f253edac4b98526, (v) => $32d597b82a7aad64$export$1f253edac4b98526 = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-Icon", () => $32d597b82a7aad64$export$d374b04f30360026, (v) => $32d597b82a7aad64$export$d374b04f30360026 = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-Tabs-selectionIndicator", () => $32d597b82a7aad64$export$a115f9117a19563e, (v) => $32d597b82a7aad64$export$a115f9117a19563e = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-Tabs--horizontal", () => $32d597b82a7aad64$export$229f1ac4d3d38f0e, (v) => $32d597b82a7aad64$export$229f1ac4d3d38f0e = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-Tabs--compact", () => $32d597b82a7aad64$export$e15c78184ee03abe, (v) => $32d597b82a7aad64$export$e15c78184ee03abe = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-Tabs--isCollapsed", () => $32d597b82a7aad64$export$ea514cd060922d6b, (v) => $32d597b82a7aad64$export$ea514cd060922d6b = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-Tabs--quiet", () => $32d597b82a7aad64$export$831e830d5828d42b, (v) => $32d597b82a7aad64$export$831e830d5828d42b = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-Tabs--vertical", () => $32d597b82a7aad64$export$e415005be3957065, (v) => $32d597b82a7aad64$export$e415005be3957065 = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-TabsPanel-collapseWrapper", () => $32d597b82a7aad64$export$e09869956b4c96f2, (v) => $32d597b82a7aad64$export$e09869956b4c96f2 = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-TabsPanel-tabs", () => $32d597b82a7aad64$export$4a157a691c854e23, (v) => $32d597b82a7aad64$export$4a157a691c854e23 = v);
$parcel$export($32d597b82a7aad64$exports, "spectrum-TabsPanel-tabpanel", () => $32d597b82a7aad64$export$acb5da0970288266, (v) => $32d597b82a7aad64$export$acb5da0970288266 = v);
$parcel$export($32d597b82a7aad64$exports, "focus-ring", () => $32d597b82a7aad64$export$f39a09f249340e2a, (v) => $32d597b82a7aad64$export$f39a09f249340e2a = v);
$parcel$export($32d597b82a7aad64$exports, "is-hovered", () => $32d597b82a7aad64$export$b8813cd5d7824ce7, (v) => $32d597b82a7aad64$export$b8813cd5d7824ce7 = v);
$parcel$export($32d597b82a7aad64$exports, "is-selected", () => $32d597b82a7aad64$export$1e0fb04f31d3c22a, (v) => $32d597b82a7aad64$export$1e0fb04f31d3c22a = v);
var $32d597b82a7aad64$export$34f1a65c57bb7bb;
var $32d597b82a7aad64$export$778fb839eaa0528c;
var $32d597b82a7aad64$export$9a6e2fb72c648072;
var $32d597b82a7aad64$export$35c105c0d9c8e56f;
var $32d597b82a7aad64$export$123e4c01a5440a3e;
var $32d597b82a7aad64$export$d35bc1e505d1ebbf;
var $32d597b82a7aad64$export$1f253edac4b98526;
var $32d597b82a7aad64$export$d374b04f30360026;
var $32d597b82a7aad64$export$a115f9117a19563e;
var $32d597b82a7aad64$export$229f1ac4d3d38f0e;
var $32d597b82a7aad64$export$e15c78184ee03abe;
var $32d597b82a7aad64$export$ea514cd060922d6b;
var $32d597b82a7aad64$export$831e830d5828d42b;
var $32d597b82a7aad64$export$e415005be3957065;
var $32d597b82a7aad64$export$e09869956b4c96f2;
var $32d597b82a7aad64$export$4a157a691c854e23;
var $32d597b82a7aad64$export$acb5da0970288266;
var $32d597b82a7aad64$export$f39a09f249340e2a;
var $32d597b82a7aad64$export$b8813cd5d7824ce7;
var $32d597b82a7aad64$export$1e0fb04f31d3c22a;
$32d597b82a7aad64$export$34f1a65c57bb7bb = "spectrum-TabsPanel_3e140b";
$32d597b82a7aad64$export$778fb839eaa0528c = "spectrum-TabsPanel--horizontal_3e140b";
$32d597b82a7aad64$export$9a6e2fb72c648072 = "spectrum-TabsPanel--vertical_3e140b";
$32d597b82a7aad64$export$35c105c0d9c8e56f = "spectrum-Tabs_3e140b";
$32d597b82a7aad64$export$123e4c01a5440a3e = "spectrum-Tabs-item_3e140b";
$32d597b82a7aad64$export$d35bc1e505d1ebbf = "is-disabled_3e140b";
$32d597b82a7aad64$export$1f253edac4b98526 = "spectrum-Tabs-itemLabel_3e140b";
$32d597b82a7aad64$export$d374b04f30360026 = "spectrum-Icon_3e140b";
$32d597b82a7aad64$export$a115f9117a19563e = "spectrum-Tabs-selectionIndicator_3e140b";
$32d597b82a7aad64$export$229f1ac4d3d38f0e = "spectrum-Tabs--horizontal_3e140b";
$32d597b82a7aad64$export$e15c78184ee03abe = "spectrum-Tabs--compact_3e140b";
$32d597b82a7aad64$export$ea514cd060922d6b = "spectrum-Tabs--isCollapsed_3e140b";
$32d597b82a7aad64$export$831e830d5828d42b = "spectrum-Tabs--quiet_3e140b";
$32d597b82a7aad64$export$e415005be3957065 = "spectrum-Tabs--vertical_3e140b";
$32d597b82a7aad64$export$e09869956b4c96f2 = "spectrum-TabsPanel-collapseWrapper_3e140b";
$32d597b82a7aad64$export$4a157a691c854e23 = "spectrum-TabsPanel-tabs_3e140b";
$32d597b82a7aad64$export$acb5da0970288266 = "spectrum-TabsPanel-tabpanel_3e140b";
$32d597b82a7aad64$export$f39a09f249340e2a = "focus-ring_3e140b";
$32d597b82a7aad64$export$b8813cd5d7824ce7 = "is-hovered_3e140b";
$32d597b82a7aad64$export$1e0fb04f31d3c22a = "is-selected_3e140b";










const $65ab70ddbed1564b$var$TabContext = /*#__PURE__*/ $cLzzU$react.createContext(null);
function $65ab70ddbed1564b$var$Tabs(props, ref) {
    props = $cLzzU$useProviderProps(props);
    let { orientation: orientation = 'horizontal' , density: density = 'regular' , children: children , ...otherProps } = props;
    let domRef = $cLzzU$useDOMRef(ref);
    let tablistRef = $cLzzU$useRef();
    let wrapperRef = $cLzzU$useRef();
    let { direction: direction  } = $cLzzU$useLocale();
    let { styleProps: styleProps  } = $cLzzU$useStyleProps(otherProps);
    let [collapse, setCollapse] = $cLzzU$useValueEffect(false);
    let [selectedTab1, setSelectedTab] = $cLzzU$useState();
    const [tabListState, setTabListState] = $cLzzU$useState(null);
    $cLzzU$useEffect(()=>{
        if (tablistRef.current) {
            let selectedTab = tablistRef.current.querySelector(`[data-key="${tabListState === null || tabListState === void 0 ? void 0 : tabListState.selectedKey}"]`);
            if (selectedTab != null) setSelectedTab(selectedTab);
        }
    // collapse is in the dep array so selectedTab can be updated for TabLine positioning
    }, [
        children,
        tabListState === null || tabListState === void 0 ? void 0 : tabListState.selectedKey,
        collapse,
        tablistRef
    ]);
    let checkShouldCollapse = $cLzzU$useCallback(()=>{
        let computeShouldCollapse = ()=>{
            if (wrapperRef.current) {
                let tabsComponent = wrapperRef.current;
                let tabs = tablistRef.current.querySelectorAll('[role="tab"]');
                let lastTab = tabs[tabs.length - 1];
                let end = direction === 'rtl' ? 'left' : 'right';
                let farEdgeTabList = tabsComponent.getBoundingClientRect()[end];
                let farEdgeLastTab = lastTab === null || lastTab === void 0 ? void 0 : lastTab.getBoundingClientRect()[end];
                let shouldCollapse = direction === 'rtl' ? farEdgeLastTab < farEdgeTabList : farEdgeTabList < farEdgeLastTab;
                return shouldCollapse;
            }
        };
        if (orientation !== 'vertical') setCollapse(function*() {
            // Make Tabs render in non-collapsed state
            yield false;
            // Compute if Tabs should collapse and update
            yield computeShouldCollapse();
        });
    }, [
        tablistRef,
        wrapperRef,
        direction,
        orientation,
        setCollapse
    ]);
    $cLzzU$useEffect(()=>{
        checkShouldCollapse();
    }, [
        children,
        checkShouldCollapse
    ]);
    $cLzzU$useResizeObserver({
        ref: wrapperRef,
        onResize: checkShouldCollapse
    });
    let tabPanelProps = {
        'aria-labelledby': undefined
    };
    // When the tabs are collapsed, the tabPanel should be labelled by the Picker button element.
    let collapsibleTabListId = $cLzzU$useId();
    if (collapse && orientation !== 'vertical') tabPanelProps['aria-labelledby'] = collapsibleTabListId;
    return(/*#__PURE__*/ $cLzzU$react.createElement($65ab70ddbed1564b$var$TabContext.Provider, {
        value: {
            tabProps: {
                ...props,
                orientation: orientation,
                density: density
            },
            tabState: {
                tabListState: tabListState,
                setTabListState: setTabListState,
                selectedTab: selectedTab1,
                collapse: collapse
            },
            refs: {
                tablistRef: tablistRef,
                wrapperRef: wrapperRef
            },
            tabPanelProps: tabPanelProps
        }
    }, /*#__PURE__*/ $cLzzU$react.createElement("div", {
        ...$cLzzU$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef,
        className: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-TabsPanel', `spectrum-TabsPanel--${orientation}`, styleProps.className)
    }, props.children)));
}
// @private
function $65ab70ddbed1564b$var$Tab(props) {
    let { item: item , state: state , isDisabled: propsDisabled  } = props;
    let { key: key , rendered: rendered  } = item;
    let isDisabled = propsDisabled || state.disabledKeys.has(key);
    let ref = $cLzzU$useRef();
    let { tabProps: tabProps  } = $cLzzU$useTab({
        key: key,
        isDisabled: isDisabled
    }, state, ref);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $cLzzU$useHover({
        ...props
    });
    let isSelected = state.selectedKey === key;
    let domProps = $cLzzU$filterDOMProps(item.props);
    delete domProps.id;
    return(/*#__PURE__*/ $cLzzU$react.createElement($cLzzU$FocusRing, {
        focusRingClass: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'focus-ring')
    }, /*#__PURE__*/ $cLzzU$react.createElement("div", {
        ...$cLzzU$mergeProps(tabProps, hoverProps, domProps),
        ref: ref,
        className: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-Tabs-item', {
            'is-selected': isSelected,
            'is-disabled': isDisabled,
            'is-hovered': isHovered
        })
    }, /*#__PURE__*/ $cLzzU$react.createElement($cLzzU$SlotProvider, {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-Icon')
            },
            text: {
                UNSAFE_className: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-Tabs-itemLabel')
            }
        }
    }, typeof rendered === 'string' ? /*#__PURE__*/ $cLzzU$react.createElement($cLzzU$Text, null, rendered) : rendered))));
}
// @private
function $65ab70ddbed1564b$var$TabLine(props) {
    let { orientation: orientation , selectedTab: // Is either the tab node (non-collapsed) or the picker node (collapsed)
    selectedTab , selectedKey: // selectedKey is provided so that the TabLine styles are updated when the TabPicker's width updates from a selection change
    selectedKey  } = props;
    let { direction: direction  } = $cLzzU$useLocale();
    let { scale: scale  } = $cLzzU$useProvider();
    let [style, setStyle] = $cLzzU$useState({
        width: undefined,
        height: undefined
    });
    $cLzzU$useLayoutEffect(()=>{
        if (selectedTab) {
            var ref;
            let styleObj = {
                transform: undefined,
                width: undefined,
                height: undefined
            };
            // In RTL, calculate the transform from the right edge of the tablist so that resizing the window doesn't break the Tabline position due to offsetLeft changes
            let offset = direction === 'rtl' ? -1 * (((ref = selectedTab.offsetParent) === null || ref === void 0 ? void 0 : ref.offsetWidth) - selectedTab.offsetWidth - selectedTab.offsetLeft) : selectedTab.offsetLeft;
            styleObj.transform = orientation === 'vertical' ? `translateY(${selectedTab.offsetTop}px)` : `translateX(${offset}px)`;
            if (orientation === 'horizontal') styleObj.width = `${selectedTab.offsetWidth}px`;
            else styleObj.height = `${selectedTab.offsetHeight}px`;
            setStyle(styleObj);
        }
    }, [
        direction,
        setStyle,
        selectedTab,
        orientation,
        scale,
        selectedKey
    ]);
    return(/*#__PURE__*/ $cLzzU$react.createElement("div", {
        className: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-Tabs-selectionIndicator'),
        role: "presentation",
        style: style
    }));
}
function $65ab70ddbed1564b$export$e51a686c67fdaa2d(props) {
    const tabContext = $cLzzU$useContext($65ab70ddbed1564b$var$TabContext);
    const { refs: refs , tabState: tabState , tabProps: tabProps , tabPanelProps: tabPanelProps  } = tabContext;
    const { isQuiet: isQuiet , density: density , isDisabled: isDisabled , orientation: orientation  } = tabProps;
    const { selectedTab: selectedTab , collapse: collapse , setTabListState: setTabListState  } = tabState;
    const { tablistRef: tablistRef , wrapperRef: wrapperRef  } = refs;
    // Pass original Tab props but override children to create the collection.
    const state = $cLzzU$useTabListState({
        ...tabProps,
        children: props.children
    });
    let { styleProps: styleProps  } = $cLzzU$useStyleProps(props);
    const { tabListProps: tabListProps  } = $cLzzU$useTabList({
        ...tabProps,
        ...props
    }, state, tablistRef);
    $cLzzU$useEffect(()=>{
        // Passing back to root as useTabPanel needs the TabListState
        setTabListState(state);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        state.disabledKeys,
        state.selectedItem,
        state.selectedKey,
        props.children
    ]);
    let stylePropsForVertical = orientation === 'vertical' ? styleProps : {
    };
    let tabListclassName = $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-TabsPanel-tabs');
    const tabContent = /*#__PURE__*/ $cLzzU$react.createElement("div", {
        ...stylePropsForVertical,
        ...tabListProps,
        ref: tablistRef,
        className: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-Tabs', `spectrum-Tabs--${orientation}`, tabListclassName, {
            'spectrum-Tabs--quiet': isQuiet,
            ['spectrum-Tabs--compact']: density === 'compact'
        }, orientation === 'vertical' && styleProps.className)
    }, [
        ...state.collection
    ].map((item)=>/*#__PURE__*/ $cLzzU$react.createElement($65ab70ddbed1564b$var$Tab, {
            key: item.key,
            item: item,
            state: state,
            isDisabled: isDisabled,
            orientation: orientation
        })
    ), /*#__PURE__*/ $cLzzU$react.createElement($65ab70ddbed1564b$var$TabLine, {
        orientation: orientation,
        selectedTab: selectedTab
    }));
    if (orientation === 'vertical') return tabContent;
    else return(/*#__PURE__*/ $cLzzU$react.createElement("div", {
        ...styleProps,
        ref: wrapperRef,
        className: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-TabsPanel-collapseWrapper', styleProps.className)
    }, collapse ? /*#__PURE__*/ $cLzzU$react.createElement($65ab70ddbed1564b$var$TabPicker, {
        ...props,
        ...tabProps,
        id: tabPanelProps['aria-labelledby'],
        state: state,
        className: tabListclassName
    }) : tabContent));
}
function $65ab70ddbed1564b$export$5dae8d435677f210(props) {
    const { tabState: tabState , tabProps: tabProps  } = $cLzzU$useContext($65ab70ddbed1564b$var$TabContext);
    const { tabListState: tabListState  } = tabState;
    const factory = (nodes)=>new $cLzzU$ListCollection(nodes)
    ;
    const collection = $cLzzU$useCollection({
        items: tabProps.items,
        ...props
    }, factory, {
        suppressTextValueWarning: true
    });
    const selectedItem = tabListState ? collection.getItem(tabListState.selectedKey) : null;
    return(/*#__PURE__*/ $cLzzU$react.createElement($65ab70ddbed1564b$var$TabPanel, {
        ...props,
        key: tabListState === null || tabListState === void 0 ? void 0 : tabListState.selectedKey
    }, selectedItem && selectedItem.props.children));
}
// @private
function $65ab70ddbed1564b$var$TabPanel(props) {
    const { tabState: tabState , tabPanelProps: ctxTabPanelProps  } = $cLzzU$useContext($65ab70ddbed1564b$var$TabContext);
    const { tabListState: tabListState  } = tabState;
    let ref = $cLzzU$useRef();
    const { tabPanelProps: tabPanelProps  } = $cLzzU$useTabPanel(props, tabListState, ref);
    let { styleProps: styleProps  } = $cLzzU$useStyleProps(props);
    if (ctxTabPanelProps['aria-labelledby']) tabPanelProps['aria-labelledby'] = ctxTabPanelProps['aria-labelledby'];
    return(/*#__PURE__*/ $cLzzU$react.createElement($cLzzU$FocusRing, {
        focusRingClass: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'focus-ring')
    }, /*#__PURE__*/ $cLzzU$react.createElement("div", {
        ...styleProps,
        ...tabPanelProps,
        ref: ref,
        className: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-TabsPanel-tabpanel', styleProps.className)
    }, props.children)));
}
function $65ab70ddbed1564b$var$TabPicker(props) {
    let { isDisabled: isDisabled , isQuiet: isQuiet , state: state , 'aria-labelledby': ariaLabeledBy , 'aria-label': ariaLabel , density: density , className: className , id: id  } = props;
    let ref = $cLzzU$useRef();
    let [pickerNode, setPickerNode] = $cLzzU$useState(null);
    $cLzzU$useEffect(()=>{
        let node = $cLzzU$unwrapDOMRef(ref);
        setPickerNode(node.current);
    }, [
        ref
    ]);
    let items = [
        ...state.collection
    ].map((item)=>({
            rendered: item.rendered,
            textValue: item.textValue,
            id: item.key
        })
    );
    let pickerProps = {
        'aria-labelledby': ariaLabeledBy,
        'aria-label': ariaLabel
    };
    // TODO: Figure out if tabListProps should go onto the div here, v2 doesn't do it
    return(/*#__PURE__*/ $cLzzU$react.createElement("div", {
        className: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-Tabs', 'spectrum-Tabs--horizontal', 'spectrum-Tabs--isCollapsed', {
            'spectrum-Tabs--quiet': isQuiet,
            ['spectrum-Tabs--compact']: density === 'compact'
        }, className)
    }, /*#__PURE__*/ $cLzzU$react.createElement($cLzzU$SlotProvider, {
        slots: {
            icon: {
                size: 'S',
                UNSAFE_className: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'spectrum-Icon')
            },
            button: {
                focusRingClass: $cLzzU$classNames((/*@__PURE__*/$parcel$interopDefault($32d597b82a7aad64$exports)), 'focus-ring')
            }
        }
    }, /*#__PURE__*/ $cLzzU$react.createElement($cLzzU$Picker, {
        ...pickerProps,
        id: id,
        items: items,
        ref: ref,
        isQuiet: true,
        isDisabled: isDisabled,
        selectedKey: state.selectedKey,
        disabledKeys: state.disabledKeys,
        onSelectionChange: state.setSelectedKey
    }, (item)=>/*#__PURE__*/ $cLzzU$react.createElement($cLzzU$Item, {
            textValue: item.textValue
        }, item.rendered)
    ), pickerNode && /*#__PURE__*/ $cLzzU$react.createElement($65ab70ddbed1564b$var$TabLine, {
        orientation: "horizontal",
        selectedTab: pickerNode,
        selectedKey: state.selectedKey
    }))));
}
/**
 * Tabs organize content into multiple sections and allow users to navigate between them. The content under the set of tabs should be related and form a coherent unit.
 */ // forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $65ab70ddbed1564b$export$b2539bed5023c21c = /*#__PURE__*/ $cLzzU$react.forwardRef($65ab70ddbed1564b$var$Tabs);





export {$7e899e8977e9ac07$re_export$Item as Item, $65ab70ddbed1564b$export$e51a686c67fdaa2d as TabList, $65ab70ddbed1564b$export$5dae8d435677f210 as TabPanels, $65ab70ddbed1564b$export$b2539bed5023c21c as Tabs};
