// @react-spectrum/calendar/dist/module.js
import "./main.css";
import {createCalendar as $40VnT$createCalendar, endOfMonth as $40VnT$endOfMonth, getWeeksInMonth as $40VnT$getWeeksInMonth, isSameMonth as $40VnT$isSameMonth, isSameDay as $40VnT$isSameDay, getDayOfWeek as $40VnT$getDayOfWeek, isToday as $40VnT$isToday} from "@internationalized/date";
import {createDOMRef as $40VnT$createDOMRef, useStyleProps as $40VnT$useStyleProps, classNames as $40VnT$classNames} from "@react-spectrum/utils";
import $40VnT$react, {useMemo as $40VnT$useMemo, useRef as $40VnT$useRef, useImperativeHandle as $40VnT$useImperativeHandle} from "react";
import {useCalendar as $40VnT$useCalendar, useCalendarGrid as $40VnT$useCalendarGrid, useCalendarCell as $40VnT$useCalendarCell, useRangeCalendar as $40VnT$useRangeCalendar} from "@react-aria/calendar";
import {useCalendarState as $40VnT$useCalendarState, useRangeCalendarState as $40VnT$useRangeCalendarState} from "@react-stately/calendar";
import {useLocale as $40VnT$useLocale, useLocalizedStringFormatter as $40VnT$useLocalizedStringFormatter, useDateFormatter as $40VnT$useDateFormatter} from "@react-aria/i18n";
import {useProviderProps as $40VnT$useProviderProps} from "@react-spectrum/provider";
import {ActionButton as $40VnT$ActionButton} from "@react-spectrum/button";
import $40VnT$spectrumiconsuiChevronLeftLarge from "@spectrum-icons/ui/ChevronLeftLarge";
import $40VnT$spectrumiconsuiChevronRightLarge from "@spectrum-icons/ui/ChevronRightLarge";
import {HelpText as $40VnT$HelpText} from "@react-spectrum/label";
import {VisuallyHidden as $40VnT$VisuallyHidden} from "@react-aria/visually-hidden";
import {mergeProps as $40VnT$mergeProps} from "@react-aria/utils";
import {useFocusRing as $40VnT$useFocusRing} from "@react-aria/focus";
import {useHover as $40VnT$useHover} from "@react-aria/interactions";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}






var $52c51313300b3b3e$exports = {};

$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar", () => $52c51313300b3b3e$export$2d9837ea36f7a875, (v) => $52c51313300b3b3e$export$2d9837ea36f7a875 = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar--padded", () => $52c51313300b3b3e$export$8d90e3a3d21fe428, (v) => $52c51313300b3b3e$export$8d90e3a3d21fe428 = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-header", () => $52c51313300b3b3e$export$7efb9b56af6e2d09, (v) => $52c51313300b3b3e$export$7efb9b56af6e2d09 = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-monthHeader", () => $52c51313300b3b3e$export$185018e73e57145f, (v) => $52c51313300b3b3e$export$185018e73e57145f = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-title", () => $52c51313300b3b3e$export$98ee27555a2b7c01, (v) => $52c51313300b3b3e$export$98ee27555a2b7c01 = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-prevMonth", () => $52c51313300b3b3e$export$d4c1c2a202a50db2, (v) => $52c51313300b3b3e$export$d4c1c2a202a50db2 = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-nextMonth", () => $52c51313300b3b3e$export$7876229ab722b448, (v) => $52c51313300b3b3e$export$7876229ab722b448 = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-months", () => $52c51313300b3b3e$export$5dacd232bfef6687, (v) => $52c51313300b3b3e$export$5dacd232bfef6687 = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-dayOfWeek", () => $52c51313300b3b3e$export$7ff21fe81b21fa51, (v) => $52c51313300b3b3e$export$7ff21fe81b21fa51 = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-body", () => $52c51313300b3b3e$export$33407a73687c0e62, (v) => $52c51313300b3b3e$export$33407a73687c0e62 = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-table", () => $52c51313300b3b3e$export$27f4c38933cceedf, (v) => $52c51313300b3b3e$export$27f4c38933cceedf = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-tableCell", () => $52c51313300b3b3e$export$292b81cfd590634d, (v) => $52c51313300b3b3e$export$292b81cfd590634d = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-date", () => $52c51313300b3b3e$export$c20fe43d14f8fa36, (v) => $52c51313300b3b3e$export$c20fe43d14f8fa36 = v);
$parcel$export($52c51313300b3b3e$exports, "spectrum-Calendar-dateText", () => $52c51313300b3b3e$export$cc48a6e73fdb7719, (v) => $52c51313300b3b3e$export$cc48a6e73fdb7719 = v);
$parcel$export($52c51313300b3b3e$exports, "is-disabled", () => $52c51313300b3b3e$export$d35bc1e505d1ebbf, (v) => $52c51313300b3b3e$export$d35bc1e505d1ebbf = v);
$parcel$export($52c51313300b3b3e$exports, "is-unavailable", () => $52c51313300b3b3e$export$38b08f1df825800e, (v) => $52c51313300b3b3e$export$38b08f1df825800e = v);
$parcel$export($52c51313300b3b3e$exports, "is-outsideMonth", () => $52c51313300b3b3e$export$93ff5b06817b3d41, (v) => $52c51313300b3b3e$export$93ff5b06817b3d41 = v);
$parcel$export($52c51313300b3b3e$exports, "is-focused", () => $52c51313300b3b3e$export$e7dc768d35940237, (v) => $52c51313300b3b3e$export$e7dc768d35940237 = v);
$parcel$export($52c51313300b3b3e$exports, "is-range-selection", () => $52c51313300b3b3e$export$cd2839d8bb0c6407, (v) => $52c51313300b3b3e$export$cd2839d8bb0c6407 = v);
$parcel$export($52c51313300b3b3e$exports, "is-range-start", () => $52c51313300b3b3e$export$4ce7d3d24e42e411, (v) => $52c51313300b3b3e$export$4ce7d3d24e42e411 = v);
$parcel$export($52c51313300b3b3e$exports, "is-selection-start", () => $52c51313300b3b3e$export$f0091c9d6b8c19d3, (v) => $52c51313300b3b3e$export$f0091c9d6b8c19d3 = v);
$parcel$export($52c51313300b3b3e$exports, "is-range-end", () => $52c51313300b3b3e$export$52890a3cb1ab46b, (v) => $52c51313300b3b3e$export$52890a3cb1ab46b = v);
$parcel$export($52c51313300b3b3e$exports, "is-selection-end", () => $52c51313300b3b3e$export$50722ae5663230f9, (v) => $52c51313300b3b3e$export$50722ae5663230f9 = v);
$parcel$export($52c51313300b3b3e$exports, "is-invalid", () => $52c51313300b3b3e$export$14a30de2866685fa, (v) => $52c51313300b3b3e$export$14a30de2866685fa = v);
$parcel$export($52c51313300b3b3e$exports, "is-today", () => $52c51313300b3b3e$export$189f6cc7eaf68d8b, (v) => $52c51313300b3b3e$export$189f6cc7eaf68d8b = v);
$parcel$export($52c51313300b3b3e$exports, "is-selected", () => $52c51313300b3b3e$export$1e0fb04f31d3c22a, (v) => $52c51313300b3b3e$export$1e0fb04f31d3c22a = v);
$parcel$export($52c51313300b3b3e$exports, "is-hovered", () => $52c51313300b3b3e$export$b8813cd5d7824ce7, (v) => $52c51313300b3b3e$export$b8813cd5d7824ce7 = v);
$parcel$export($52c51313300b3b3e$exports, "is-pressed", () => $52c51313300b3b3e$export$15b73a232798b8d3, (v) => $52c51313300b3b3e$export$15b73a232798b8d3 = v);
var $52c51313300b3b3e$export$2d9837ea36f7a875;
var $52c51313300b3b3e$export$8d90e3a3d21fe428;
var $52c51313300b3b3e$export$7efb9b56af6e2d09;
var $52c51313300b3b3e$export$185018e73e57145f;
var $52c51313300b3b3e$export$98ee27555a2b7c01;
var $52c51313300b3b3e$export$d4c1c2a202a50db2;
var $52c51313300b3b3e$export$7876229ab722b448;
var $52c51313300b3b3e$export$5dacd232bfef6687;
var $52c51313300b3b3e$export$7ff21fe81b21fa51;
var $52c51313300b3b3e$export$33407a73687c0e62;
var $52c51313300b3b3e$export$27f4c38933cceedf;
var $52c51313300b3b3e$export$292b81cfd590634d;
var $52c51313300b3b3e$export$c20fe43d14f8fa36;
var $52c51313300b3b3e$export$cc48a6e73fdb7719;
var $52c51313300b3b3e$export$d35bc1e505d1ebbf;
var $52c51313300b3b3e$export$38b08f1df825800e;
var $52c51313300b3b3e$export$93ff5b06817b3d41;
var $52c51313300b3b3e$export$e7dc768d35940237;
var $52c51313300b3b3e$export$cd2839d8bb0c6407;
var $52c51313300b3b3e$export$4ce7d3d24e42e411;
var $52c51313300b3b3e$export$f0091c9d6b8c19d3;
var $52c51313300b3b3e$export$52890a3cb1ab46b;
var $52c51313300b3b3e$export$50722ae5663230f9;
var $52c51313300b3b3e$export$14a30de2866685fa;
var $52c51313300b3b3e$export$189f6cc7eaf68d8b;
var $52c51313300b3b3e$export$1e0fb04f31d3c22a;
var $52c51313300b3b3e$export$b8813cd5d7824ce7;
var $52c51313300b3b3e$export$15b73a232798b8d3;
$52c51313300b3b3e$export$2d9837ea36f7a875 = "spectrum-Calendar_6099b6";
$52c51313300b3b3e$export$8d90e3a3d21fe428 = "spectrum-Calendar--padded_6099b6";
$52c51313300b3b3e$export$7efb9b56af6e2d09 = "spectrum-Calendar-header_6099b6";
$52c51313300b3b3e$export$185018e73e57145f = "spectrum-Calendar-monthHeader_6099b6";
$52c51313300b3b3e$export$98ee27555a2b7c01 = "spectrum-Calendar-title_6099b6";
$52c51313300b3b3e$export$d4c1c2a202a50db2 = "spectrum-Calendar-prevMonth_6099b6";
$52c51313300b3b3e$export$7876229ab722b448 = "spectrum-Calendar-nextMonth_6099b6";
$52c51313300b3b3e$export$5dacd232bfef6687 = "spectrum-Calendar-months_6099b6";
$52c51313300b3b3e$export$7ff21fe81b21fa51 = "spectrum-Calendar-dayOfWeek_6099b6";
$52c51313300b3b3e$export$33407a73687c0e62 = "spectrum-Calendar-body_6099b6";
$52c51313300b3b3e$export$27f4c38933cceedf = "spectrum-Calendar-table_6099b6";
$52c51313300b3b3e$export$292b81cfd590634d = "spectrum-Calendar-tableCell_6099b6";
$52c51313300b3b3e$export$c20fe43d14f8fa36 = "spectrum-Calendar-date_6099b6";
$52c51313300b3b3e$export$cc48a6e73fdb7719 = "spectrum-Calendar-dateText_6099b6";
$52c51313300b3b3e$export$d35bc1e505d1ebbf = "is-disabled_6099b6";
$52c51313300b3b3e$export$38b08f1df825800e = "is-unavailable_6099b6";
$52c51313300b3b3e$export$93ff5b06817b3d41 = "is-outsideMonth_6099b6";
$52c51313300b3b3e$export$e7dc768d35940237 = "is-focused_6099b6";
$52c51313300b3b3e$export$cd2839d8bb0c6407 = "is-range-selection_6099b6";
$52c51313300b3b3e$export$4ce7d3d24e42e411 = "is-range-start_6099b6";
$52c51313300b3b3e$export$f0091c9d6b8c19d3 = "is-selection-start_6099b6";
$52c51313300b3b3e$export$52890a3cb1ab46b = "is-range-end_6099b6";
$52c51313300b3b3e$export$50722ae5663230f9 = "is-selection-end_6099b6";
$52c51313300b3b3e$export$14a30de2866685fa = "is-invalid_6099b6";
$52c51313300b3b3e$export$189f6cc7eaf68d8b = "is-today_6099b6";
$52c51313300b3b3e$export$1e0fb04f31d3c22a = "is-selected_6099b6";
$52c51313300b3b3e$export$b8813cd5d7824ce7 = "is-hovered_6099b6";
$52c51313300b3b3e$export$15b73a232798b8d3 = "is-pressed_6099b6";





function $453c556b9066da11$export$5d847498420df57b({ state: state , currentMonth: currentMonth , ...props }) {
    let ref = $40VnT$useRef();
    let { cellProps: cellProps , buttonProps: buttonProps , isPressed: isPressed , isSelected: isSelected , isDisabled: isDisabled , isFocused: isFocused , isInvalid: isInvalid , formattedDate: formattedDate  } = $40VnT$useCalendarCell({
        ...props,
        isDisabled: !$40VnT$isSameMonth(props.date, currentMonth)
    }, state, ref);
    let isUnavailable = state.isCellUnavailable(props.date) && !isDisabled;
    let isLastSelectedBeforeDisabled = !isDisabled && !isInvalid && state.isCellUnavailable(props.date.add({
        days: 1
    }));
    let isFirstSelectedAfterDisabled = !isDisabled && !isInvalid && state.isCellUnavailable(props.date.subtract({
        days: 1
    }));
    let highlightedRange = 'highlightedRange' in state && state.highlightedRange;
    let isSelectionStart = isSelected && highlightedRange && $40VnT$isSameDay(props.date, highlightedRange.start);
    let isSelectionEnd = isSelected && highlightedRange && $40VnT$isSameDay(props.date, highlightedRange.end);
    let { locale: locale  } = $40VnT$useLocale();
    let dayOfWeek = $40VnT$getDayOfWeek(props.date, locale);
    let isRangeStart = isSelected && (isFirstSelectedAfterDisabled || dayOfWeek === 0 || props.date.day === 1);
    let isRangeEnd = isSelected && (isLastSelectedBeforeDisabled || dayOfWeek === 6 || props.date.day === currentMonth.calendar.getDaysInMonth(currentMonth));
    let { focusProps: focusProps , isFocusVisible: isFocusVisible  } = $40VnT$useFocusRing();
    let { hoverProps: hoverProps , isHovered: isHovered  } = $40VnT$useHover({
        isDisabled: isDisabled || isUnavailable || state.isReadOnly
    });
    return(/*#__PURE__*/ $40VnT$react.createElement("td", {
        ...cellProps,
        className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-tableCell')
    }, /*#__PURE__*/ $40VnT$react.createElement("span", {
        ...$40VnT$mergeProps(buttonProps, hoverProps, focusProps),
        ref: ref,
        className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-date', {
            'is-today': $40VnT$isToday(props.date, state.timeZone),
            'is-selected': isSelected,
            'is-focused': isFocused && isFocusVisible,
            // Style disabled (i.e. out of min/max range), but selected dates as unavailable
            // since it is more clear than trying to dim the selection.
            'is-disabled': isDisabled && !isInvalid,
            'is-unavailable': isUnavailable || isInvalid && isDisabled,
            'is-outsideMonth': !$40VnT$isSameMonth(props.date, currentMonth),
            'is-range-start': isRangeStart,
            'is-range-end': isRangeEnd,
            'is-range-selection': isSelected && 'highlightedRange' in state,
            'is-selection-start': isSelectionStart,
            'is-selection-end': isSelectionEnd,
            'is-hovered': isHovered,
            'is-pressed': isPressed && !state.isReadOnly,
            'is-invalid': isInvalid
        })
    }, /*#__PURE__*/ $40VnT$react.createElement("span", {
        className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-dateText')
    }, /*#__PURE__*/ $40VnT$react.createElement("span", null, formattedDate)))));
}








function $faa2f3867773cfe5$export$26e2752316b9a375(props) {
    let { state: state , startDate: startDate  } = props;
    let { gridProps: gridProps , headerProps: headerProps , weekDays: weekDays  } = $40VnT$useCalendarGrid({
        ...props,
        endDate: $40VnT$endOfMonth(startDate)
    }, state);
    let { locale: locale  } = $40VnT$useLocale();
    let weeksInMonth = $40VnT$getWeeksInMonth(startDate, locale);
    return(/*#__PURE__*/ $40VnT$react.createElement("table", {
        ...gridProps,
        className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-body', 'spectrum-Calendar-table')
    }, /*#__PURE__*/ $40VnT$react.createElement("thead", headerProps, /*#__PURE__*/ $40VnT$react.createElement("tr", null, weekDays.map((day, index)=>/*#__PURE__*/ $40VnT$react.createElement("th", {
            key: index,
            className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-tableCell')
        }, /*#__PURE__*/ $40VnT$react.createElement("span", {
            className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-dayOfWeek')
        }, day))
    ))), /*#__PURE__*/ $40VnT$react.createElement("tbody", null, [
        ...new Array(weeksInMonth).keys()
    ].map((weekIndex)=>/*#__PURE__*/ $40VnT$react.createElement("tr", {
            key: weekIndex
        }, state.getDatesInWeek(weekIndex, startDate).map((date, i)=>date ? /*#__PURE__*/ $40VnT$react.createElement($453c556b9066da11$export$5d847498420df57b, {
                key: i,
                state: state,
                date: date,
                currentMonth: startDate
            }) : /*#__PURE__*/ $40VnT$react.createElement("td", {
                key: i
            })
        ))
    ))));
}






var $e0a80792abac0d48$exports = {};
var $c1fdee2284f63122$exports = {};
$c1fdee2284f63122$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `التاريخ`,
            other: `التواريخ`
        })} المحدد/المحددة غير متوفرة.`
};


var $81326fc5b8464f3e$exports = {};
$81326fc5b8464f3e$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Избраната дата не е налична`,
            other: `Избраните дати не са налични`
        })}.`
};


var $61bec992b8f61b70$exports = {};
$61bec992b8f61b70$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Vybrané datum není`,
            other: `Vybraná data nejsou`
        })} k dispozici.`
};


var $92192aa23ec84b2d$exports = {};
$92192aa23ec84b2d$exports = {
    "invalidSelection": (args, formatter)=>`Valgte ${formatter.plural(args.selectedCount, {
            one: `dato`,
            other: `datoer`
        })} ikke ledige.`
};


var $4db8629ea44949b0$exports = {};
$4db8629ea44949b0$exports = {
    "invalidSelection": (args, formatter)=>`Ausgewählte(s) ${formatter.plural(args.selectedCount, {
            one: `Datum`,
            other: `Daten`
        })} nicht verfügbar.`
};


var $d1795c2b69d00f6c$exports = {};
$d1795c2b69d00f6c$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Η επιλεγμένη ημερομηνία δεν είναι διαθέσιμη`,
            other: `Οι επιλεγμένες ημερομηνίες δεν είναι διαθέσιμες`
        })}.`
};


var $16d2015735d51a41$exports = {};
$16d2015735d51a41$exports = {
    "invalidSelection": (args, formatter)=>`Selected ${formatter.plural(args.selectedCount, {
            one: `date`,
            other: `dates`
        })} unavailable.`
};


var $96cd1e101026681b$exports = {};
$96cd1e101026681b$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `La fecha seleccionada no está disponible`,
            other: `Las fechas seleccionadas no están disponibles`
        })}.`
};


var $1133d3628ea10323$exports = {};
$1133d3628ea10323$exports = {
    "invalidSelection": (args, formatter)=>`Valitud ${formatter.plural(args.selectedCount, {
            one: `kuupäev`,
            other: `kuupäevad`
        })} pole saadaval.`
};


var $53163229660d53e4$exports = {};
$53163229660d53e4$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Valittu päivämäärä ei ole`,
            other: `Valitut päivämäärät eivät ole`
        })} saatavilla.`
};


var $d0808f7b535fb9b6$exports = {};
$d0808f7b535fb9b6$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `La date sélectionnée est indisponible`,
            other: `Les dates sélectionnées sont indisponibles`
        })}.`
};


var $658f594fbca84b9b$exports = {};
$658f594fbca84b9b$exports = {
    "invalidSelection": (args, formatter)=>`נבחר ${formatter.plural(args.selectedCount, {
            one: `תאריך`,
            other: `תאריכים`
        })} אינו זמין.`
};


var $0b62e8a6414bc20c$exports = {};
$0b62e8a6414bc20c$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Odabrani datum nije dostupan`,
            other: `Odabrani datumi nisu dostupni`
        })}.`
};


var $132d1b97de231314$exports = {};
$132d1b97de231314$exports = {
    "invalidSelection": (args, formatter)=>`A kijelölt ${formatter.plural(args.selectedCount, {
            one: `dátum nem érhető el`,
            other: `dátumok nem érhetőek el`
        })}.`
};


var $5653a55498966c6c$exports = {};
$5653a55498966c6c$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Data selezionata non disponibile`,
            other: `Date selezionate non disponibili`
        })}.`
};


var $243f99057b9b0af0$exports = {};
$243f99057b9b0af0$exports = {
    "invalidSelection": (args, formatter)=>`選択した${formatter.plural(args.selectedCount, {
            one: `日付`,
            other: `日付`
        })}は使用できません。`
};


var $9dbe802a4c1b47ed$exports = {};
$9dbe802a4c1b47ed$exports = {
    "invalidSelection": (args, formatter)=>`선택한 ${formatter.plural(args.selectedCount, {
            one: `일자를`,
            other: `일자를`
        })} 사용할 수 없습니다.`
};


var $ed6bec4d46a5be1b$exports = {};
$ed6bec4d46a5be1b$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Datos`,
            other: `Datų`
        })} pasirinkimas negalimas.`
};


var $e203254b4140c831$exports = {};
$e203254b4140c831$exports = {
    "invalidSelection": (args, formatter)=>`Atlasītais(-ie) ${formatter.plural(args.selectedCount, {
            one: `date`,
            other: `dates`
        })} nav pieejams(-i).`
};


var $87004dc1feacb708$exports = {};
$87004dc1feacb708$exports = {
    "invalidSelection": (args, formatter)=>`Valgt(e) ${formatter.plural(args.selectedCount, {
            one: `dato`,
            other: `datoer`
        })} er utilgjengelig.`
};


var $fd392c0c0b12ff84$exports = {};
$fd392c0c0b12ff84$exports = {
    "invalidSelection": (args, formatter)=>`Geselecteerde ${formatter.plural(args.selectedCount, {
            one: `datum`,
            other: `datums`
        })} niet beschikbaar.`
};


var $02f96c467435edbf$exports = {};
$02f96c467435edbf$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Wybrana data jest niedostępna`,
            other: `Wybrane daty są niedostępne`
        })}.`
};


var $a516ff2265dc1cde$exports = {};
$a516ff2265dc1cde$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Data`,
            other: `Datas`
        })} selecionada(s) indisponível(is).`
};


var $2a8fe7dc6b71b8d6$exports = {};
$2a8fe7dc6b71b8d6$exports = {
    "invalidSelection": (args, formatter)=>`A(s) ${formatter.plural(args.selectedCount, {
            one: `data`,
            other: `datas`
        })} selecionada(s) não está(ão) disponível/disponível(eis).`
};


var $459c636d0d282e8f$exports = {};
$459c636d0d282e8f$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Data selectată nu este disponibilă`,
            other: `Datele selectate nu sunt disponibile`
        })}.`
};


var $d49b96bc4e237385$exports = {};
$d49b96bc4e237385$exports = {
    "invalidSelection": (args, formatter)=>`Выбранные ${formatter.plural(args.selectedCount, {
            one: `дата`,
            other: `даты`
        })} недоступны.`
};


var $14f44f430d85943b$exports = {};
$14f44f430d85943b$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Vybratý dátum`,
            other: `Vybraté dátumy`
        })}: nedostupné.`
};


var $ced45866328d3850$exports = {};
$ced45866328d3850$exports = {
    "invalidSelection": (args, formatter)=>`Izbrano ${formatter.plural(args.selectedCount, {
            one: `date`,
            other: `dates`
        })} ni na voljo.`
};


var $46417484d64689f8$exports = {};
$46417484d64689f8$exports = {
    "invalidSelection": (args, formatter)=>`Izabrani ${formatter.plural(args.selectedCount, {
            one: `date`,
            other: `dates`
        })} nije dostupan.`
};


var $9d8ccb39096f4575$exports = {};
$9d8ccb39096f4575$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Valt datum är inte tillgängligt.`,
            other: `Valda datum är inte tillgängliga.`
        })}`
};


var $8e9c1a10ed1cc18a$exports = {};
$8e9c1a10ed1cc18a$exports = {
    "invalidSelection": (args, formatter)=>`Seçilen ${formatter.plural(args.selectedCount, {
            one: `tarih`,
            other: `tarihler`
        })} müsait değil.`
};


var $84c3bf1d3eaf1f59$exports = {};
$84c3bf1d3eaf1f59$exports = {
    "invalidSelection": (args, formatter)=>`${formatter.plural(args.selectedCount, {
            one: `Вибрана дата недоступна`,
            other: `Вибрані дати недоступні`
        })}.`
};


var $0212afd59952ac06$exports = {};
$0212afd59952ac06$exports = {
    "invalidSelection": (args, formatter)=>`选定的 ${formatter.plural(args.selectedCount, {
            one: `个日期`,
            other: `个日期`
        })} 不可用。`
};


var $87077630a4e30349$exports = {};
$87077630a4e30349$exports = {
    "invalidSelection": (args, formatter)=>`所選的${formatter.plural(args.selectedCount, {
            one: `日期`,
            other: `日期`
        })}無法使用。`
};


$e0a80792abac0d48$exports = {
    "ar-AE": $c1fdee2284f63122$exports,
    "bg-BG": $81326fc5b8464f3e$exports,
    "cs-CZ": $61bec992b8f61b70$exports,
    "da-DK": $92192aa23ec84b2d$exports,
    "de-DE": $4db8629ea44949b0$exports,
    "el-GR": $d1795c2b69d00f6c$exports,
    "en-US": $16d2015735d51a41$exports,
    "es-ES": $96cd1e101026681b$exports,
    "et-EE": $1133d3628ea10323$exports,
    "fi-FI": $53163229660d53e4$exports,
    "fr-FR": $d0808f7b535fb9b6$exports,
    "he-IL": $658f594fbca84b9b$exports,
    "hr-HR": $0b62e8a6414bc20c$exports,
    "hu-HU": $132d1b97de231314$exports,
    "it-IT": $5653a55498966c6c$exports,
    "ja-JP": $243f99057b9b0af0$exports,
    "ko-KR": $9dbe802a4c1b47ed$exports,
    "lt-LT": $ed6bec4d46a5be1b$exports,
    "lv-LV": $e203254b4140c831$exports,
    "nb-NO": $87004dc1feacb708$exports,
    "nl-NL": $fd392c0c0b12ff84$exports,
    "pl-PL": $02f96c467435edbf$exports,
    "pt-BR": $a516ff2265dc1cde$exports,
    "pt-PT": $2a8fe7dc6b71b8d6$exports,
    "ro-RO": $459c636d0d282e8f$exports,
    "ru-RU": $d49b96bc4e237385$exports,
    "sk-SK": $14f44f430d85943b$exports,
    "sl-SI": $ced45866328d3850$exports,
    "sr-SP": $46417484d64689f8$exports,
    "sv-SE": $9d8ccb39096f4575$exports,
    "tr-TR": $8e9c1a10ed1cc18a$exports,
    "uk-UA": $84c3bf1d3eaf1f59$exports,
    "zh-CN": $0212afd59952ac06$exports,
    "zh-TW": $87077630a4e30349$exports
};






function $62cd1e80f0dffb66$export$bfd52a43017368fe(props) {
    let { state: state , calendarProps: calendarProps , nextButtonProps: nextButtonProps , prevButtonProps: prevButtonProps , errorMessageProps: errorMessageProps , calendarRef: ref , visibleMonths: visibleMonths = 1  } = props;
    let { styleProps: styleProps  } = $40VnT$useStyleProps(props);
    let stringFormatter = $40VnT$useLocalizedStringFormatter((/*@__PURE__*/$parcel$interopDefault($e0a80792abac0d48$exports)));
    let { direction: direction  } = $40VnT$useLocale();
    let currentMonth = state.visibleRange.start;
    let monthDateFormatter = $40VnT$useDateFormatter({
        month: 'long',
        year: 'numeric',
        era: currentMonth.calendar.identifier === 'gregory' && currentMonth.era === 'BC' ? 'short' : undefined,
        calendar: currentMonth.calendar.identifier,
        timeZone: state.timeZone
    });
    let titles = [];
    let calendars = [];
    for(let i = 0; i < visibleMonths; i++){
        let d = currentMonth.add({
            months: i
        });
        titles.push(/*#__PURE__*/ $40VnT$react.createElement("div", {
            key: i,
            className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-monthHeader')
        }, i === 0 && /*#__PURE__*/ $40VnT$react.createElement($40VnT$ActionButton, {
            ...prevButtonProps,
            UNSAFE_className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-prevMonth'),
            isQuiet: true
        }, direction === 'rtl' ? /*#__PURE__*/ $40VnT$react.createElement($40VnT$spectrumiconsuiChevronRightLarge, null) : /*#__PURE__*/ $40VnT$react.createElement($40VnT$spectrumiconsuiChevronLeftLarge, null)), /*#__PURE__*/ $40VnT$react.createElement("h2", {
            // We have a visually hidden heading describing the entire visible range,
            // and the calendar itself describes the individual month
            // so we don't need to repeat that here for screen reader users.
            "aria-hidden": true,
            className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-title')
        }, monthDateFormatter.format(d.toDate(state.timeZone))), i === visibleMonths - 1 && /*#__PURE__*/ $40VnT$react.createElement($40VnT$ActionButton, {
            ...nextButtonProps,
            UNSAFE_className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-nextMonth'),
            isQuiet: true
        }, direction === 'rtl' ? /*#__PURE__*/ $40VnT$react.createElement($40VnT$spectrumiconsuiChevronLeftLarge, null) : /*#__PURE__*/ $40VnT$react.createElement($40VnT$spectrumiconsuiChevronRightLarge, null))));
        calendars.push(/*#__PURE__*/ $40VnT$react.createElement($faa2f3867773cfe5$export$26e2752316b9a375, {
            ...props,
            key: i,
            state: state,
            startDate: d
        }));
    }
    return(/*#__PURE__*/ $40VnT$react.createElement("div", {
        ...styleProps,
        ...calendarProps,
        ref: ref,
        className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar', styleProps.className)
    }, /*#__PURE__*/ $40VnT$react.createElement($40VnT$VisuallyHidden, null, /*#__PURE__*/ $40VnT$react.createElement("h2", null, calendarProps['aria-label'])), /*#__PURE__*/ $40VnT$react.createElement("div", {
        className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-header')
    }, titles), /*#__PURE__*/ $40VnT$react.createElement("div", {
        className: $40VnT$classNames((/*@__PURE__*/$parcel$interopDefault($52c51313300b3b3e$exports)), 'spectrum-Calendar-months')
    }, calendars), /*#__PURE__*/ $40VnT$react.createElement($40VnT$VisuallyHidden, null, /*#__PURE__*/ $40VnT$react.createElement("button", {
        "aria-label": nextButtonProps['aria-label'],
        disabled: nextButtonProps.isDisabled,
        onClick: ()=>state.focusNextPage()
        ,
        tabIndex: -1
    })), state.validationState === 'invalid' && /*#__PURE__*/ $40VnT$react.createElement($40VnT$HelpText, {
        showErrorIcon: true,
        errorMessage: props.errorMessage || stringFormatter.format('invalidSelection', {
            selectedCount: 'highlightedRange' in state ? 2 : 1
        }),
        errorMessageProps: errorMessageProps,
        validationState: "invalid",
        // Intentionally a global class name so it can be targeted in DatePicker CSS...
        UNSAFE_className: "spectrum-Calendar-helpText"
    })));
}









function $d2b8d4577dd88671$var$Calendar(props, ref) {
    props = $40VnT$useProviderProps(props);
    let { visibleMonths: visibleMonths = 1  } = props;
    let visibleDuration = $40VnT$useMemo(()=>({
            months: visibleMonths
        })
    , [
        visibleMonths
    ]);
    let { locale: locale  } = $40VnT$useLocale();
    let state = $40VnT$useCalendarState({
        ...props,
        locale: locale,
        visibleDuration: visibleDuration,
        createCalendar: $40VnT$createCalendar
    });
    let domRef = $40VnT$useRef();
    $40VnT$useImperativeHandle(ref, ()=>({
            ...$40VnT$createDOMRef(domRef),
            focus () {
                state.setFocused(true);
            }
        })
    );
    let { calendarProps: calendarProps , prevButtonProps: prevButtonProps , nextButtonProps: nextButtonProps , errorMessageProps: errorMessageProps  } = $40VnT$useCalendar(props, state);
    return(/*#__PURE__*/ $40VnT$react.createElement($62cd1e80f0dffb66$export$bfd52a43017368fe, {
        ...props,
        state: state,
        calendarRef: domRef,
        calendarProps: calendarProps,
        prevButtonProps: prevButtonProps,
        nextButtonProps: nextButtonProps,
        errorMessageProps: errorMessageProps
    }));
}
/**
 * Calendars display a grid of days in one or more months and allow users to select a single date.
 */ const $d2b8d4577dd88671$export$e1aef45b828286de = /*#__PURE__*/ $40VnT$react.forwardRef($d2b8d4577dd88671$var$Calendar);










function $8b1e8eddbf3b15de$var$RangeCalendar(props, ref) {
    props = $40VnT$useProviderProps(props);
    let { visibleMonths: visibleMonths = 1  } = props;
    let visibleDuration = $40VnT$useMemo(()=>({
            months: visibleMonths
        })
    , [
        visibleMonths
    ]);
    let { locale: locale  } = $40VnT$useLocale();
    let state = $40VnT$useRangeCalendarState({
        ...props,
        locale: locale,
        visibleDuration: visibleDuration,
        createCalendar: $40VnT$createCalendar
    });
    let domRef = $40VnT$useRef();
    $40VnT$useImperativeHandle(ref, ()=>({
            ...$40VnT$createDOMRef(domRef),
            focus () {
                state.setFocused(true);
            }
        })
    );
    let { calendarProps: calendarProps , prevButtonProps: prevButtonProps , nextButtonProps: nextButtonProps , errorMessageProps: errorMessageProps  } = $40VnT$useRangeCalendar(props, state, domRef);
    return(/*#__PURE__*/ $40VnT$react.createElement($62cd1e80f0dffb66$export$bfd52a43017368fe, {
        ...props,
        state: state,
        calendarRef: domRef,
        calendarProps: calendarProps,
        prevButtonProps: prevButtonProps,
        nextButtonProps: nextButtonProps,
        errorMessageProps: errorMessageProps
    }));
}
/**
 * RangeCalendars display a grid of days in one or more months and allow users to select a contiguous range of dates.
 */ const $8b1e8eddbf3b15de$export$a4f5c8b89d277a8d = /*#__PURE__*/ $40VnT$react.forwardRef($8b1e8eddbf3b15de$var$RangeCalendar);




export {$d2b8d4577dd88671$export$e1aef45b828286de as Calendar, $8b1e8eddbf3b15de$export$a4f5c8b89d277a8d as RangeCalendar};
