// @react-aria/progress/dist/module.js
import {filterDOMProps as $e3Dei$filterDOMProps, clamp as $e3Dei$clamp, mergeProps as $e3Dei$mergeProps} from "@react-aria/utils";
import {useLabel as $e3Dei$useLabel} from "@react-aria/label";
import {useNumberFormatter as $e3Dei$useNumberFormatter} from "@react-aria/i18n";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $204d9ebcedfb8806$exports = {};

$parcel$export($204d9ebcedfb8806$exports, "useProgressBar", () => $204d9ebcedfb8806$export$ed5abd763a836edc);



function $204d9ebcedfb8806$export$ed5abd763a836edc(props) {
    let { value: value = 0 , minValue: minValue = 0 , maxValue: maxValue = 100 , valueLabel: valueLabel , isIndeterminate: isIndeterminate , formatOptions: formatOptions = {
        style: 'percent'
    }  } = props;
    let domProps = $e3Dei$filterDOMProps(props, {
        labelable: true
    });
    let { labelProps: labelProps , fieldProps: fieldProps  } = $e3Dei$useLabel({
        ...props,
        // Progress bar is not an HTML input element so it
        // shouldn't be labeled by a <label> element.
        labelElementType: 'span'
    });
    value = $e3Dei$clamp(value, minValue, maxValue);
    let percentage = (value - minValue) / (maxValue - minValue);
    let formatter = $e3Dei$useNumberFormatter(formatOptions);
    if (!isIndeterminate && !valueLabel) {
        let valueToFormat = formatOptions.style === 'percent' ? percentage : value;
        valueLabel = formatter.format(valueToFormat);
    }
    return {
        progressBarProps: $e3Dei$mergeProps(domProps, {
            ...fieldProps,
            'aria-valuenow': isIndeterminate ? undefined : value,
            'aria-valuemin': minValue,
            'aria-valuemax': maxValue,
            'aria-valuetext': isIndeterminate ? undefined : valueLabel,
            role: 'progressbar'
        }),
        labelProps: labelProps
    };
}




export {$204d9ebcedfb8806$export$ed5abd763a836edc as useProgressBar};
