// @react-spectrum/listbox/dist/main.js
require("./main.css");
var $3Fwf2$reactstatelycollections = require("@react-stately/collections");
var $3Fwf2$react = require("react");
var $3Fwf2$reactspectrumutils = require("@react-spectrum/utils");
var $3Fwf2$reactstatelylist = require("@react-stately/list");
var $3Fwf2$reactarialistbox = require("@react-aria/listbox");
var $3Fwf2$reactstatelylayout = require("@react-stately/layout");
var $3Fwf2$reactariautils = require("@react-aria/utils");
var $3Fwf2$reactspectrumprogress = require("@react-spectrum/progress");
var $3Fwf2$reactariai18n = require("@react-aria/i18n");
var $3Fwf2$reactspectrumprovider = require("@react-spectrum/provider");
var $3Fwf2$reactariavirtualizer = require("@react-aria/virtualizer");
var $3Fwf2$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $3Fwf2$reactariafocus = require("@react-aria/focus");
var $3Fwf2$reactspectrumlayout = require("@react-spectrum/layout");
var $3Fwf2$reactariainteractions = require("@react-aria/interactions");
var $3Fwf2$reactspectrumtext = require("@react-spectrum/text");
var $3Fwf2$reactariaseparator = require("@react-aria/separator");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Item", () => $3Fwf2$reactstatelycollections.Item);
$parcel$export(module.exports, "Section", () => $3Fwf2$reactstatelycollections.Section);
var $582d86e45be55635$exports = {};

$parcel$export($582d86e45be55635$exports, "ListBox", () => $582d86e45be55635$export$41f133550aa26f48);
var $60eb4b34c53310cb$exports = {};

$parcel$export($60eb4b34c53310cb$exports, "useListBoxLayout", () => $60eb4b34c53310cb$export$25768ea656ae32a7);
$parcel$export($60eb4b34c53310cb$exports, "ListBoxBase", () => $60eb4b34c53310cb$export$1afdcf349979fb7e);


var $b48c40096cc25703$exports = {};
var $db2f31a656e206dc$exports = {};
$db2f31a656e206dc$exports = JSON.parse("{\"loading\":\"جارٍ التحميل...\",\"loadingMore\":\"جارٍ تحميل المزيد...\"}");


var $5fc5de12245ee37a$exports = {};
$5fc5de12245ee37a$exports = JSON.parse("{\"loading\":\"Зареждане...\",\"loadingMore\":\"Зареждане на още...\"}");


var $c6716434befdc578$exports = {};
$c6716434befdc578$exports = JSON.parse("{\"loading\":\"Načítání...\",\"loadingMore\":\"Načítání dalších...\"}");


var $aff0838a0af4442c$exports = {};
$aff0838a0af4442c$exports = JSON.parse("{\"loading\":\"Indlæser ...\",\"loadingMore\":\"Indlæser flere ...\"}");


var $e23661cbd2fcf9d9$exports = {};
$e23661cbd2fcf9d9$exports = JSON.parse("{\"loading\":\"Laden...\",\"loadingMore\":\"Mehr laden ...\"}");


var $0665dd51687e6f20$exports = {};
$0665dd51687e6f20$exports = JSON.parse("{\"loading\":\"Φόρτωση...\",\"loadingMore\":\"Φόρτωση περισσότερων...\"}");


var $9db92f31bd81178a$exports = {};
$9db92f31bd81178a$exports = JSON.parse("{\"loading\":\"Loading…\",\"loadingMore\":\"Loading more…\"}");


var $b85a375140bc142b$exports = {};
$b85a375140bc142b$exports = JSON.parse("{\"loading\":\"Cargando…\",\"loadingMore\":\"Cargando más…\"}");


var $9d80b9bee9643813$exports = {};
$9d80b9bee9643813$exports = JSON.parse("{\"loading\":\"Laadimine...\",\"loadingMore\":\"Laadi rohkem...\"}");


var $32fac20f569c89b2$exports = {};
$32fac20f569c89b2$exports = JSON.parse("{\"loading\":\"Ladataan…\",\"loadingMore\":\"Ladataan lisää…\"}");


var $1ae83646fec0a00f$exports = {};
$1ae83646fec0a00f$exports = JSON.parse("{\"loading\":\"Chargement...\",\"loadingMore\":\"Chargement supplémentaire...\"}");


var $1e59e08ce00d3a62$exports = {};
$1e59e08ce00d3a62$exports = JSON.parse("{\"loading\":\"טוען...\",\"loadingMore\":\"טוען עוד...\"}");


var $e55a93edef01c480$exports = {};
$e55a93edef01c480$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"loadingMore\":\"Učitavam još...\"}");


var $c479dd99077bc32f$exports = {};
$c479dd99077bc32f$exports = JSON.parse("{\"loading\":\"Betöltés folyamatban…\",\"loadingMore\":\"Továbbiak betöltése folyamatban…\"}");


var $93e538a0db81bf51$exports = {};
$93e538a0db81bf51$exports = JSON.parse("{\"loading\":\"Caricamento...\",\"loadingMore\":\"Caricamento altri...\"}");


var $ba01846c1b5c1a2b$exports = {};
$ba01846c1b5c1a2b$exports = JSON.parse("{\"loading\":\"読み込み中...\",\"loadingMore\":\"さらに読み込み中...\"}");


var $9efc98866ffb3d03$exports = {};
$9efc98866ffb3d03$exports = JSON.parse("{\"loading\":\"로드 중\",\"loadingMore\":\"추가 로드 중\"}");


var $72e1b4485a8dff00$exports = {};
$72e1b4485a8dff00$exports = JSON.parse("{\"loading\":\"Įkeliama...\",\"loadingMore\":\"Įkeliama daugiau...\"}");


var $9d82f7842d0956d5$exports = {};
$9d82f7842d0956d5$exports = JSON.parse("{\"loading\":\"Notiek ielāde...\",\"loadingMore\":\"Tiek ielādēts vēl...\"}");


var $2eeedcf868421556$exports = {};
$2eeedcf868421556$exports = JSON.parse("{\"loading\":\"Laster inn ...\",\"loadingMore\":\"Laster inn flere ...\"}");


var $ce5a0f667a7b7bff$exports = {};
$ce5a0f667a7b7bff$exports = JSON.parse("{\"loading\":\"Laden...\",\"loadingMore\":\"Meer laden...\"}");


var $9700a49ad8081602$exports = {};
$9700a49ad8081602$exports = JSON.parse("{\"loading\":\"Ładowanie...\",\"loadingMore\":\"Wczytywanie większej liczby...\"}");


var $65c2d0648e10b51a$exports = {};
$65c2d0648e10b51a$exports = JSON.parse("{\"loading\":\"Carregando...\",\"loadingMore\":\"Carregando mais...\"}");


var $ce0c2fb246a5350d$exports = {};
$ce0c2fb246a5350d$exports = JSON.parse("{\"loading\":\"A carregar...\",\"loadingMore\":\"A carregar mais...\"}");


var $8af4a376429f1f9a$exports = {};
$8af4a376429f1f9a$exports = JSON.parse("{\"loading\":\"Se încarcă...\",\"loadingMore\":\"Se încarcă mai multe...\"}");


var $99024a2f9ca1b2c9$exports = {};
$99024a2f9ca1b2c9$exports = JSON.parse("{\"loading\":\"Загрузка...\",\"loadingMore\":\"Дополнительная загрузка...\"}");


var $304201c7620897f0$exports = {};
$304201c7620897f0$exports = JSON.parse("{\"loading\":\"Načítava sa...\",\"loadingMore\":\"Načítava sa viac...\"}");


var $713bd39bd769c331$exports = {};
$713bd39bd769c331$exports = JSON.parse("{\"loading\":\"Nalaganje...\",\"loadingMore\":\"Nalaganje več vsebine...\"}");


var $9864766ffc224b5e$exports = {};
$9864766ffc224b5e$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"loadingMore\":\"Učitavam još...\"}");


var $b3456f989b077e88$exports = {};
$b3456f989b077e88$exports = JSON.parse("{\"loading\":\"Läser in...\",\"loadingMore\":\"Läser in mer...\"}");


var $5255c7a45247b8dc$exports = {};
$5255c7a45247b8dc$exports = JSON.parse("{\"loading\":\"Yükleniyor...\",\"loadingMore\":\"Daha fazla yükleniyor...\"}");


var $8ee6f6af8918bfbf$exports = {};
$8ee6f6af8918bfbf$exports = JSON.parse("{\"loading\":\"Завантаження…\",\"loadingMore\":\"Завантаження інших об’єктів...\"}");


var $024aed1a793198cd$exports = {};
$024aed1a793198cd$exports = JSON.parse("{\"loading\":\"正在加载...\",\"loadingMore\":\"正在加载更多...\"}");


var $be53ff8daaa52db0$exports = {};
$be53ff8daaa52db0$exports = JSON.parse("{\"loading\":\"正在載入\",\"loadingMore\":\"正在載入更多…\"}");


$b48c40096cc25703$exports = {
    "ar-AE": $db2f31a656e206dc$exports,
    "bg-BG": $5fc5de12245ee37a$exports,
    "cs-CZ": $c6716434befdc578$exports,
    "da-DK": $aff0838a0af4442c$exports,
    "de-DE": $e23661cbd2fcf9d9$exports,
    "el-GR": $0665dd51687e6f20$exports,
    "en-US": $9db92f31bd81178a$exports,
    "es-ES": $b85a375140bc142b$exports,
    "et-EE": $9d80b9bee9643813$exports,
    "fi-FI": $32fac20f569c89b2$exports,
    "fr-FR": $1ae83646fec0a00f$exports,
    "he-IL": $1e59e08ce00d3a62$exports,
    "hr-HR": $e55a93edef01c480$exports,
    "hu-HU": $c479dd99077bc32f$exports,
    "it-IT": $93e538a0db81bf51$exports,
    "ja-JP": $ba01846c1b5c1a2b$exports,
    "ko-KR": $9efc98866ffb3d03$exports,
    "lt-LT": $72e1b4485a8dff00$exports,
    "lv-LV": $9d82f7842d0956d5$exports,
    "nb-NO": $2eeedcf868421556$exports,
    "nl-NL": $ce5a0f667a7b7bff$exports,
    "pl-PL": $9700a49ad8081602$exports,
    "pt-BR": $65c2d0648e10b51a$exports,
    "pt-PT": $ce0c2fb246a5350d$exports,
    "ro-RO": $8af4a376429f1f9a$exports,
    "ru-RU": $99024a2f9ca1b2c9$exports,
    "sk-SK": $304201c7620897f0$exports,
    "sl-SI": $713bd39bd769c331$exports,
    "sr-SP": $9864766ffc224b5e$exports,
    "sv-SE": $b3456f989b077e88$exports,
    "tr-TR": $5255c7a45247b8dc$exports,
    "uk-UA": $8ee6f6af8918bfbf$exports,
    "zh-CN": $024aed1a793198cd$exports,
    "zh-TW": $be53ff8daaa52db0$exports
};



const $611e93c7b7c0413c$export$7ff8f37d2d81a48d = ($parcel$interopDefault($3Fwf2$react)).createContext(null);










var $3deee0fd2b46e64f$exports = {};

$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-popover", () => $3deee0fd2b46e64f$export$d959f30786125b, (v) => $3deee0fd2b46e64f$export$d959f30786125b = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu", () => $3deee0fd2b46e64f$export$fae6af6decd50ac5, (v) => $3deee0fd2b46e64f$export$fae6af6decd50ac5 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-sectionHeading", () => $3deee0fd2b46e64f$export$c20ea4b35020fbdd, (v) => $3deee0fd2b46e64f$export$c20ea4b35020fbdd = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-checkmark", () => $3deee0fd2b46e64f$export$8e15a726e4f7e62d, (v) => $3deee0fd2b46e64f$export$8e15a726e4f7e62d = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-item", () => $3deee0fd2b46e64f$export$45faad328e6c532b, (v) => $3deee0fd2b46e64f$export$45faad328e6c532b = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-selected", () => $3deee0fd2b46e64f$export$1e0fb04f31d3c22a, (v) => $3deee0fd2b46e64f$export$1e0fb04f31d3c22a = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Icon", () => $3deee0fd2b46e64f$export$d374b04f30360026, (v) => $3deee0fd2b46e64f$export$d374b04f30360026 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemLabel", () => $3deee0fd2b46e64f$export$1aa6419c5c650567, (v) => $3deee0fd2b46e64f$export$1aa6419c5c650567 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemIcon", () => $3deee0fd2b46e64f$export$f431001e1bb54633, (v) => $3deee0fd2b46e64f$export$f431001e1bb54633 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemLabel--wrapping", () => $3deee0fd2b46e64f$export$465092164a8e1d98, (v) => $3deee0fd2b46e64f$export$465092164a8e1d98 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-divider", () => $3deee0fd2b46e64f$export$e892026f534efed8, (v) => $3deee0fd2b46e64f$export$e892026f534efed8 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemGrid", () => $3deee0fd2b46e64f$export$ee5f52784f57f9c8, (v) => $3deee0fd2b46e64f$export$ee5f52784f57f9c8 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-selectable", () => $3deee0fd2b46e64f$export$fbd22ba224473193, (v) => $3deee0fd2b46e64f$export$fbd22ba224473193 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-end", () => $3deee0fd2b46e64f$export$926a1bb74ed65667, (v) => $3deee0fd2b46e64f$export$926a1bb74ed65667 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-icon", () => $3deee0fd2b46e64f$export$cbbae63ef9e44c7a, (v) => $3deee0fd2b46e64f$export$cbbae63ef9e44c7a = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-description", () => $3deee0fd2b46e64f$export$36f1275051e39ead, (v) => $3deee0fd2b46e64f$export$36f1275051e39ead = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-keyboard", () => $3deee0fd2b46e64f$export$4f9817a7dff6ba0f, (v) => $3deee0fd2b46e64f$export$4f9817a7dff6ba0f = v);
$parcel$export($3deee0fd2b46e64f$exports, "focus-ring", () => $3deee0fd2b46e64f$export$f39a09f249340e2a, (v) => $3deee0fd2b46e64f$export$f39a09f249340e2a = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-focused", () => $3deee0fd2b46e64f$export$e7dc768d35940237, (v) => $3deee0fd2b46e64f$export$e7dc768d35940237 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-hovered", () => $3deee0fd2b46e64f$export$b8813cd5d7824ce7, (v) => $3deee0fd2b46e64f$export$b8813cd5d7824ce7 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-highlighted", () => $3deee0fd2b46e64f$export$451c613d0772eb99, (v) => $3deee0fd2b46e64f$export$451c613d0772eb99 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-open", () => $3deee0fd2b46e64f$export$a9781837241c946d, (v) => $3deee0fd2b46e64f$export$a9781837241c946d = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-active", () => $3deee0fd2b46e64f$export$20fd0f7cd4e6112f, (v) => $3deee0fd2b46e64f$export$20fd0f7cd4e6112f = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-disabled", () => $3deee0fd2b46e64f$export$d35bc1e505d1ebbf, (v) => $3deee0fd2b46e64f$export$d35bc1e505d1ebbf = v);
var $3deee0fd2b46e64f$export$d959f30786125b;
var $3deee0fd2b46e64f$export$fae6af6decd50ac5;
var $3deee0fd2b46e64f$export$c20ea4b35020fbdd;
var $3deee0fd2b46e64f$export$8e15a726e4f7e62d;
var $3deee0fd2b46e64f$export$45faad328e6c532b;
var $3deee0fd2b46e64f$export$1e0fb04f31d3c22a;
var $3deee0fd2b46e64f$export$d374b04f30360026;
var $3deee0fd2b46e64f$export$1aa6419c5c650567;
var $3deee0fd2b46e64f$export$f431001e1bb54633;
var $3deee0fd2b46e64f$export$465092164a8e1d98;
var $3deee0fd2b46e64f$export$e892026f534efed8;
var $3deee0fd2b46e64f$export$ee5f52784f57f9c8;
var $3deee0fd2b46e64f$export$fbd22ba224473193;
var $3deee0fd2b46e64f$export$926a1bb74ed65667;
var $3deee0fd2b46e64f$export$cbbae63ef9e44c7a;
var $3deee0fd2b46e64f$export$36f1275051e39ead;
var $3deee0fd2b46e64f$export$4f9817a7dff6ba0f;
var $3deee0fd2b46e64f$export$f39a09f249340e2a;
var $3deee0fd2b46e64f$export$e7dc768d35940237;
var $3deee0fd2b46e64f$export$b8813cd5d7824ce7;
var $3deee0fd2b46e64f$export$451c613d0772eb99;
var $3deee0fd2b46e64f$export$a9781837241c946d;
var $3deee0fd2b46e64f$export$20fd0f7cd4e6112f;
var $3deee0fd2b46e64f$export$d35bc1e505d1ebbf;
$3deee0fd2b46e64f$export$d959f30786125b = "spectrum-Menu-popover_7ea47c";
$3deee0fd2b46e64f$export$fae6af6decd50ac5 = "spectrum-Menu_7ea47c";
$3deee0fd2b46e64f$export$c20ea4b35020fbdd = "spectrum-Menu-sectionHeading_7ea47c";
$3deee0fd2b46e64f$export$8e15a726e4f7e62d = "spectrum-Menu-checkmark_7ea47c";
$3deee0fd2b46e64f$export$45faad328e6c532b = "spectrum-Menu-item_7ea47c";
$3deee0fd2b46e64f$export$1e0fb04f31d3c22a = "is-selected_7ea47c";
$3deee0fd2b46e64f$export$d374b04f30360026 = "spectrum-Icon_7ea47c";
$3deee0fd2b46e64f$export$1aa6419c5c650567 = "spectrum-Menu-itemLabel_7ea47c";
$3deee0fd2b46e64f$export$f431001e1bb54633 = "spectrum-Menu-itemIcon_7ea47c";
$3deee0fd2b46e64f$export$465092164a8e1d98 = "spectrum-Menu-itemLabel--wrapping_7ea47c";
$3deee0fd2b46e64f$export$e892026f534efed8 = "spectrum-Menu-divider_7ea47c";
$3deee0fd2b46e64f$export$ee5f52784f57f9c8 = "spectrum-Menu-itemGrid_7ea47c";
$3deee0fd2b46e64f$export$fbd22ba224473193 = "is-selectable_7ea47c";
$3deee0fd2b46e64f$export$926a1bb74ed65667 = "spectrum-Menu-end_7ea47c";
$3deee0fd2b46e64f$export$cbbae63ef9e44c7a = "spectrum-Menu-icon_7ea47c";
$3deee0fd2b46e64f$export$36f1275051e39ead = "spectrum-Menu-description_7ea47c";
$3deee0fd2b46e64f$export$4f9817a7dff6ba0f = "spectrum-Menu-keyboard_7ea47c";
$3deee0fd2b46e64f$export$f39a09f249340e2a = "focus-ring_7ea47c";
$3deee0fd2b46e64f$export$e7dc768d35940237 = "is-focused_7ea47c";
$3deee0fd2b46e64f$export$b8813cd5d7824ce7 = "is-hovered_7ea47c";
$3deee0fd2b46e64f$export$451c613d0772eb99 = "is-highlighted_7ea47c";
$3deee0fd2b46e64f$export$a9781837241c946d = "is-open_7ea47c";
$3deee0fd2b46e64f$export$20fd0f7cd4e6112f = "is-active_7ea47c";
$3deee0fd2b46e64f$export$d35bc1e505d1ebbf = "is-disabled_7ea47c";





function $2205d250ee1cb1ac$export$feb3b6b552c14a12(props) {
    let { item: item , shouldSelectOnPressUp: shouldSelectOnPressUp , shouldFocusOnHover: shouldFocusOnHover , shouldUseVirtualFocus: shouldUseVirtualFocus  } = props;
    let { rendered: rendered , key: key  } = item;
    let state = $3Fwf2$react.useContext($611e93c7b7c0413c$export$7ff8f37d2d81a48d);
    let ref = $3Fwf2$react.useRef();
    let { optionProps: optionProps , labelProps: labelProps , descriptionProps: descriptionProps , isSelected: isSelected , isDisabled: isDisabled , isFocused: isFocused  } = $3Fwf2$reactarialistbox.useOption({
        'aria-label': item['aria-label'],
        key: key,
        shouldSelectOnPressUp: shouldSelectOnPressUp,
        shouldFocusOnHover: shouldFocusOnHover,
        isVirtualized: true,
        shouldUseVirtualFocus: shouldUseVirtualFocus
    }, state, ref);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $3Fwf2$reactariainteractions.useHover({
        ...props,
        isDisabled: isDisabled
    });
    let contents = typeof rendered === 'string' ? /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($3Fwf2$reactspectrumtext.Text, null, rendered) : rendered;
    let isKeyboardModality = $3Fwf2$reactariainteractions.isFocusVisible();
    return(/*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($3Fwf2$reactariafocus.FocusRing, {
        focusRingClass: $3Fwf2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement("div", {
        ...$3Fwf2$reactariautils.mergeProps(optionProps, shouldFocusOnHover ? {
        } : hoverProps),
        ref: ref,
        className: $3Fwf2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)), 'spectrum-Menu-item', {
            // If using virtual focus, apply focused styles to the item when the user is interacting with keyboard modality
            'is-focused': shouldUseVirtualFocus && isFocused && isKeyboardModality,
            'is-disabled': isDisabled,
            'is-selected': isSelected,
            'is-selectable': state.selectionManager.selectionMode !== 'none',
            // When shouldFocusOnHover is false, apply hover styles both when hovered with the mouse.
            // Otherwise, apply hover styles when focused using non-keyboard modality.
            'is-hovered': isHovered && !shouldFocusOnHover || isFocused && !isKeyboardModality
        })
    }, /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($3Fwf2$reactspectrumlayout.Grid, {
        UNSAFE_className: $3Fwf2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)), 'spectrum-Menu-itemGrid')
    }, /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($3Fwf2$reactspectrumutils.ClearSlots, null, /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($3Fwf2$reactspectrumutils.SlotProvider, {
        slots: {
            text: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))['spectrum-Menu-itemLabel'],
                ...labelProps
            },
            icon: {
                size: 'S',
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))['spectrum-Menu-icon']
            },
            description: {
                UNSAFE_className: (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))['spectrum-Menu-description'],
                ...descriptionProps
            }
        }
    }, contents, isSelected && /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement(($parcel$interopDefault($3Fwf2$spectrumiconsuiCheckmarkMedium)), {
        slot: "checkmark",
        UNSAFE_className: $3Fwf2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)), 'spectrum-Menu-checkmark')
    })))))));
}










function $0f5fd8483f48ef09$export$dca12b0bb56e4fc(props) {
    let { children: children , reusableView: reusableView , header: header  } = props;
    let item = reusableView.content;
    let { headingProps: headingProps , groupProps: groupProps  } = $3Fwf2$reactarialistbox.useListBoxSection({
        heading: item.rendered,
        'aria-label': item['aria-label']
    });
    let { separatorProps: separatorProps  } = $3Fwf2$reactariaseparator.useSeparator({
        elementType: 'li'
    });
    let headerRef = $3Fwf2$react.useRef();
    $3Fwf2$reactariavirtualizer.useVirtualizerItem({
        reusableView: header,
        ref: headerRef
    });
    let { direction: direction  } = $3Fwf2$reactariai18n.useLocale();
    let state = $3Fwf2$react.useContext($611e93c7b7c0413c$export$7ff8f37d2d81a48d);
    return(/*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($3Fwf2$react.Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement("div", {
        role: "presentation",
        ref: headerRef,
        style: $3Fwf2$reactariavirtualizer.layoutInfoToStyle(header.layoutInfo, direction)
    }, item.key !== state.collection.getFirstKey() && /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement("div", {
        ...separatorProps,
        className: $3Fwf2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)), 'spectrum-Menu-divider')
    }), item.rendered && /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement("div", {
        ...headingProps,
        className: $3Fwf2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)), 'spectrum-Menu-sectionHeading')
    }, item.rendered)), /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement("div", {
        ...groupProps,
        style: $3Fwf2$reactariavirtualizer.layoutInfoToStyle(reusableView.layoutInfo, direction),
        className: $3Fwf2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)), 'spectrum-Menu')
    }, children)));
}










function $60eb4b34c53310cb$export$25768ea656ae32a7(state) {
    let { scale: scale  } = $3Fwf2$reactspectrumprovider.useProvider();
    let collator = $3Fwf2$reactariai18n.useCollator({
        usage: 'search',
        sensitivity: 'base'
    });
    let layout = $3Fwf2$react.useMemo(()=>new $3Fwf2$reactstatelylayout.ListLayout({
            estimatedRowHeight: scale === 'large' ? 48 : 32,
            estimatedHeadingHeight: scale === 'large' ? 33 : 26,
            padding: scale === 'large' ? 5 : 4,
            loaderHeight: 40,
            placeholderHeight: scale === 'large' ? 48 : 32,
            collator: collator
        })
    , [
        collator,
        scale
    ]);
    layout.collection = state.collection;
    layout.disabledKeys = state.disabledKeys;
    return layout;
}
/** @private */ function $60eb4b34c53310cb$var$ListBoxBase(props, ref) {
    let { layout: layout , state: state , shouldSelectOnPressUp: shouldSelectOnPressUp , focusOnPointerEnter: focusOnPointerEnter , shouldUseVirtualFocus: shouldUseVirtualFocus , domProps: domProps = {
    } , transitionDuration: transitionDuration = 0 , onScroll: onScroll  } = props;
    let { listBoxProps: listBoxProps  } = $3Fwf2$reactarialistbox.useListBox({
        ...props,
        keyboardDelegate: layout,
        isVirtualized: true
    }, state, ref);
    let { styleProps: styleProps  } = $3Fwf2$reactspectrumutils.useStyleProps(props);
    let formatMessage = $3Fwf2$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($b48c40096cc25703$exports)));
    // Sync loading state into the layout.
    layout.isLoading = props.isLoading;
    let renderWrapper = (parent, reusableView, children, renderChildren)=>{
        if (reusableView.viewType === 'section') return(/*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($0f5fd8483f48ef09$export$dca12b0bb56e4fc, {
            key: reusableView.key,
            reusableView: reusableView,
            header: children.find((c)=>c.viewType === 'header'
            )
        }, renderChildren(children.filter((c)=>c.viewType === 'item'
        ))));
        return(/*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($3Fwf2$reactariavirtualizer.VirtualizerItem, {
            key: reusableView.key,
            reusableView: reusableView,
            parent: parent
        }));
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($611e93c7b7c0413c$export$7ff8f37d2d81a48d.Provider, {
        value: state
    }, /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($3Fwf2$reactariavirtualizer.Virtualizer, {
        ...styleProps,
        ...$3Fwf2$reactariautils.mergeProps(listBoxProps, domProps),
        ref: ref,
        focusedKey: state.selectionManager.focusedKey,
        sizeToFit: "height",
        scrollDirection: "vertical",
        className: $3Fwf2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)), 'spectrum-Menu', styleProps.className),
        layout: layout,
        collection: state.collection,
        renderWrapper: renderWrapper,
        transitionDuration: transitionDuration,
        isLoading: props.isLoading,
        onLoadMore: props.onLoadMore,
        shouldUseVirtualFocus: shouldUseVirtualFocus,
        onScroll: onScroll
    }, (type, item)=>{
        if (type === 'item') return(/*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($2205d250ee1cb1ac$export$feb3b6b552c14a12, {
            item: item,
            shouldSelectOnPressUp: shouldSelectOnPressUp,
            shouldFocusOnHover: focusOnPointerEnter,
            shouldUseVirtualFocus: shouldUseVirtualFocus
        }));
        else if (type === 'loader') return(// aria-selected isn't needed here since this option is not selectable.
        // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
        /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement("div", {
            role: "option",
            style: {
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
                height: '100%'
            }
        }, /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($3Fwf2$reactspectrumprogress.ProgressCircle, {
            isIndeterminate: true,
            size: "S",
            "aria-label": state.collection.size > 0 ? formatMessage('loadingMore') : formatMessage('loading'),
            UNSAFE_className: $3Fwf2$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)), 'spectrum-Dropdown-progressCircle')
        })));
        else if (type === 'placeholder') {
            let emptyState = props.renderEmptyState ? props.renderEmptyState() : null;
            if (emptyState == null) return null;
            return(/*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement("div", {
                // aria-selected isn't needed here since this option is not selectable.
                // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
                role: "option"
            }, emptyState));
        }
    })));
}
// forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $60eb4b34c53310cb$export$1afdcf349979fb7e = /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).forwardRef($60eb4b34c53310cb$var$ListBoxBase);





function $582d86e45be55635$var$ListBox(props, ref) {
    let state = $3Fwf2$reactstatelylist.useListState(props);
    let layout = $60eb4b34c53310cb$export$25768ea656ae32a7(state);
    let domRef = $3Fwf2$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).createElement($60eb4b34c53310cb$export$1afdcf349979fb7e, {
        ...props,
        ref: domRef,
        state: state,
        layout: layout
    }));
}
// forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
/**
 * A list of options that can allow selection of one or more.
 */ const $582d86e45be55635$export$41f133550aa26f48 = /*#__PURE__*/ ($parcel$interopDefault($3Fwf2$react)).forwardRef($582d86e45be55635$var$ListBox);




$parcel$exportWildcard(module.exports, $582d86e45be55635$exports);
$parcel$exportWildcard(module.exports, $60eb4b34c53310cb$exports);
