// @react-aria/switch/dist/main.js
var $3VG2V$reactariatoggle = require("@react-aria/toggle");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $e64bc6eb3aa7bd8d$exports = {};

$parcel$export($e64bc6eb3aa7bd8d$exports, "useSwitch", () => $e64bc6eb3aa7bd8d$export$d853f7095ae95f88);

function $e64bc6eb3aa7bd8d$export$d853f7095ae95f88(props, state, ref) {
    let { inputProps: inputProps  } = $3VG2V$reactariatoggle.useToggle(props, state, ref);
    let { isSelected: isSelected  } = state;
    return {
        inputProps: {
            ...inputProps,
            role: 'switch',
            checked: isSelected,
            'aria-checked': isSelected
        }
    };
}


$parcel$exportWildcard(module.exports, $e64bc6eb3aa7bd8d$exports);
