// @react-spectrum/meter/dist/main.js
require("./main.css");
var $6eMp1$reactspectrumutils = require("@react-spectrum/utils");
var $6eMp1$reactspectrumprogress = require("@react-spectrum/progress");
var $6eMp1$react = require("react");
var $6eMp1$reactariameter = require("@react-aria/meter");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $fc7fd823390a6f61$exports = {};

$parcel$export($fc7fd823390a6f61$exports, "Meter", () => $fc7fd823390a6f61$export$62e3ae2a4090b879);



var $9f78a6c009f32d0a$exports = {};

$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader", () => $9f78a6c009f32d0a$export$162ba336fb0c239f, (v) => $9f78a6c009f32d0a$export$162ba336fb0c239f = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--indeterminate", () => $9f78a6c009f32d0a$export$1249dfbb98223794, (v) => $9f78a6c009f32d0a$export$1249dfbb98223794 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--sideLabel", () => $9f78a6c009f32d0a$export$99af8802cc2a10dc, (v) => $9f78a6c009f32d0a$export$99af8802cc2a10dc = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-track", () => $9f78a6c009f32d0a$export$3e2e13fb57b5c74b, (v) => $9f78a6c009f32d0a$export$3e2e13fb57b5c74b = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-fill", () => $9f78a6c009f32d0a$export$84c9ee7bc724b71b, (v) => $9f78a6c009f32d0a$export$84c9ee7bc724b71b = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-label", () => $9f78a6c009f32d0a$export$5379ba198163be86, (v) => $9f78a6c009f32d0a$export$5379ba198163be86 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-percentage", () => $9f78a6c009f32d0a$export$ecc9c825de9d1a18, (v) => $9f78a6c009f32d0a$export$ecc9c825de9d1a18 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--small", () => $9f78a6c009f32d0a$export$b945305960ad74a7, (v) => $9f78a6c009f32d0a$export$b945305960ad74a7 = v);
$parcel$export($9f78a6c009f32d0a$exports, "indeterminate-loop-ltr", () => $9f78a6c009f32d0a$export$a889038a25113dd4, (v) => $9f78a6c009f32d0a$export$a889038a25113dd4 = v);
$parcel$export($9f78a6c009f32d0a$exports, "indeterminate-loop-rtl", () => $9f78a6c009f32d0a$export$aeff243cf3356f00, (v) => $9f78a6c009f32d0a$export$aeff243cf3356f00 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--overBackground", () => $9f78a6c009f32d0a$export$55612d0bd9b5d41, (v) => $9f78a6c009f32d0a$export$55612d0bd9b5d41 = v);
$parcel$export($9f78a6c009f32d0a$exports, "is-positive", () => $9f78a6c009f32d0a$export$aea9fd919467f03a, (v) => $9f78a6c009f32d0a$export$aea9fd919467f03a = v);
$parcel$export($9f78a6c009f32d0a$exports, "is-warning", () => $9f78a6c009f32d0a$export$d4ebda6218901908, (v) => $9f78a6c009f32d0a$export$d4ebda6218901908 = v);
$parcel$export($9f78a6c009f32d0a$exports, "is-critical", () => $9f78a6c009f32d0a$export$a1f608a9810da50f, (v) => $9f78a6c009f32d0a$export$a1f608a9810da50f = v);
var $9f78a6c009f32d0a$export$162ba336fb0c239f;
var $9f78a6c009f32d0a$export$1249dfbb98223794;
var $9f78a6c009f32d0a$export$99af8802cc2a10dc;
var $9f78a6c009f32d0a$export$3e2e13fb57b5c74b;
var $9f78a6c009f32d0a$export$84c9ee7bc724b71b;
var $9f78a6c009f32d0a$export$5379ba198163be86;
var $9f78a6c009f32d0a$export$ecc9c825de9d1a18;
var $9f78a6c009f32d0a$export$b945305960ad74a7;
var $9f78a6c009f32d0a$export$a889038a25113dd4;
var $9f78a6c009f32d0a$export$aeff243cf3356f00;
var $9f78a6c009f32d0a$export$55612d0bd9b5d41;
var $9f78a6c009f32d0a$export$aea9fd919467f03a;
var $9f78a6c009f32d0a$export$d4ebda6218901908;
var $9f78a6c009f32d0a$export$a1f608a9810da50f;
$9f78a6c009f32d0a$export$162ba336fb0c239f = "spectrum-BarLoader_48541f";
$9f78a6c009f32d0a$export$1249dfbb98223794 = "spectrum-BarLoader--indeterminate_48541f";
$9f78a6c009f32d0a$export$99af8802cc2a10dc = "spectrum-BarLoader--sideLabel_48541f";
$9f78a6c009f32d0a$export$3e2e13fb57b5c74b = "spectrum-BarLoader-track_48541f";
$9f78a6c009f32d0a$export$84c9ee7bc724b71b = "spectrum-BarLoader-fill_48541f";
$9f78a6c009f32d0a$export$5379ba198163be86 = "spectrum-BarLoader-label_48541f";
$9f78a6c009f32d0a$export$ecc9c825de9d1a18 = "spectrum-BarLoader-percentage_48541f";
$9f78a6c009f32d0a$export$b945305960ad74a7 = "spectrum-BarLoader--small_48541f";
$9f78a6c009f32d0a$export$a889038a25113dd4 = "indeterminate-loop-ltr_48541f";
$9f78a6c009f32d0a$export$aeff243cf3356f00 = "indeterminate-loop-rtl_48541f";
$9f78a6c009f32d0a$export$55612d0bd9b5d41 = "spectrum-BarLoader--overBackground_48541f";
$9f78a6c009f32d0a$export$aea9fd919467f03a = "is-positive_48541f";
$9f78a6c009f32d0a$export$d4ebda6218901908 = "is-warning_48541f";
$9f78a6c009f32d0a$export$a1f608a9810da50f = "is-critical_48541f";



function $fc7fd823390a6f61$var$Meter(props, ref) {
    let { variant: variant , ...otherProps } = props;
    const { meterProps: meterProps , labelProps: labelProps  } = $6eMp1$reactariameter.useMeter(props);
    return(/*#__PURE__*/ ($parcel$interopDefault($6eMp1$react)).createElement($6eMp1$reactspectrumprogress.ProgressBarBase, {
        ...otherProps,
        ref: ref,
        barProps: meterProps,
        labelProps: labelProps,
        barClassName: $6eMp1$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($9f78a6c009f32d0a$exports)), {
            'is-positive': variant === 'positive',
            'is-warning': variant === 'warning',
            'is-critical': variant === 'critical'
        })
    }));
}
/**
 * Meters are visual representations of a quantity or an achievement.
 * Their progress is determined by user actions, rather than system actions.
 */ let $fc7fd823390a6f61$export$62e3ae2a4090b879 = /*#__PURE__*/ ($parcel$interopDefault($6eMp1$react)).forwardRef($fc7fd823390a6f61$var$Meter);


$parcel$exportWildcard(module.exports, $fc7fd823390a6f61$exports);
