// @react-aria/separator/dist/main.js
var $dDFJl$reactariautils = require("@react-aria/utils");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $8fe2a70229d5746b$exports = {};

$parcel$export($8fe2a70229d5746b$exports, "useSeparator", () => $8fe2a70229d5746b$export$52210f68a14655d0);

function $8fe2a70229d5746b$export$52210f68a14655d0(props) {
    let domProps = $dDFJl$reactariautils.filterDOMProps(props, {
        labelable: true
    });
    let ariaOrientation;
    // if orientation is horizontal, aria-orientation default is horizontal, so we leave it undefined
    // if it's vertical, we need to specify it
    if (props.orientation === 'vertical') ariaOrientation = 'vertical';
    // hr elements implicitly have role = separator and a horizontal orientation
    if (props.elementType !== 'hr') return {
        separatorProps: {
            ...domProps,
            role: 'separator',
            'aria-orientation': ariaOrientation
        }
    };
    return {
        separatorProps: domProps
    };
}


$parcel$exportWildcard(module.exports, $8fe2a70229d5746b$exports);
