// @react-spectrum/view/dist/main.js
var $3YgEH$reactspectrumutils = require("@react-spectrum/utils");
var $3YgEH$reactariautils = require("@react-aria/utils");
var $3YgEH$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $6c848be8e8bffe98$exports = {};

$parcel$export($6c848be8e8bffe98$exports, "View", () => $6c848be8e8bffe98$export$27a5bd065ad55220);



function $6c848be8e8bffe98$var$View(props, ref) {
    props = $3YgEH$reactspectrumutils.useSlotProps(props);
    let { elementType: ElementType = 'div' , children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $3YgEH$reactspectrumutils.useStyleProps(props, $3YgEH$reactspectrumutils.viewStyleProps);
    let domRef = $3YgEH$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($3YgEH$react)).createElement(ElementType, {
        ...$3YgEH$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($3YgEH$react)).createElement($3YgEH$reactspectrumutils.ClearSlots, null, children)));
}
/**
 * View is a general purpose container with no specific semantics that can be used for custom styling purposes.
 * It supports Spectrum style props to ensure consistency with other Spectrum components.
 */ const $6c848be8e8bffe98$export$27a5bd065ad55220 = /*#__PURE__*/ $3YgEH$react.forwardRef($6c848be8e8bffe98$var$View);


var $2d8ae89d4e4c1d41$exports = {};

$parcel$export($2d8ae89d4e4c1d41$exports, "Content", () => $2d8ae89d4e4c1d41$export$7c6e2c02157bb7d2);




function $2d8ae89d4e4c1d41$var$Content(props, ref) {
    props = $3YgEH$reactspectrumutils.useSlotProps(props, 'content');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $3YgEH$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $3YgEH$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($3YgEH$react)).createElement("section", {
        ...$3YgEH$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($3YgEH$react)).createElement($3YgEH$reactspectrumutils.ClearSlots, null, children)));
}
/**
 * Content represents the primary content within a Spectrum container.
 */ const $2d8ae89d4e4c1d41$export$7c6e2c02157bb7d2 = /*#__PURE__*/ $3YgEH$react.forwardRef($2d8ae89d4e4c1d41$var$Content);


var $e266c0b33fd8525d$exports = {};

$parcel$export($e266c0b33fd8525d$exports, "Footer", () => $e266c0b33fd8525d$export$a06f1c675e846f6f);



function $e266c0b33fd8525d$var$Footer(props, ref) {
    props = $3YgEH$reactspectrumutils.useSlotProps(props, 'footer');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $3YgEH$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $3YgEH$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($3YgEH$react)).createElement("footer", {
        ...$3YgEH$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($3YgEH$react)).createElement($3YgEH$reactspectrumutils.ClearSlots, null, children)));
}
/**
 * Footer represents a footer within a Spectrum container.
 */ const $e266c0b33fd8525d$export$a06f1c675e846f6f = /*#__PURE__*/ $3YgEH$react.forwardRef($e266c0b33fd8525d$var$Footer);


var $62d558b43ecd7a77$exports = {};

$parcel$export($62d558b43ecd7a77$exports, "Header", () => $62d558b43ecd7a77$export$8b251419efc915eb);



function $62d558b43ecd7a77$var$Header(props, ref) {
    props = $3YgEH$reactspectrumutils.useSlotProps(props, 'header');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $3YgEH$reactspectrumutils.useStyleProps(otherProps);
    let domRef = $3YgEH$reactspectrumutils.useDOMRef(ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($3YgEH$react)).createElement("header", {
        ...$3YgEH$reactariautils.filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($3YgEH$react)).createElement($3YgEH$reactspectrumutils.ClearSlots, null, children)));
}
/**
 * Header represents a header within a Spectrum container.
 */ const $62d558b43ecd7a77$export$8b251419efc915eb = /*#__PURE__*/ $3YgEH$react.forwardRef($62d558b43ecd7a77$var$Header);


$parcel$exportWildcard(module.exports, $6c848be8e8bffe98$exports);
$parcel$exportWildcard(module.exports, $2d8ae89d4e4c1d41$exports);
$parcel$exportWildcard(module.exports, $e266c0b33fd8525d$exports);
$parcel$exportWildcard(module.exports, $62d558b43ecd7a77$exports);
