// @react-spectrum/view/dist/module.js
import {useSlotProps as $bJbxN$useSlotProps, useStyleProps as $bJbxN$useStyleProps, viewStyleProps as $bJbxN$viewStyleProps, useDOMRef as $bJbxN$useDOMRef, ClearSlots as $bJbxN$ClearSlots} from "@react-spectrum/utils";
import {filterDOMProps as $bJbxN$filterDOMProps} from "@react-aria/utils";
import $bJbxN$react, {forwardRef as $bJbxN$forwardRef} from "react";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $b9606c0c41d55371$exports = {};

$parcel$export($b9606c0c41d55371$exports, "View", () => $b9606c0c41d55371$export$27a5bd065ad55220);



function $b9606c0c41d55371$var$View(props, ref) {
    props = $bJbxN$useSlotProps(props);
    let { elementType: ElementType = 'div' , children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $bJbxN$useStyleProps(props, $bJbxN$viewStyleProps);
    let domRef = $bJbxN$useDOMRef(ref);
    return(/*#__PURE__*/ $bJbxN$react.createElement(ElementType, {
        ...$bJbxN$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ $bJbxN$react.createElement($bJbxN$ClearSlots, null, children)));
}
/**
 * View is a general purpose container with no specific semantics that can be used for custom styling purposes.
 * It supports Spectrum style props to ensure consistency with other Spectrum components.
 */ const $b9606c0c41d55371$export$27a5bd065ad55220 = /*#__PURE__*/ $bJbxN$forwardRef($b9606c0c41d55371$var$View);


var $0a7c18a2ef74b280$exports = {};

$parcel$export($0a7c18a2ef74b280$exports, "Content", () => $0a7c18a2ef74b280$export$7c6e2c02157bb7d2);




function $0a7c18a2ef74b280$var$Content(props, ref) {
    props = $bJbxN$useSlotProps(props, 'content');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $bJbxN$useStyleProps(otherProps);
    let domRef = $bJbxN$useDOMRef(ref);
    return(/*#__PURE__*/ $bJbxN$react.createElement("section", {
        ...$bJbxN$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ $bJbxN$react.createElement($bJbxN$ClearSlots, null, children)));
}
/**
 * Content represents the primary content within a Spectrum container.
 */ const $0a7c18a2ef74b280$export$7c6e2c02157bb7d2 = /*#__PURE__*/ $bJbxN$forwardRef($0a7c18a2ef74b280$var$Content);


var $8d68cc89c1f9278e$exports = {};

$parcel$export($8d68cc89c1f9278e$exports, "Footer", () => $8d68cc89c1f9278e$export$a06f1c675e846f6f);



function $8d68cc89c1f9278e$var$Footer(props, ref) {
    props = $bJbxN$useSlotProps(props, 'footer');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $bJbxN$useStyleProps(otherProps);
    let domRef = $bJbxN$useDOMRef(ref);
    return(/*#__PURE__*/ $bJbxN$react.createElement("footer", {
        ...$bJbxN$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ $bJbxN$react.createElement($bJbxN$ClearSlots, null, children)));
}
/**
 * Footer represents a footer within a Spectrum container.
 */ const $8d68cc89c1f9278e$export$a06f1c675e846f6f = /*#__PURE__*/ $bJbxN$forwardRef($8d68cc89c1f9278e$var$Footer);


var $aedcc4886d2a392a$exports = {};

$parcel$export($aedcc4886d2a392a$exports, "Header", () => $aedcc4886d2a392a$export$8b251419efc915eb);



function $aedcc4886d2a392a$var$Header(props, ref) {
    props = $bJbxN$useSlotProps(props, 'header');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $bJbxN$useStyleProps(otherProps);
    let domRef = $bJbxN$useDOMRef(ref);
    return(/*#__PURE__*/ $bJbxN$react.createElement("header", {
        ...$bJbxN$filterDOMProps(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ $bJbxN$react.createElement($bJbxN$ClearSlots, null, children)));
}
/**
 * Header represents a header within a Spectrum container.
 */ const $aedcc4886d2a392a$export$8b251419efc915eb = /*#__PURE__*/ $bJbxN$forwardRef($aedcc4886d2a392a$var$Header);




export {$b9606c0c41d55371$export$27a5bd065ad55220 as View, $0a7c18a2ef74b280$export$7c6e2c02157bb7d2 as Content, $8d68cc89c1f9278e$export$a06f1c675e846f6f as Footer, $aedcc4886d2a392a$export$8b251419efc915eb as Header};
