// @react-spectrum/table/dist/main.js
require("./main.css");
var $3fpzo$reactstatelytable = require("@react-stately/table");
var $3fpzo$spectrumiconsuiArrowDownSmall = require("@spectrum-icons/ui/ArrowDownSmall");
var $3fpzo$reactariautils = require("@react-aria/utils");
var $3fpzo$reactspectrumcheckbox = require("@react-spectrum/checkbox");
var $3fpzo$reactspectrumutils = require("@react-spectrum/utils");
var $3fpzo$reactariafocus = require("@react-aria/focus");
var $3fpzo$reactspectrummenu = require("@react-spectrum/menu");
var $3fpzo$reactariavirtualizer = require("@react-aria/virtualizer");
var $3fpzo$reactspectrumprogress = require("@react-spectrum/progress");
var $3fpzo$react = require("react");
var $3fpzo$reactstatelyvirtualizer = require("@react-stately/virtualizer");
var $3fpzo$reactstatelylayout = require("@react-stately/layout");
var $3fpzo$reactspectrumtooltip = require("@react-spectrum/tooltip");
var $3fpzo$reactariabutton = require("@react-aria/button");
var $3fpzo$reactariainteractions = require("@react-aria/interactions");
var $3fpzo$reactariai18n = require("@react-aria/i18n");
var $3fpzo$reactspectrumprovider = require("@react-spectrum/provider");
var $3fpzo$reactariatable = require("@react-aria/table");
var $3fpzo$reactariavisuallyhidden = require("@react-aria/visually-hidden");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Column", () => $4b0b515c47252574$export$816b5d811295e6bc);
$parcel$export(module.exports, "TableView", () => $be8a9c035a82f788$export$b3c27e869d856b7);
$parcel$export(module.exports, "TableHeader", () => $3fpzo$reactstatelytable.TableHeader);
$parcel$export(module.exports, "TableBody", () => $3fpzo$reactstatelytable.TableBody);
$parcel$export(module.exports, "Section", () => $3fpzo$reactstatelytable.Section);
$parcel$export(module.exports, "Row", () => $3fpzo$reactstatelytable.Row);
$parcel$export(module.exports, "Cell", () => $3fpzo$reactstatelytable.Cell);






var $a89862f958371476$exports = {};
var $aea5dbc70156d941$exports = {};
$aea5dbc70156d941$exports = {
    "columnResizer": `أداة تغيير حجم العمود`,
    "loading": `جارٍ التحميل...`,
    "loadingMore": `جارٍ تحميل المزيد...`,
    "resizeColumn": `تغيير حجم العمود`,
    "sortAscending": `فرز بترتيب تصاعدي`,
    "sortDescending": `فرز بترتيب تنازلي`
};


var $11f14c1e8be327ec$exports = {};
$11f14c1e8be327ec$exports = {
    "columnResizer": `Преоразмеряване на колони`,
    "loading": `Зареждане...`,
    "loadingMore": `Зареждане на още...`,
    "resizeColumn": `Преоразмеряване на колона`,
    "sortAscending": `Възходящо сортиране`,
    "sortDescending": `Низходящо сортиране `
};


var $2ecdfa3a3f4780a9$exports = {};
$2ecdfa3a3f4780a9$exports = {
    "columnResizer": `Změna velikosti sloupce`,
    "loading": `Načítání...`,
    "loadingMore": `Načítání dalších...`,
    "resizeColumn": `Změnit velikost sloupce`,
    "sortAscending": `Seřadit vzestupně`,
    "sortDescending": `Seřadit sestupně`
};


var $5adae494c3f47d1c$exports = {};
$5adae494c3f47d1c$exports = {
    "columnResizer": `Kolonneændring`,
    "loading": `Indlæser ...`,
    "loadingMore": `Indlæser flere ...`,
    "resizeColumn": `Tilpas størrelse på kolonne`,
    "sortAscending": `Sorter stigende`,
    "sortDescending": `Sorter faldende`
};


var $6239e14fe242b5cd$exports = {};
$6239e14fe242b5cd$exports = {
    "columnResizer": `Spaltenanpassung`,
    "loading": `Laden...`,
    "loadingMore": `Mehr laden ...`,
    "resizeColumn": `Spaltengröße ändern`,
    "sortAscending": `Aufsteigend sortieren`,
    "sortDescending": `Absteigend sortieren`
};


var $548083df36594e1d$exports = {};
$548083df36594e1d$exports = {
    "columnResizer": `Αλλαγή μεγέθους στήλης`,
    "loading": `Φόρτωση...`,
    "loadingMore": `Φόρτωση περισσότερων...`,
    "resizeColumn": `Αλλαγή μεγέθους στήλης`,
    "sortAscending": `Ταξινόμηση κατά αύξουσα σειρά`,
    "sortDescending": `Ταξινόμηση κατά φθίνουσα σειρά`
};


var $b5db09171b92c95b$exports = {};
$b5db09171b92c95b$exports = {
    "loading": `Loading…`,
    "loadingMore": `Loading more…`,
    "sortAscending": `Sort Ascending`,
    "sortDescending": `Sort Descending`,
    "resizeColumn": `Resize column`,
    "columnResizer": `Column resizer`
};


var $4da957af4062e6c6$exports = {};
$4da957af4062e6c6$exports = {
    "columnResizer": `Redimensionador de columnas`,
    "loading": `Cargando…`,
    "loadingMore": `Cargando más…`,
    "resizeColumn": `Cambiar el tamaño de la columna`,
    "sortAscending": `Orden de subida`,
    "sortDescending": `Orden de bajada`
};


var $74726a0aedf994f5$exports = {};
$74726a0aedf994f5$exports = {
    "columnResizer": `Veeru suuruse muutja`,
    "loading": `Laadimine...`,
    "loadingMore": `Laadi rohkem...`,
    "resizeColumn": `Muuda veeru suurust`,
    "sortAscending": `Sordi kasvavalt`,
    "sortDescending": `Sordi kahanevalt`
};


var $801f6ae5b3d3b235$exports = {};
$801f6ae5b3d3b235$exports = {
    "columnResizer": `Sarakekoon muuttaja`,
    "loading": `Ladataan…`,
    "loadingMore": `Ladataan lisää…`,
    "resizeColumn": `Muuta sarakkeen kokoa`,
    "sortAscending": `Lajittelujärjestys: nouseva`,
    "sortDescending": `Lajittelujärjestys: laskeva`
};


var $9360047b0141de84$exports = {};
$9360047b0141de84$exports = {
    "columnResizer": `Redimensionnement de colonne`,
    "loading": `Chargement...`,
    "loadingMore": `Chargement supplémentaire...`,
    "resizeColumn": `Redimensionner la colonne`,
    "sortAscending": `Trier par ordre croissant`,
    "sortDescending": `Trier par ordre décroissant`
};


var $3cf53c7fc4feae86$exports = {};
$3cf53c7fc4feae86$exports = {
    "columnResizer": `שינוי גודל עמודה`,
    "loading": `טוען...`,
    "loadingMore": `טוען עוד...`,
    "resizeColumn": `שנה את גודל העמודה`,
    "sortAscending": `מיין בסדר עולה`,
    "sortDescending": `מיין בסדר יורד`
};


var $450e2334eb2be6c9$exports = {};
$450e2334eb2be6c9$exports = {
    "columnResizer": `Alat za promjenu veličine stupca`,
    "loading": `Učitavam...`,
    "loadingMore": `Učitavam još...`,
    "resizeColumn": `Promijeni veličinu stupca`,
    "sortAscending": `Sortiraj uzlazno`,
    "sortDescending": `Sortiraj silazno`
};


var $858276b4ec508f2f$exports = {};
$858276b4ec508f2f$exports = {
    "columnResizer": `Oszlopátméretező`,
    "loading": `Betöltés folyamatban…`,
    "loadingMore": `Továbbiak betöltése folyamatban…`,
    "resizeColumn": `Oszlop átméretezése`,
    "sortAscending": `Növekvő rendezés`,
    "sortDescending": `Csökkenő rendezés`
};


var $a7e2289ac11d4bcc$exports = {};
$a7e2289ac11d4bcc$exports = {
    "columnResizer": `Ridimensionamento colonne`,
    "loading": `Caricamento...`,
    "loadingMore": `Caricamento altri...`,
    "resizeColumn": `Ridimensiona colonna`,
    "sortAscending": `Ordinamento crescente`,
    "sortDescending": `Ordinamento decrescente`
};


var $069e94bb941acd04$exports = {};
$069e94bb941acd04$exports = {
    "columnResizer": `列リサイザー`,
    "loading": `読み込み中...`,
    "loadingMore": `さらに読み込み中...`,
    "resizeColumn": `列幅を変更`,
    "sortAscending": `昇順に並べ替え`,
    "sortDescending": `降順に並べ替え`
};


var $6df8b1c85e0d6c1c$exports = {};
$6df8b1c85e0d6c1c$exports = {
    "columnResizer": `열 크기 조정기`,
    "loading": `로드 중`,
    "loadingMore": `추가 로드 중`,
    "resizeColumn": `열 크기 조정`,
    "sortAscending": `오름차순 정렬`,
    "sortDescending": `내림차순 정렬`
};


var $e339442a63cca57f$exports = {};
$e339442a63cca57f$exports = {
    "columnResizer": `Stulpelio dydžio keitiklis`,
    "loading": `Įkeliama...`,
    "loadingMore": `Įkeliama daugiau...`,
    "resizeColumn": `Keisti stulpelio dydį`,
    "sortAscending": `Rikiuoti didėjimo tvarka`,
    "sortDescending": `Rikiuoti mažėjimo tvarka`
};


var $8d629efbd88ee34d$exports = {};
$8d629efbd88ee34d$exports = {
    "columnResizer": `Kolonnas izmēru mainītājs`,
    "loading": `Notiek ielāde...`,
    "loadingMore": `Tiek ielādēts vēl...`,
    "resizeColumn": `Mainīt kolonnas lielumu`,
    "sortAscending": `Kārtot augošā secībā`,
    "sortDescending": `Kārtot dilstošā secībā`
};


var $c0daac194b2d01a3$exports = {};
$c0daac194b2d01a3$exports = {
    "columnResizer": `Størrelsesendring av kolonne`,
    "loading": `Laster inn ...`,
    "loadingMore": `Laster inn flere ...`,
    "resizeColumn": `Endre størrelse på kolonne`,
    "sortAscending": `Sorter stigende`,
    "sortDescending": `Sorter synkende`
};


var $39589c06d5afb6db$exports = {};
$39589c06d5afb6db$exports = {
    "columnResizer": `Groottewijziging van kolom`,
    "loading": `Laden...`,
    "loadingMore": `Meer laden...`,
    "resizeColumn": `Kolomgrootte wijzigen`,
    "sortAscending": `Oplopend sorteren`,
    "sortDescending": `Aflopend sorteren`
};


var $4333ec237d5afa31$exports = {};
$4333ec237d5afa31$exports = {
    "columnResizer": `Narzędzie zmiany rozmiaru kolumny`,
    "loading": `Ładowanie...`,
    "loadingMore": `Wczytywanie większej liczby...`,
    "resizeColumn": `Zmień rozmiar kolumny`,
    "sortAscending": `Sortuj rosnąco`,
    "sortDescending": `Sortuj malejąco`
};


var $5aa66c54d994ad4c$exports = {};
$5aa66c54d994ad4c$exports = {
    "columnResizer": `Redimensionamento de colunas`,
    "loading": `Carregando...`,
    "loadingMore": `Carregando mais...`,
    "resizeColumn": `Redimensionar coluna`,
    "sortAscending": `Ordenar por ordem crescente`,
    "sortDescending": `Ordenar por ordem decrescente`
};


var $d2901017a4a40e5f$exports = {};
$d2901017a4a40e5f$exports = {
    "columnResizer": `Redimensionador de coluna`,
    "loading": `A carregar...`,
    "loadingMore": `A carregar mais...`,
    "resizeColumn": `Redimensionar coluna`,
    "sortAscending": `Ordenar por ordem ascendente`,
    "sortDescending": `Ordenar por ordem decrescente`
};


var $b247d60aff277eee$exports = {};
$b247d60aff277eee$exports = {
    "columnResizer": `Instrument redimensionare coloane`,
    "loading": `Se încarcă...`,
    "loadingMore": `Se încarcă mai multe...`,
    "resizeColumn": `Redimensionați coloana`,
    "sortAscending": `Sortați crescător`,
    "sortDescending": `Sortați descrescător`
};


var $2f6a66990fe0eb2b$exports = {};
$2f6a66990fe0eb2b$exports = {
    "columnResizer": `Средство изменения размера столбцов`,
    "loading": `Загрузка...`,
    "loadingMore": `Дополнительная загрузка...`,
    "resizeColumn": `Изменить размер столбца`,
    "sortAscending": `Сортировать по возрастанию`,
    "sortDescending": `Сортировать по убыванию`
};


var $1e8c28cc0013ef8e$exports = {};
$1e8c28cc0013ef8e$exports = {
    "columnResizer": `Nástroj na zmenu veľkosti stĺpcov`,
    "loading": `Načítava sa...`,
    "loadingMore": `Načítava sa viac...`,
    "resizeColumn": `Zmeniť veľkosť stĺpca`,
    "sortAscending": `Zoradiť vzostupne`,
    "sortDescending": `Zoradiť zostupne`
};


var $a63729a5f7450d5e$exports = {};
$a63729a5f7450d5e$exports = {
    "columnResizer": `Prilagojevalnik velikosti stolpcev`,
    "loading": `Nalaganje...`,
    "loadingMore": `Nalaganje več vsebine...`,
    "resizeColumn": `Spremeni velikost stolpca`,
    "sortAscending": `Razvrsti naraščajoče`,
    "sortDescending": `Razvrsti padajoče`
};


var $9c992f998f269d4d$exports = {};
$9c992f998f269d4d$exports = {
    "columnResizer": `Alat za promenu veličine kolone`,
    "loading": `Učitavam...`,
    "loadingMore": `Učitavam još...`,
    "resizeColumn": `Promeni veličinu kolone`,
    "sortAscending": `Sortiraj po rastućem redosledu`,
    "sortDescending": `Sortiraj po opadajućem redosledu`
};


var $45c12baedb14591e$exports = {};
$45c12baedb14591e$exports = {
    "columnResizer": `Ändra storlek på kolumn`,
    "loading": `Läser in...`,
    "loadingMore": `Läser in mer...`,
    "resizeColumn": `Ändra storlek på kolumn`,
    "sortAscending": `Sortera i stigande ordning`,
    "sortDescending": `Sortera i fallande ordning`
};


var $d960f28e0cef8a8c$exports = {};
$d960f28e0cef8a8c$exports = {
    "columnResizer": `Yeniden sütun boyutlandırıcı`,
    "loading": `Yükleniyor...`,
    "loadingMore": `Daha fazla yükleniyor...`,
    "resizeColumn": `Sütunu yeniden boyutlandır`,
    "sortAscending": `Artan Sıralama`,
    "sortDescending": `Azalan Sıralama`
};


var $693364411fbb8580$exports = {};
$693364411fbb8580$exports = {
    "columnResizer": `Засіб змінення розміру стовпця`,
    "loading": `Завантаження…`,
    "loadingMore": `Завантаження інших об’єктів...`,
    "resizeColumn": `Змінити розмір стовпця`,
    "sortAscending": `Сортувати за зростанням`,
    "sortDescending": `Сортувати за спаданням`
};


var $85fd0a65720e1eb6$exports = {};
$85fd0a65720e1eb6$exports = {
    "columnResizer": `列尺寸调整器`,
    "loading": `正在加载...`,
    "loadingMore": `正在加载更多...`,
    "resizeColumn": `调整列大小`,
    "sortAscending": `升序排序`,
    "sortDescending": `降序排序`
};


var $0624a686d9964a9c$exports = {};
$0624a686d9964a9c$exports = {
    "columnResizer": `欄大小調整器`,
    "loading": `正在載入`,
    "loadingMore": `正在載入更多…`,
    "resizeColumn": `調整欄大小`,
    "sortAscending": `升序排序`,
    "sortDescending": `降序排序`
};


$a89862f958371476$exports = {
    "ar-AE": $aea5dbc70156d941$exports,
    "bg-BG": $11f14c1e8be327ec$exports,
    "cs-CZ": $2ecdfa3a3f4780a9$exports,
    "da-DK": $5adae494c3f47d1c$exports,
    "de-DE": $6239e14fe242b5cd$exports,
    "el-GR": $548083df36594e1d$exports,
    "en-US": $b5db09171b92c95b$exports,
    "es-ES": $4da957af4062e6c6$exports,
    "et-EE": $74726a0aedf994f5$exports,
    "fi-FI": $801f6ae5b3d3b235$exports,
    "fr-FR": $9360047b0141de84$exports,
    "he-IL": $3cf53c7fc4feae86$exports,
    "hr-HR": $450e2334eb2be6c9$exports,
    "hu-HU": $858276b4ec508f2f$exports,
    "it-IT": $a7e2289ac11d4bcc$exports,
    "ja-JP": $069e94bb941acd04$exports,
    "ko-KR": $6df8b1c85e0d6c1c$exports,
    "lt-LT": $e339442a63cca57f$exports,
    "lv-LV": $8d629efbd88ee34d$exports,
    "nb-NO": $c0daac194b2d01a3$exports,
    "nl-NL": $39589c06d5afb6db$exports,
    "pl-PL": $4333ec237d5afa31$exports,
    "pt-BR": $5aa66c54d994ad4c$exports,
    "pt-PT": $d2901017a4a40e5f$exports,
    "ro-RO": $b247d60aff277eee$exports,
    "ru-RU": $2f6a66990fe0eb2b$exports,
    "sk-SK": $1e8c28cc0013ef8e$exports,
    "sl-SI": $a63729a5f7450d5e$exports,
    "sr-SP": $9c992f998f269d4d$exports,
    "sv-SE": $45c12baedb14591e$exports,
    "tr-TR": $d960f28e0cef8a8c$exports,
    "uk-UA": $693364411fbb8580$exports,
    "zh-CN": $85fd0a65720e1eb6$exports,
    "zh-TW": $0624a686d9964a9c$exports
};











var $982885d0a34882ea$exports = {};

$parcel$export($982885d0a34882ea$exports, "spectrum-Table", () => $982885d0a34882ea$export$455c59d2b5bafef0, (v) => $982885d0a34882ea$export$455c59d2b5bafef0 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-sortedIcon", () => $982885d0a34882ea$export$7d48ded8ceef1e62, (v) => $982885d0a34882ea$export$7d48ded8ceef1e62 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-headWrapper", () => $982885d0a34882ea$export$a08929593e9a2fb2, (v) => $982885d0a34882ea$export$a08929593e9a2fb2 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-headCellContents", () => $982885d0a34882ea$export$ce66f3b8316b228e, (v) => $982885d0a34882ea$export$ce66f3b8316b228e = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-headCell", () => $982885d0a34882ea$export$f64307d6c29e2f55, (v) => $982885d0a34882ea$export$f64307d6c29e2f55 = v);
$parcel$export($982885d0a34882ea$exports, "is-sortable", () => $982885d0a34882ea$export$c5f84948d72c03fb, (v) => $982885d0a34882ea$export$c5f84948d72c03fb = v);
$parcel$export($982885d0a34882ea$exports, "is-sorted-asc", () => $982885d0a34882ea$export$24499792d6b87cb, (v) => $982885d0a34882ea$export$24499792d6b87cb = v);
$parcel$export($982885d0a34882ea$exports, "is-sorted-desc", () => $982885d0a34882ea$export$cda6d7b632f895cc, (v) => $982885d0a34882ea$export$cda6d7b632f895cc = v);
$parcel$export($982885d0a34882ea$exports, "is-resizable", () => $982885d0a34882ea$export$d20629d2e587d4f, (v) => $982885d0a34882ea$export$d20629d2e587d4f = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-headCellButton", () => $982885d0a34882ea$export$231a910aceaf76df, (v) => $982885d0a34882ea$export$231a910aceaf76df = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-columnResizer", () => $982885d0a34882ea$export$a1a03b9bd1868758, (v) => $982885d0a34882ea$export$a1a03b9bd1868758 = v);
$parcel$export($982885d0a34882ea$exports, "focus-ring", () => $982885d0a34882ea$export$f39a09f249340e2a, (v) => $982885d0a34882ea$export$f39a09f249340e2a = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-cell--alignCenter", () => $982885d0a34882ea$export$7e9bb4c3d8ec70e2, (v) => $982885d0a34882ea$export$7e9bb4c3d8ec70e2 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-cell--alignEnd", () => $982885d0a34882ea$export$94db5e2e979379b6, (v) => $982885d0a34882ea$export$94db5e2e979379b6 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-body", () => $982885d0a34882ea$export$c841a58af70827c7, (v) => $982885d0a34882ea$export$c841a58af70827c7 = v);
$parcel$export($982885d0a34882ea$exports, "is-drop-target", () => $982885d0a34882ea$export$d8467f38d2d00bd0, (v) => $982885d0a34882ea$export$d8467f38d2d00bd0 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-row", () => $982885d0a34882ea$export$fefa57dcd4648052, (v) => $982885d0a34882ea$export$fefa57dcd4648052 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table--quiet", () => $982885d0a34882ea$export$80f179961079d85e, (v) => $982885d0a34882ea$export$80f179961079d85e = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-cell", () => $982885d0a34882ea$export$c073926c283bb618, (v) => $982885d0a34882ea$export$c073926c283bb618 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table--compact", () => $982885d0a34882ea$export$bcb43fe016211cab, (v) => $982885d0a34882ea$export$bcb43fe016211cab = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table--spacious", () => $982885d0a34882ea$export$eef21b0715bc0914, (v) => $982885d0a34882ea$export$eef21b0715bc0914 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-cellContents", () => $982885d0a34882ea$export$683061182a3353d3, (v) => $982885d0a34882ea$export$683061182a3353d3 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-cell--hideHeader", () => $982885d0a34882ea$export$8ff2ab45f5f0045d, (v) => $982885d0a34882ea$export$8ff2ab45f5f0045d = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table--wrap", () => $982885d0a34882ea$export$9d4c255d6285f0ec, (v) => $982885d0a34882ea$export$9d4c255d6285f0ec = v);
$parcel$export($982885d0a34882ea$exports, "is-focused", () => $982885d0a34882ea$export$e7dc768d35940237, (v) => $982885d0a34882ea$export$e7dc768d35940237 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-cell--divider", () => $982885d0a34882ea$export$9531650467e5785b, (v) => $982885d0a34882ea$export$9531650467e5785b = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-cellWrapper", () => $982885d0a34882ea$export$fe4389976827e927, (v) => $982885d0a34882ea$export$fe4389976827e927 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table--loadingMore", () => $982885d0a34882ea$export$51b307035a9062f1, (v) => $982885d0a34882ea$export$51b307035a9062f1 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-row--lastRow", () => $982885d0a34882ea$export$6bc25dc2c214dc40, (v) => $982885d0a34882ea$export$6bc25dc2c214dc40 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-row--isFlushBottom", () => $982885d0a34882ea$export$c544a9cedcdf7e74, (v) => $982885d0a34882ea$export$c544a9cedcdf7e74 = v);
$parcel$export($982885d0a34882ea$exports, "is-selected", () => $982885d0a34882ea$export$1e0fb04f31d3c22a, (v) => $982885d0a34882ea$export$1e0fb04f31d3c22a = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-row--firstRow", () => $982885d0a34882ea$export$71150fb6934bcef6, (v) => $982885d0a34882ea$export$71150fb6934bcef6 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table--isVerticalScrollbarVisible", () => $982885d0a34882ea$export$f3cbf113923c0570, (v) => $982885d0a34882ea$export$f3cbf113923c0570 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table--isHorizontalScrollbarVisible", () => $982885d0a34882ea$export$eb2b06f9a223c9ae, (v) => $982885d0a34882ea$export$eb2b06f9a223c9ae = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-checkboxCell", () => $982885d0a34882ea$export$ae17a7501a9fd0b3, (v) => $982885d0a34882ea$export$ae17a7501a9fd0b3 = v);
$parcel$export($982885d0a34882ea$exports, "spectrum-Table-checkbox", () => $982885d0a34882ea$export$9c494845d6df55df, (v) => $982885d0a34882ea$export$9c494845d6df55df = v);
$parcel$export($982885d0a34882ea$exports, "is-hovered", () => $982885d0a34882ea$export$b8813cd5d7824ce7, (v) => $982885d0a34882ea$export$b8813cd5d7824ce7 = v);
$parcel$export($982885d0a34882ea$exports, "is-active", () => $982885d0a34882ea$export$20fd0f7cd4e6112f, (v) => $982885d0a34882ea$export$20fd0f7cd4e6112f = v);
$parcel$export($982885d0a34882ea$exports, "is-disabled", () => $982885d0a34882ea$export$d35bc1e505d1ebbf, (v) => $982885d0a34882ea$export$d35bc1e505d1ebbf = v);
$parcel$export($982885d0a34882ea$exports, "is-next-selected", () => $982885d0a34882ea$export$53aa811f86efbfad, (v) => $982885d0a34882ea$export$53aa811f86efbfad = v);
var $982885d0a34882ea$export$455c59d2b5bafef0;
var $982885d0a34882ea$export$7d48ded8ceef1e62;
var $982885d0a34882ea$export$a08929593e9a2fb2;
var $982885d0a34882ea$export$ce66f3b8316b228e;
var $982885d0a34882ea$export$f64307d6c29e2f55;
var $982885d0a34882ea$export$c5f84948d72c03fb;
var $982885d0a34882ea$export$24499792d6b87cb;
var $982885d0a34882ea$export$cda6d7b632f895cc;
var $982885d0a34882ea$export$d20629d2e587d4f;
var $982885d0a34882ea$export$231a910aceaf76df;
var $982885d0a34882ea$export$a1a03b9bd1868758;
var $982885d0a34882ea$export$f39a09f249340e2a;
var $982885d0a34882ea$export$7e9bb4c3d8ec70e2;
var $982885d0a34882ea$export$94db5e2e979379b6;
var $982885d0a34882ea$export$c841a58af70827c7;
var $982885d0a34882ea$export$d8467f38d2d00bd0;
var $982885d0a34882ea$export$fefa57dcd4648052;
var $982885d0a34882ea$export$80f179961079d85e;
var $982885d0a34882ea$export$c073926c283bb618;
var $982885d0a34882ea$export$bcb43fe016211cab;
var $982885d0a34882ea$export$eef21b0715bc0914;
var $982885d0a34882ea$export$683061182a3353d3;
var $982885d0a34882ea$export$8ff2ab45f5f0045d;
var $982885d0a34882ea$export$9d4c255d6285f0ec;
var $982885d0a34882ea$export$e7dc768d35940237;
var $982885d0a34882ea$export$9531650467e5785b;
var $982885d0a34882ea$export$fe4389976827e927;
var $982885d0a34882ea$export$51b307035a9062f1;
var $982885d0a34882ea$export$6bc25dc2c214dc40;
var $982885d0a34882ea$export$c544a9cedcdf7e74;
var $982885d0a34882ea$export$1e0fb04f31d3c22a;
var $982885d0a34882ea$export$71150fb6934bcef6;
var $982885d0a34882ea$export$f3cbf113923c0570;
var $982885d0a34882ea$export$eb2b06f9a223c9ae;
var $982885d0a34882ea$export$ae17a7501a9fd0b3;
var $982885d0a34882ea$export$9c494845d6df55df;
var $982885d0a34882ea$export$b8813cd5d7824ce7;
var $982885d0a34882ea$export$20fd0f7cd4e6112f;
var $982885d0a34882ea$export$d35bc1e505d1ebbf;
var $982885d0a34882ea$export$53aa811f86efbfad;
$982885d0a34882ea$export$455c59d2b5bafef0 = "spectrum-Table_d0fc74";
$982885d0a34882ea$export$7d48ded8ceef1e62 = "spectrum-Table-sortedIcon_d0fc74";
$982885d0a34882ea$export$a08929593e9a2fb2 = "spectrum-Table-headWrapper_d0fc74";
$982885d0a34882ea$export$ce66f3b8316b228e = "spectrum-Table-headCellContents_d0fc74";
$982885d0a34882ea$export$f64307d6c29e2f55 = "spectrum-Table-headCell_d0fc74";
$982885d0a34882ea$export$c5f84948d72c03fb = "is-sortable_d0fc74";
$982885d0a34882ea$export$24499792d6b87cb = "is-sorted-asc_d0fc74";
$982885d0a34882ea$export$cda6d7b632f895cc = "is-sorted-desc_d0fc74";
$982885d0a34882ea$export$d20629d2e587d4f = "is-resizable_d0fc74";
$982885d0a34882ea$export$231a910aceaf76df = "spectrum-Table-headCellButton_d0fc74";
$982885d0a34882ea$export$a1a03b9bd1868758 = "spectrum-Table-columnResizer_d0fc74";
$982885d0a34882ea$export$f39a09f249340e2a = "focus-ring_d0fc74";
$982885d0a34882ea$export$7e9bb4c3d8ec70e2 = "spectrum-Table-cell--alignCenter_d0fc74";
$982885d0a34882ea$export$94db5e2e979379b6 = "spectrum-Table-cell--alignEnd_d0fc74";
$982885d0a34882ea$export$c841a58af70827c7 = "spectrum-Table-body_d0fc74";
$982885d0a34882ea$export$d8467f38d2d00bd0 = "is-drop-target_d0fc74";
$982885d0a34882ea$export$fefa57dcd4648052 = "spectrum-Table-row_d0fc74";
$982885d0a34882ea$export$80f179961079d85e = "spectrum-Table--quiet_d0fc74";
$982885d0a34882ea$export$c073926c283bb618 = "spectrum-Table-cell_d0fc74";
$982885d0a34882ea$export$bcb43fe016211cab = "spectrum-Table--compact_d0fc74";
$982885d0a34882ea$export$eef21b0715bc0914 = "spectrum-Table--spacious_d0fc74";
$982885d0a34882ea$export$683061182a3353d3 = "spectrum-Table-cellContents_d0fc74";
$982885d0a34882ea$export$8ff2ab45f5f0045d = "spectrum-Table-cell--hideHeader_d0fc74";
$982885d0a34882ea$export$9d4c255d6285f0ec = "spectrum-Table--wrap_d0fc74";
$982885d0a34882ea$export$e7dc768d35940237 = "is-focused_d0fc74";
$982885d0a34882ea$export$9531650467e5785b = "spectrum-Table-cell--divider_d0fc74";
$982885d0a34882ea$export$fe4389976827e927 = "spectrum-Table-cellWrapper_d0fc74";
$982885d0a34882ea$export$51b307035a9062f1 = "spectrum-Table--loadingMore_d0fc74";
$982885d0a34882ea$export$6bc25dc2c214dc40 = "spectrum-Table-row--lastRow_d0fc74";
$982885d0a34882ea$export$c544a9cedcdf7e74 = "spectrum-Table-row--isFlushBottom_d0fc74";
$982885d0a34882ea$export$1e0fb04f31d3c22a = "is-selected_d0fc74";
$982885d0a34882ea$export$71150fb6934bcef6 = "spectrum-Table-row--firstRow_d0fc74";
$982885d0a34882ea$export$f3cbf113923c0570 = "spectrum-Table--isVerticalScrollbarVisible_d0fc74";
$982885d0a34882ea$export$eb2b06f9a223c9ae = "spectrum-Table--isHorizontalScrollbarVisible_d0fc74";
$982885d0a34882ea$export$ae17a7501a9fd0b3 = "spectrum-Table-checkboxCell_d0fc74";
$982885d0a34882ea$export$9c494845d6df55df = "spectrum-Table-checkbox_d0fc74";
$982885d0a34882ea$export$b8813cd5d7824ce7 = "is-hovered_d0fc74";
$982885d0a34882ea$export$20fd0f7cd4e6112f = "is-active_d0fc74";
$982885d0a34882ea$export$d35bc1e505d1ebbf = "is-disabled_d0fc74";
$982885d0a34882ea$export$53aa811f86efbfad = "is-next-selected_d0fc74";






function $2877766f94c67a67$var$Resizer(props, ref) {
    let { column: column , showResizer: showResizer  } = props;
    let { state: state , columnState: columnState  } = $be8a9c035a82f788$export$3cb274deb6c2d854();
    let stringFormatter = $3fpzo$reactariai18n.useLocalizedStringFormatter((/*@__PURE__*/$parcel$interopDefault($a89862f958371476$exports)));
    let { direction: direction  } = $3fpzo$reactariai18n.useLocale();
    let { inputProps: inputProps , resizerProps: resizerProps  } = $3fpzo$reactariatable.useTableColumnResize({
        ...props,
        label: stringFormatter.format('columnResizer')
    }, state, columnState, ref);
    let style = {
        cursor: undefined,
        height: '100%',
        display: showResizer ? undefined : 'none',
        touchAction: 'none'
    };
    if (columnState.getColumnMinWidth(column.key) >= columnState.getColumnWidth(column.key)) style.cursor = direction === 'rtl' ? 'w-resize' : 'e-resize';
    else if (columnState.getColumnMaxWidth(column.key) <= columnState.getColumnWidth(column.key)) style.cursor = direction === 'rtl' ? 'e-resize' : 'w-resize';
    else style.cursor = 'col-resize';
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariafocus.FocusRing, {
        within: true,
        focusRingClass: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        role: "presentation",
        style: style,
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-columnResizer'),
        ...resizerProps
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariavisuallyhidden.VisuallyHidden, null, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("input", {
        ref: ref,
        type: "range",
        ...inputProps
    })))));
}
const $2877766f94c67a67$export$48a76196cafe3b93 = /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).forwardRef($2877766f94c67a67$var$Resizer);



var $4afcd54cfd94dbb9$exports = {};

$parcel$export($4afcd54cfd94dbb9$exports, "react-spectrum-Table", () => $4afcd54cfd94dbb9$export$8cc7e6e5f955c192, (v) => $4afcd54cfd94dbb9$export$8cc7e6e5f955c192 = v);
$parcel$export($4afcd54cfd94dbb9$exports, "react-spectrum-Table-cell", () => $4afcd54cfd94dbb9$export$19015e98090cea63, (v) => $4afcd54cfd94dbb9$export$19015e98090cea63 = v);
$parcel$export($4afcd54cfd94dbb9$exports, "react-spectrum-Table-cellWrapper", () => $4afcd54cfd94dbb9$export$ea1d9f40910c509, (v) => $4afcd54cfd94dbb9$export$ea1d9f40910c509 = v);
$parcel$export($4afcd54cfd94dbb9$exports, "react-spectrum-Table-cell--alignStart", () => $4afcd54cfd94dbb9$export$25d7f56ba6d0d442, (v) => $4afcd54cfd94dbb9$export$25d7f56ba6d0d442 = v);
$parcel$export($4afcd54cfd94dbb9$exports, "react-spectrum-Table-cell--alignCenter", () => $4afcd54cfd94dbb9$export$14101ea2699002fc, (v) => $4afcd54cfd94dbb9$export$14101ea2699002fc = v);
$parcel$export($4afcd54cfd94dbb9$exports, "react-spectrum-Table-cell--alignEnd", () => $4afcd54cfd94dbb9$export$c594335f8535fc47, (v) => $4afcd54cfd94dbb9$export$c594335f8535fc47 = v);
$parcel$export($4afcd54cfd94dbb9$exports, "react-spectrum-Table-centeredWrapper", () => $4afcd54cfd94dbb9$export$8bffb31c98aaf8d5, (v) => $4afcd54cfd94dbb9$export$8bffb31c98aaf8d5 = v);
var $4afcd54cfd94dbb9$export$8cc7e6e5f955c192;
var $4afcd54cfd94dbb9$export$19015e98090cea63;
var $4afcd54cfd94dbb9$export$ea1d9f40910c509;
var $4afcd54cfd94dbb9$export$25d7f56ba6d0d442;
var $4afcd54cfd94dbb9$export$14101ea2699002fc;
var $4afcd54cfd94dbb9$export$c594335f8535fc47;
var $4afcd54cfd94dbb9$export$8bffb31c98aaf8d5;
$4afcd54cfd94dbb9$export$8cc7e6e5f955c192 = "react-spectrum-Table_e844a7";
$4afcd54cfd94dbb9$export$19015e98090cea63 = "react-spectrum-Table-cell_e844a7";
$4afcd54cfd94dbb9$export$ea1d9f40910c509 = "react-spectrum-Table-cellWrapper_e844a7";
$4afcd54cfd94dbb9$export$25d7f56ba6d0d442 = "react-spectrum-Table-cell--alignStart_e844a7";
$4afcd54cfd94dbb9$export$14101ea2699002fc = "react-spectrum-Table-cell--alignCenter_e844a7";
$4afcd54cfd94dbb9$export$c594335f8535fc47 = "react-spectrum-Table-cell--alignEnd_e844a7";
$4afcd54cfd94dbb9$export$8bffb31c98aaf8d5 = "react-spectrum-Table-centeredWrapper_e844a7";











const $be8a9c035a82f788$var$DEFAULT_HEADER_HEIGHT = {
    medium: 34,
    large: 40
};
const $be8a9c035a82f788$var$DEFAULT_HIDE_HEADER_CELL_WIDTH = {
    medium: 36,
    large: 44
};
const $be8a9c035a82f788$var$ROW_HEIGHTS = {
    compact: {
        medium: 32,
        large: 40
    },
    regular: {
        medium: 40,
        large: 50
    },
    spacious: {
        medium: 48,
        large: 60
    }
};
const $be8a9c035a82f788$var$SELECTION_CELL_DEFAULT_WIDTH = {
    medium: 38,
    large: 48
};
const $be8a9c035a82f788$var$TableContext = /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createContext(null);
function $be8a9c035a82f788$export$3cb274deb6c2d854() {
    return $3fpzo$react.useContext($be8a9c035a82f788$var$TableContext);
}
function $be8a9c035a82f788$var$TableView(props, ref) {
    props = $3fpzo$reactspectrumprovider.useProviderProps(props);
    let { isQuiet: isQuiet , onAction: onAction  } = props;
    let { styleProps: styleProps  } = $3fpzo$reactspectrumutils.useStyleProps(props);
    let [showSelectionCheckboxes, setShowSelectionCheckboxes] = $3fpzo$react.useState(props.selectionStyle !== 'highlight');
    let { direction: direction  } = $3fpzo$reactariai18n.useLocale();
    let { scale: scale  } = $3fpzo$reactspectrumprovider.useProvider();
    const getDefaultWidth = $3fpzo$react.useCallback(({ hideHeader: hideHeader , isSelectionCell: isSelectionCell , showDivider: showDivider  })=>{
        if (hideHeader) {
            let width = $be8a9c035a82f788$var$DEFAULT_HIDE_HEADER_CELL_WIDTH[scale];
            return showDivider ? width + 1 : width;
        } else if (isSelectionCell) return $be8a9c035a82f788$var$SELECTION_CELL_DEFAULT_WIDTH[scale];
    }, [
        scale
    ]);
    let state = $3fpzo$reactstatelytable.useTableState({
        ...props,
        showSelectionCheckboxes: showSelectionCheckboxes,
        selectionBehavior: props.selectionStyle === 'highlight' ? 'replace' : 'toggle'
    });
    const columnState = $3fpzo$reactstatelytable.useTableColumnResizeState({
        ...props,
        getDefaultWidth: getDefaultWidth
    }, state.collection);
    // If the selection behavior changes in state, we need to update showSelectionCheckboxes here due to the circular dependency...
    let shouldShowCheckboxes = state.selectionManager.selectionBehavior !== 'replace';
    if (shouldShowCheckboxes !== showSelectionCheckboxes) setShowSelectionCheckboxes(shouldShowCheckboxes);
    let domRef = $3fpzo$reactspectrumutils.useDOMRef(ref);
    let bodyRef = $3fpzo$react.useRef();
    let stringFormatter = $3fpzo$reactariai18n.useLocalizedStringFormatter((/*@__PURE__*/$parcel$interopDefault($a89862f958371476$exports)));
    let density = props.density || 'regular';
    let layout = $3fpzo$react.useMemo(()=>new $3fpzo$reactstatelylayout.TableLayout({
            // If props.rowHeight is auto, then use estimated heights based on scale, otherwise use fixed heights.
            rowHeight: props.overflowMode === 'wrap' ? null : $be8a9c035a82f788$var$ROW_HEIGHTS[density][scale],
            estimatedRowHeight: props.overflowMode === 'wrap' ? $be8a9c035a82f788$var$ROW_HEIGHTS[density][scale] : null,
            headingHeight: props.overflowMode === 'wrap' ? null : $be8a9c035a82f788$var$DEFAULT_HEADER_HEIGHT[scale],
            estimatedHeadingHeight: props.overflowMode === 'wrap' ? $be8a9c035a82f788$var$DEFAULT_HEADER_HEIGHT[scale] : null
        })
    , [
        props.overflowMode,
        scale,
        density
    ]);
    layout.collection = state.collection;
    layout.getColumnWidth = columnState.getColumnWidth;
    let { gridProps: gridProps  } = $3fpzo$reactariatable.useTable({
        ...props,
        isVirtualized: true,
        layout: layout,
        onRowAction: onAction
    }, state, domRef);
    let [headerRowHovered, setHeaderRowHovered] = $3fpzo$react.useState(false);
    let renderWrapper = (parent, reusableView, children, renderChildren)=>{
        let style = $3fpzo$reactariavirtualizer.layoutInfoToStyle(reusableView.layoutInfo, direction, parent && parent.layoutInfo);
        if (style.overflow === 'hidden') style.overflow = 'visible'; // needed to support position: sticky
        if (reusableView.viewType === 'rowgroup') return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableRowGroup, {
            key: reusableView.key,
            style: style
        }, renderChildren(children)));
        if (reusableView.viewType === 'header') return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableHeader, {
            key: reusableView.key,
            style: style
        }, renderChildren(children)));
        if (reusableView.viewType === 'row') return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableRow, {
            key: reusableView.key,
            item: reusableView.content,
            style: style,
            hasActions: onAction
        }, renderChildren(children)));
        if (reusableView.viewType === 'headerrow') return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableHeaderRow, {
            onHoverChange: setHeaderRowHovered,
            key: reusableView.key,
            style: style,
            item: reusableView.content
        }, renderChildren(children)));
        return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariavirtualizer.VirtualizerItem, {
            key: reusableView.key,
            reusableView: reusableView,
            parent: parent,
            className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-cellWrapper', $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($4afcd54cfd94dbb9$exports)), {
                'react-spectrum-Table-cellWrapper': !reusableView.layoutInfo.estimatedSize
            }))
        }));
    };
    let renderView = (type, item)=>{
        switch(type){
            case 'header':
            case 'rowgroup':
            case 'section':
            case 'row':
            case 'headerrow':
                return null;
            case 'cell':
                if (item.props.isSelectionCell) return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableCheckboxCell, {
                    cell: item
                }));
                return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableCell, {
                    cell: item
                }));
            case 'placeholder':
                // TODO: move to react-aria?
                return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
                    role: "gridcell",
                    "aria-colindex": item.index + 1,
                    "aria-colspan": item.colspan > 1 ? item.colspan : null
                }));
            case 'column':
                if (item.props.isSelectionCell) return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableSelectAllCell, {
                    column: item
                }));
                if (item.props.hideHeader) return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactspectrumtooltip.TooltipTrigger, {
                    placement: "top",
                    trigger: "focus"
                }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableColumnHeader, {
                    column: item
                }), /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactspectrumtooltip.Tooltip, {
                    placement: "top"
                }, item.rendered)));
                if (item.props.allowsResizing) return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$ResizableTableColumnHeader, {
                    tableRef: domRef,
                    column: item
                }));
                return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableColumnHeader, {
                    column: item
                }));
            case 'loader':
                return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$CenteredWrapper, null, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactspectrumprogress.ProgressCircle, {
                    isIndeterminate: true,
                    "aria-label": state.collection.size > 0 ? stringFormatter.format('loadingMore') : stringFormatter.format('loading')
                })));
            case 'empty':
                {
                    let emptyState = props.renderEmptyState ? props.renderEmptyState() : null;
                    if (emptyState == null) return null;
                    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$CenteredWrapper, null, emptyState));
                }
        }
    };
    let [isVerticalScrollbarVisible, setVerticalScollbarVisible] = $3fpzo$react.useState(false);
    let [isHorizontalScrollbarVisible, setHorizontalScollbarVisible] = $3fpzo$react.useState(false);
    let viewport = $3fpzo$react.useRef({
        x: 0,
        y: 0,
        width: 0,
        height: 0
    });
    let onVisibleRectChange = $3fpzo$react.useCallback((e)=>{
        if (viewport.current.width === e.width && viewport.current.height === e.height) return;
        viewport.current = e;
        if (bodyRef.current) {
            setVerticalScollbarVisible(bodyRef.current.clientWidth + 2 < bodyRef.current.offsetWidth);
            setHorizontalScollbarVisible(bodyRef.current.clientHeight + 2 < bodyRef.current.offsetHeight);
        }
    }, []);
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableContext.Provider, {
        value: {
            state: state,
            layout: layout,
            columnState: columnState,
            headerRowHovered: headerRowHovered
        }
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableVirtualizer, {
        ...gridProps,
        ...styleProps,
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table', `spectrum-Table--${density}`, {
            'spectrum-Table--quiet': isQuiet,
            'spectrum-Table--wrap': props.overflowMode === 'wrap',
            'spectrum-Table--loadingMore': state.collection.body.props.loadingState === 'loadingMore',
            'spectrum-Table--isVerticalScrollbarVisible': isVerticalScrollbarVisible,
            'spectrum-Table--isHorizontalScrollbarVisible': isHorizontalScrollbarVisible
        }, $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($4afcd54cfd94dbb9$exports)), 'react-spectrum-Table')),
        layout: layout,
        collection: state.collection,
        focusedKey: state.selectionManager.focusedKey,
        renderView: renderView,
        renderWrapper: renderWrapper,
        setTableWidth: columnState.setTableWidth,
        onVisibleRectChange: onVisibleRectChange,
        domRef: domRef,
        bodyRef: bodyRef,
        getColumnWidth: columnState.getColumnWidth
    })));
}
// This is a custom Virtualizer that also has a header that syncs its scroll position with the body.
function $be8a9c035a82f788$var$TableVirtualizer({ layout: layout , collection: collection , focusedKey: focusedKey , renderView: renderView , renderWrapper: renderWrapper , domRef: domRef , bodyRef: bodyRef , setTableWidth: setTableWidth , getColumnWidth: getColumnWidth , onVisibleRectChange: onVisibleRectChangeProp , ...otherProps }) {
    var ref;
    let { direction: direction  } = $3fpzo$reactariai18n.useLocale();
    let headerRef = $3fpzo$react.useRef();
    let loadingState = collection.body.props.loadingState;
    let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
    let onLoadMore = collection.body.props.onLoadMore;
    let state = $3fpzo$reactstatelyvirtualizer.useVirtualizerState({
        layout: layout,
        collection: collection,
        renderView: renderView,
        renderWrapper: renderWrapper,
        onVisibleRectChange (rect) {
            bodyRef.current.scrollTop = rect.y;
            $3fpzo$reactariavirtualizer.setScrollLeft(bodyRef.current, direction, rect.x);
        },
        transitionDuration: isLoading ? 160 : 220
    });
    let { virtualizerProps: virtualizerProps  } = $3fpzo$reactariavirtualizer.useVirtualizer({
        focusedKey: focusedKey,
        scrollToItem (key) {
            let item = collection.getItem(key);
            let column = collection.columns[0];
            state.virtualizer.scrollToItem(key, {
                duration: 0,
                // Prevent scrolling to the top when clicking on column headers.
                shouldScrollY: (item === null || item === void 0 ? void 0 : item.type) !== 'column',
                // Offset scroll position by width of selection cell
                // (which is sticky and will overlap the cell we're scrolling to).
                offsetX: column.props.isSelectionCell ? layout.getColumnWidth(column.key) : 0
            });
        }
    }, state, domRef);
    // If columnwidths change, need to relayout.
    $3fpzo$reactariautils.useLayoutEffect(()=>{
        state.virtualizer.relayoutNow({
            sizeChanged: true
        });
    }, [
        getColumnWidth,
        state.virtualizer
    ]);
    let headerHeight = ((ref = layout.getLayoutInfo('header')) === null || ref === void 0 ? void 0 : ref.rect.height) || 0;
    let visibleRect = state.virtualizer.visibleRect;
    // Sync the scroll position from the table body to the header container.
    let onScroll = $3fpzo$react.useCallback(()=>{
        headerRef.current.scrollLeft = bodyRef.current.scrollLeft;
    }, [
        bodyRef
    ]);
    let onVisibleRectChange = $3fpzo$react.useCallback((rect)=>{
        setTableWidth(rect.width);
        state.setVisibleRect(rect);
        if (!isLoading && onLoadMore) {
            let scrollOffset = state.virtualizer.contentSize.height - rect.height * 2;
            if (rect.y > scrollOffset) onLoadMore();
        }
    }, [
        onLoadMore,
        isLoading,
        state.setVisibleRect,
        state.virtualizer
    ]);
    $3fpzo$reactariautils.useLayoutEffect(()=>{
        if (!isLoading && onLoadMore && !state.isAnimating) {
            if (state.contentSize.height <= state.virtualizer.visibleRect.height) onLoadMore();
        }
    }, [
        state.contentSize,
        state.virtualizer,
        state.isAnimating,
        onLoadMore,
        isLoading
    ]);
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        ...$3fpzo$reactariautils.mergeProps(otherProps, virtualizerProps),
        ref: domRef
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        role: "presentation",
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-headWrapper'),
        style: {
            width: visibleRect.width,
            height: headerHeight,
            overflow: 'hidden',
            position: 'relative',
            willChange: state.isScrolling ? 'scroll-position' : '',
            transition: state.isAnimating ? `none ${state.virtualizer.transitionDuration}ms` : undefined
        },
        ref: headerRef
    }, state.visibleViews[0]), /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariavirtualizer.ScrollView, {
        role: "presentation",
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-body'),
        style: {
            flex: 1
        },
        innerStyle: {
            overflow: 'visible',
            transition: state.isAnimating ? `none ${state.virtualizer.transitionDuration}ms` : undefined
        },
        ref: bodyRef,
        contentSize: state.contentSize,
        onVisibleRectChange: $3fpzo$reactariautils.chain(onVisibleRectChange, onVisibleRectChangeProp),
        onScrollStart: state.startScrolling,
        onScrollEnd: state.endScrolling,
        onScroll: onScroll
    }, state.visibleViews[1])));
}
function $be8a9c035a82f788$var$TableHeader({ children: children , ...otherProps }) {
    let { rowGroupProps: rowGroupProps  } = $3fpzo$reactariatable.useTableRowGroup();
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        ...rowGroupProps,
        ...otherProps,
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-head')
    }, children));
}
function $be8a9c035a82f788$var$TableColumnHeader(props) {
    var ref, ref1, ref2, ref3;
    let { column: column  } = props;
    let ref4 = $3fpzo$react.useRef(null);
    let { state: state  } = $be8a9c035a82f788$export$3cb274deb6c2d854();
    let { columnHeaderProps: columnHeaderProps  } = $3fpzo$reactariatable.useTableColumnHeader({
        node: column,
        isVirtualized: true
    }, state, ref4);
    let columnProps = column.props;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $3fpzo$reactariainteractions.useHover(props);
    const allProps = [
        columnHeaderProps,
        hoverProps
    ];
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariafocus.FocusRing, {
        focusRingClass: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        ...$3fpzo$reactariautils.mergeProps(...allProps),
        ref: ref4,
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-headCell', {
            'is-resizable': columnProps.allowsResizing,
            'is-sortable': columnProps.allowsSorting,
            'is-sorted-desc': ((ref = state.sortDescriptor) === null || ref === void 0 ? void 0 : ref.column) === column.key && ((ref1 = state.sortDescriptor) === null || ref1 === void 0 ? void 0 : ref1.direction) === 'descending',
            'is-sorted-asc': ((ref2 = state.sortDescriptor) === null || ref2 === void 0 ? void 0 : ref2.column) === column.key && ((ref3 = state.sortDescriptor) === null || ref3 === void 0 ? void 0 : ref3.direction) === 'ascending',
            'is-hovered': isHovered,
            'spectrum-Table-cell--hideHeader': columnProps.hideHeader
        }, $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($4afcd54cfd94dbb9$exports)), 'react-spectrum-Table-cell', {
            'react-spectrum-Table-cell--alignCenter': columnProps.align === 'center' || column.colspan > 1,
            'react-spectrum-Table-cell--alignEnd': columnProps.align === 'end'
        }))
    }, columnProps.hideHeader ? /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariavisuallyhidden.VisuallyHidden, null, column.rendered) : /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-headCellContents')
    }, column.rendered), columnProps.allowsSorting && /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement(($parcel$interopDefault($3fpzo$spectrumiconsuiArrowDownSmall)), {
        UNSAFE_className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-sortedIcon')
    }))));
}
let $be8a9c035a82f788$var$_TableColumnHeaderButton = (props, ref)=>{
    let domRef = $3fpzo$reactspectrumutils.useFocusableRef(ref);
    let { buttonProps: buttonProps  } = $3fpzo$reactariabutton.useButton({
        ...props,
        elementType: 'div'
    }, domRef);
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-headCellContents')
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariafocus.FocusRing, {
        focusRingClass: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-headCellButton'),
        ...buttonProps,
        ref: domRef
    }, props.children))));
};
let $be8a9c035a82f788$var$TableColumnHeaderButton = /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).forwardRef($be8a9c035a82f788$var$_TableColumnHeaderButton);
function $be8a9c035a82f788$var$ResizableTableColumnHeader(props) {
    var ref, ref5, ref6, ref7, ref8;
    let { column: column  } = props;
    let ref9 = $3fpzo$react.useRef(null);
    let triggerRef = $3fpzo$react.useRef(null);
    let resizingRef = $3fpzo$react.useRef(null);
    let { state: state , columnState: columnState , headerRowHovered: headerRowHovered  } = $be8a9c035a82f788$export$3cb274deb6c2d854();
    let stringFormatter = $3fpzo$reactariai18n.useLocalizedStringFormatter((/*@__PURE__*/$parcel$interopDefault($a89862f958371476$exports)));
    let { columnHeaderProps: columnHeaderProps  } = $3fpzo$reactariatable.useTableColumnHeader({
        node: column,
        isVirtualized: true,
        hasMenu: true
    }, state, ref9);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $3fpzo$reactariainteractions.useHover(props);
    const allProps = [
        columnHeaderProps,
        hoverProps
    ];
    let columnProps = column.props;
    if (columnProps.width && columnProps.allowsResizing) throw new Error('Controlled state is not yet supported with column resizing. Please use defaultWidth for uncontrolled column resizing or remove the allowsResizing prop.');
    const onMenuSelect = (key)=>{
        switch(key){
            case 'sort-asc':
                state.sort(column.key, 'ascending');
                break;
            case 'sort-desc':
                state.sort(column.key, 'descending');
                break;
            case 'resize':
                columnState.onColumnResizeStart(column);
                break;
        }
    };
    let allowsSorting = (ref = column.props) === null || ref === void 0 ? void 0 : ref.allowsSorting;
    let items = $3fpzo$react.useMemo(()=>{
        let options = [
            allowsSorting ? {
                label: stringFormatter.format('sortAscending'),
                id: 'sort-asc'
            } : undefined,
            allowsSorting ? {
                label: stringFormatter.format('sortDescending'),
                id: 'sort-desc'
            } : undefined,
            {
                label: stringFormatter.format('resizeColumn'),
                id: 'resize'
            }
        ];
        return options;
    }, [
        allowsSorting
    ]);
    $3fpzo$react.useEffect(()=>{
        if (columnState.currentlyResizingColumn === column.key) // focusSafely won't actually focus because the focus moves from the menuitem to the body during the after transition wait
        // without the immediate timeout, Android Chrome doesn't move focus to the resizer
        setTimeout(()=>{
            resizingRef.current.focus();
        }, 0);
    }, [
        columnState.currentlyResizingColumn,
        column.key
    ]);
    let showResizer = headerRowHovered || columnState.currentlyResizingColumn != null;
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariafocus.FocusRing, {
        focusRingClass: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        ...$3fpzo$reactariautils.mergeProps(...allProps),
        ref: ref9,
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-headCell', {
            'is-resizable': columnProps.allowsResizing,
            'is-sortable': columnProps.allowsSorting,
            'is-sorted-desc': ((ref5 = state.sortDescriptor) === null || ref5 === void 0 ? void 0 : ref5.column) === column.key && ((ref6 = state.sortDescriptor) === null || ref6 === void 0 ? void 0 : ref6.direction) === 'descending',
            'is-sorted-asc': ((ref7 = state.sortDescriptor) === null || ref7 === void 0 ? void 0 : ref7.column) === column.key && ((ref8 = state.sortDescriptor) === null || ref8 === void 0 ? void 0 : ref8.direction) === 'ascending',
            'is-hovered': isHovered,
            'spectrum-Table-cell--hideHeader': columnProps.hideHeader
        }, $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($4afcd54cfd94dbb9$exports)), 'react-spectrum-Table-cell', {
            'react-spectrum-Table-cell--alignCenter': columnProps.align === 'center' || column.colspan > 1,
            'react-spectrum-Table-cell--alignEnd': columnProps.align === 'end'
        }))
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactspectrummenu.MenuTrigger, null, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($be8a9c035a82f788$var$TableColumnHeaderButton, {
        ref: triggerRef
    }, columnProps.hideHeader ? /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariavisuallyhidden.VisuallyHidden, null, column.rendered) : column.rendered, columnProps.allowsSorting && /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement(($parcel$interopDefault($3fpzo$spectrumiconsuiArrowDownSmall)), {
        UNSAFE_className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-sortedIcon')
    })), /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactspectrummenu.Menu, {
        onAction: onMenuSelect,
        minWidth: "size-2000",
        items: items
    }, (item)=>/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactspectrummenu.Item, null, item.label)
    )), /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($2877766f94c67a67$export$48a76196cafe3b93, {
        ref: resizingRef,
        column: column,
        showResizer: showResizer,
        triggerRef: $3fpzo$reactspectrumutils.useUnwrapDOMRef(triggerRef)
    }))));
}
function $be8a9c035a82f788$var$TableSelectAllCell({ column: column  }) {
    let ref = $3fpzo$react.useRef();
    let { state: state  } = $be8a9c035a82f788$export$3cb274deb6c2d854();
    let isSingleSelectionMode = state.selectionManager.selectionMode === 'single';
    let { columnHeaderProps: columnHeaderProps  } = $3fpzo$reactariatable.useTableColumnHeader({
        node: column,
        isVirtualized: true
    }, state, ref);
    let { checkboxProps: checkboxProps  } = $3fpzo$reactariatable.useTableSelectAllCheckbox(state);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $3fpzo$reactariainteractions.useHover({
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariafocus.FocusRing, {
        focusRingClass: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        ...$3fpzo$reactariautils.mergeProps(columnHeaderProps, hoverProps),
        ref: ref,
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-headCell', 'spectrum-Table-checkboxCell', {
            'is-hovered': isHovered
        })
    }, /*
            In single selection mode, the checkbox will be hidden.
            So to avoid leaving a column header with no accessible content,
            we use a VisuallyHidden component to include the aria-label from the checkbox,
            which for single selection will be "Select."
          */ isSingleSelectionMode && /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariavisuallyhidden.VisuallyHidden, null, checkboxProps['aria-label']), /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactspectrumcheckbox.Checkbox, {
        ...checkboxProps,
        isDisabled: isSingleSelectionMode,
        isEmphasized: true,
        UNSAFE_style: isSingleSelectionMode ? {
            visibility: 'hidden'
        } : undefined,
        UNSAFE_className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-checkbox')
    }))));
}
function $be8a9c035a82f788$var$TableRowGroup({ children: children , ...otherProps }) {
    let { rowGroupProps: rowGroupProps  } = $3fpzo$reactariatable.useTableRowGroup();
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        ...rowGroupProps,
        ...otherProps
    }, children));
}
function $be8a9c035a82f788$var$TableRow({ item: item , children: children , hasActions: hasActions , ...otherProps }) {
    var ref, ref10, ref11;
    let ref12 = $3fpzo$react.useRef();
    let { state: state , layout: layout  } = $be8a9c035a82f788$export$3cb274deb6c2d854();
    let allowsInteraction = state.selectionManager.selectionMode !== 'none' || hasActions;
    let isDisabled = !allowsInteraction || state.disabledKeys.has(item.key);
    let isSelected = state.selectionManager.isSelected(item.key);
    let { rowProps: rowProps  } = $3fpzo$reactariatable.useTableRow({
        node: item,
        isVirtualized: true
    }, state, ref12);
    let { pressProps: pressProps , isPressed: isPressed  } = $3fpzo$reactariainteractions.usePress({
        isDisabled: isDisabled
    });
    // The row should show the focus background style when any cell inside it is focused.
    // If the row itself is focused, then it should have a blue focus indicator on the left.
    let { isFocusVisible: isFocusVisibleWithin , focusProps: focusWithinProps  } = $3fpzo$reactariafocus.useFocusRing({
        within: true
    });
    let { isFocusVisible: isFocusVisible , focusProps: focusProps  } = $3fpzo$reactariafocus.useFocusRing();
    let { hoverProps: hoverProps , isHovered: isHovered  } = $3fpzo$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    let props = $3fpzo$reactariautils.mergeProps(rowProps, otherProps, focusWithinProps, focusProps, hoverProps, pressProps);
    let isFirstRow = ((ref = state.collection.rows.find((row)=>row.level === 1
    )) === null || ref === void 0 ? void 0 : ref.key) === item.key;
    let isLastRow = item.nextKey == null;
    // Figure out if the TableView content is equal or greater in height to the container. If so, we'll need to round the bottom
    // border corners of the last row when selected.
    let isFlushWithContainerBottom = false;
    if (isLastRow) {
        if (((ref10 = layout.getContentSize()) === null || ref10 === void 0 ? void 0 : ref10.height) >= ((ref11 = layout.virtualizer) === null || ref11 === void 0 ? void 0 : ref11.getVisibleRect().height)) isFlushWithContainerBottom = true;
    }
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        ...props,
        ref: ref12,
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-row', {
            'is-active': isPressed,
            'is-selected': isSelected,
            'spectrum-Table-row--highlightSelection': state.selectionManager.selectionBehavior === 'replace',
            'is-next-selected': state.selectionManager.isSelected(item.nextKey),
            'is-focused': isFocusVisibleWithin,
            'focus-ring': isFocusVisible,
            'is-hovered': isHovered,
            'is-disabled': isDisabled,
            'spectrum-Table-row--firstRow': isFirstRow,
            'spectrum-Table-row--lastRow': isLastRow,
            'spectrum-Table-row--isFlushBottom': isFlushWithContainerBottom
        })
    }, children));
}
function $be8a9c035a82f788$var$TableHeaderRow({ item: item , children: children , style: style , ...props }) {
    let { state: state  } = $be8a9c035a82f788$export$3cb274deb6c2d854();
    let ref = $3fpzo$react.useRef();
    let { rowProps: rowProps  } = $3fpzo$reactariatable.useTableHeaderRow({
        node: item,
        isVirtualized: true
    }, state, ref);
    let { hoverProps: hoverProps  } = $3fpzo$reactariainteractions.useHover(props);
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        ...$3fpzo$reactariautils.mergeProps(rowProps, hoverProps),
        ref: ref,
        style: style
    }, children));
}
function $be8a9c035a82f788$var$TableCheckboxCell({ cell: cell  }) {
    let ref = $3fpzo$react.useRef();
    let { state: state  } = $be8a9c035a82f788$export$3cb274deb6c2d854();
    let isDisabled = state.disabledKeys.has(cell.parentKey);
    let { gridCellProps: gridCellProps  } = $3fpzo$reactariatable.useTableCell({
        node: cell,
        isVirtualized: true
    }, state, ref);
    let { checkboxProps: checkboxProps  } = $3fpzo$reactariatable.useTableSelectionCheckbox({
        key: cell.parentKey
    }, state);
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariafocus.FocusRing, {
        focusRingClass: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        ...gridCellProps,
        ref: ref,
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-cell', 'spectrum-Table-checkboxCell', {
            'is-disabled': isDisabled
        }, $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($4afcd54cfd94dbb9$exports)), 'react-spectrum-Table-cell'))
    }, state.selectionManager.selectionMode !== 'none' && /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactspectrumcheckbox.Checkbox, {
        ...checkboxProps,
        isEmphasized: true,
        isDisabled: isDisabled,
        UNSAFE_className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-checkbox')
    }))));
}
function $be8a9c035a82f788$var$TableCell({ cell: cell  }) {
    let { state: state  } = $be8a9c035a82f788$export$3cb274deb6c2d854();
    let ref = $3fpzo$react.useRef();
    let columnProps = cell.column.props;
    let isDisabled = state.disabledKeys.has(cell.parentKey);
    let { gridCellProps: gridCellProps  } = $3fpzo$reactariatable.useTableCell({
        node: cell,
        isVirtualized: true
    }, state, ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement($3fpzo$reactariafocus.FocusRing, {
        focusRingClass: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'focus-ring')
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        ...gridCellProps,
        ref: ref,
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-cell', {
            'spectrum-Table-cell--divider': columnProps.showDivider && cell.column.nextKey !== null,
            'spectrum-Table-cell--hideHeader': columnProps.hideHeader,
            'is-disabled': isDisabled
        }, $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($4afcd54cfd94dbb9$exports)), 'react-spectrum-Table-cell', {
            'react-spectrum-Table-cell--alignStart': columnProps.align === 'start',
            'react-spectrum-Table-cell--alignCenter': columnProps.align === 'center',
            'react-spectrum-Table-cell--alignEnd': columnProps.align === 'end'
        }))
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("span", {
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($982885d0a34882ea$exports)), 'spectrum-Table-cellContents')
    }, cell.rendered))));
}
function $be8a9c035a82f788$var$CenteredWrapper({ children: children  }) {
    let { state: state  } = $be8a9c035a82f788$export$3cb274deb6c2d854();
    return(/*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        role: "row",
        "aria-rowindex": state.collection.headerRows.length + state.collection.size + 1,
        className: $3fpzo$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($4afcd54cfd94dbb9$exports)), 'react-spectrum-Table-centeredWrapper')
    }, /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).createElement("div", {
        role: "rowheader",
        "aria-colspan": state.collection.columns.length
    }, children)));
}
/**
 * Tables are containers for displaying information. They allow users to quickly scan, sort, compare, and take action on large amounts of data.
 */ const $be8a9c035a82f788$export$b3c27e869d856b7 = /*#__PURE__*/ ($parcel$interopDefault($3fpzo$react)).forwardRef($be8a9c035a82f788$var$TableView);



// Override TS for Column to support spectrum specific props.
const $4b0b515c47252574$export$816b5d811295e6bc = $3fpzo$reactstatelytable.Column;
