// @react-spectrum/illustratedmessage/dist/main.js
require("./main.css");
var $6dAMz$reactspectrumutils = require("@react-spectrum/utils");
var $6dAMz$reactariautils = require("@react-aria/utils");
var $6dAMz$reactspectrumlayout = require("@react-spectrum/layout");
var $6dAMz$react = require("react");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $90e6e9abe7fc12ca$exports = {};

$parcel$export($90e6e9abe7fc12ca$exports, "IllustratedMessage", () => $90e6e9abe7fc12ca$export$406dbc84c317ece0);




var $e7750c05f28ceb4f$exports = {};

$parcel$export($e7750c05f28ceb4f$exports, "spectrum-IllustratedMessage", () => $e7750c05f28ceb4f$export$e22c6753c239ec31, (v) => $e7750c05f28ceb4f$export$e22c6753c239ec31 = v);
$parcel$export($e7750c05f28ceb4f$exports, "spectrum-IllustratedMessage-heading", () => $e7750c05f28ceb4f$export$5df23faa5ed344c4, (v) => $e7750c05f28ceb4f$export$5df23faa5ed344c4 = v);
$parcel$export($e7750c05f28ceb4f$exports, "spectrum-IllustratedMessage-description", () => $e7750c05f28ceb4f$export$cf9d959111789fe7, (v) => $e7750c05f28ceb4f$export$cf9d959111789fe7 = v);
$parcel$export($e7750c05f28ceb4f$exports, "spectrum-IllustratedMessage--cta", () => $e7750c05f28ceb4f$export$29744d89c9a76d50, (v) => $e7750c05f28ceb4f$export$29744d89c9a76d50 = v);
var $e7750c05f28ceb4f$export$e22c6753c239ec31;
var $e7750c05f28ceb4f$export$5df23faa5ed344c4;
var $e7750c05f28ceb4f$export$cf9d959111789fe7;
var $e7750c05f28ceb4f$export$29744d89c9a76d50;
$e7750c05f28ceb4f$export$e22c6753c239ec31 = "spectrum-IllustratedMessage_02bd98";
$e7750c05f28ceb4f$export$5df23faa5ed344c4 = "spectrum-IllustratedMessage-heading_02bd98";
$e7750c05f28ceb4f$export$cf9d959111789fe7 = "spectrum-IllustratedMessage-description_02bd98";
$e7750c05f28ceb4f$export$29744d89c9a76d50 = "spectrum-IllustratedMessage--cta_02bd98";


function $90e6e9abe7fc12ca$var$IllustratedMessage(props, ref) {
    props = $6dAMz$reactspectrumutils.useSlotProps(props, 'illustration');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $6dAMz$reactspectrumutils.useStyleProps(otherProps);
    let headingClassName = $6dAMz$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e7750c05f28ceb4f$exports)), 'spectrum-IllustratedMessage-heading');
    let contentClassName = $6dAMz$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e7750c05f28ceb4f$exports)), 'spectrum-IllustratedMessage-description');
    let slots = {
        heading: {
            UNSAFE_className: headingClassName
        },
        content: {
            UNSAFE_className: contentClassName
        }
    };
    return(/*#__PURE__*/ ($parcel$interopDefault($6dAMz$react)).createElement($6dAMz$reactspectrumlayout.Flex, {
        ...$6dAMz$reactariautils.filterDOMProps(otherProps),
        UNSAFE_style: styleProps.style,
        isHidden: styleProps.hidden,
        UNSAFE_className: $6dAMz$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($e7750c05f28ceb4f$exports)), 'spectrum-IllustratedMessage', styleProps.className),
        ref: ref
    }, /*#__PURE__*/ ($parcel$interopDefault($6dAMz$react)).createElement($6dAMz$reactspectrumutils.SlotProvider, {
        slots: slots
    }, children)));
}
/**
 * An IllustratedMessage displays an illustration and a message, usually
 * for an empty state or an error page.
 */ let $90e6e9abe7fc12ca$export$406dbc84c317ece0 = /*#__PURE__*/ $6dAMz$react.forwardRef($90e6e9abe7fc12ca$var$IllustratedMessage);


$parcel$exportWildcard(module.exports, $90e6e9abe7fc12ca$exports);
