// @react-spectrum/illustratedmessage/dist/module.js
import "./main.css";
import {useSlotProps as $RWtz0$useSlotProps, useStyleProps as $RWtz0$useStyleProps, classNames as $RWtz0$classNames, SlotProvider as $RWtz0$SlotProvider} from "@react-spectrum/utils";
import {filterDOMProps as $RWtz0$filterDOMProps} from "@react-aria/utils";
import {Flex as $RWtz0$Flex} from "@react-spectrum/layout";
import $RWtz0$react, {forwardRef as $RWtz0$forwardRef} from "react";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $ccaf5c86c16f58c0$exports = {};

$parcel$export($ccaf5c86c16f58c0$exports, "IllustratedMessage", () => $ccaf5c86c16f58c0$export$406dbc84c317ece0);




var $4c3e55dfb549f378$exports = {};

$parcel$export($4c3e55dfb549f378$exports, "spectrum-IllustratedMessage", () => $4c3e55dfb549f378$export$e22c6753c239ec31, (v) => $4c3e55dfb549f378$export$e22c6753c239ec31 = v);
$parcel$export($4c3e55dfb549f378$exports, "spectrum-IllustratedMessage-heading", () => $4c3e55dfb549f378$export$5df23faa5ed344c4, (v) => $4c3e55dfb549f378$export$5df23faa5ed344c4 = v);
$parcel$export($4c3e55dfb549f378$exports, "spectrum-IllustratedMessage-description", () => $4c3e55dfb549f378$export$cf9d959111789fe7, (v) => $4c3e55dfb549f378$export$cf9d959111789fe7 = v);
$parcel$export($4c3e55dfb549f378$exports, "spectrum-IllustratedMessage--cta", () => $4c3e55dfb549f378$export$29744d89c9a76d50, (v) => $4c3e55dfb549f378$export$29744d89c9a76d50 = v);
var $4c3e55dfb549f378$export$e22c6753c239ec31;
var $4c3e55dfb549f378$export$5df23faa5ed344c4;
var $4c3e55dfb549f378$export$cf9d959111789fe7;
var $4c3e55dfb549f378$export$29744d89c9a76d50;
$4c3e55dfb549f378$export$e22c6753c239ec31 = "spectrum-IllustratedMessage_02bd98";
$4c3e55dfb549f378$export$5df23faa5ed344c4 = "spectrum-IllustratedMessage-heading_02bd98";
$4c3e55dfb549f378$export$cf9d959111789fe7 = "spectrum-IllustratedMessage-description_02bd98";
$4c3e55dfb549f378$export$29744d89c9a76d50 = "spectrum-IllustratedMessage--cta_02bd98";


function $ccaf5c86c16f58c0$var$IllustratedMessage(props, ref) {
    props = $RWtz0$useSlotProps(props, 'illustration');
    let { children: children , ...otherProps } = props;
    let { styleProps: styleProps  } = $RWtz0$useStyleProps(otherProps);
    let headingClassName = $RWtz0$classNames((/*@__PURE__*/$parcel$interopDefault($4c3e55dfb549f378$exports)), 'spectrum-IllustratedMessage-heading');
    let contentClassName = $RWtz0$classNames((/*@__PURE__*/$parcel$interopDefault($4c3e55dfb549f378$exports)), 'spectrum-IllustratedMessage-description');
    let slots = {
        heading: {
            UNSAFE_className: headingClassName
        },
        content: {
            UNSAFE_className: contentClassName
        }
    };
    return(/*#__PURE__*/ $RWtz0$react.createElement($RWtz0$Flex, {
        ...$RWtz0$filterDOMProps(otherProps),
        UNSAFE_style: styleProps.style,
        isHidden: styleProps.hidden,
        UNSAFE_className: $RWtz0$classNames((/*@__PURE__*/$parcel$interopDefault($4c3e55dfb549f378$exports)), 'spectrum-IllustratedMessage', styleProps.className),
        ref: ref
    }, /*#__PURE__*/ $RWtz0$react.createElement($RWtz0$SlotProvider, {
        slots: slots
    }, children)));
}
/**
 * An IllustratedMessage displays an illustration and a message, usually
 * for an empty state or an error page.
 */ let $ccaf5c86c16f58c0$export$406dbc84c317ece0 = /*#__PURE__*/ $RWtz0$forwardRef($ccaf5c86c16f58c0$var$IllustratedMessage);




export {$ccaf5c86c16f58c0$export$406dbc84c317ece0 as IllustratedMessage};
