// @react-spectrum/picker/dist/module.js
import "./main.css";
import {Item as $23896c8062ba3967$re_export$Item, Section as $23896c8062ba3967$re_export$Section} from "@react-stately/collections";
import $3toCD$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import $3toCD$spectrumiconsuiChevronDownMedium from "@spectrum-icons/ui/ChevronDownMedium";
import {useSlotProps as $3toCD$useSlotProps, useDOMRef as $3toCD$useDOMRef, useUnwrapDOMRef as $3toCD$useUnwrapDOMRef, useIsMobileDevice as $3toCD$useIsMobileDevice, dimensionValue as $3toCD$dimensionValue, classNames as $3toCD$classNames, SlotProvider as $3toCD$SlotProvider} from "@react-spectrum/utils";
import {useOverlayPosition as $3toCD$useOverlayPosition, DismissButton as $3toCD$DismissButton} from "@react-aria/overlays";
import {Field as $3toCD$Field} from "@react-spectrum/label";
import {FieldButton as $3toCD$FieldButton} from "@react-spectrum/button";
import {FocusScope as $3toCD$FocusScope} from "@react-aria/focus";
import {useSelect as $3toCD$useSelect, HiddenSelect as $3toCD$HiddenSelect} from "@react-aria/select";
import {useListBoxLayout as $3toCD$useListBoxLayout, ListBoxBase as $3toCD$ListBoxBase} from "@react-spectrum/listbox";
import {useLayoutEffect as $3toCD$useLayoutEffect, useResizeObserver as $3toCD$useResizeObserver, mergeProps as $3toCD$mergeProps} from "@react-aria/utils";
import {Tray as $3toCD$Tray, Popover as $3toCD$Popover} from "@react-spectrum/overlays";
import {useHover as $3toCD$useHover, PressResponder as $3toCD$PressResponder} from "@react-aria/interactions";
import {ProgressCircle as $3toCD$ProgressCircle} from "@react-spectrum/progress";
import $3toCD$react, {useRef as $3toCD$useRef, useState as $3toCD$useState, useCallback as $3toCD$useCallback} from "react";
import {Text as $3toCD$Text} from "@react-spectrum/text";
import {useMessageFormatter as $3toCD$useMessageFormatter} from "@react-aria/i18n";
import {useProviderProps as $3toCD$useProviderProps, useProvider as $3toCD$useProvider} from "@react-spectrum/provider";
import {useSelectState as $3toCD$useSelectState} from "@react-stately/select";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $51664a9007a3c6f9$exports = {};

$parcel$export($51664a9007a3c6f9$exports, "Picker", () => $51664a9007a3c6f9$export$ba25329847403e11);








var $23df90b22503709f$exports = {};
var $d1a348ee160b1c35$exports = {};
$d1a348ee160b1c35$exports = JSON.parse("{\"loading\":\"جارٍ التحميل...\",\"placeholder\":\"حدد خيارًا...\"}");


var $6f4b5567d8931b2c$exports = {};
$6f4b5567d8931b2c$exports = JSON.parse("{\"loading\":\"Зареждане...\",\"placeholder\":\"Изберете опция\"}");


var $413f5c9b05a53d70$exports = {};
$413f5c9b05a53d70$exports = JSON.parse("{\"loading\":\"Načítání...\",\"placeholder\":\"Vyberte vhodnou možnost...\"}");


var $23214587b45d0c7a$exports = {};
$23214587b45d0c7a$exports = JSON.parse("{\"loading\":\"Indlæser ...\",\"placeholder\":\"Vælg en mulighed ...\"}");


var $291e2fb947064c57$exports = {};
$291e2fb947064c57$exports = JSON.parse("{\"loading\":\"Laden...\",\"placeholder\":\"Option auswählen...\"}");


var $c5247287ce725677$exports = {};
$c5247287ce725677$exports = JSON.parse("{\"loading\":\"Φόρτωση...\",\"placeholder\":\"Επιλέξτε…\"}");


var $3d85b21727aef233$exports = {};
$3d85b21727aef233$exports = JSON.parse("{\"placeholder\":\"Select an option…\",\"loading\":\"Loading…\"}");


var $573c338a19d2e44a$exports = {};
$573c338a19d2e44a$exports = JSON.parse("{\"loading\":\"Cargando…\",\"placeholder\":\"Seleccione una opción…\"}");


var $52537a7562a0fcd4$exports = {};
$52537a7562a0fcd4$exports = JSON.parse("{\"loading\":\"Laadimine...\",\"placeholder\":\"Valige valikuline...\"}");


var $f128a549f8c2b170$exports = {};
$f128a549f8c2b170$exports = JSON.parse("{\"loading\":\"Ladataan…\",\"placeholder\":\"Valitse vaihtoehto...\"}");


var $f69e7a23857afc72$exports = {};
$f69e7a23857afc72$exports = JSON.parse("{\"loading\":\"Chargement...\",\"placeholder\":\"Sélectionnez une option...\"}");


var $cf880887f8515ea2$exports = {};
$cf880887f8515ea2$exports = JSON.parse("{\"loading\":\"טוען...\",\"placeholder\":\"בחר אפשרות...\"}");


var $1f679bbb79879af2$exports = {};
$1f679bbb79879af2$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"placeholder\":\"Odaberite opciju\"}");


var $fc9bc2ec180ca71d$exports = {};
$fc9bc2ec180ca71d$exports = JSON.parse("{\"loading\":\"Betöltés folyamatban…\",\"placeholder\":\"Válasszon egy opciót…\"}");


var $9fbc1ff7e87407ac$exports = {};
$9fbc1ff7e87407ac$exports = JSON.parse("{\"loading\":\"Caricamento...\",\"placeholder\":\"Seleziona un’opzione...\"}");


var $96673b78dbaf44b6$exports = {};
$96673b78dbaf44b6$exports = JSON.parse("{\"loading\":\"読み込み中...\",\"placeholder\":\"オプションを選択...\"}");


var $beddb5ded972a868$exports = {};
$beddb5ded972a868$exports = JSON.parse("{\"loading\":\"로드 중\",\"placeholder\":\"선택 사항 선택\"}");


var $2ac19a654f19cde2$exports = {};
$2ac19a654f19cde2$exports = JSON.parse("{\"loading\":\"Įkeliama...\",\"placeholder\":\"Pasirinkite parinktį...\"}");


var $844be78af2d57459$exports = {};
$844be78af2d57459$exports = JSON.parse("{\"loading\":\"Notiek ielāde...\",\"placeholder\":\"Atlasiet opciju...\"}");


var $42b05097f4e27f73$exports = {};
$42b05097f4e27f73$exports = JSON.parse("{\"loading\":\"Laster inn ...\",\"placeholder\":\"Velg et alternativ...\"}");


var $74dd3f417665b06f$exports = {};
$74dd3f417665b06f$exports = JSON.parse("{\"loading\":\"Laden...\",\"placeholder\":\"Optie selecteren...\"}");


var $6221651ca0306be3$exports = {};
$6221651ca0306be3$exports = JSON.parse("{\"loading\":\"Ładowanie...\",\"placeholder\":\"Wybierz opcję...\"}");


var $45bd58553e21ec53$exports = {};
$45bd58553e21ec53$exports = JSON.parse("{\"loading\":\"Carregando...\",\"placeholder\":\"Selecione uma opção...\"}");


var $7ae7985d84009e3a$exports = {};
$7ae7985d84009e3a$exports = JSON.parse("{\"loading\":\"A carregar...\",\"placeholder\":\"Selecionar uma opção...\"}");


var $c83f9725f6024f47$exports = {};
$c83f9725f6024f47$exports = JSON.parse("{\"loading\":\"Se încarcă...\",\"placeholder\":\"Selectați o opțiune\"}");


var $0fb522ae7b51fefd$exports = {};
$0fb522ae7b51fefd$exports = JSON.parse("{\"loading\":\"Загрузка...\",\"placeholder\":\"Выбрать параметр...\"}");


var $5ca94db2a021a726$exports = {};
$5ca94db2a021a726$exports = JSON.parse("{\"loading\":\"Načítava sa...\",\"placeholder\":\"Vyberte možnosť...\"}");


var $2c7a753ed24e9851$exports = {};
$2c7a753ed24e9851$exports = JSON.parse("{\"loading\":\"Nalaganje...\",\"placeholder\":\"Izberite možnost\"}");


var $14a1d1750c9de591$exports = {};
$14a1d1750c9de591$exports = JSON.parse("{\"loading\":\"Učitavam...\",\"placeholder\":\"Izaberite opciju\"}");


var $bf3f89a84488b301$exports = {};
$bf3f89a84488b301$exports = JSON.parse("{\"loading\":\"Läser in...\",\"placeholder\":\"Välj ett alternativ...\"}");


var $1ff90abddf42fc72$exports = {};
$1ff90abddf42fc72$exports = JSON.parse("{\"loading\":\"Yükleniyor...\",\"placeholder\":\"Bir seçim yapın…\"}");


var $4f07d8c17926a679$exports = {};
$4f07d8c17926a679$exports = JSON.parse("{\"loading\":\"Завантаження…\",\"placeholder\":\"Виберіть опцію...\"}");


var $07a8d3b903fa886f$exports = {};
$07a8d3b903fa886f$exports = JSON.parse("{\"loading\":\"正在加载...\",\"placeholder\":\"选择一个选项...\"}");


var $e1416a5539091e36$exports = {};
$e1416a5539091e36$exports = JSON.parse("{\"loading\":\"正在載入\",\"placeholder\":\"選取一個選項\"}");


$23df90b22503709f$exports = {
    "ar-AE": $d1a348ee160b1c35$exports,
    "bg-BG": $6f4b5567d8931b2c$exports,
    "cs-CZ": $413f5c9b05a53d70$exports,
    "da-DK": $23214587b45d0c7a$exports,
    "de-DE": $291e2fb947064c57$exports,
    "el-GR": $c5247287ce725677$exports,
    "en-US": $3d85b21727aef233$exports,
    "es-ES": $573c338a19d2e44a$exports,
    "et-EE": $52537a7562a0fcd4$exports,
    "fi-FI": $f128a549f8c2b170$exports,
    "fr-FR": $f69e7a23857afc72$exports,
    "he-IL": $cf880887f8515ea2$exports,
    "hr-HR": $1f679bbb79879af2$exports,
    "hu-HU": $fc9bc2ec180ca71d$exports,
    "it-IT": $9fbc1ff7e87407ac$exports,
    "ja-JP": $96673b78dbaf44b6$exports,
    "ko-KR": $beddb5ded972a868$exports,
    "lt-LT": $2ac19a654f19cde2$exports,
    "lv-LV": $844be78af2d57459$exports,
    "nb-NO": $42b05097f4e27f73$exports,
    "nl-NL": $74dd3f417665b06f$exports,
    "pl-PL": $6221651ca0306be3$exports,
    "pt-BR": $45bd58553e21ec53$exports,
    "pt-PT": $7ae7985d84009e3a$exports,
    "ro-RO": $c83f9725f6024f47$exports,
    "ru-RU": $0fb522ae7b51fefd$exports,
    "sk-SK": $5ca94db2a021a726$exports,
    "sl-SI": $2c7a753ed24e9851$exports,
    "sr-SP": $14a1d1750c9de591$exports,
    "sv-SE": $bf3f89a84488b301$exports,
    "tr-TR": $1ff90abddf42fc72$exports,
    "uk-UA": $4f07d8c17926a679$exports,
    "zh-CN": $07a8d3b903fa886f$exports,
    "zh-TW": $e1416a5539091e36$exports
};








var $3c14925a8a65017e$exports = {};

$parcel$export($3c14925a8a65017e$exports, "spectrum-Dropdown", () => $3c14925a8a65017e$export$f33c6833ddc773bd, (v) => $3c14925a8a65017e$export$f33c6833ddc773bd = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Dropdown-chevron", () => $3c14925a8a65017e$export$abbb0cf96dbbb8a0, (v) => $3c14925a8a65017e$export$abbb0cf96dbbb8a0 = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Dropdown-trigger", () => $3c14925a8a65017e$export$37a6d27a7cb47ce5, (v) => $3c14925a8a65017e$export$37a6d27a7cb47ce5 = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Dropdown-label", () => $3c14925a8a65017e$export$1235a00795b885e7, (v) => $3c14925a8a65017e$export$1235a00795b885e7 = v);
$parcel$export($3c14925a8a65017e$exports, "is-placeholder", () => $3c14925a8a65017e$export$e5b2f5233e4e5194, (v) => $3c14925a8a65017e$export$e5b2f5233e4e5194 = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Icon", () => $3c14925a8a65017e$export$d374b04f30360026, (v) => $3c14925a8a65017e$export$d374b04f30360026 = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Dropdown-invalidIcon", () => $3c14925a8a65017e$export$a32fdb5a063ea755, (v) => $3c14925a8a65017e$export$a32fdb5a063ea755 = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Dropdown-progressCircle", () => $3c14925a8a65017e$export$18281c34cabbd13f, (v) => $3c14925a8a65017e$export$18281c34cabbd13f = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Dropdown--quiet", () => $3c14925a8a65017e$export$b2306417a70c7a62, (v) => $3c14925a8a65017e$export$b2306417a70c7a62 = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Dropdown-popover--quiet", () => $3c14925a8a65017e$export$cbeaf986973c0633, (v) => $3c14925a8a65017e$export$cbeaf986973c0633 = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Field", () => $3c14925a8a65017e$export$f6d480ae1e56eba0, (v) => $3c14925a8a65017e$export$f6d480ae1e56eba0 = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Dropdown-fieldWrapper--quiet", () => $3c14925a8a65017e$export$66ddc2410b0e9eba, (v) => $3c14925a8a65017e$export$66ddc2410b0e9eba = v);
$parcel$export($3c14925a8a65017e$exports, "spectrum-Dropdown-fieldWrapper--positionSide", () => $3c14925a8a65017e$export$c0b4e9596538c7b1, (v) => $3c14925a8a65017e$export$c0b4e9596538c7b1 = v);
$parcel$export($3c14925a8a65017e$exports, "is-hovered", () => $3c14925a8a65017e$export$b8813cd5d7824ce7, (v) => $3c14925a8a65017e$export$b8813cd5d7824ce7 = v);
$parcel$export($3c14925a8a65017e$exports, "is-selected", () => $3c14925a8a65017e$export$1e0fb04f31d3c22a, (v) => $3c14925a8a65017e$export$1e0fb04f31d3c22a = v);
$parcel$export($3c14925a8a65017e$exports, "is-invalid", () => $3c14925a8a65017e$export$14a30de2866685fa, (v) => $3c14925a8a65017e$export$14a30de2866685fa = v);
$parcel$export($3c14925a8a65017e$exports, "is-disabled", () => $3c14925a8a65017e$export$d35bc1e505d1ebbf, (v) => $3c14925a8a65017e$export$d35bc1e505d1ebbf = v);
$parcel$export($3c14925a8a65017e$exports, "focus-ring", () => $3c14925a8a65017e$export$f39a09f249340e2a, (v) => $3c14925a8a65017e$export$f39a09f249340e2a = v);
var $3c14925a8a65017e$export$f33c6833ddc773bd;
var $3c14925a8a65017e$export$abbb0cf96dbbb8a0;
var $3c14925a8a65017e$export$37a6d27a7cb47ce5;
var $3c14925a8a65017e$export$1235a00795b885e7;
var $3c14925a8a65017e$export$e5b2f5233e4e5194;
var $3c14925a8a65017e$export$d374b04f30360026;
var $3c14925a8a65017e$export$a32fdb5a063ea755;
var $3c14925a8a65017e$export$18281c34cabbd13f;
var $3c14925a8a65017e$export$b2306417a70c7a62;
var $3c14925a8a65017e$export$cbeaf986973c0633;
var $3c14925a8a65017e$export$f6d480ae1e56eba0;
var $3c14925a8a65017e$export$66ddc2410b0e9eba;
var $3c14925a8a65017e$export$c0b4e9596538c7b1;
var $3c14925a8a65017e$export$b8813cd5d7824ce7;
var $3c14925a8a65017e$export$1e0fb04f31d3c22a;
var $3c14925a8a65017e$export$14a30de2866685fa;
var $3c14925a8a65017e$export$d35bc1e505d1ebbf;
var $3c14925a8a65017e$export$f39a09f249340e2a;
$3c14925a8a65017e$export$f33c6833ddc773bd = "spectrum-Dropdown_6d8ec0";
$3c14925a8a65017e$export$abbb0cf96dbbb8a0 = "spectrum-Dropdown-chevron_6d8ec0";
$3c14925a8a65017e$export$37a6d27a7cb47ce5 = "spectrum-Dropdown-trigger_6d8ec0";
$3c14925a8a65017e$export$1235a00795b885e7 = "spectrum-Dropdown-label_6d8ec0";
$3c14925a8a65017e$export$e5b2f5233e4e5194 = "is-placeholder_6d8ec0";
$3c14925a8a65017e$export$d374b04f30360026 = "spectrum-Icon_6d8ec0";
$3c14925a8a65017e$export$a32fdb5a063ea755 = "spectrum-Dropdown-invalidIcon_6d8ec0";
$3c14925a8a65017e$export$18281c34cabbd13f = "spectrum-Dropdown-progressCircle_6d8ec0";
$3c14925a8a65017e$export$b2306417a70c7a62 = "spectrum-Dropdown--quiet_6d8ec0";
$3c14925a8a65017e$export$cbeaf986973c0633 = "spectrum-Dropdown-popover--quiet_6d8ec0";
$3c14925a8a65017e$export$f6d480ae1e56eba0 = "spectrum-Field_6d8ec0";
$3c14925a8a65017e$export$66ddc2410b0e9eba = "spectrum-Dropdown-fieldWrapper--quiet_6d8ec0";
$3c14925a8a65017e$export$c0b4e9596538c7b1 = "spectrum-Dropdown-fieldWrapper--positionSide_6d8ec0";
$3c14925a8a65017e$export$b8813cd5d7824ce7 = "is-hovered_6d8ec0";
$3c14925a8a65017e$export$1e0fb04f31d3c22a = "is-selected_6d8ec0";
$3c14925a8a65017e$export$14a30de2866685fa = "is-invalid_6d8ec0";
$3c14925a8a65017e$export$d35bc1e505d1ebbf = "is-disabled_6d8ec0";
$3c14925a8a65017e$export$f39a09f249340e2a = "focus-ring_6d8ec0";






function $51664a9007a3c6f9$var$Picker(props, ref) {
    props = $3toCD$useSlotProps(props, 'picker');
    props = $3toCD$useProviderProps(props);
    let formatMessage = $3toCD$useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($23df90b22503709f$exports)));
    let { autoComplete: autoComplete , isDisabled: isDisabled , direction: direction = 'bottom' , align: align = 'start' , shouldFlip: shouldFlip = true , placeholder: placeholder = formatMessage('placeholder') , validationState: validationState , isQuiet: isQuiet , label: label , labelPosition: labelPosition = 'top' , menuWidth: menuWidth , name: name , autoFocus: autoFocus  } = props;
    let state = $3toCD$useSelectState(props);
    let domRef = $3toCD$useDOMRef(ref);
    let popoverRef = $3toCD$useRef();
    let unwrappedPopoverRef = $3toCD$useUnwrapDOMRef(popoverRef);
    let triggerRef = $3toCD$useRef();
    let unwrappedTriggerRef = $3toCD$useUnwrapDOMRef(triggerRef);
    let listboxRef = $3toCD$useRef();
    // We create the listbox layout in Picker and pass it to ListBoxBase below
    // so that the layout information can be cached even while the listbox is not mounted.
    // We also use the layout as the keyboard delegate for type to select.
    let layout = $3toCD$useListBoxLayout(state);
    let { labelProps: labelProps , triggerProps: triggerProps , valueProps: valueProps , menuProps: menuProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $3toCD$useSelect({
        ...props,
        keyboardDelegate: layout
    }, state, unwrappedTriggerRef);
    let isMobile = $3toCD$useIsMobileDevice();
    let { overlayProps: overlayProps , placement: placement , updatePosition: updatePosition  } = $3toCD$useOverlayPosition({
        targetRef: unwrappedTriggerRef,
        overlayRef: unwrappedPopoverRef,
        scrollRef: listboxRef,
        placement: `${direction} ${align}`,
        shouldFlip: shouldFlip,
        isOpen: state.isOpen && !isMobile,
        onClose: state.close
    });
    let { hoverProps: hoverProps , isHovered: isHovered  } = $3toCD$useHover({
        isDisabled: isDisabled
    });
    // Update position once the ListBox has rendered. This ensures that
    // it flips properly when it doesn't fit in the available space.
    // TODO: add ResizeObserver to useOverlayPosition so we don't need this.
    $3toCD$useLayoutEffect(()=>{
        if (state.isOpen) requestAnimationFrame(()=>{
            updatePosition();
        });
    }, [
        state.isOpen,
        updatePosition
    ]);
    let isLoadingInitial = props.isLoading && state.collection.size === 0;
    let isLoadingMore = props.isLoading && state.collection.size > 0;
    // On small screen devices, the listbox is rendered in a tray, otherwise a popover.
    let listbox = /*#__PURE__*/ $3toCD$react.createElement($3toCD$FocusScope, {
        restoreFocus: true,
        contain: isMobile
    }, /*#__PURE__*/ $3toCD$react.createElement($3toCD$DismissButton, {
        onDismiss: ()=>state.close()
    }), /*#__PURE__*/ $3toCD$react.createElement($3toCD$ListBoxBase, {
        ...menuProps,
        ref: listboxRef,
        disallowEmptySelection: true,
        autoFocus: state.focusStrategy || true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        width: isMobile ? '100%' : undefined,
        // Set max height: inherit so Tray scrolling works
        UNSAFE_style: {
            maxHeight: 'inherit'
        },
        isLoading: isLoadingMore,
        onLoadMore: props.onLoadMore
    }), /*#__PURE__*/ $3toCD$react.createElement($3toCD$DismissButton, {
        onDismiss: ()=>state.close()
    }));
    // Measure the width of the button to inform the width of the menu (below).
    let [buttonWidth, setButtonWidth] = $3toCD$useState(null);
    let { scale: scale  } = $3toCD$useProvider();
    let onResize = $3toCD$useCallback(()=>{
        if (!isMobile && unwrappedTriggerRef.current) {
            let width = unwrappedTriggerRef.current.offsetWidth;
            setButtonWidth(width);
        }
    }, [
        unwrappedTriggerRef,
        setButtonWidth,
        isMobile
    ]);
    $3toCD$useResizeObserver({
        ref: unwrappedTriggerRef,
        onResize: onResize
    });
    $3toCD$useLayoutEffect(onResize, [
        scale,
        state.selectedKey,
        onResize
    ]);
    let overlay;
    if (isMobile) overlay = /*#__PURE__*/ $3toCD$react.createElement($3toCD$Tray, {
        isOpen: state.isOpen,
        onClose: state.close
    }, listbox);
    else {
        // If quiet, use the default width, otherwise match the width of the button. This can be overridden by the menuWidth prop.
        // Always have a minimum width of the button width. When quiet, there is an extra offset to add.
        // Not using style props for this because they don't support `calc`.
        let width = isQuiet ? null : buttonWidth;
        let style = {
            ...overlayProps.style,
            width: menuWidth ? $3toCD$dimensionValue(menuWidth) : width,
            minWidth: isQuiet ? `calc(${buttonWidth}px + calc(2 * var(--spectrum-dropdown-quiet-offset)))` : buttonWidth
        };
        overlay = /*#__PURE__*/ $3toCD$react.createElement($3toCD$Popover, {
            isOpen: state.isOpen,
            UNSAFE_style: style,
            UNSAFE_className: $3toCD$classNames((/*@__PURE__*/$parcel$interopDefault($3c14925a8a65017e$exports)), 'spectrum-Dropdown-popover', {
                'spectrum-Dropdown-popover--quiet': isQuiet
            }),
            ref: popoverRef,
            placement: placement,
            hideArrow: true,
            shouldCloseOnBlur: true,
            onClose: state.close
        }, listbox);
    }
    let contents = state.selectedItem ? state.selectedItem.rendered : placeholder;
    if (typeof contents === 'string') contents = /*#__PURE__*/ $3toCD$react.createElement($3toCD$Text, null, contents);
    let picker = /*#__PURE__*/ $3toCD$react.createElement("div", {
        className: $3toCD$classNames((/*@__PURE__*/$parcel$interopDefault($3c14925a8a65017e$exports)), 'spectrum-Dropdown', {
            'is-invalid': validationState === 'invalid',
            'is-disabled': isDisabled,
            'spectrum-Dropdown--quiet': isQuiet
        })
    }, /*#__PURE__*/ $3toCD$react.createElement($3toCD$HiddenSelect, {
        autoComplete: autoComplete,
        isDisabled: isDisabled,
        state: state,
        triggerRef: unwrappedTriggerRef,
        label: label,
        name: name
    }), /*#__PURE__*/ $3toCD$react.createElement($3toCD$PressResponder, $3toCD$mergeProps(hoverProps, triggerProps), /*#__PURE__*/ $3toCD$react.createElement($3toCD$FieldButton, {
        ref: triggerRef,
        isActive: state.isOpen,
        isQuiet: isQuiet,
        isDisabled: isDisabled,
        validationState: validationState,
        autoFocus: autoFocus,
        UNSAFE_className: $3toCD$classNames((/*@__PURE__*/$parcel$interopDefault($3c14925a8a65017e$exports)), 'spectrum-Dropdown-trigger', {
            'is-hovered': isHovered
        })
    }, /*#__PURE__*/ $3toCD$react.createElement($3toCD$SlotProvider, {
        slots: {
            icon: {
                UNSAFE_className: $3toCD$classNames((/*@__PURE__*/$parcel$interopDefault($3c14925a8a65017e$exports)), 'spectrum-Icon'),
                size: 'S'
            },
            text: {
                ...valueProps,
                UNSAFE_className: $3toCD$classNames((/*@__PURE__*/$parcel$interopDefault($3c14925a8a65017e$exports)), 'spectrum-Dropdown-label', {
                    'is-placeholder': !state.selectedItem
                })
            },
            description: {
                isHidden: true
            }
        }
    }, contents), isLoadingInitial && /*#__PURE__*/ $3toCD$react.createElement($3toCD$ProgressCircle, {
        isIndeterminate: true,
        size: "S",
        "aria-label": formatMessage('loading'),
        UNSAFE_className: $3toCD$classNames((/*@__PURE__*/$parcel$interopDefault($3c14925a8a65017e$exports)), 'spectrum-Dropdown-progressCircle')
    }), validationState === 'invalid' && !isLoadingInitial && /*#__PURE__*/ $3toCD$react.createElement($3toCD$spectrumiconsuiAlertMedium, {
        UNSAFE_className: $3toCD$classNames((/*@__PURE__*/$parcel$interopDefault($3c14925a8a65017e$exports)), 'spectrum-Dropdown-invalidIcon')
    }), /*#__PURE__*/ $3toCD$react.createElement($3toCD$spectrumiconsuiChevronDownMedium, {
        UNSAFE_className: $3toCD$classNames((/*@__PURE__*/$parcel$interopDefault($3c14925a8a65017e$exports)), 'spectrum-Dropdown-chevron')
    }))), state.collection.size === 0 ? null : overlay);
    let wrapperClassName = label ? $3toCD$classNames((/*@__PURE__*/$parcel$interopDefault($3c14925a8a65017e$exports)), 'spectrum-Field', {
        'spectrum-Dropdown-fieldWrapper--quiet': isQuiet,
        'spectrum-Dropdown-fieldWrapper--positionSide': labelPosition === 'side'
    }) : '';
    return(/*#__PURE__*/ $3toCD$react.createElement($3toCD$Field, {
        ...props,
        ref: domRef,
        wrapperClassName: wrapperClassName,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        showErrorIcon: false,
        includeNecessityIndicatorInAccessibilityName: true,
        elementType: "span"
    }, picker));
}
/**
 * Pickers allow users to choose a single option from a collapsible list of options when space is limited.
 */ // forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $51664a9007a3c6f9$export$ba25329847403e11 = /*#__PURE__*/ $3toCD$react.forwardRef($51664a9007a3c6f9$var$Picker);





export {$23896c8062ba3967$re_export$Item as Item, $23896c8062ba3967$re_export$Section as Section, $51664a9007a3c6f9$export$ba25329847403e11 as Picker};
