// @react-spectrum/datepicker/dist/main.js
require("./main.css");
var $9kQ8p$reactspectrumcalendar = require("@react-spectrum/calendar");
var $9kQ8p$spectrumiconsworkflowCalendar = require("@spectrum-icons/workflow/Calendar");
var $9kQ8p$reactspectrumutils = require("@react-spectrum/utils");
var $9kQ8p$reactspectrumview = require("@react-spectrum/view");
var $9kQ8p$reactspectrumdialog = require("@react-spectrum/dialog");
var $9kQ8p$reactspectrumlabel = require("@react-spectrum/label");
var $9kQ8p$reactspectrumbutton = require("@react-spectrum/button");
var $9kQ8p$reactariautils = require("@react-aria/utils");
var $9kQ8p$react = require("react");
var $9kQ8p$reactariadatepicker = require("@react-aria/datepicker");
var $9kQ8p$reactstatelydatepicker = require("@react-stately/datepicker");
var $9kQ8p$reactariafocus = require("@react-aria/focus");
var $9kQ8p$reactariainteractions = require("@react-aria/interactions");
var $9kQ8p$reactariai18n = require("@react-aria/i18n");
var $9kQ8p$reactspectrumprovider = require("@react-spectrum/provider");
var $9kQ8p$internationalizeddate = require("@internationalized/date");
var $9kQ8p$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $9kQ8p$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $9kQ8p$reactspectrumlayout = require("@react-spectrum/layout");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "DatePicker", () => $5c1ddf85d7c27889$export$5109c6dd95d8fb00);
$parcel$export(module.exports, "DateRangePicker", () => $16cafb71dac155a4$export$17334619f3ac2224);
$parcel$export(module.exports, "TimeField", () => $740ff83729a8f317$export$5eaee2322dd727eb);
$parcel$export(module.exports, "DateField", () => $1ab74d2cd1dc2e45$export$d9781c7894a82487);








var $91c3db8c6a21ca71$exports = {};

$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-fieldWrapper", () => $91c3db8c6a21ca71$export$48501287c6ce4708, (v) => $91c3db8c6a21ca71$export$48501287c6ce4708 = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-startField", () => $91c3db8c6a21ca71$export$eec05853f5fc7ef8, (v) => $91c3db8c6a21ca71$export$eec05853f5fc7ef8 = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-endField", () => $91c3db8c6a21ca71$export$b41789438b49c852, (v) => $91c3db8c6a21ca71$export$b41789438b49c852 = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-field", () => $91c3db8c6a21ca71$export$8cbece9fb909350d, (v) => $91c3db8c6a21ca71$export$8cbece9fb909350d = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-input", () => $91c3db8c6a21ca71$export$4cdb39b006dc840a, (v) => $91c3db8c6a21ca71$export$4cdb39b006dc840a = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-DateField", () => $91c3db8c6a21ca71$export$c96d01b692101b8a, (v) => $91c3db8c6a21ca71$export$c96d01b692101b8a = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-TimeField", () => $91c3db8c6a21ca71$export$61c201f1ba261729, (v) => $91c3db8c6a21ca71$export$61c201f1ba261729 = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-TimeField-fieldWrapper", () => $91c3db8c6a21ca71$export$3008240fa163ee96, (v) => $91c3db8c6a21ca71$export$3008240fa163ee96 = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-inputContents", () => $91c3db8c6a21ca71$export$609545772ae0466a, (v) => $91c3db8c6a21ca71$export$609545772ae0466a = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-rangeDash", () => $91c3db8c6a21ca71$export$1ae39afd2cf49425, (v) => $91c3db8c6a21ca71$export$1ae39afd2cf49425 = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-segments", () => $91c3db8c6a21ca71$export$b9c34f0306ac2ffc, (v) => $91c3db8c6a21ca71$export$b9c34f0306ac2ffc = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-literal", () => $91c3db8c6a21ca71$export$24a265734c16a355, (v) => $91c3db8c6a21ca71$export$24a265734c16a355 = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-DatePicker-cell", () => $91c3db8c6a21ca71$export$947f5cb8a73639a4, (v) => $91c3db8c6a21ca71$export$947f5cb8a73639a4 = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-DatePicker-placeholder", () => $91c3db8c6a21ca71$export$d7b1df8540d8ce55, (v) => $91c3db8c6a21ca71$export$d7b1df8540d8ce55 = v);
$parcel$export($91c3db8c6a21ca71$exports, "is-placeholder", () => $91c3db8c6a21ca71$export$e5b2f5233e4e5194, (v) => $91c3db8c6a21ca71$export$e5b2f5233e4e5194 = v);
$parcel$export($91c3db8c6a21ca71$exports, "is-read-only", () => $91c3db8c6a21ca71$export$dddce3ae23a6e3ea, (v) => $91c3db8c6a21ca71$export$dddce3ae23a6e3ea = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-dialog", () => $91c3db8c6a21ca71$export$7c3cb625bb792356, (v) => $91c3db8c6a21ca71$export$7c3cb625bb792356 = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-dialogContent", () => $91c3db8c6a21ca71$export$94c97d6b6a927921, (v) => $91c3db8c6a21ca71$export$94c97d6b6a927921 = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-calendar", () => $91c3db8c6a21ca71$export$640bfc6f7362791a, (v) => $91c3db8c6a21ca71$export$640bfc6f7362791a = v);
$parcel$export($91c3db8c6a21ca71$exports, "is-invalid", () => $91c3db8c6a21ca71$export$14a30de2866685fa, (v) => $91c3db8c6a21ca71$export$14a30de2866685fa = v);
$parcel$export($91c3db8c6a21ca71$exports, "react-spectrum-Datepicker-timeFields", () => $91c3db8c6a21ca71$export$8d47712a13573e39, (v) => $91c3db8c6a21ca71$export$8d47712a13573e39 = v);
var $91c3db8c6a21ca71$export$48501287c6ce4708;
var $91c3db8c6a21ca71$export$eec05853f5fc7ef8;
var $91c3db8c6a21ca71$export$b41789438b49c852;
var $91c3db8c6a21ca71$export$8cbece9fb909350d;
var $91c3db8c6a21ca71$export$4cdb39b006dc840a;
var $91c3db8c6a21ca71$export$c96d01b692101b8a;
var $91c3db8c6a21ca71$export$61c201f1ba261729;
var $91c3db8c6a21ca71$export$3008240fa163ee96;
var $91c3db8c6a21ca71$export$609545772ae0466a;
var $91c3db8c6a21ca71$export$1ae39afd2cf49425;
var $91c3db8c6a21ca71$export$b9c34f0306ac2ffc;
var $91c3db8c6a21ca71$export$24a265734c16a355;
var $91c3db8c6a21ca71$export$947f5cb8a73639a4;
var $91c3db8c6a21ca71$export$d7b1df8540d8ce55;
var $91c3db8c6a21ca71$export$e5b2f5233e4e5194;
var $91c3db8c6a21ca71$export$dddce3ae23a6e3ea;
var $91c3db8c6a21ca71$export$7c3cb625bb792356;
var $91c3db8c6a21ca71$export$94c97d6b6a927921;
var $91c3db8c6a21ca71$export$640bfc6f7362791a;
var $91c3db8c6a21ca71$export$14a30de2866685fa;
var $91c3db8c6a21ca71$export$8d47712a13573e39;
$91c3db8c6a21ca71$export$48501287c6ce4708 = "react-spectrum-Datepicker-fieldWrapper_f02f90";
$91c3db8c6a21ca71$export$eec05853f5fc7ef8 = "react-spectrum-Datepicker-startField_f02f90";
$91c3db8c6a21ca71$export$b41789438b49c852 = "react-spectrum-Datepicker-endField_f02f90";
$91c3db8c6a21ca71$export$8cbece9fb909350d = "react-spectrum-Datepicker-field_f02f90";
$91c3db8c6a21ca71$export$4cdb39b006dc840a = "react-spectrum-Datepicker-input_f02f90";
$91c3db8c6a21ca71$export$c96d01b692101b8a = "react-spectrum-DateField_f02f90";
$91c3db8c6a21ca71$export$61c201f1ba261729 = "react-spectrum-TimeField_f02f90";
$91c3db8c6a21ca71$export$3008240fa163ee96 = "react-spectrum-TimeField-fieldWrapper_f02f90";
$91c3db8c6a21ca71$export$609545772ae0466a = "react-spectrum-Datepicker-inputContents_f02f90";
$91c3db8c6a21ca71$export$1ae39afd2cf49425 = "react-spectrum-Datepicker-rangeDash_f02f90";
$91c3db8c6a21ca71$export$b9c34f0306ac2ffc = "react-spectrum-Datepicker-segments_f02f90";
$91c3db8c6a21ca71$export$24a265734c16a355 = "react-spectrum-Datepicker-literal_f02f90";
$91c3db8c6a21ca71$export$947f5cb8a73639a4 = "react-spectrum-DatePicker-cell_f02f90";
$91c3db8c6a21ca71$export$d7b1df8540d8ce55 = "react-spectrum-DatePicker-placeholder_f02f90";
$91c3db8c6a21ca71$export$e5b2f5233e4e5194 = "is-placeholder_f02f90";
$91c3db8c6a21ca71$export$dddce3ae23a6e3ea = "is-read-only_f02f90";
$91c3db8c6a21ca71$export$7c3cb625bb792356 = "react-spectrum-Datepicker-dialog_f02f90";
$91c3db8c6a21ca71$export$94c97d6b6a927921 = "react-spectrum-Datepicker-dialogContent_f02f90";
$91c3db8c6a21ca71$export$640bfc6f7362791a = "react-spectrum-Datepicker-calendar_f02f90";
$91c3db8c6a21ca71$export$14a30de2866685fa = "is-invalid_f02f90";
$91c3db8c6a21ca71$export$8d47712a13573e39 = "react-spectrum-Datepicker-timeFields_f02f90";



function $edac158d20ebe4a9$export$6388987c5223b54e({ segment: segment , state: state , ...otherProps }) {
    switch(segment.type){
        // A separator, e.g. punctuation
        case 'literal':
            return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($edac158d20ebe4a9$var$LiteralSegment, {
                segment: segment
            }));
        // Editable segment
        default:
            return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($edac158d20ebe4a9$var$EditableSegment, {
                segment: segment,
                state: state,
                ...otherProps
            }));
    }
}
function $edac158d20ebe4a9$var$LiteralSegment({ segment: segment  }) {
    return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("span", {
        "aria-hidden": "true",
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-literal'),
        "data-testid": segment.type === 'literal' ? undefined : segment.type
    }, segment.text));
}
function $edac158d20ebe4a9$var$EditableSegment({ segment: segment , state: state  }) {
    let ref = $9kQ8p$react.useRef();
    let { segmentProps: segmentProps  } = $9kQ8p$reactariadatepicker.useDateSegment(segment, state, ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        ...segmentProps,
        ref: ref,
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-DatePicker-cell', {
            'is-placeholder': segment.isPlaceholder,
            'is-read-only': !segment.isEditable
        }),
        style: {
            ...segmentProps.style,
            minWidth: segment.maxValue != null ? String(segment.maxValue).length + 'ch' : null
        },
        "data-testid": segment.type
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("span", {
        "aria-hidden": "true",
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-DatePicker-placeholder')
    }, segment.placeholder), segment.isPlaceholder ? '' : segment.text));
}







function $bd22dc8dad9522bc$export$34dc4cfa15ead1(props) {
    let { isDisabled: isDisabled , isReadOnly: isReadOnly , isRequired: isRequired , inputClassName: inputClassName  } = props;
    let ref = $9kQ8p$react.useRef();
    let { locale: locale  } = $9kQ8p$reactariai18n.useLocale();
    let state = $9kQ8p$reactstatelydatepicker.useDateFieldState({
        ...props,
        locale: locale,
        createCalendar: $9kQ8p$internationalizeddate.createCalendar
    });
    let { fieldProps: fieldProps  } = $9kQ8p$reactariadatepicker.useDateField(props, state, ref);
    return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        ...fieldProps,
        "data-testid": props['data-testid'],
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-segments', inputClassName),
        ref: ref
    }, state.segments.map((segment, i)=>/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($edac158d20ebe4a9$export$6388987c5223b54e, {
            key: i,
            segment: segment,
            state: state,
            isDisabled: isDisabled,
            isReadOnly: isReadOnly,
            isRequired: isRequired
        })
    )));
}













var $40cb3a00c193680f$exports = {};

$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-input", () => $40cb3a00c193680f$export$1ac009feb2b0bd0c, (v) => $40cb3a00c193680f$export$1ac009feb2b0bd0c = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield", () => $40cb3a00c193680f$export$2c8a10299d8b3418, (v) => $40cb3a00c193680f$export$2c8a10299d8b3418 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield--quiet", () => $40cb3a00c193680f$export$bffd5e3d61a81737, (v) => $40cb3a00c193680f$export$bffd5e3d61a81737 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield--multiline", () => $40cb3a00c193680f$export$9699f1c538a448a0, (v) => $40cb3a00c193680f$export$9699f1c538a448a0 = v);
$parcel$export($40cb3a00c193680f$exports, "is-placeholder", () => $40cb3a00c193680f$export$e5b2f5233e4e5194, (v) => $40cb3a00c193680f$export$e5b2f5233e4e5194 = v);
$parcel$export($40cb3a00c193680f$exports, "is-hovered", () => $40cb3a00c193680f$export$b8813cd5d7824ce7, (v) => $40cb3a00c193680f$export$b8813cd5d7824ce7 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield--valid", () => $40cb3a00c193680f$export$c2abbc3f970170b7, (v) => $40cb3a00c193680f$export$c2abbc3f970170b7 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield--invalid", () => $40cb3a00c193680f$export$58e50b10d30123df, (v) => $40cb3a00c193680f$export$58e50b10d30123df = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield--loadable", () => $40cb3a00c193680f$export$14fd30fea6f1348c, (v) => $40cb3a00c193680f$export$14fd30fea6f1348c = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-wrapper", () => $40cb3a00c193680f$export$dd1165e0e6012973, (v) => $40cb3a00c193680f$export$dd1165e0e6012973 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-validationIcon", () => $40cb3a00c193680f$export$cf0ead5b44db0da3, (v) => $40cb3a00c193680f$export$cf0ead5b44db0da3 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-icon", () => $40cb3a00c193680f$export$222d1ccd1870be1d, (v) => $40cb3a00c193680f$export$222d1ccd1870be1d = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-inputIcon", () => $40cb3a00c193680f$export$c1cdcaa5fe76a871, (v) => $40cb3a00c193680f$export$c1cdcaa5fe76a871 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-circleLoader", () => $40cb3a00c193680f$export$82a249fb4d6127, (v) => $40cb3a00c193680f$export$82a249fb4d6127 = v);
$parcel$export($40cb3a00c193680f$exports, "is-focused", () => $40cb3a00c193680f$export$e7dc768d35940237, (v) => $40cb3a00c193680f$export$e7dc768d35940237 = v);
$parcel$export($40cb3a00c193680f$exports, "focus-ring", () => $40cb3a00c193680f$export$f39a09f249340e2a, (v) => $40cb3a00c193680f$export$f39a09f249340e2a = v);
$parcel$export($40cb3a00c193680f$exports, "is-disabled", () => $40cb3a00c193680f$export$d35bc1e505d1ebbf, (v) => $40cb3a00c193680f$export$d35bc1e505d1ebbf = v);
var $40cb3a00c193680f$export$1ac009feb2b0bd0c;
var $40cb3a00c193680f$export$2c8a10299d8b3418;
var $40cb3a00c193680f$export$bffd5e3d61a81737;
var $40cb3a00c193680f$export$9699f1c538a448a0;
var $40cb3a00c193680f$export$e5b2f5233e4e5194;
var $40cb3a00c193680f$export$b8813cd5d7824ce7;
var $40cb3a00c193680f$export$c2abbc3f970170b7;
var $40cb3a00c193680f$export$58e50b10d30123df;
var $40cb3a00c193680f$export$14fd30fea6f1348c;
var $40cb3a00c193680f$export$dd1165e0e6012973;
var $40cb3a00c193680f$export$cf0ead5b44db0da3;
var $40cb3a00c193680f$export$222d1ccd1870be1d;
var $40cb3a00c193680f$export$c1cdcaa5fe76a871;
var $40cb3a00c193680f$export$82a249fb4d6127;
var $40cb3a00c193680f$export$e7dc768d35940237;
var $40cb3a00c193680f$export$f39a09f249340e2a;
var $40cb3a00c193680f$export$d35bc1e505d1ebbf;
$40cb3a00c193680f$export$1ac009feb2b0bd0c = "spectrum-Textfield-input_73bc77";
$40cb3a00c193680f$export$2c8a10299d8b3418 = "spectrum-Textfield_73bc77";
$40cb3a00c193680f$export$bffd5e3d61a81737 = "spectrum-Textfield--quiet_73bc77";
$40cb3a00c193680f$export$9699f1c538a448a0 = "spectrum-Textfield--multiline_73bc77";
$40cb3a00c193680f$export$e5b2f5233e4e5194 = "is-placeholder_73bc77";
$40cb3a00c193680f$export$b8813cd5d7824ce7 = "is-hovered_73bc77";
$40cb3a00c193680f$export$c2abbc3f970170b7 = "spectrum-Textfield--valid_73bc77";
$40cb3a00c193680f$export$58e50b10d30123df = "spectrum-Textfield--invalid_73bc77";
$40cb3a00c193680f$export$14fd30fea6f1348c = "spectrum-Textfield--loadable_73bc77";
$40cb3a00c193680f$export$dd1165e0e6012973 = "spectrum-Textfield-wrapper_73bc77";
$40cb3a00c193680f$export$cf0ead5b44db0da3 = "spectrum-Textfield-validationIcon_73bc77";
$40cb3a00c193680f$export$222d1ccd1870be1d = "spectrum-Textfield-icon_73bc77";
$40cb3a00c193680f$export$c1cdcaa5fe76a871 = "spectrum-Textfield-inputIcon_73bc77";
$40cb3a00c193680f$export$82a249fb4d6127 = "spectrum-Textfield-circleLoader_73bc77";
$40cb3a00c193680f$export$e7dc768d35940237 = "is-focused_73bc77";
$40cb3a00c193680f$export$f39a09f249340e2a = "focus-ring_73bc77";
$40cb3a00c193680f$export$d35bc1e505d1ebbf = "is-disabled_73bc77";


function $c3ebc4931d9b9977$var$Input(props, ref) {
    let inputRef = $9kQ8p$react.useRef(null);
    let { isDisabled: isDisabled , isQuiet: isQuiet , inputClassName: inputClassName , validationState: validationState , children: children , fieldProps: fieldProps , className: className , style: style  } = props;
    // Reserve padding for the error icon when the width of the input is unconstrained.
    // When constrained, don't reserve space because adding it only when invalid will
    // not cause a layout shift.
    let [reservePadding1, setReservePadding] = $9kQ8p$reactspectrumutils.useValueEffect(false);
    let onResize = $9kQ8p$react.useCallback(()=>setReservePadding(function*(reservePadding) {
            if (inputRef.current) {
                if (reservePadding) // Try to collapse padding if the content is clipped.
                {
                    if (inputRef.current.scrollWidth > inputRef.current.offsetWidth) {
                        let width = inputRef.current.parentElement.offsetWidth;
                        yield false;
                        // If removing padding causes a layout shift, add it back.
                        if (inputRef.current.parentElement.offsetWidth !== width) yield true;
                    }
                } else // Try to add padding if the content is not clipped.
                if (inputRef.current.offsetWidth >= inputRef.current.scrollWidth) {
                    let width = inputRef.current.parentElement.offsetWidth;
                    yield true;
                    // If adding padding does not change the width (i.e. width is constrained), remove it again.
                    if (inputRef.current.parentElement.offsetWidth === width) yield false;
                }
            }
        })
    , [
        inputRef,
        setReservePadding
    ]);
    $9kQ8p$reactariautils.useLayoutEffect(onResize, [
        onResize
    ]);
    $9kQ8p$reactariautils.useResizeObserver({
        ref: inputRef,
        onResize: onResize
    });
    // We also need to listen for resize events of the window so we can detect
    // when there is enough space for the padding to be re-added. Ideally we'd
    // use a resize observer on a parent element, but it's hard to know _what_
    // parent element.
    $9kQ8p$reactariautils.useEvent($9kQ8p$react.useRef(typeof window !== 'undefined' ? window : null), 'resize', onResize);
    let isInvalid = validationState === 'invalid';
    let textfieldClass = $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($40cb3a00c193680f$exports)), 'spectrum-Textfield', {
        'spectrum-Textfield--invalid': isInvalid,
        'spectrum-Textfield--valid': validationState === 'valid',
        'spectrum-Textfield--quiet': isQuiet
    }, $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-field'), className);
    let inputClass = $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($40cb3a00c193680f$exports)), 'spectrum-Textfield-input', {
        'is-disabled': isDisabled,
        'is-invalid': isInvalid
    }, reservePadding1 && $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-input'), inputClassName);
    let iconClass = $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($40cb3a00c193680f$exports)), 'spectrum-Textfield-validationIcon');
    let validationIcon = null;
    if (validationState === 'invalid') validationIcon = /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement(($parcel$interopDefault($9kQ8p$spectrumiconsuiAlertMedium)), {
        "data-testid": "invalid-icon",
        UNSAFE_className: iconClass
    });
    else if (validationState === 'valid') validationIcon = /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement(($parcel$interopDefault($9kQ8p$spectrumiconsuiCheckmarkMedium)), {
        "data-testid": "valid-icon",
        UNSAFE_className: iconClass
    });
    return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        role: "presentation",
        ...fieldProps,
        className: textfieldClass,
        style: style
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactariafocus.FocusRing, {
        focusClass: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($40cb3a00c193680f$exports)), 'is-focused'),
        focusRingClass: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($40cb3a00c193680f$exports)), 'focus-ring'),
        isTextInput: true,
        within: true
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        role: "presentation",
        className: inputClass
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        role: "presentation",
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-inputContents'),
        ref: $9kQ8p$reactariautils.mergeRefs(ref, inputRef)
    }, children))), validationIcon));
}
const $c3ebc4931d9b9977$export$f5b8910cec6cf069 = /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).forwardRef($c3ebc4931d9b9977$var$Input);


var $dc9e5db3fad3e690$exports = {};
var $4390d311cc480168$exports = {};
$4390d311cc480168$exports = {
    "time": `Time`,
    "startTime": `Start time`,
    "endTime": `End time`
};


$dc9e5db3fad3e690$exports = {
    "en-US": $4390d311cc480168$exports
};





var $2f86633ce5b04f1e$exports = {};

$parcel$export($2f86633ce5b04f1e$exports, "spectrum-InputGroup", () => $2f86633ce5b04f1e$export$5be359c1227eedcb, (v) => $2f86633ce5b04f1e$export$5be359c1227eedcb = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-FieldButton", () => $2f86633ce5b04f1e$export$a024c99ecf6b5741, (v) => $2f86633ce5b04f1e$export$a024c99ecf6b5741 = v);
$parcel$export($2f86633ce5b04f1e$exports, "is-disabled", () => $2f86633ce5b04f1e$export$d35bc1e505d1ebbf, (v) => $2f86633ce5b04f1e$export$d35bc1e505d1ebbf = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-InputGroup--invalid", () => $2f86633ce5b04f1e$export$3b48509739a9cda0, (v) => $2f86633ce5b04f1e$export$3b48509739a9cda0 = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-InputGroup--quiet", () => $2f86633ce5b04f1e$export$ebf265ffc062029b, (v) => $2f86633ce5b04f1e$export$ebf265ffc062029b = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-InputGroup-field", () => $2f86633ce5b04f1e$export$ea77459595e26bec, (v) => $2f86633ce5b04f1e$export$ea77459595e26bec = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-InputGroup-input-validationIcon", () => $2f86633ce5b04f1e$export$f7ab13ee113f4884, (v) => $2f86633ce5b04f1e$export$f7ab13ee113f4884 = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-InputGroup-input-circleLoader", () => $2f86633ce5b04f1e$export$dcc9e00cdad967fa, (v) => $2f86633ce5b04f1e$export$dcc9e00cdad967fa = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-InputGroup-input", () => $2f86633ce5b04f1e$export$92706f5c6fe4074, (v) => $2f86633ce5b04f1e$export$92706f5c6fe4074 = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-InputGroup-icon", () => $2f86633ce5b04f1e$export$b91b6af799fd93a8, (v) => $2f86633ce5b04f1e$export$b91b6af799fd93a8 = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-InputGroup-popover--quiet", () => $2f86633ce5b04f1e$export$5e116178f9b1556a, (v) => $2f86633ce5b04f1e$export$5e116178f9b1556a = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-Datepicker--range", () => $2f86633ce5b04f1e$export$1d8b0b29e11d7cd, (v) => $2f86633ce5b04f1e$export$1d8b0b29e11d7cd = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-Datepicker--datetimeRange", () => $2f86633ce5b04f1e$export$58dd7cd23c92699f, (v) => $2f86633ce5b04f1e$export$58dd7cd23c92699f = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-Datepicker-startField", () => $2f86633ce5b04f1e$export$80f9c838a8406469, (v) => $2f86633ce5b04f1e$export$80f9c838a8406469 = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-Datepicker-endField", () => $2f86633ce5b04f1e$export$1c72fb41fa4f21ee, (v) => $2f86633ce5b04f1e$export$1c72fb41fa4f21ee = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-Datepicker--rangeDash", () => $2f86633ce5b04f1e$export$202113e883898dc, (v) => $2f86633ce5b04f1e$export$202113e883898dc = v);
$parcel$export($2f86633ce5b04f1e$exports, "is-focused", () => $2f86633ce5b04f1e$export$e7dc768d35940237, (v) => $2f86633ce5b04f1e$export$e7dc768d35940237 = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-Datepicker-focusRing", () => $2f86633ce5b04f1e$export$eb262b15b9261fde, (v) => $2f86633ce5b04f1e$export$eb262b15b9261fde = v);
$parcel$export($2f86633ce5b04f1e$exports, "is-hovered", () => $2f86633ce5b04f1e$export$b8813cd5d7824ce7, (v) => $2f86633ce5b04f1e$export$b8813cd5d7824ce7 = v);
$parcel$export($2f86633ce5b04f1e$exports, "focus-ring", () => $2f86633ce5b04f1e$export$f39a09f249340e2a, (v) => $2f86633ce5b04f1e$export$f39a09f249340e2a = v);
$parcel$export($2f86633ce5b04f1e$exports, "is-selected", () => $2f86633ce5b04f1e$export$1e0fb04f31d3c22a, (v) => $2f86633ce5b04f1e$export$1e0fb04f31d3c22a = v);
$parcel$export($2f86633ce5b04f1e$exports, "spectrum-FieldButton--invalid", () => $2f86633ce5b04f1e$export$c9f503f672e8a3c1, (v) => $2f86633ce5b04f1e$export$c9f503f672e8a3c1 = v);
var $2f86633ce5b04f1e$export$5be359c1227eedcb;
var $2f86633ce5b04f1e$export$a024c99ecf6b5741;
var $2f86633ce5b04f1e$export$d35bc1e505d1ebbf;
var $2f86633ce5b04f1e$export$3b48509739a9cda0;
var $2f86633ce5b04f1e$export$ebf265ffc062029b;
var $2f86633ce5b04f1e$export$ea77459595e26bec;
var $2f86633ce5b04f1e$export$f7ab13ee113f4884;
var $2f86633ce5b04f1e$export$dcc9e00cdad967fa;
var $2f86633ce5b04f1e$export$92706f5c6fe4074;
var $2f86633ce5b04f1e$export$b91b6af799fd93a8;
var $2f86633ce5b04f1e$export$5e116178f9b1556a;
var $2f86633ce5b04f1e$export$1d8b0b29e11d7cd;
var $2f86633ce5b04f1e$export$58dd7cd23c92699f;
var $2f86633ce5b04f1e$export$80f9c838a8406469;
var $2f86633ce5b04f1e$export$1c72fb41fa4f21ee;
var $2f86633ce5b04f1e$export$202113e883898dc;
var $2f86633ce5b04f1e$export$e7dc768d35940237;
var $2f86633ce5b04f1e$export$eb262b15b9261fde;
var $2f86633ce5b04f1e$export$b8813cd5d7824ce7;
var $2f86633ce5b04f1e$export$f39a09f249340e2a;
var $2f86633ce5b04f1e$export$1e0fb04f31d3c22a;
var $2f86633ce5b04f1e$export$c9f503f672e8a3c1;
$2f86633ce5b04f1e$export$5be359c1227eedcb = "spectrum-InputGroup_a0942f";
$2f86633ce5b04f1e$export$a024c99ecf6b5741 = "spectrum-FieldButton_a0942f";
$2f86633ce5b04f1e$export$d35bc1e505d1ebbf = "is-disabled_a0942f";
$2f86633ce5b04f1e$export$3b48509739a9cda0 = "spectrum-InputGroup--invalid_a0942f";
$2f86633ce5b04f1e$export$ebf265ffc062029b = "spectrum-InputGroup--quiet_a0942f";
$2f86633ce5b04f1e$export$ea77459595e26bec = "spectrum-InputGroup-field_a0942f";
$2f86633ce5b04f1e$export$f7ab13ee113f4884 = "spectrum-InputGroup-input-validationIcon_a0942f";
$2f86633ce5b04f1e$export$dcc9e00cdad967fa = "spectrum-InputGroup-input-circleLoader_a0942f";
$2f86633ce5b04f1e$export$92706f5c6fe4074 = "spectrum-InputGroup-input_a0942f";
$2f86633ce5b04f1e$export$b91b6af799fd93a8 = "spectrum-InputGroup-icon_a0942f";
$2f86633ce5b04f1e$export$5e116178f9b1556a = "spectrum-InputGroup-popover--quiet_a0942f";
$2f86633ce5b04f1e$export$1d8b0b29e11d7cd = "spectrum-Datepicker--range_a0942f";
$2f86633ce5b04f1e$export$58dd7cd23c92699f = "spectrum-Datepicker--datetimeRange_a0942f";
$2f86633ce5b04f1e$export$80f9c838a8406469 = "spectrum-Datepicker-startField_a0942f";
$2f86633ce5b04f1e$export$1c72fb41fa4f21ee = "spectrum-Datepicker-endField_a0942f";
$2f86633ce5b04f1e$export$202113e883898dc = "spectrum-Datepicker--rangeDash_a0942f";
$2f86633ce5b04f1e$export$e7dc768d35940237 = "is-focused_a0942f";
$2f86633ce5b04f1e$export$eb262b15b9261fde = "spectrum-Datepicker-focusRing_a0942f";
$2f86633ce5b04f1e$export$b8813cd5d7824ce7 = "is-hovered_a0942f";
$2f86633ce5b04f1e$export$f39a09f249340e2a = "focus-ring_a0942f";
$2f86633ce5b04f1e$export$1e0fb04f31d3c22a = "is-selected_a0942f";
$2f86633ce5b04f1e$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_a0942f";















function $4d2e419f522adb48$export$322f4580ccd8dde6(props) {
    let formatter = $9kQ8p$reactariai18n.useDateFormatter({
        dateStyle: 'short'
    });
    let displayNames = $9kQ8p$reactariadatepicker.useDisplayNames();
    return $9kQ8p$react.useMemo(()=>{
        if (props.description) return props.description;
        if (props.showFormatHelpText) return formatter.formatToParts(new Date()).map((s)=>{
            if (s.type === 'literal') return s.value;
            return displayNames.of(s.type);
        }).join(' ');
        return '';
    }, [
        props.description,
        props.showFormatHelpText,
        formatter,
        displayNames
    ]);
}
function $4d2e419f522adb48$export$12ce2869ce471b1f(maxVisibleMonths) {
    let { scale: scale  } = $9kQ8p$reactspectrumprovider.useProvider();
    let [visibleMonths, setVisibleMonths] = $9kQ8p$react.useState($4d2e419f522adb48$var$getVisibleMonths(scale));
    $9kQ8p$reactariautils.useLayoutEffect(()=>{
        let onResize = ()=>setVisibleMonths($4d2e419f522adb48$var$getVisibleMonths(scale))
        ;
        onResize();
        window.addEventListener('resize', onResize);
        return ()=>{
            window.removeEventListener('resize', onResize);
        };
    }, [
        scale
    ]);
    return Math.max(1, Math.min(visibleMonths, maxVisibleMonths, 3));
}
function $4d2e419f522adb48$var$getVisibleMonths(scale) {
    if (typeof window === 'undefined') return 1;
    let monthWidth = scale === 'large' ? 336 : 280;
    let gap = scale === 'large' ? 30 : 24;
    let popoverPadding = scale === 'large' ? 32 : 48;
    return Math.floor((window.innerWidth - popoverPadding * 2) / (monthWidth + gap));
}
function $4d2e419f522adb48$export$71a23a36270e4bf0(ref) {
    let domRef = $9kQ8p$react.useRef();
    $9kQ8p$react.useImperativeHandle(ref, ()=>({
            ...$9kQ8p$reactspectrumutils.createDOMRef(domRef),
            focus () {
                $9kQ8p$reactariafocus.createFocusManager(domRef).focusFirst({
                    tabbable: true
                });
            }
        })
    );
    return domRef;
}






function $740ff83729a8f317$var$TimeField(props, ref) {
    props = $9kQ8p$reactspectrumprovider.useProviderProps(props);
    let { autoFocus: autoFocus , isDisabled: isDisabled , isReadOnly: isReadOnly , isRequired: isRequired , isQuiet: isQuiet  } = props;
    let domRef = $4d2e419f522adb48$export$71a23a36270e4bf0(ref);
    let { locale: locale  } = $9kQ8p$reactariai18n.useLocale();
    let state = $9kQ8p$reactstatelydatepicker.useTimeFieldState({
        ...props,
        locale: locale
    });
    let { labelProps: labelProps , fieldProps: fieldProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $9kQ8p$reactariadatepicker.useTimeField(props, state, domRef);
    return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumlabel.Field, {
        ...props,
        ref: domRef,
        elementType: "span",
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: state.validationState,
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-TimeField-fieldWrapper')
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($c3ebc4931d9b9977$export$f5b8910cec6cf069, {
        fieldProps: fieldProps,
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        autoFocus: autoFocus,
        validationState: state.validationState,
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-TimeField')
    }, state.segments.map((segment, i)=>/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($edac158d20ebe4a9$export$6388987c5223b54e, {
            key: i,
            segment: segment,
            state: state,
            isDisabled: isDisabled,
            isReadOnly: isReadOnly,
            isRequired: isRequired
        })
    ))));
}
/**
 * TimeFields allow users to enter and edit time values using a keyboard.
 * Each part of the time is displayed in an individually editable segment.
 */ const $740ff83729a8f317$export$5eaee2322dd727eb = /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).forwardRef($740ff83729a8f317$var$TimeField);









function $5c1ddf85d7c27889$var$DatePicker(props, ref) {
    props = $9kQ8p$reactspectrumprovider.useProviderProps(props);
    let { autoFocus: autoFocus , isQuiet: isQuiet , isDisabled: isDisabled , isReadOnly: isReadOnly , placeholderValue: placeholderValue , maxVisibleMonths: maxVisibleMonths = 1  } = props;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $9kQ8p$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    let targetRef = $9kQ8p$react.useRef();
    let state = $9kQ8p$reactstatelydatepicker.useDatePickerState({
        ...props,
        shouldCloseOnSelect: ()=>!state.hasTime
    });
    let { groupProps: groupProps , labelProps: labelProps , fieldProps: fieldProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps , buttonProps: buttonProps , dialogProps: dialogProps , calendarProps: calendarProps  } = $9kQ8p$reactariadatepicker.useDatePicker(props, state, targetRef);
    let { isOpen: isOpen , setOpen: setOpen  } = state;
    let { direction: direction  } = $9kQ8p$reactariai18n.useLocale();
    let domRef = $4d2e419f522adb48$export$71a23a36270e4bf0(ref);
    let stringFormatter = $9kQ8p$reactariai18n.useLocalizedStringFormatter((/*@__PURE__*/$parcel$interopDefault($dc9e5db3fad3e690$exports)));
    let { isFocused: isFocused , isFocusVisible: isFocusVisible , focusProps: focusProps  } = $9kQ8p$reactariafocus.useFocusRing({
        within: true,
        isTextInput: true,
        autoFocus: autoFocus
    });
    let { isFocused: isFocusedButton , focusProps: focusPropsButton  } = $9kQ8p$reactariafocus.useFocusRing({
        within: false,
        isTextInput: false,
        autoFocus: autoFocus
    });
    let className = $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($2f86633ce5b04f1e$exports)), 'spectrum-InputGroup', {
        'spectrum-InputGroup--quiet': isQuiet,
        'spectrum-InputGroup--invalid': state.validationState === 'invalid',
        'is-disabled': isDisabled,
        'is-hovered': isHovered,
        'is-focused': isFocused,
        'focus-ring': isFocusVisible && !isFocusedButton
    });
    let fieldClassName = $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($2f86633ce5b04f1e$exports)), 'spectrum-InputGroup-input', {
        'is-disabled': isDisabled,
        'is-invalid': state.validationState === 'invalid'
    });
    // Note: this description is intentionally not passed to useDatePicker.
    // The format help text is unnecessary for screen reader users because each segment already has a label.
    let description = $4d2e419f522adb48$export$322f4580ccd8dde6(props);
    if (description && !props.description) descriptionProps.id = null;
    let placeholder = placeholderValue;
    let timePlaceholder = placeholder && 'hour' in placeholder ? placeholder : null;
    let timeMinValue = props.minValue && 'hour' in props.minValue ? props.minValue : null;
    let timeMaxValue = props.maxValue && 'hour' in props.maxValue ? props.maxValue : null;
    let timeGranularity = state.granularity === 'hour' || state.granularity === 'minute' || state.granularity === 'second' ? state.granularity : null;
    let showTimeField = !!timeGranularity;
    let visibleMonths = $4d2e419f522adb48$export$12ce2869ce471b1f(maxVisibleMonths);
    return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumlabel.Field, {
        ...props,
        ref: domRef,
        elementType: "span",
        description: description,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: state.validationState,
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-fieldWrapper')
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        ...$9kQ8p$reactariautils.mergeProps(groupProps, hoverProps, focusProps),
        className: className,
        ref: targetRef
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($c3ebc4931d9b9977$export$f5b8910cec6cf069, {
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: state.validationState,
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($2f86633ce5b04f1e$exports)), 'spectrum-InputGroup-field'),
        inputClassName: fieldClassName
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($bd22dc8dad9522bc$export$34dc4cfa15ead1, {
        ...fieldProps,
        "data-testid": "date-field",
        isQuiet: isQuiet
    })), /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumdialog.DialogTrigger, {
        type: "popover",
        mobileType: "tray",
        placement: direction === 'rtl' ? 'bottom right' : 'bottom left',
        targetRef: targetRef,
        hideArrow: true,
        isOpen: isOpen,
        onOpenChange: setOpen,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumbutton.FieldButton, {
        ...$9kQ8p$reactariautils.mergeProps(buttonProps, focusPropsButton),
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($2f86633ce5b04f1e$exports)), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: state.validationState,
        isDisabled: isDisabled || isReadOnly
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement(($parcel$interopDefault($9kQ8p$spectrumiconsworkflowCalendar)), null)), /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumdialog.Dialog, {
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-dialog'),
        ...dialogProps
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumview.Content, null, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-dialogContent')
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumcalendar.Calendar, {
        ...calendarProps,
        visibleMonths: visibleMonths,
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-calendar', {
            'is-invalid': state.validationState === 'invalid'
        })
    }), showTimeField && /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-timeFields')
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($740ff83729a8f317$export$5eaee2322dd727eb, {
        label: stringFormatter.format('time'),
        value: state.timeValue,
        onChange: state.setTimeValue,
        placeholderValue: timePlaceholder,
        granularity: timeGranularity,
        minValue: timeMinValue,
        maxValue: timeMaxValue,
        hourCycle: props.hourCycle,
        hideTimeZone: props.hideTimeZone,
        marginTop: "size-100"
    })))))))));
}
/**
 * DatePickers combine a DateField and a Calendar popover to allow users to enter or select a date and time value.
 */ const $5c1ddf85d7c27889$export$5109c6dd95d8fb00 = /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).forwardRef($5c1ddf85d7c27889$var$DatePicker);

























function $16cafb71dac155a4$var$DateRangePicker(props, ref) {
    var ref1, ref2;
    props = $9kQ8p$reactspectrumprovider.useProviderProps(props);
    let { isQuiet: isQuiet , isDisabled: isDisabled , isReadOnly: isReadOnly , autoFocus: autoFocus , placeholderValue: placeholderValue , maxVisibleMonths: maxVisibleMonths = 1  } = props;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $9kQ8p$reactariainteractions.useHover({
        isDisabled: isDisabled
    });
    let targetRef = $9kQ8p$react.useRef();
    let state = $9kQ8p$reactstatelydatepicker.useDateRangePickerState({
        ...props,
        shouldCloseOnSelect: ()=>!state.hasTime
    });
    let { labelProps: labelProps , groupProps: groupProps , buttonProps: buttonProps , dialogProps: dialogProps , startFieldProps: startFieldProps , endFieldProps: endFieldProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps , calendarProps: calendarProps  } = $9kQ8p$reactariadatepicker.useDateRangePicker(props, state, targetRef);
    let { isOpen: isOpen , setOpen: setOpen  } = state;
    let { direction: direction  } = $9kQ8p$reactariai18n.useLocale();
    let domRef = $4d2e419f522adb48$export$71a23a36270e4bf0(ref);
    let stringFormatter = $9kQ8p$reactariai18n.useLocalizedStringFormatter((/*@__PURE__*/$parcel$interopDefault($dc9e5db3fad3e690$exports)));
    let { isFocused: isFocused , isFocusVisible: isFocusVisible , focusProps: focusProps  } = $9kQ8p$reactariafocus.useFocusRing({
        within: true,
        isTextInput: true,
        autoFocus: autoFocus
    });
    let { isFocused: isFocusedButton , focusProps: focusPropsButton  } = $9kQ8p$reactariafocus.useFocusRing({
        within: false,
        isTextInput: false,
        autoFocus: autoFocus
    });
    let className = $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($2f86633ce5b04f1e$exports)), 'spectrum-InputGroup', {
        'spectrum-InputGroup--quiet': isQuiet,
        'spectrum-InputGroup--invalid': state.validationState === 'invalid',
        'is-disabled': isDisabled,
        'is-hovered': isHovered,
        'is-focused': isFocused,
        'focus-ring': isFocusVisible && !isFocusedButton
    });
    let fieldClassName = $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($2f86633ce5b04f1e$exports)), 'spectrum-InputGroup-input', {
        'is-disabled': isDisabled,
        'is-invalid': state.validationState === 'invalid'
    });
    // Note: this description is intentionally not passed to useDatePicker.
    // The format help text is unnecessary for screen reader users because each segment already has a label.
    let description = $4d2e419f522adb48$export$322f4580ccd8dde6(props);
    if (description && !props.description) descriptionProps.id = null;
    let placeholder = placeholderValue;
    let timePlaceholder = placeholder && 'hour' in placeholder ? placeholder : null;
    let timeMinValue = props.minValue && 'hour' in props.minValue ? props.minValue : null;
    let timeMaxValue = props.maxValue && 'hour' in props.maxValue ? props.maxValue : null;
    let timeGranularity = state.granularity === 'hour' || state.granularity === 'minute' || state.granularity === 'second' ? state.granularity : null;
    let showTimeField = !!timeGranularity;
    let visibleMonths = $4d2e419f522adb48$export$12ce2869ce471b1f(maxVisibleMonths);
    return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumlabel.Field, {
        ...props,
        ref: domRef,
        elementType: "span",
        description: description,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: state.validationState,
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-fieldWrapper')
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        ...$9kQ8p$reactariautils.mergeProps(groupProps, hoverProps, focusProps),
        className: className,
        ref: targetRef
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($c3ebc4931d9b9977$export$f5b8910cec6cf069, {
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: state.validationState,
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($2f86633ce5b04f1e$exports)), 'spectrum-InputGroup-field'),
        inputClassName: fieldClassName
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($bd22dc8dad9522bc$export$34dc4cfa15ead1, {
        ...startFieldProps,
        "data-testid": "start-date",
        isQuiet: props.isQuiet,
        inputClassName: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-startField')
    }), /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($16cafb71dac155a4$var$DateRangeDash, null), /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($bd22dc8dad9522bc$export$34dc4cfa15ead1, {
        ...endFieldProps,
        "data-testid": "end-date",
        isQuiet: props.isQuiet,
        inputClassName: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($2f86633ce5b04f1e$exports)), 'spectrum-Datepicker-endField', $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-endField'))
    })), /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumdialog.DialogTrigger, {
        type: "popover",
        mobileType: "tray",
        placement: direction === 'rtl' ? 'bottom right' : 'bottom left',
        targetRef: targetRef,
        hideArrow: true,
        isOpen: isOpen,
        onOpenChange: setOpen,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumbutton.FieldButton, {
        ...$9kQ8p$reactariautils.mergeProps(buttonProps, focusPropsButton),
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($2f86633ce5b04f1e$exports)), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: state.validationState,
        isDisabled: isDisabled || isReadOnly
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement(($parcel$interopDefault($9kQ8p$spectrumiconsworkflowCalendar)), null)), /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumdialog.Dialog, {
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-dialog'),
        ...dialogProps
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumview.Content, null, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-dialogContent')
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumcalendar.RangeCalendar, {
        ...calendarProps,
        visibleMonths: visibleMonths,
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-calendar', {
            'is-invalid': state.validationState === 'invalid'
        })
    }), showTimeField && /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumlayout.Flex, {
        gap: "size-100",
        marginTop: "size-100",
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-timeFields')
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($740ff83729a8f317$export$5eaee2322dd727eb, {
        label: stringFormatter.format('startTime'),
        value: ((ref1 = state.timeRange) === null || ref1 === void 0 ? void 0 : ref1.start) || null,
        onChange: (v)=>state.setTime('start', v)
        ,
        placeholderValue: timePlaceholder,
        granularity: timeGranularity,
        minValue: timeMinValue,
        maxValue: timeMaxValue,
        hourCycle: props.hourCycle,
        hideTimeZone: props.hideTimeZone,
        flex: true
    }), /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($740ff83729a8f317$export$5eaee2322dd727eb, {
        label: stringFormatter.format('endTime'),
        value: ((ref2 = state.timeRange) === null || ref2 === void 0 ? void 0 : ref2.end) || null,
        onChange: (v)=>state.setTime('end', v)
        ,
        placeholderValue: timePlaceholder,
        granularity: timeGranularity,
        minValue: timeMinValue,
        maxValue: timeMaxValue,
        hourCycle: props.hourCycle,
        hideTimeZone: props.hideTimeZone,
        flex: true
    })))))))));
}
function $16cafb71dac155a4$var$DateRangeDash() {
    return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement("div", {
        "aria-hidden": "true",
        "data-testid": "date-range-dash",
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-rangeDash')
    }));
}
/**
 * DateRangePickers combine two DateFields and a RangeCalendar popover to allow users
 * to enter or select a date and time range.
 */ const $16cafb71dac155a4$export$17334619f3ac2224 = /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).forwardRef($16cafb71dac155a4$var$DateRangePicker);















function $1ab74d2cd1dc2e45$var$DateField(props, ref) {
    props = $9kQ8p$reactspectrumprovider.useProviderProps(props);
    let { autoFocus: autoFocus , isDisabled: isDisabled , isReadOnly: isReadOnly , isRequired: isRequired , isQuiet: isQuiet  } = props;
    let domRef = $4d2e419f522adb48$export$71a23a36270e4bf0(ref);
    let { locale: locale  } = $9kQ8p$reactariai18n.useLocale();
    let state = $9kQ8p$reactstatelydatepicker.useDateFieldState({
        ...props,
        locale: locale,
        createCalendar: $9kQ8p$internationalizeddate.createCalendar
    });
    let { labelProps: labelProps , fieldProps: fieldProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $9kQ8p$reactariadatepicker.useDateField(props, state, domRef);
    // Note: this description is intentionally not passed to useDatePicker.
    // The format help text is unnecessary for screen reader users because each segment already has a label.
    let description = $4d2e419f522adb48$export$322f4580ccd8dde6(props);
    if (description && !props.description) descriptionProps.id = null;
    return(/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($9kQ8p$reactspectrumlabel.Field, {
        ...props,
        ref: domRef,
        elementType: "span",
        description: description,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: state.validationState,
        UNSAFE_className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-Datepicker-fieldWrapper')
    }, /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($c3ebc4931d9b9977$export$f5b8910cec6cf069, {
        fieldProps: fieldProps,
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        autoFocus: autoFocus,
        validationState: state.validationState,
        className: $9kQ8p$reactspectrumutils.classNames((/*@__PURE__*/$parcel$interopDefault($91c3db8c6a21ca71$exports)), 'react-spectrum-DateField')
    }, state.segments.map((segment, i)=>/*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).createElement($edac158d20ebe4a9$export$6388987c5223b54e, {
            key: i,
            segment: segment,
            state: state,
            isDisabled: isDisabled,
            isReadOnly: isReadOnly,
            isRequired: isRequired
        })
    ))));
}
/**
 * DateFields allow users to enter and edit date and time values using a keyboard.
 * Each part of a date value is displayed in an individually editable segment.
 */ const $1ab74d2cd1dc2e45$export$d9781c7894a82487 = /*#__PURE__*/ ($parcel$interopDefault($9kQ8p$react)).forwardRef($1ab74d2cd1dc2e45$var$DateField);
