// @react-spectrum/datepicker/dist/module.js
import "./main.css";
import {Calendar as $aayOp$Calendar, RangeCalendar as $aayOp$RangeCalendar} from "@react-spectrum/calendar";
import $aayOp$spectrumiconsworkflowCalendar from "@spectrum-icons/workflow/Calendar";
import {classNames as $aayOp$classNames, useValueEffect as $aayOp$useValueEffect, createDOMRef as $aayOp$createDOMRef} from "@react-spectrum/utils";
import {Content as $aayOp$Content} from "@react-spectrum/view";
import {DialogTrigger as $aayOp$DialogTrigger, Dialog as $aayOp$Dialog} from "@react-spectrum/dialog";
import {Field as $aayOp$Field} from "@react-spectrum/label";
import {FieldButton as $aayOp$FieldButton} from "@react-spectrum/button";
import {mergeProps as $aayOp$mergeProps, useLayoutEffect as $aayOp$useLayoutEffect, useResizeObserver as $aayOp$useResizeObserver, useEvent as $aayOp$useEvent, mergeRefs as $aayOp$mergeRefs} from "@react-aria/utils";
import $aayOp$react, {useRef as $aayOp$useRef, useCallback as $aayOp$useCallback, useMemo as $aayOp$useMemo, useState as $aayOp$useState, useImperativeHandle as $aayOp$useImperativeHandle} from "react";
import {useDatePicker as $aayOp$useDatePicker, useDateField as $aayOp$useDateField, useDateSegment as $aayOp$useDateSegment, useTimeField as $aayOp$useTimeField, useDisplayNames as $aayOp$useDisplayNames, useDateRangePicker as $aayOp$useDateRangePicker} from "@react-aria/datepicker";
import {useDatePickerState as $aayOp$useDatePickerState, useDateFieldState as $aayOp$useDateFieldState, useTimeFieldState as $aayOp$useTimeFieldState, useDateRangePickerState as $aayOp$useDateRangePickerState} from "@react-stately/datepicker";
import {useFocusRing as $aayOp$useFocusRing, FocusRing as $aayOp$FocusRing, createFocusManager as $aayOp$createFocusManager} from "@react-aria/focus";
import {useHover as $aayOp$useHover} from "@react-aria/interactions";
import {useLocale as $aayOp$useLocale, useLocalizedStringFormatter as $aayOp$useLocalizedStringFormatter, useDateFormatter as $aayOp$useDateFormatter} from "@react-aria/i18n";
import {useProviderProps as $aayOp$useProviderProps, useProvider as $aayOp$useProvider} from "@react-spectrum/provider";
import {createCalendar as $aayOp$createCalendar} from "@internationalized/date";
import $aayOp$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import $aayOp$spectrumiconsuiCheckmarkMedium from "@spectrum-icons/ui/CheckmarkMedium";
import {Flex as $aayOp$Flex} from "@react-spectrum/layout";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}








var $465eb3a688ef3440$exports = {};

$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-fieldWrapper", () => $465eb3a688ef3440$export$48501287c6ce4708, (v) => $465eb3a688ef3440$export$48501287c6ce4708 = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-startField", () => $465eb3a688ef3440$export$eec05853f5fc7ef8, (v) => $465eb3a688ef3440$export$eec05853f5fc7ef8 = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-endField", () => $465eb3a688ef3440$export$b41789438b49c852, (v) => $465eb3a688ef3440$export$b41789438b49c852 = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-field", () => $465eb3a688ef3440$export$8cbece9fb909350d, (v) => $465eb3a688ef3440$export$8cbece9fb909350d = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-input", () => $465eb3a688ef3440$export$4cdb39b006dc840a, (v) => $465eb3a688ef3440$export$4cdb39b006dc840a = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-DateField", () => $465eb3a688ef3440$export$c96d01b692101b8a, (v) => $465eb3a688ef3440$export$c96d01b692101b8a = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-TimeField", () => $465eb3a688ef3440$export$61c201f1ba261729, (v) => $465eb3a688ef3440$export$61c201f1ba261729 = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-TimeField-fieldWrapper", () => $465eb3a688ef3440$export$3008240fa163ee96, (v) => $465eb3a688ef3440$export$3008240fa163ee96 = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-inputContents", () => $465eb3a688ef3440$export$609545772ae0466a, (v) => $465eb3a688ef3440$export$609545772ae0466a = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-rangeDash", () => $465eb3a688ef3440$export$1ae39afd2cf49425, (v) => $465eb3a688ef3440$export$1ae39afd2cf49425 = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-segments", () => $465eb3a688ef3440$export$b9c34f0306ac2ffc, (v) => $465eb3a688ef3440$export$b9c34f0306ac2ffc = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-literal", () => $465eb3a688ef3440$export$24a265734c16a355, (v) => $465eb3a688ef3440$export$24a265734c16a355 = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-DatePicker-cell", () => $465eb3a688ef3440$export$947f5cb8a73639a4, (v) => $465eb3a688ef3440$export$947f5cb8a73639a4 = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-DatePicker-placeholder", () => $465eb3a688ef3440$export$d7b1df8540d8ce55, (v) => $465eb3a688ef3440$export$d7b1df8540d8ce55 = v);
$parcel$export($465eb3a688ef3440$exports, "is-placeholder", () => $465eb3a688ef3440$export$e5b2f5233e4e5194, (v) => $465eb3a688ef3440$export$e5b2f5233e4e5194 = v);
$parcel$export($465eb3a688ef3440$exports, "is-read-only", () => $465eb3a688ef3440$export$dddce3ae23a6e3ea, (v) => $465eb3a688ef3440$export$dddce3ae23a6e3ea = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-dialog", () => $465eb3a688ef3440$export$7c3cb625bb792356, (v) => $465eb3a688ef3440$export$7c3cb625bb792356 = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-dialogContent", () => $465eb3a688ef3440$export$94c97d6b6a927921, (v) => $465eb3a688ef3440$export$94c97d6b6a927921 = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-calendar", () => $465eb3a688ef3440$export$640bfc6f7362791a, (v) => $465eb3a688ef3440$export$640bfc6f7362791a = v);
$parcel$export($465eb3a688ef3440$exports, "is-invalid", () => $465eb3a688ef3440$export$14a30de2866685fa, (v) => $465eb3a688ef3440$export$14a30de2866685fa = v);
$parcel$export($465eb3a688ef3440$exports, "react-spectrum-Datepicker-timeFields", () => $465eb3a688ef3440$export$8d47712a13573e39, (v) => $465eb3a688ef3440$export$8d47712a13573e39 = v);
var $465eb3a688ef3440$export$48501287c6ce4708;
var $465eb3a688ef3440$export$eec05853f5fc7ef8;
var $465eb3a688ef3440$export$b41789438b49c852;
var $465eb3a688ef3440$export$8cbece9fb909350d;
var $465eb3a688ef3440$export$4cdb39b006dc840a;
var $465eb3a688ef3440$export$c96d01b692101b8a;
var $465eb3a688ef3440$export$61c201f1ba261729;
var $465eb3a688ef3440$export$3008240fa163ee96;
var $465eb3a688ef3440$export$609545772ae0466a;
var $465eb3a688ef3440$export$1ae39afd2cf49425;
var $465eb3a688ef3440$export$b9c34f0306ac2ffc;
var $465eb3a688ef3440$export$24a265734c16a355;
var $465eb3a688ef3440$export$947f5cb8a73639a4;
var $465eb3a688ef3440$export$d7b1df8540d8ce55;
var $465eb3a688ef3440$export$e5b2f5233e4e5194;
var $465eb3a688ef3440$export$dddce3ae23a6e3ea;
var $465eb3a688ef3440$export$7c3cb625bb792356;
var $465eb3a688ef3440$export$94c97d6b6a927921;
var $465eb3a688ef3440$export$640bfc6f7362791a;
var $465eb3a688ef3440$export$14a30de2866685fa;
var $465eb3a688ef3440$export$8d47712a13573e39;
$465eb3a688ef3440$export$48501287c6ce4708 = "react-spectrum-Datepicker-fieldWrapper_f02f90";
$465eb3a688ef3440$export$eec05853f5fc7ef8 = "react-spectrum-Datepicker-startField_f02f90";
$465eb3a688ef3440$export$b41789438b49c852 = "react-spectrum-Datepicker-endField_f02f90";
$465eb3a688ef3440$export$8cbece9fb909350d = "react-spectrum-Datepicker-field_f02f90";
$465eb3a688ef3440$export$4cdb39b006dc840a = "react-spectrum-Datepicker-input_f02f90";
$465eb3a688ef3440$export$c96d01b692101b8a = "react-spectrum-DateField_f02f90";
$465eb3a688ef3440$export$61c201f1ba261729 = "react-spectrum-TimeField_f02f90";
$465eb3a688ef3440$export$3008240fa163ee96 = "react-spectrum-TimeField-fieldWrapper_f02f90";
$465eb3a688ef3440$export$609545772ae0466a = "react-spectrum-Datepicker-inputContents_f02f90";
$465eb3a688ef3440$export$1ae39afd2cf49425 = "react-spectrum-Datepicker-rangeDash_f02f90";
$465eb3a688ef3440$export$b9c34f0306ac2ffc = "react-spectrum-Datepicker-segments_f02f90";
$465eb3a688ef3440$export$24a265734c16a355 = "react-spectrum-Datepicker-literal_f02f90";
$465eb3a688ef3440$export$947f5cb8a73639a4 = "react-spectrum-DatePicker-cell_f02f90";
$465eb3a688ef3440$export$d7b1df8540d8ce55 = "react-spectrum-DatePicker-placeholder_f02f90";
$465eb3a688ef3440$export$e5b2f5233e4e5194 = "is-placeholder_f02f90";
$465eb3a688ef3440$export$dddce3ae23a6e3ea = "is-read-only_f02f90";
$465eb3a688ef3440$export$7c3cb625bb792356 = "react-spectrum-Datepicker-dialog_f02f90";
$465eb3a688ef3440$export$94c97d6b6a927921 = "react-spectrum-Datepicker-dialogContent_f02f90";
$465eb3a688ef3440$export$640bfc6f7362791a = "react-spectrum-Datepicker-calendar_f02f90";
$465eb3a688ef3440$export$14a30de2866685fa = "is-invalid_f02f90";
$465eb3a688ef3440$export$8d47712a13573e39 = "react-spectrum-Datepicker-timeFields_f02f90";



function $cfa347761f172638$export$6388987c5223b54e({ segment: segment , state: state , ...otherProps }) {
    switch(segment.type){
        // A separator, e.g. punctuation
        case 'literal':
            return(/*#__PURE__*/ $aayOp$react.createElement($cfa347761f172638$var$LiteralSegment, {
                segment: segment
            }));
        // Editable segment
        default:
            return(/*#__PURE__*/ $aayOp$react.createElement($cfa347761f172638$var$EditableSegment, {
                segment: segment,
                state: state,
                ...otherProps
            }));
    }
}
function $cfa347761f172638$var$LiteralSegment({ segment: segment  }) {
    return(/*#__PURE__*/ $aayOp$react.createElement("span", {
        "aria-hidden": "true",
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-literal'),
        "data-testid": segment.type === 'literal' ? undefined : segment.type
    }, segment.text));
}
function $cfa347761f172638$var$EditableSegment({ segment: segment , state: state  }) {
    let ref = $aayOp$useRef();
    let { segmentProps: segmentProps  } = $aayOp$useDateSegment(segment, state, ref);
    return(/*#__PURE__*/ $aayOp$react.createElement("div", {
        ...segmentProps,
        ref: ref,
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-DatePicker-cell', {
            'is-placeholder': segment.isPlaceholder,
            'is-read-only': !segment.isEditable
        }),
        style: {
            ...segmentProps.style,
            minWidth: segment.maxValue != null ? String(segment.maxValue).length + 'ch' : null
        },
        "data-testid": segment.type
    }, /*#__PURE__*/ $aayOp$react.createElement("span", {
        "aria-hidden": "true",
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-DatePicker-placeholder')
    }, segment.placeholder), segment.isPlaceholder ? '' : segment.text));
}







function $d2bb88c82604b25f$export$34dc4cfa15ead1(props) {
    let { isDisabled: isDisabled , isReadOnly: isReadOnly , isRequired: isRequired , inputClassName: inputClassName  } = props;
    let ref = $aayOp$useRef();
    let { locale: locale  } = $aayOp$useLocale();
    let state = $aayOp$useDateFieldState({
        ...props,
        locale: locale,
        createCalendar: $aayOp$createCalendar
    });
    let { fieldProps: fieldProps  } = $aayOp$useDateField(props, state, ref);
    return(/*#__PURE__*/ $aayOp$react.createElement("div", {
        ...fieldProps,
        "data-testid": props['data-testid'],
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-segments', inputClassName),
        ref: ref
    }, state.segments.map((segment, i)=>/*#__PURE__*/ $aayOp$react.createElement($cfa347761f172638$export$6388987c5223b54e, {
            key: i,
            segment: segment,
            state: state,
            isDisabled: isDisabled,
            isReadOnly: isReadOnly,
            isRequired: isRequired
        })
    )));
}













var $c14528ddc146317f$exports = {};

$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-input", () => $c14528ddc146317f$export$1ac009feb2b0bd0c, (v) => $c14528ddc146317f$export$1ac009feb2b0bd0c = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield", () => $c14528ddc146317f$export$2c8a10299d8b3418, (v) => $c14528ddc146317f$export$2c8a10299d8b3418 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield--quiet", () => $c14528ddc146317f$export$bffd5e3d61a81737, (v) => $c14528ddc146317f$export$bffd5e3d61a81737 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield--multiline", () => $c14528ddc146317f$export$9699f1c538a448a0, (v) => $c14528ddc146317f$export$9699f1c538a448a0 = v);
$parcel$export($c14528ddc146317f$exports, "is-placeholder", () => $c14528ddc146317f$export$e5b2f5233e4e5194, (v) => $c14528ddc146317f$export$e5b2f5233e4e5194 = v);
$parcel$export($c14528ddc146317f$exports, "is-hovered", () => $c14528ddc146317f$export$b8813cd5d7824ce7, (v) => $c14528ddc146317f$export$b8813cd5d7824ce7 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield--valid", () => $c14528ddc146317f$export$c2abbc3f970170b7, (v) => $c14528ddc146317f$export$c2abbc3f970170b7 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield--invalid", () => $c14528ddc146317f$export$58e50b10d30123df, (v) => $c14528ddc146317f$export$58e50b10d30123df = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield--loadable", () => $c14528ddc146317f$export$14fd30fea6f1348c, (v) => $c14528ddc146317f$export$14fd30fea6f1348c = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-wrapper", () => $c14528ddc146317f$export$dd1165e0e6012973, (v) => $c14528ddc146317f$export$dd1165e0e6012973 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-validationIcon", () => $c14528ddc146317f$export$cf0ead5b44db0da3, (v) => $c14528ddc146317f$export$cf0ead5b44db0da3 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-icon", () => $c14528ddc146317f$export$222d1ccd1870be1d, (v) => $c14528ddc146317f$export$222d1ccd1870be1d = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-inputIcon", () => $c14528ddc146317f$export$c1cdcaa5fe76a871, (v) => $c14528ddc146317f$export$c1cdcaa5fe76a871 = v);
$parcel$export($c14528ddc146317f$exports, "spectrum-Textfield-circleLoader", () => $c14528ddc146317f$export$82a249fb4d6127, (v) => $c14528ddc146317f$export$82a249fb4d6127 = v);
$parcel$export($c14528ddc146317f$exports, "is-focused", () => $c14528ddc146317f$export$e7dc768d35940237, (v) => $c14528ddc146317f$export$e7dc768d35940237 = v);
$parcel$export($c14528ddc146317f$exports, "focus-ring", () => $c14528ddc146317f$export$f39a09f249340e2a, (v) => $c14528ddc146317f$export$f39a09f249340e2a = v);
$parcel$export($c14528ddc146317f$exports, "is-disabled", () => $c14528ddc146317f$export$d35bc1e505d1ebbf, (v) => $c14528ddc146317f$export$d35bc1e505d1ebbf = v);
var $c14528ddc146317f$export$1ac009feb2b0bd0c;
var $c14528ddc146317f$export$2c8a10299d8b3418;
var $c14528ddc146317f$export$bffd5e3d61a81737;
var $c14528ddc146317f$export$9699f1c538a448a0;
var $c14528ddc146317f$export$e5b2f5233e4e5194;
var $c14528ddc146317f$export$b8813cd5d7824ce7;
var $c14528ddc146317f$export$c2abbc3f970170b7;
var $c14528ddc146317f$export$58e50b10d30123df;
var $c14528ddc146317f$export$14fd30fea6f1348c;
var $c14528ddc146317f$export$dd1165e0e6012973;
var $c14528ddc146317f$export$cf0ead5b44db0da3;
var $c14528ddc146317f$export$222d1ccd1870be1d;
var $c14528ddc146317f$export$c1cdcaa5fe76a871;
var $c14528ddc146317f$export$82a249fb4d6127;
var $c14528ddc146317f$export$e7dc768d35940237;
var $c14528ddc146317f$export$f39a09f249340e2a;
var $c14528ddc146317f$export$d35bc1e505d1ebbf;
$c14528ddc146317f$export$1ac009feb2b0bd0c = "spectrum-Textfield-input_73bc77";
$c14528ddc146317f$export$2c8a10299d8b3418 = "spectrum-Textfield_73bc77";
$c14528ddc146317f$export$bffd5e3d61a81737 = "spectrum-Textfield--quiet_73bc77";
$c14528ddc146317f$export$9699f1c538a448a0 = "spectrum-Textfield--multiline_73bc77";
$c14528ddc146317f$export$e5b2f5233e4e5194 = "is-placeholder_73bc77";
$c14528ddc146317f$export$b8813cd5d7824ce7 = "is-hovered_73bc77";
$c14528ddc146317f$export$c2abbc3f970170b7 = "spectrum-Textfield--valid_73bc77";
$c14528ddc146317f$export$58e50b10d30123df = "spectrum-Textfield--invalid_73bc77";
$c14528ddc146317f$export$14fd30fea6f1348c = "spectrum-Textfield--loadable_73bc77";
$c14528ddc146317f$export$dd1165e0e6012973 = "spectrum-Textfield-wrapper_73bc77";
$c14528ddc146317f$export$cf0ead5b44db0da3 = "spectrum-Textfield-validationIcon_73bc77";
$c14528ddc146317f$export$222d1ccd1870be1d = "spectrum-Textfield-icon_73bc77";
$c14528ddc146317f$export$c1cdcaa5fe76a871 = "spectrum-Textfield-inputIcon_73bc77";
$c14528ddc146317f$export$82a249fb4d6127 = "spectrum-Textfield-circleLoader_73bc77";
$c14528ddc146317f$export$e7dc768d35940237 = "is-focused_73bc77";
$c14528ddc146317f$export$f39a09f249340e2a = "focus-ring_73bc77";
$c14528ddc146317f$export$d35bc1e505d1ebbf = "is-disabled_73bc77";


function $79348162c55d687f$var$Input(props, ref) {
    let inputRef = $aayOp$useRef(null);
    let { isDisabled: isDisabled , isQuiet: isQuiet , inputClassName: inputClassName , validationState: validationState , children: children , fieldProps: fieldProps , className: className , style: style  } = props;
    // Reserve padding for the error icon when the width of the input is unconstrained.
    // When constrained, don't reserve space because adding it only when invalid will
    // not cause a layout shift.
    let [reservePadding1, setReservePadding] = $aayOp$useValueEffect(false);
    let onResize = $aayOp$useCallback(()=>setReservePadding(function*(reservePadding) {
            if (inputRef.current) {
                if (reservePadding) // Try to collapse padding if the content is clipped.
                {
                    if (inputRef.current.scrollWidth > inputRef.current.offsetWidth) {
                        let width = inputRef.current.parentElement.offsetWidth;
                        yield false;
                        // If removing padding causes a layout shift, add it back.
                        if (inputRef.current.parentElement.offsetWidth !== width) yield true;
                    }
                } else // Try to add padding if the content is not clipped.
                if (inputRef.current.offsetWidth >= inputRef.current.scrollWidth) {
                    let width = inputRef.current.parentElement.offsetWidth;
                    yield true;
                    // If adding padding does not change the width (i.e. width is constrained), remove it again.
                    if (inputRef.current.parentElement.offsetWidth === width) yield false;
                }
            }
        })
    , [
        inputRef,
        setReservePadding
    ]);
    $aayOp$useLayoutEffect(onResize, [
        onResize
    ]);
    $aayOp$useResizeObserver({
        ref: inputRef,
        onResize: onResize
    });
    // We also need to listen for resize events of the window so we can detect
    // when there is enough space for the padding to be re-added. Ideally we'd
    // use a resize observer on a parent element, but it's hard to know _what_
    // parent element.
    $aayOp$useEvent($aayOp$useRef(typeof window !== 'undefined' ? window : null), 'resize', onResize);
    let isInvalid = validationState === 'invalid';
    let textfieldClass = $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($c14528ddc146317f$exports)), 'spectrum-Textfield', {
        'spectrum-Textfield--invalid': isInvalid,
        'spectrum-Textfield--valid': validationState === 'valid',
        'spectrum-Textfield--quiet': isQuiet
    }, $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-field'), className);
    let inputClass = $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($c14528ddc146317f$exports)), 'spectrum-Textfield-input', {
        'is-disabled': isDisabled,
        'is-invalid': isInvalid
    }, reservePadding1 && $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-input'), inputClassName);
    let iconClass = $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($c14528ddc146317f$exports)), 'spectrum-Textfield-validationIcon');
    let validationIcon = null;
    if (validationState === 'invalid') validationIcon = /*#__PURE__*/ $aayOp$react.createElement($aayOp$spectrumiconsuiAlertMedium, {
        "data-testid": "invalid-icon",
        UNSAFE_className: iconClass
    });
    else if (validationState === 'valid') validationIcon = /*#__PURE__*/ $aayOp$react.createElement($aayOp$spectrumiconsuiCheckmarkMedium, {
        "data-testid": "valid-icon",
        UNSAFE_className: iconClass
    });
    return(/*#__PURE__*/ $aayOp$react.createElement("div", {
        role: "presentation",
        ...fieldProps,
        className: textfieldClass,
        style: style
    }, /*#__PURE__*/ $aayOp$react.createElement($aayOp$FocusRing, {
        focusClass: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($c14528ddc146317f$exports)), 'is-focused'),
        focusRingClass: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($c14528ddc146317f$exports)), 'focus-ring'),
        isTextInput: true,
        within: true
    }, /*#__PURE__*/ $aayOp$react.createElement("div", {
        role: "presentation",
        className: inputClass
    }, /*#__PURE__*/ $aayOp$react.createElement("div", {
        role: "presentation",
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-inputContents'),
        ref: $aayOp$mergeRefs(ref, inputRef)
    }, children))), validationIcon));
}
const $79348162c55d687f$export$f5b8910cec6cf069 = /*#__PURE__*/ $aayOp$react.forwardRef($79348162c55d687f$var$Input);


var $ac1a7d8d9ccff97f$exports = {};
var $1729b68fc5d3f126$exports = {};
$1729b68fc5d3f126$exports = {
    "time": `Time`,
    "startTime": `Start time`,
    "endTime": `End time`
};


$ac1a7d8d9ccff97f$exports = {
    "en-US": $1729b68fc5d3f126$exports
};





var $8fb52a80fe52b1c7$exports = {};

$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup", () => $8fb52a80fe52b1c7$export$5be359c1227eedcb, (v) => $8fb52a80fe52b1c7$export$5be359c1227eedcb = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-FieldButton", () => $8fb52a80fe52b1c7$export$a024c99ecf6b5741, (v) => $8fb52a80fe52b1c7$export$a024c99ecf6b5741 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "is-disabled", () => $8fb52a80fe52b1c7$export$d35bc1e505d1ebbf, (v) => $8fb52a80fe52b1c7$export$d35bc1e505d1ebbf = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup--invalid", () => $8fb52a80fe52b1c7$export$3b48509739a9cda0, (v) => $8fb52a80fe52b1c7$export$3b48509739a9cda0 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup--quiet", () => $8fb52a80fe52b1c7$export$ebf265ffc062029b, (v) => $8fb52a80fe52b1c7$export$ebf265ffc062029b = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-field", () => $8fb52a80fe52b1c7$export$ea77459595e26bec, (v) => $8fb52a80fe52b1c7$export$ea77459595e26bec = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-input-validationIcon", () => $8fb52a80fe52b1c7$export$f7ab13ee113f4884, (v) => $8fb52a80fe52b1c7$export$f7ab13ee113f4884 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-input-circleLoader", () => $8fb52a80fe52b1c7$export$dcc9e00cdad967fa, (v) => $8fb52a80fe52b1c7$export$dcc9e00cdad967fa = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-input", () => $8fb52a80fe52b1c7$export$92706f5c6fe4074, (v) => $8fb52a80fe52b1c7$export$92706f5c6fe4074 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-icon", () => $8fb52a80fe52b1c7$export$b91b6af799fd93a8, (v) => $8fb52a80fe52b1c7$export$b91b6af799fd93a8 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-InputGroup-popover--quiet", () => $8fb52a80fe52b1c7$export$5e116178f9b1556a, (v) => $8fb52a80fe52b1c7$export$5e116178f9b1556a = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker--range", () => $8fb52a80fe52b1c7$export$1d8b0b29e11d7cd, (v) => $8fb52a80fe52b1c7$export$1d8b0b29e11d7cd = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker--datetimeRange", () => $8fb52a80fe52b1c7$export$58dd7cd23c92699f, (v) => $8fb52a80fe52b1c7$export$58dd7cd23c92699f = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker-startField", () => $8fb52a80fe52b1c7$export$80f9c838a8406469, (v) => $8fb52a80fe52b1c7$export$80f9c838a8406469 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker-endField", () => $8fb52a80fe52b1c7$export$1c72fb41fa4f21ee, (v) => $8fb52a80fe52b1c7$export$1c72fb41fa4f21ee = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker--rangeDash", () => $8fb52a80fe52b1c7$export$202113e883898dc, (v) => $8fb52a80fe52b1c7$export$202113e883898dc = v);
$parcel$export($8fb52a80fe52b1c7$exports, "is-focused", () => $8fb52a80fe52b1c7$export$e7dc768d35940237, (v) => $8fb52a80fe52b1c7$export$e7dc768d35940237 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-Datepicker-focusRing", () => $8fb52a80fe52b1c7$export$eb262b15b9261fde, (v) => $8fb52a80fe52b1c7$export$eb262b15b9261fde = v);
$parcel$export($8fb52a80fe52b1c7$exports, "is-hovered", () => $8fb52a80fe52b1c7$export$b8813cd5d7824ce7, (v) => $8fb52a80fe52b1c7$export$b8813cd5d7824ce7 = v);
$parcel$export($8fb52a80fe52b1c7$exports, "focus-ring", () => $8fb52a80fe52b1c7$export$f39a09f249340e2a, (v) => $8fb52a80fe52b1c7$export$f39a09f249340e2a = v);
$parcel$export($8fb52a80fe52b1c7$exports, "is-selected", () => $8fb52a80fe52b1c7$export$1e0fb04f31d3c22a, (v) => $8fb52a80fe52b1c7$export$1e0fb04f31d3c22a = v);
$parcel$export($8fb52a80fe52b1c7$exports, "spectrum-FieldButton--invalid", () => $8fb52a80fe52b1c7$export$c9f503f672e8a3c1, (v) => $8fb52a80fe52b1c7$export$c9f503f672e8a3c1 = v);
var $8fb52a80fe52b1c7$export$5be359c1227eedcb;
var $8fb52a80fe52b1c7$export$a024c99ecf6b5741;
var $8fb52a80fe52b1c7$export$d35bc1e505d1ebbf;
var $8fb52a80fe52b1c7$export$3b48509739a9cda0;
var $8fb52a80fe52b1c7$export$ebf265ffc062029b;
var $8fb52a80fe52b1c7$export$ea77459595e26bec;
var $8fb52a80fe52b1c7$export$f7ab13ee113f4884;
var $8fb52a80fe52b1c7$export$dcc9e00cdad967fa;
var $8fb52a80fe52b1c7$export$92706f5c6fe4074;
var $8fb52a80fe52b1c7$export$b91b6af799fd93a8;
var $8fb52a80fe52b1c7$export$5e116178f9b1556a;
var $8fb52a80fe52b1c7$export$1d8b0b29e11d7cd;
var $8fb52a80fe52b1c7$export$58dd7cd23c92699f;
var $8fb52a80fe52b1c7$export$80f9c838a8406469;
var $8fb52a80fe52b1c7$export$1c72fb41fa4f21ee;
var $8fb52a80fe52b1c7$export$202113e883898dc;
var $8fb52a80fe52b1c7$export$e7dc768d35940237;
var $8fb52a80fe52b1c7$export$eb262b15b9261fde;
var $8fb52a80fe52b1c7$export$b8813cd5d7824ce7;
var $8fb52a80fe52b1c7$export$f39a09f249340e2a;
var $8fb52a80fe52b1c7$export$1e0fb04f31d3c22a;
var $8fb52a80fe52b1c7$export$c9f503f672e8a3c1;
$8fb52a80fe52b1c7$export$5be359c1227eedcb = "spectrum-InputGroup_a0942f";
$8fb52a80fe52b1c7$export$a024c99ecf6b5741 = "spectrum-FieldButton_a0942f";
$8fb52a80fe52b1c7$export$d35bc1e505d1ebbf = "is-disabled_a0942f";
$8fb52a80fe52b1c7$export$3b48509739a9cda0 = "spectrum-InputGroup--invalid_a0942f";
$8fb52a80fe52b1c7$export$ebf265ffc062029b = "spectrum-InputGroup--quiet_a0942f";
$8fb52a80fe52b1c7$export$ea77459595e26bec = "spectrum-InputGroup-field_a0942f";
$8fb52a80fe52b1c7$export$f7ab13ee113f4884 = "spectrum-InputGroup-input-validationIcon_a0942f";
$8fb52a80fe52b1c7$export$dcc9e00cdad967fa = "spectrum-InputGroup-input-circleLoader_a0942f";
$8fb52a80fe52b1c7$export$92706f5c6fe4074 = "spectrum-InputGroup-input_a0942f";
$8fb52a80fe52b1c7$export$b91b6af799fd93a8 = "spectrum-InputGroup-icon_a0942f";
$8fb52a80fe52b1c7$export$5e116178f9b1556a = "spectrum-InputGroup-popover--quiet_a0942f";
$8fb52a80fe52b1c7$export$1d8b0b29e11d7cd = "spectrum-Datepicker--range_a0942f";
$8fb52a80fe52b1c7$export$58dd7cd23c92699f = "spectrum-Datepicker--datetimeRange_a0942f";
$8fb52a80fe52b1c7$export$80f9c838a8406469 = "spectrum-Datepicker-startField_a0942f";
$8fb52a80fe52b1c7$export$1c72fb41fa4f21ee = "spectrum-Datepicker-endField_a0942f";
$8fb52a80fe52b1c7$export$202113e883898dc = "spectrum-Datepicker--rangeDash_a0942f";
$8fb52a80fe52b1c7$export$e7dc768d35940237 = "is-focused_a0942f";
$8fb52a80fe52b1c7$export$eb262b15b9261fde = "spectrum-Datepicker-focusRing_a0942f";
$8fb52a80fe52b1c7$export$b8813cd5d7824ce7 = "is-hovered_a0942f";
$8fb52a80fe52b1c7$export$f39a09f249340e2a = "focus-ring_a0942f";
$8fb52a80fe52b1c7$export$1e0fb04f31d3c22a = "is-selected_a0942f";
$8fb52a80fe52b1c7$export$c9f503f672e8a3c1 = "spectrum-FieldButton--invalid_a0942f";















function $04e96200274b03de$export$322f4580ccd8dde6(props) {
    let formatter = $aayOp$useDateFormatter({
        dateStyle: 'short'
    });
    let displayNames = $aayOp$useDisplayNames();
    return $aayOp$useMemo(()=>{
        if (props.description) return props.description;
        if (props.showFormatHelpText) return formatter.formatToParts(new Date()).map((s)=>{
            if (s.type === 'literal') return s.value;
            return displayNames.of(s.type);
        }).join(' ');
        return '';
    }, [
        props.description,
        props.showFormatHelpText,
        formatter,
        displayNames
    ]);
}
function $04e96200274b03de$export$12ce2869ce471b1f(maxVisibleMonths) {
    let { scale: scale  } = $aayOp$useProvider();
    let [visibleMonths, setVisibleMonths] = $aayOp$useState($04e96200274b03de$var$getVisibleMonths(scale));
    $aayOp$useLayoutEffect(()=>{
        let onResize = ()=>setVisibleMonths($04e96200274b03de$var$getVisibleMonths(scale))
        ;
        onResize();
        window.addEventListener('resize', onResize);
        return ()=>{
            window.removeEventListener('resize', onResize);
        };
    }, [
        scale
    ]);
    return Math.max(1, Math.min(visibleMonths, maxVisibleMonths, 3));
}
function $04e96200274b03de$var$getVisibleMonths(scale) {
    if (typeof window === 'undefined') return 1;
    let monthWidth = scale === 'large' ? 336 : 280;
    let gap = scale === 'large' ? 30 : 24;
    let popoverPadding = scale === 'large' ? 32 : 48;
    return Math.floor((window.innerWidth - popoverPadding * 2) / (monthWidth + gap));
}
function $04e96200274b03de$export$71a23a36270e4bf0(ref) {
    let domRef = $aayOp$useRef();
    $aayOp$useImperativeHandle(ref, ()=>({
            ...$aayOp$createDOMRef(domRef),
            focus () {
                $aayOp$createFocusManager(domRef).focusFirst({
                    tabbable: true
                });
            }
        })
    );
    return domRef;
}






function $f0a04554754386b6$var$TimeField(props, ref) {
    props = $aayOp$useProviderProps(props);
    let { autoFocus: autoFocus , isDisabled: isDisabled , isReadOnly: isReadOnly , isRequired: isRequired , isQuiet: isQuiet  } = props;
    let domRef = $04e96200274b03de$export$71a23a36270e4bf0(ref);
    let { locale: locale  } = $aayOp$useLocale();
    let state = $aayOp$useTimeFieldState({
        ...props,
        locale: locale
    });
    let { labelProps: labelProps , fieldProps: fieldProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $aayOp$useTimeField(props, state, domRef);
    return(/*#__PURE__*/ $aayOp$react.createElement($aayOp$Field, {
        ...props,
        ref: domRef,
        elementType: "span",
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: state.validationState,
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-TimeField-fieldWrapper')
    }, /*#__PURE__*/ $aayOp$react.createElement($79348162c55d687f$export$f5b8910cec6cf069, {
        fieldProps: fieldProps,
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        autoFocus: autoFocus,
        validationState: state.validationState,
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-TimeField')
    }, state.segments.map((segment, i)=>/*#__PURE__*/ $aayOp$react.createElement($cfa347761f172638$export$6388987c5223b54e, {
            key: i,
            segment: segment,
            state: state,
            isDisabled: isDisabled,
            isReadOnly: isReadOnly,
            isRequired: isRequired
        })
    ))));
}
/**
 * TimeFields allow users to enter and edit time values using a keyboard.
 * Each part of the time is displayed in an individually editable segment.
 */ const $f0a04554754386b6$export$5eaee2322dd727eb = /*#__PURE__*/ $aayOp$react.forwardRef($f0a04554754386b6$var$TimeField);









function $beaeefa091999363$var$DatePicker(props, ref) {
    props = $aayOp$useProviderProps(props);
    let { autoFocus: autoFocus , isQuiet: isQuiet , isDisabled: isDisabled , isReadOnly: isReadOnly , placeholderValue: placeholderValue , maxVisibleMonths: maxVisibleMonths = 1  } = props;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $aayOp$useHover({
        isDisabled: isDisabled
    });
    let targetRef = $aayOp$useRef();
    let state = $aayOp$useDatePickerState({
        ...props,
        shouldCloseOnSelect: ()=>!state.hasTime
    });
    let { groupProps: groupProps , labelProps: labelProps , fieldProps: fieldProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps , buttonProps: buttonProps , dialogProps: dialogProps , calendarProps: calendarProps  } = $aayOp$useDatePicker(props, state, targetRef);
    let { isOpen: isOpen , setOpen: setOpen  } = state;
    let { direction: direction  } = $aayOp$useLocale();
    let domRef = $04e96200274b03de$export$71a23a36270e4bf0(ref);
    let stringFormatter = $aayOp$useLocalizedStringFormatter((/*@__PURE__*/$parcel$interopDefault($ac1a7d8d9ccff97f$exports)));
    let { isFocused: isFocused , isFocusVisible: isFocusVisible , focusProps: focusProps  } = $aayOp$useFocusRing({
        within: true,
        isTextInput: true,
        autoFocus: autoFocus
    });
    let { isFocused: isFocusedButton , focusProps: focusPropsButton  } = $aayOp$useFocusRing({
        within: false,
        isTextInput: false,
        autoFocus: autoFocus
    });
    let className = $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup', {
        'spectrum-InputGroup--quiet': isQuiet,
        'spectrum-InputGroup--invalid': state.validationState === 'invalid',
        'is-disabled': isDisabled,
        'is-hovered': isHovered,
        'is-focused': isFocused,
        'focus-ring': isFocusVisible && !isFocusedButton
    });
    let fieldClassName = $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-input', {
        'is-disabled': isDisabled,
        'is-invalid': state.validationState === 'invalid'
    });
    // Note: this description is intentionally not passed to useDatePicker.
    // The format help text is unnecessary for screen reader users because each segment already has a label.
    let description = $04e96200274b03de$export$322f4580ccd8dde6(props);
    if (description && !props.description) descriptionProps.id = null;
    let placeholder = placeholderValue;
    let timePlaceholder = placeholder && 'hour' in placeholder ? placeholder : null;
    let timeMinValue = props.minValue && 'hour' in props.minValue ? props.minValue : null;
    let timeMaxValue = props.maxValue && 'hour' in props.maxValue ? props.maxValue : null;
    let timeGranularity = state.granularity === 'hour' || state.granularity === 'minute' || state.granularity === 'second' ? state.granularity : null;
    let showTimeField = !!timeGranularity;
    let visibleMonths = $04e96200274b03de$export$12ce2869ce471b1f(maxVisibleMonths);
    return(/*#__PURE__*/ $aayOp$react.createElement($aayOp$Field, {
        ...props,
        ref: domRef,
        elementType: "span",
        description: description,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: state.validationState,
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-fieldWrapper')
    }, /*#__PURE__*/ $aayOp$react.createElement("div", {
        ...$aayOp$mergeProps(groupProps, hoverProps, focusProps),
        className: className,
        ref: targetRef
    }, /*#__PURE__*/ $aayOp$react.createElement($79348162c55d687f$export$f5b8910cec6cf069, {
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: state.validationState,
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-field'),
        inputClassName: fieldClassName
    }, /*#__PURE__*/ $aayOp$react.createElement($d2bb88c82604b25f$export$34dc4cfa15ead1, {
        ...fieldProps,
        "data-testid": "date-field",
        isQuiet: isQuiet
    })), /*#__PURE__*/ $aayOp$react.createElement($aayOp$DialogTrigger, {
        type: "popover",
        mobileType: "tray",
        placement: direction === 'rtl' ? 'bottom right' : 'bottom left',
        targetRef: targetRef,
        hideArrow: true,
        isOpen: isOpen,
        onOpenChange: setOpen,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ $aayOp$react.createElement($aayOp$FieldButton, {
        ...$aayOp$mergeProps(buttonProps, focusPropsButton),
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: state.validationState,
        isDisabled: isDisabled || isReadOnly
    }, /*#__PURE__*/ $aayOp$react.createElement($aayOp$spectrumiconsworkflowCalendar, null)), /*#__PURE__*/ $aayOp$react.createElement($aayOp$Dialog, {
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-dialog'),
        ...dialogProps
    }, /*#__PURE__*/ $aayOp$react.createElement($aayOp$Content, null, /*#__PURE__*/ $aayOp$react.createElement("div", {
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-dialogContent')
    }, /*#__PURE__*/ $aayOp$react.createElement($aayOp$Calendar, {
        ...calendarProps,
        visibleMonths: visibleMonths,
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-calendar', {
            'is-invalid': state.validationState === 'invalid'
        })
    }), showTimeField && /*#__PURE__*/ $aayOp$react.createElement("div", {
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-timeFields')
    }, /*#__PURE__*/ $aayOp$react.createElement($f0a04554754386b6$export$5eaee2322dd727eb, {
        label: stringFormatter.format('time'),
        value: state.timeValue,
        onChange: state.setTimeValue,
        placeholderValue: timePlaceholder,
        granularity: timeGranularity,
        minValue: timeMinValue,
        maxValue: timeMaxValue,
        hourCycle: props.hourCycle,
        hideTimeZone: props.hideTimeZone,
        marginTop: "size-100"
    })))))))));
}
/**
 * DatePickers combine a DateField and a Calendar popover to allow users to enter or select a date and time value.
 */ const $beaeefa091999363$export$5109c6dd95d8fb00 = /*#__PURE__*/ $aayOp$react.forwardRef($beaeefa091999363$var$DatePicker);

























function $cb301300011a98f7$var$DateRangePicker(props, ref) {
    var ref1, ref2;
    props = $aayOp$useProviderProps(props);
    let { isQuiet: isQuiet , isDisabled: isDisabled , isReadOnly: isReadOnly , autoFocus: autoFocus , placeholderValue: placeholderValue , maxVisibleMonths: maxVisibleMonths = 1  } = props;
    let { hoverProps: hoverProps , isHovered: isHovered  } = $aayOp$useHover({
        isDisabled: isDisabled
    });
    let targetRef = $aayOp$useRef();
    let state = $aayOp$useDateRangePickerState({
        ...props,
        shouldCloseOnSelect: ()=>!state.hasTime
    });
    let { labelProps: labelProps , groupProps: groupProps , buttonProps: buttonProps , dialogProps: dialogProps , startFieldProps: startFieldProps , endFieldProps: endFieldProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps , calendarProps: calendarProps  } = $aayOp$useDateRangePicker(props, state, targetRef);
    let { isOpen: isOpen , setOpen: setOpen  } = state;
    let { direction: direction  } = $aayOp$useLocale();
    let domRef = $04e96200274b03de$export$71a23a36270e4bf0(ref);
    let stringFormatter = $aayOp$useLocalizedStringFormatter((/*@__PURE__*/$parcel$interopDefault($ac1a7d8d9ccff97f$exports)));
    let { isFocused: isFocused , isFocusVisible: isFocusVisible , focusProps: focusProps  } = $aayOp$useFocusRing({
        within: true,
        isTextInput: true,
        autoFocus: autoFocus
    });
    let { isFocused: isFocusedButton , focusProps: focusPropsButton  } = $aayOp$useFocusRing({
        within: false,
        isTextInput: false,
        autoFocus: autoFocus
    });
    let className = $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup', {
        'spectrum-InputGroup--quiet': isQuiet,
        'spectrum-InputGroup--invalid': state.validationState === 'invalid',
        'is-disabled': isDisabled,
        'is-hovered': isHovered,
        'is-focused': isFocused,
        'focus-ring': isFocusVisible && !isFocusedButton
    });
    let fieldClassName = $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-input', {
        'is-disabled': isDisabled,
        'is-invalid': state.validationState === 'invalid'
    });
    // Note: this description is intentionally not passed to useDatePicker.
    // The format help text is unnecessary for screen reader users because each segment already has a label.
    let description = $04e96200274b03de$export$322f4580ccd8dde6(props);
    if (description && !props.description) descriptionProps.id = null;
    let placeholder = placeholderValue;
    let timePlaceholder = placeholder && 'hour' in placeholder ? placeholder : null;
    let timeMinValue = props.minValue && 'hour' in props.minValue ? props.minValue : null;
    let timeMaxValue = props.maxValue && 'hour' in props.maxValue ? props.maxValue : null;
    let timeGranularity = state.granularity === 'hour' || state.granularity === 'minute' || state.granularity === 'second' ? state.granularity : null;
    let showTimeField = !!timeGranularity;
    let visibleMonths = $04e96200274b03de$export$12ce2869ce471b1f(maxVisibleMonths);
    return(/*#__PURE__*/ $aayOp$react.createElement($aayOp$Field, {
        ...props,
        ref: domRef,
        elementType: "span",
        description: description,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: state.validationState,
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-fieldWrapper')
    }, /*#__PURE__*/ $aayOp$react.createElement("div", {
        ...$aayOp$mergeProps(groupProps, hoverProps, focusProps),
        className: className,
        ref: targetRef
    }, /*#__PURE__*/ $aayOp$react.createElement($79348162c55d687f$export$f5b8910cec6cf069, {
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        validationState: state.validationState,
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-InputGroup-field'),
        inputClassName: fieldClassName
    }, /*#__PURE__*/ $aayOp$react.createElement($d2bb88c82604b25f$export$34dc4cfa15ead1, {
        ...startFieldProps,
        "data-testid": "start-date",
        isQuiet: props.isQuiet,
        inputClassName: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-startField')
    }), /*#__PURE__*/ $aayOp$react.createElement($cb301300011a98f7$var$DateRangeDash, null), /*#__PURE__*/ $aayOp$react.createElement($d2bb88c82604b25f$export$34dc4cfa15ead1, {
        ...endFieldProps,
        "data-testid": "end-date",
        isQuiet: props.isQuiet,
        inputClassName: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-Datepicker-endField', $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-endField'))
    })), /*#__PURE__*/ $aayOp$react.createElement($aayOp$DialogTrigger, {
        type: "popover",
        mobileType: "tray",
        placement: direction === 'rtl' ? 'bottom right' : 'bottom left',
        targetRef: targetRef,
        hideArrow: true,
        isOpen: isOpen,
        onOpenChange: setOpen,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ $aayOp$react.createElement($aayOp$FieldButton, {
        ...$aayOp$mergeProps(buttonProps, focusPropsButton),
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($8fb52a80fe52b1c7$exports)), 'spectrum-FieldButton'),
        isQuiet: isQuiet,
        validationState: state.validationState,
        isDisabled: isDisabled || isReadOnly
    }, /*#__PURE__*/ $aayOp$react.createElement($aayOp$spectrumiconsworkflowCalendar, null)), /*#__PURE__*/ $aayOp$react.createElement($aayOp$Dialog, {
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-dialog'),
        ...dialogProps
    }, /*#__PURE__*/ $aayOp$react.createElement($aayOp$Content, null, /*#__PURE__*/ $aayOp$react.createElement("div", {
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-dialogContent')
    }, /*#__PURE__*/ $aayOp$react.createElement($aayOp$RangeCalendar, {
        ...calendarProps,
        visibleMonths: visibleMonths,
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-calendar', {
            'is-invalid': state.validationState === 'invalid'
        })
    }), showTimeField && /*#__PURE__*/ $aayOp$react.createElement($aayOp$Flex, {
        gap: "size-100",
        marginTop: "size-100",
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-timeFields')
    }, /*#__PURE__*/ $aayOp$react.createElement($f0a04554754386b6$export$5eaee2322dd727eb, {
        label: stringFormatter.format('startTime'),
        value: ((ref1 = state.timeRange) === null || ref1 === void 0 ? void 0 : ref1.start) || null,
        onChange: (v)=>state.setTime('start', v)
        ,
        placeholderValue: timePlaceholder,
        granularity: timeGranularity,
        minValue: timeMinValue,
        maxValue: timeMaxValue,
        hourCycle: props.hourCycle,
        hideTimeZone: props.hideTimeZone,
        flex: true
    }), /*#__PURE__*/ $aayOp$react.createElement($f0a04554754386b6$export$5eaee2322dd727eb, {
        label: stringFormatter.format('endTime'),
        value: ((ref2 = state.timeRange) === null || ref2 === void 0 ? void 0 : ref2.end) || null,
        onChange: (v)=>state.setTime('end', v)
        ,
        placeholderValue: timePlaceholder,
        granularity: timeGranularity,
        minValue: timeMinValue,
        maxValue: timeMaxValue,
        hourCycle: props.hourCycle,
        hideTimeZone: props.hideTimeZone,
        flex: true
    })))))))));
}
function $cb301300011a98f7$var$DateRangeDash() {
    return(/*#__PURE__*/ $aayOp$react.createElement("div", {
        "aria-hidden": "true",
        "data-testid": "date-range-dash",
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-rangeDash')
    }));
}
/**
 * DateRangePickers combine two DateFields and a RangeCalendar popover to allow users
 * to enter or select a date and time range.
 */ const $cb301300011a98f7$export$17334619f3ac2224 = /*#__PURE__*/ $aayOp$react.forwardRef($cb301300011a98f7$var$DateRangePicker);















function $0129798af6791690$var$DateField(props, ref) {
    props = $aayOp$useProviderProps(props);
    let { autoFocus: autoFocus , isDisabled: isDisabled , isReadOnly: isReadOnly , isRequired: isRequired , isQuiet: isQuiet  } = props;
    let domRef = $04e96200274b03de$export$71a23a36270e4bf0(ref);
    let { locale: locale  } = $aayOp$useLocale();
    let state = $aayOp$useDateFieldState({
        ...props,
        locale: locale,
        createCalendar: $aayOp$createCalendar
    });
    let { labelProps: labelProps , fieldProps: fieldProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = $aayOp$useDateField(props, state, domRef);
    // Note: this description is intentionally not passed to useDatePicker.
    // The format help text is unnecessary for screen reader users because each segment already has a label.
    let description = $04e96200274b03de$export$322f4580ccd8dde6(props);
    if (description && !props.description) descriptionProps.id = null;
    return(/*#__PURE__*/ $aayOp$react.createElement($aayOp$Field, {
        ...props,
        ref: domRef,
        elementType: "span",
        description: description,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        validationState: state.validationState,
        UNSAFE_className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-Datepicker-fieldWrapper')
    }, /*#__PURE__*/ $aayOp$react.createElement($79348162c55d687f$export$f5b8910cec6cf069, {
        fieldProps: fieldProps,
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        autoFocus: autoFocus,
        validationState: state.validationState,
        className: $aayOp$classNames((/*@__PURE__*/$parcel$interopDefault($465eb3a688ef3440$exports)), 'react-spectrum-DateField')
    }, state.segments.map((segment, i)=>/*#__PURE__*/ $aayOp$react.createElement($cfa347761f172638$export$6388987c5223b54e, {
            key: i,
            segment: segment,
            state: state,
            isDisabled: isDisabled,
            isReadOnly: isReadOnly,
            isRequired: isRequired
        })
    ))));
}
/**
 * DateFields allow users to enter and edit date and time values using a keyboard.
 * Each part of a date value is displayed in an individually editable segment.
 */ const $0129798af6791690$export$d9781c7894a82487 = /*#__PURE__*/ $aayOp$react.forwardRef($0129798af6791690$var$DateField);




export {$beaeefa091999363$export$5109c6dd95d8fb00 as DatePicker, $cb301300011a98f7$export$17334619f3ac2224 as DateRangePicker, $f0a04554754386b6$export$5eaee2322dd727eb as TimeField, $0129798af6791690$export$d9781c7894a82487 as DateField};
