// @react-spectrum/statuslight/dist/module.js
import "./main.css";
import {useDOMRef as $71ACv$useDOMRef, useStyleProps as $71ACv$useStyleProps, classNames as $71ACv$classNames} from "@react-spectrum/utils";
import {filterDOMProps as $71ACv$filterDOMProps} from "@react-aria/utils";
import $71ACv$react, {forwardRef as $71ACv$forwardRef} from "react";
import {useProviderProps as $71ACv$useProviderProps} from "@react-spectrum/provider";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $df8f173c8bd4ebc0$exports = {};

$parcel$export($df8f173c8bd4ebc0$exports, "StatusLight", () => $df8f173c8bd4ebc0$export$5f84c37a31c6e41c);



var $7661ca422872705f$exports = {};

$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight", () => $7661ca422872705f$export$11d3697e7e5c1bef, (v) => $7661ca422872705f$export$11d3697e7e5c1bef = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--neutral", () => $7661ca422872705f$export$bbf75dd99797a93e, (v) => $7661ca422872705f$export$bbf75dd99797a93e = v);
$parcel$export($7661ca422872705f$exports, "is-disabled", () => $7661ca422872705f$export$d35bc1e505d1ebbf, (v) => $7661ca422872705f$export$d35bc1e505d1ebbf = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--negative", () => $7661ca422872705f$export$1974c2646528dc4f, (v) => $7661ca422872705f$export$1974c2646528dc4f = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--notice", () => $7661ca422872705f$export$30a5e512ed4618b, (v) => $7661ca422872705f$export$30a5e512ed4618b = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--positive", () => $7661ca422872705f$export$141dae5af9534f54, (v) => $7661ca422872705f$export$141dae5af9534f54 = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--info", () => $7661ca422872705f$export$945a5d9a4e8494e6, (v) => $7661ca422872705f$export$945a5d9a4e8494e6 = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--active", () => $7661ca422872705f$export$b40c603f4b579265, (v) => $7661ca422872705f$export$b40c603f4b579265 = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--celery", () => $7661ca422872705f$export$e9a22f483095fe3e, (v) => $7661ca422872705f$export$e9a22f483095fe3e = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--yellow", () => $7661ca422872705f$export$b22c4af54e28c6c, (v) => $7661ca422872705f$export$b22c4af54e28c6c = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--fuchsia", () => $7661ca422872705f$export$cbaba642d0afd312, (v) => $7661ca422872705f$export$cbaba642d0afd312 = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--indigo", () => $7661ca422872705f$export$8b77e0c07c11635a, (v) => $7661ca422872705f$export$8b77e0c07c11635a = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--seafoam", () => $7661ca422872705f$export$d14bcd2d3ae8238c, (v) => $7661ca422872705f$export$d14bcd2d3ae8238c = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--chartreuse", () => $7661ca422872705f$export$f05bd1f10c0cc15f, (v) => $7661ca422872705f$export$f05bd1f10c0cc15f = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--magenta", () => $7661ca422872705f$export$43990cc5ba9a0059, (v) => $7661ca422872705f$export$43990cc5ba9a0059 = v);
$parcel$export($7661ca422872705f$exports, "spectrum-StatusLight--purple", () => $7661ca422872705f$export$975baced4f8182b4, (v) => $7661ca422872705f$export$975baced4f8182b4 = v);
var $7661ca422872705f$export$11d3697e7e5c1bef;
var $7661ca422872705f$export$bbf75dd99797a93e;
var $7661ca422872705f$export$d35bc1e505d1ebbf;
var $7661ca422872705f$export$1974c2646528dc4f;
var $7661ca422872705f$export$30a5e512ed4618b;
var $7661ca422872705f$export$141dae5af9534f54;
var $7661ca422872705f$export$945a5d9a4e8494e6;
var $7661ca422872705f$export$b40c603f4b579265;
var $7661ca422872705f$export$e9a22f483095fe3e;
var $7661ca422872705f$export$b22c4af54e28c6c;
var $7661ca422872705f$export$cbaba642d0afd312;
var $7661ca422872705f$export$8b77e0c07c11635a;
var $7661ca422872705f$export$d14bcd2d3ae8238c;
var $7661ca422872705f$export$f05bd1f10c0cc15f;
var $7661ca422872705f$export$43990cc5ba9a0059;
var $7661ca422872705f$export$975baced4f8182b4;
$7661ca422872705f$export$11d3697e7e5c1bef = "spectrum-StatusLight_c020b6";
$7661ca422872705f$export$bbf75dd99797a93e = "spectrum-StatusLight--neutral_c020b6";
$7661ca422872705f$export$d35bc1e505d1ebbf = "is-disabled_c020b6";
$7661ca422872705f$export$1974c2646528dc4f = "spectrum-StatusLight--negative_c020b6";
$7661ca422872705f$export$30a5e512ed4618b = "spectrum-StatusLight--notice_c020b6";
$7661ca422872705f$export$141dae5af9534f54 = "spectrum-StatusLight--positive_c020b6";
$7661ca422872705f$export$945a5d9a4e8494e6 = "spectrum-StatusLight--info_c020b6";
$7661ca422872705f$export$b40c603f4b579265 = "spectrum-StatusLight--active_c020b6";
$7661ca422872705f$export$e9a22f483095fe3e = "spectrum-StatusLight--celery_c020b6";
$7661ca422872705f$export$b22c4af54e28c6c = "spectrum-StatusLight--yellow_c020b6";
$7661ca422872705f$export$cbaba642d0afd312 = "spectrum-StatusLight--fuchsia_c020b6";
$7661ca422872705f$export$8b77e0c07c11635a = "spectrum-StatusLight--indigo_c020b6";
$7661ca422872705f$export$d14bcd2d3ae8238c = "spectrum-StatusLight--seafoam_c020b6";
$7661ca422872705f$export$f05bd1f10c0cc15f = "spectrum-StatusLight--chartreuse_c020b6";
$7661ca422872705f$export$43990cc5ba9a0059 = "spectrum-StatusLight--magenta_c020b6";
$7661ca422872705f$export$975baced4f8182b4 = "spectrum-StatusLight--purple_c020b6";



function $df8f173c8bd4ebc0$var$StatusLight(props, ref) {
    let { variant: variant , children: children , isDisabled: isDisabled , role: role , ...otherProps } = $71ACv$useProviderProps(props);
    let domRef = $71ACv$useDOMRef(ref);
    let { styleProps: styleProps  } = $71ACv$useStyleProps(otherProps);
    if (!children && !props['aria-label']) console.warn('If no children are provided, an aria-label must be specified');
    if (!role && (props['aria-label'] || props['aria-labelledby'])) console.warn('A labelled StatusLight must have a role.');
    return(/*#__PURE__*/ $71ACv$react.createElement("div", {
        ...$71ACv$filterDOMProps(otherProps, {
            labelable: !!role
        }),
        ...styleProps,
        role: role,
        className: $71ACv$classNames((/*@__PURE__*/$parcel$interopDefault($7661ca422872705f$exports)), 'spectrum-StatusLight', `spectrum-StatusLight--${variant}`, {
            'is-disabled': isDisabled
        }, styleProps.className),
        ref: domRef
    }, children));
}
/**
 * Status lights are used to color code categories and labels commonly found in data visualization.
 * When status lights have a semantic meaning, they should use semantic variant colors.
 */ let $df8f173c8bd4ebc0$export$5f84c37a31c6e41c = /*#__PURE__*/ $71ACv$forwardRef($df8f173c8bd4ebc0$var$StatusLight);




export {$df8f173c8bd4ebc0$export$5f84c37a31c6e41c as StatusLight};
