// @react-stately/tooltip/dist/main.js
var $8aPAr$react = require("react");
var $8aPAr$reactstatelyoverlays = require("@react-stately/overlays");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $3391baedd777a697$exports = {};

$parcel$export($3391baedd777a697$exports, "useTooltipTriggerState", () => $3391baedd777a697$export$4d40659c25ecb50b);


const $3391baedd777a697$var$TOOLTIP_DELAY = 1500; // this seems to be a 1.5 second delay, check with design
const $3391baedd777a697$var$TOOLTIP_COOLDOWN = 500;
let $3391baedd777a697$var$tooltips = {
};
let $3391baedd777a697$var$tooltipId = 0;
let $3391baedd777a697$var$globalWarmedUp = false;
let $3391baedd777a697$var$globalWarmUpTimeout = null;
let $3391baedd777a697$var$globalCooldownTimeout = null;
function $3391baedd777a697$export$4d40659c25ecb50b(props = {
}) {
    let { delay: delay = $3391baedd777a697$var$TOOLTIP_DELAY  } = props;
    let { isOpen: isOpen , open: open , close: close  } = $8aPAr$reactstatelyoverlays.useOverlayTriggerState(props);
    let id = $8aPAr$react.useMemo(()=>`${++$3391baedd777a697$var$tooltipId}`
    , []);
    let closeTimeout = $8aPAr$react.useRef();
    let ensureTooltipEntry = ()=>{
        $3391baedd777a697$var$tooltips[id] = hideTooltip;
    };
    let closeOpenTooltips = ()=>{
        for(let hideTooltipId in $3391baedd777a697$var$tooltips)if (hideTooltipId !== id) {
            $3391baedd777a697$var$tooltips[hideTooltipId](true);
            delete $3391baedd777a697$var$tooltips[hideTooltipId];
        }
    };
    let showTooltip = ()=>{
        clearTimeout(closeTimeout.current);
        closeTimeout.current = null;
        closeOpenTooltips();
        ensureTooltipEntry();
        $3391baedd777a697$var$globalWarmedUp = true;
        open();
        if ($3391baedd777a697$var$globalWarmUpTimeout) {
            clearTimeout($3391baedd777a697$var$globalWarmUpTimeout);
            $3391baedd777a697$var$globalWarmUpTimeout = null;
        }
        if ($3391baedd777a697$var$globalCooldownTimeout) {
            clearTimeout($3391baedd777a697$var$globalCooldownTimeout);
            $3391baedd777a697$var$globalCooldownTimeout = null;
        }
    };
    let hideTooltip = (immediate)=>{
        if (immediate) {
            clearTimeout(closeTimeout.current);
            closeTimeout.current = null;
            close();
        } else if (!closeTimeout.current) closeTimeout.current = setTimeout(()=>{
            closeTimeout.current = null;
            close();
        }, $3391baedd777a697$var$TOOLTIP_COOLDOWN);
        if ($3391baedd777a697$var$globalWarmUpTimeout) {
            clearTimeout($3391baedd777a697$var$globalWarmUpTimeout);
            $3391baedd777a697$var$globalWarmUpTimeout = null;
        }
        if ($3391baedd777a697$var$globalWarmedUp) {
            if ($3391baedd777a697$var$globalCooldownTimeout) clearTimeout($3391baedd777a697$var$globalCooldownTimeout);
            $3391baedd777a697$var$globalCooldownTimeout = setTimeout(()=>{
                delete $3391baedd777a697$var$tooltips[id];
                $3391baedd777a697$var$globalCooldownTimeout = null;
                $3391baedd777a697$var$globalWarmedUp = false;
            }, $3391baedd777a697$var$TOOLTIP_COOLDOWN);
        }
    };
    let warmupTooltip = ()=>{
        closeOpenTooltips();
        ensureTooltipEntry();
        if (!isOpen && !$3391baedd777a697$var$globalWarmUpTimeout && !$3391baedd777a697$var$globalWarmedUp) $3391baedd777a697$var$globalWarmUpTimeout = setTimeout(()=>{
            $3391baedd777a697$var$globalWarmUpTimeout = null;
            $3391baedd777a697$var$globalWarmedUp = true;
            showTooltip();
        }, delay);
        else if (!isOpen) showTooltip();
    };
    // eslint-disable-next-line arrow-body-style
    $8aPAr$react.useEffect(()=>{
        return ()=>{
            clearTimeout(closeTimeout.current);
            let tooltip = $3391baedd777a697$var$tooltips[id];
            if (tooltip) delete $3391baedd777a697$var$tooltips[id];
        };
    }, [
        id
    ]);
    return {
        isOpen: isOpen,
        open: (immediate)=>{
            if (!immediate && delay > 0 && !closeTimeout.current) warmupTooltip();
            else showTooltip();
        },
        close: hideTooltip
    };
}


$parcel$exportWildcard(module.exports, $3391baedd777a697$exports);
