// @react-spectrum/link/dist/module.js
import "./main.css";
import {useSlotProps as $11MUV$useSlotProps, useStyleProps as $11MUV$useStyleProps, classNames as $11MUV$classNames, getWrappedElement as $11MUV$getWrappedElement} from "@react-spectrum/utils";
import {FocusRing as $11MUV$FocusRing} from "@react-aria/focus";
import {mergeProps as $11MUV$mergeProps} from "@react-aria/utils";
import $11MUV$react, {useRef as $11MUV$useRef} from "react";
import {useHover as $11MUV$useHover} from "@react-aria/interactions";
import {useLink as $11MUV$useLink} from "@react-aria/link";
import {useProviderProps as $11MUV$useProviderProps} from "@react-spectrum/provider";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $3c5235ac12f2c9bb$exports = {};

$parcel$export($3c5235ac12f2c9bb$exports, "Link", () => $3c5235ac12f2c9bb$export$a6c7ac8248d6e38a);




var $4564c6c10d1690b3$exports = {};

$parcel$export($4564c6c10d1690b3$exports, "spectrum-Link", () => $4564c6c10d1690b3$export$d85dd8b94975f388, (v) => $4564c6c10d1690b3$export$d85dd8b94975f388 = v);
$parcel$export($4564c6c10d1690b3$exports, "is-hovered", () => $4564c6c10d1690b3$export$b8813cd5d7824ce7, (v) => $4564c6c10d1690b3$export$b8813cd5d7824ce7 = v);
$parcel$export($4564c6c10d1690b3$exports, "focus-ring", () => $4564c6c10d1690b3$export$f39a09f249340e2a, (v) => $4564c6c10d1690b3$export$f39a09f249340e2a = v);
$parcel$export($4564c6c10d1690b3$exports, "is-disabled", () => $4564c6c10d1690b3$export$d35bc1e505d1ebbf, (v) => $4564c6c10d1690b3$export$d35bc1e505d1ebbf = v);
$parcel$export($4564c6c10d1690b3$exports, "spectrum-Link--secondary", () => $4564c6c10d1690b3$export$ab09ed292dcb9a79, (v) => $4564c6c10d1690b3$export$ab09ed292dcb9a79 = v);
$parcel$export($4564c6c10d1690b3$exports, "spectrum-Link--overBackground", () => $4564c6c10d1690b3$export$679b6267148797da, (v) => $4564c6c10d1690b3$export$679b6267148797da = v);
$parcel$export($4564c6c10d1690b3$exports, "spectrum-Link--quiet", () => $4564c6c10d1690b3$export$81921ea93435e997, (v) => $4564c6c10d1690b3$export$81921ea93435e997 = v);
var $4564c6c10d1690b3$export$d85dd8b94975f388;
var $4564c6c10d1690b3$export$b8813cd5d7824ce7;
var $4564c6c10d1690b3$export$f39a09f249340e2a;
var $4564c6c10d1690b3$export$d35bc1e505d1ebbf;
var $4564c6c10d1690b3$export$ab09ed292dcb9a79;
var $4564c6c10d1690b3$export$679b6267148797da;
var $4564c6c10d1690b3$export$81921ea93435e997;
$4564c6c10d1690b3$export$d85dd8b94975f388 = "spectrum-Link_5a03c7";
$4564c6c10d1690b3$export$b8813cd5d7824ce7 = "is-hovered_5a03c7";
$4564c6c10d1690b3$export$f39a09f249340e2a = "focus-ring_5a03c7";
$4564c6c10d1690b3$export$d35bc1e505d1ebbf = "is-disabled_5a03c7";
$4564c6c10d1690b3$export$ab09ed292dcb9a79 = "spectrum-Link--secondary_5a03c7";
$4564c6c10d1690b3$export$679b6267148797da = "spectrum-Link--overBackground_5a03c7";
$4564c6c10d1690b3$export$81921ea93435e997 = "spectrum-Link--quiet_5a03c7";





function $3c5235ac12f2c9bb$export$a6c7ac8248d6e38a(props) {
    props = $11MUV$useProviderProps(props);
    props = $11MUV$useSlotProps(props, 'link');
    let { variant: variant = 'primary' , isQuiet: isQuiet , children: children , href: // @ts-ignore
    href  } = props;
    let { styleProps: styleProps  } = $11MUV$useStyleProps(props);
    let { hoverProps: hoverProps , isHovered: isHovered  } = $11MUV$useHover({
    });
    if (href) console.warn('href is deprecated, please use an anchor element as children');
    let ref = $11MUV$useRef();
    let { linkProps: linkProps  } = $11MUV$useLink({
        ...props,
        elementType: typeof children === 'string' ? 'span' : 'a'
    }, ref);
    return(/*#__PURE__*/ $11MUV$react.createElement($11MUV$FocusRing, {
        focusRingClass: $11MUV$classNames((/*@__PURE__*/$parcel$interopDefault($4564c6c10d1690b3$exports)), 'focus-ring')
    }, /*#__PURE__*/ $11MUV$react.cloneElement($11MUV$getWrappedElement(children), {
        ...styleProps,
        ...$11MUV$mergeProps(linkProps, hoverProps),
        ref: ref,
        className: $11MUV$classNames((/*@__PURE__*/$parcel$interopDefault($4564c6c10d1690b3$exports)), 'spectrum-Link', {
            'spectrum-Link--quiet': isQuiet,
            [`spectrum-Link--${variant}`]: variant,
            'is-hovered': isHovered
        }, styleProps.className)
    })));
}




export {$3c5235ac12f2c9bb$export$a6c7ac8248d6e38a as Link};
