// @react-aria/live-announcer/dist/main.js
var $9snhK$react = require("react");
var $9snhK$reactdom = require("react-dom");
var $9snhK$reactariavisuallyhidden = require("@react-aria/visually-hidden");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $97cebfa4133ebec3$exports = {};

$parcel$export($97cebfa4133ebec3$exports, "announce", () => $97cebfa4133ebec3$export$a9b970dcc4ae71a9);
$parcel$export($97cebfa4133ebec3$exports, "clearAnnouncer", () => $97cebfa4133ebec3$export$d10ae4f68404609a);
$parcel$export($97cebfa4133ebec3$exports, "destroyAnnouncer", () => $97cebfa4133ebec3$export$d8686216b8b81b2f);



/* Inspired by https://github.com/AlmeroSteyn/react-aria-live */ const $97cebfa4133ebec3$var$LIVEREGION_TIMEOUT_DELAY = 7000;
let $97cebfa4133ebec3$var$liveRegionAnnouncer = /*#__PURE__*/ ($parcel$interopDefault($9snhK$react)).createRef();
let $97cebfa4133ebec3$var$node = null;
let $97cebfa4133ebec3$var$messageId = 0;
function $97cebfa4133ebec3$export$a9b970dcc4ae71a9(message, assertiveness = 'assertive', timeout = $97cebfa4133ebec3$var$LIVEREGION_TIMEOUT_DELAY) {
    $97cebfa4133ebec3$var$ensureInstance((announcer)=>announcer.announce(message, assertiveness, timeout)
    );
}
function $97cebfa4133ebec3$export$d10ae4f68404609a(assertiveness) {
    $97cebfa4133ebec3$var$ensureInstance((announcer)=>announcer.clear(assertiveness)
    );
}
function $97cebfa4133ebec3$export$d8686216b8b81b2f() {
    if ($97cebfa4133ebec3$var$liveRegionAnnouncer.current) {
        ($parcel$interopDefault($9snhK$reactdom)).unmountComponentAtNode($97cebfa4133ebec3$var$node);
        document.body.removeChild($97cebfa4133ebec3$var$node);
        $97cebfa4133ebec3$var$node = null;
    }
}
/**
 * Ensures we only have one instance of the announcer so that we don't have elements competing.
 */ function $97cebfa4133ebec3$var$ensureInstance(callback) {
    if (!$97cebfa4133ebec3$var$liveRegionAnnouncer.current) {
        $97cebfa4133ebec3$var$node = document.createElement('div');
        $97cebfa4133ebec3$var$node.dataset.liveAnnouncer = 'true';
        document.body.prepend($97cebfa4133ebec3$var$node);
        ($parcel$interopDefault($9snhK$reactdom)).render(/*#__PURE__*/ ($parcel$interopDefault($9snhK$react)).createElement($97cebfa4133ebec3$var$LiveRegionAnnouncer, {
            ref: $97cebfa4133ebec3$var$liveRegionAnnouncer
        }), $97cebfa4133ebec3$var$node, ()=>callback($97cebfa4133ebec3$var$liveRegionAnnouncer.current)
        );
    } else callback($97cebfa4133ebec3$var$liveRegionAnnouncer.current);
}
const $97cebfa4133ebec3$var$LiveRegionAnnouncer = /*#__PURE__*/ ($parcel$interopDefault($9snhK$react)).forwardRef((_, ref)=>{
    let [assertiveMessages, setAssertiveMessages] = $9snhK$react.useState([]);
    let [politeMessages, setPoliteMessages] = $9snhK$react.useState([]);
    let clear = (assertiveness)=>{
        if (!assertiveness || assertiveness === 'assertive') setAssertiveMessages([]);
        if (!assertiveness || assertiveness === 'polite') setPoliteMessages([]);
    };
    let $97cebfa4133ebec3$export$a9b970dcc4ae71a9 = (message1, assertiveness = 'assertive', timeout = $97cebfa4133ebec3$var$LIVEREGION_TIMEOUT_DELAY)=>{
        let id = $97cebfa4133ebec3$var$messageId++;
        if (assertiveness === 'assertive') setAssertiveMessages((messages)=>[
                ...messages,
                {
                    id: id,
                    text: message1
                }
            ]
        );
        else setPoliteMessages((messages)=>[
                ...messages,
                {
                    id: id,
                    text: message1
                }
            ]
        );
        if (message1 !== '') setTimeout(()=>{
            if (assertiveness === 'assertive') setAssertiveMessages((messages)=>messages.filter((message)=>message.id !== id
                )
            );
            else setPoliteMessages((messages)=>messages.filter((message)=>message.id !== id
                )
            );
        }, timeout);
    };
    $9snhK$react.useImperativeHandle(ref, ()=>({
            announce: $97cebfa4133ebec3$export$a9b970dcc4ae71a9,
            clear: clear
        })
    );
    return(/*#__PURE__*/ ($parcel$interopDefault($9snhK$react)).createElement($9snhK$react.Fragment, null, /*#__PURE__*/ ($parcel$interopDefault($9snhK$react)).createElement($97cebfa4133ebec3$var$MessageBlock, {
        "aria-live": "assertive"
    }, assertiveMessages.map((message)=>/*#__PURE__*/ ($parcel$interopDefault($9snhK$react)).createElement("div", {
            key: message.id
        }, message.text)
    )), /*#__PURE__*/ ($parcel$interopDefault($9snhK$react)).createElement($97cebfa4133ebec3$var$MessageBlock, {
        "aria-live": "polite"
    }, politeMessages.map((message)=>/*#__PURE__*/ ($parcel$interopDefault($9snhK$react)).createElement("div", {
            key: message.id
        }, message.text)
    ))));
});
function $97cebfa4133ebec3$var$MessageBlock({ children: children , 'aria-live': ariaLive  }) {
    return(/*#__PURE__*/ ($parcel$interopDefault($9snhK$react)).createElement($9snhK$reactariavisuallyhidden.VisuallyHidden, {
        role: "log",
        "aria-live": ariaLive,
        "aria-relevant": "additions"
    }, children));
}


$parcel$exportWildcard(module.exports, $97cebfa4133ebec3$exports);
